// retell-sdk/resources/agent.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Agent = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Agent extends resource_1.APIResource {
    /**
     * Create a new agent
     *
     * @example
     * ```ts
     * const agentResponse = await client.agent.create({
     *   response_engine: {
     *     llm_id: 'llm_234sdertfsdsfsdf',
     *     type: 'retell-llm',
     *   },
     *   voice_id: 'retell-Cimo',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-agent', { body, ...options });
    }
    /**
     * Retrieve details of a specific agent
     *
     * @example
     * ```ts
     * const agentResponse = await client.agent.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(agentID, query = {}, options) {
        return this._client.get((0, path_1.path) `/get-agent/${agentID}`, { query, ...options });
    }
    /**
     * Update an existing agent's latest draft version
     *
     * @example
     * ```ts
     * const agentResponse = await client.agent.update(
     *   '16b980523634a6dc504898cda492e939',
     *   { agent_name: 'Jarvis' },
     * );
     * ```
     */
    update(agentID, params, options) {
        const { version, ...body } = params;
        return this._client.patch((0, path_1.path) `/update-agent/${agentID}`, { query: { version }, body, ...options });
    }
    /**
     * List all agents
     *
     * @example
     * ```ts
     * const agentResponses = await client.agent.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-agents', { query, ...options });
    }
    /**
     * Delete an existing agent
     *
     * @example
     * ```ts
     * await client.agent.delete(
     *   'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * );
     * ```
     */
    delete(agentID, options) {
        return this._client.delete((0, path_1.path) `/delete-agent/${agentID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get all versions of an agent
     *
     * @example
     * ```ts
     * const agentResponses = await client.agent.getVersions(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    getVersions(agentID, options) {
        return this._client.get((0, path_1.path) `/get-agent-versions/${agentID}`, options);
    }
    /**
     * Publish the latest version of the agent and create a new draft agent with newer
     * version.
     *
     * @example
     * ```ts
     * await client.agent.publish(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    publish(agentID, options) {
        return this._client.post((0, path_1.path) `/publish-agent/${agentID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Agent = Agent;
