// retell-sdk/resources/call.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Call = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Call extends resource_1.APIResource {
    /**
     * Retrieve details of a specific call
     *
     * @example
     * ```ts
     * const callResponse = await client.call.retrieve(
     *   '119c3f8e47135a29e65947eeb34cf12d',
     * );
     * ```
     */
    retrieve(callID, options) {
        return this._client.get((0, path_1.path) `/v2/get-call/${callID}`, options);
    }
    /**
     * Update metadata and sensitive data storage settings for an existing call.
     *
     * @example
     * ```ts
     * const callResponse = await client.call.update(
     *   'call_a4441234567890777c4a4a123e6',
     *   {
     *     data_storage_setting: 'everything_except_pii',
     *     metadata: {
     *       customer_id: 'cust_123',
     *       notes: 'Follow-up required',
     *     },
     *   },
     * );
     * ```
     */
    update(callID, body, options) {
        return this._client.patch((0, path_1.path) `/v2/update-call/${callID}`, { body, ...options });
    }
    /**
     * Retrieve call details
     *
     * @example
     * ```ts
     * const callResponses = await client.call.list();
     * ```
     */
    list(body, options) {
        return this._client.post('/v2/list-calls', {
            body,
            timeout: this._client._options.timeout ?? 300000,
            ...options,
        });
    }
    /**
     * Delete a specific call and its associated data
     *
     * @example
     * ```ts
     * await client.call.delete(
     *   '119c3f8e47135a29e65947eeb34cf12d',
     * );
     * ```
     */
    delete(callID, options) {
        return this._client.delete((0, path_1.path) `/v2/delete-call/${callID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Create a new outbound phone call
     *
     * @example
     * ```ts
     * const phoneCallResponse = await client.call.createPhoneCall(
     *   {
     *     from_number: '+14157774444',
     *     to_number: '+12137774445',
     *   },
     * );
     * ```
     */
    createPhoneCall(body, options) {
        return this._client.post('/v2/create-phone-call', {
            body,
            timeout: this._client._options.timeout ?? 120000,
            ...options,
        });
    }
    /**
     * Create a new web call
     *
     * @example
     * ```ts
     * const webCallResponse = await client.call.createWebCall({
     *   agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * });
     * ```
     */
    createWebCall(body, options) {
        return this._client.post('/v2/create-web-call', { body, ...options });
    }
    /**
     * Register a new phone call for custom telephony
     *
     * @example
     * ```ts
     * const phoneCallResponse =
     *   await client.call.registerPhoneCall({
     *     agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *   });
     * ```
     */
    registerPhoneCall(body, options) {
        return this._client.post('/v2/register-phone-call', { body, ...options });
    }
}
exports.Call = Call;
