// retell-sdk/resources/chat-agent.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatAgent = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class ChatAgent extends resource_1.APIResource {
    /**
     * Create a new chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.create({
     *   response_engine: {
     *     llm_id: 'llm_234sdertfsdsfsdf',
     *     type: 'retell-llm',
     *   },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-chat-agent', { body, ...options });
    }
    /**
     * Retrieve details of a specific chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(agentID, query = {}, options) {
        return this._client.get((0, path_1.path) `/get-chat-agent/${agentID}`, { query, ...options });
    }
    /**
     * Update an existing chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.update(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    update(agentID, params, options) {
        const { version, ...body } = params;
        return this._client.patch((0, path_1.path) `/update-chat-agent/${agentID}`, { query: { version }, body, ...options });
    }
    /**
     * List all chat agents
     *
     * @example
     * ```ts
     * const chatAgentResponses = await client.chatAgent.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-chat-agents', { query, ...options });
    }
    /**
     * Delete an existing chat agent
     *
     * @example
     * ```ts
     * await client.chatAgent.delete(
     *   'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * );
     * ```
     */
    delete(agentID, options) {
        return this._client.delete((0, path_1.path) `/delete-chat-agent/${agentID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get all versions of a chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponses =
     *   await client.chatAgent.getVersions(
     *     '16b980523634a6dc504898cda492e939',
     *   );
     * ```
     */
    getVersions(agentID, options) {
        return this._client.get((0, path_1.path) `/get-chat-agent-versions/${agentID}`, options);
    }
    /**
     * Publish the latest version of the chat agent and create a new draft chat agent
     * with newer version.
     *
     * @example
     * ```ts
     * await client.chatAgent.publish(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    publish(agentID, options) {
        return this._client.post((0, path_1.path) `/publish-chat-agent/${agentID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.ChatAgent = ChatAgent;
