// retell-sdk/resources/chat.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Chat extends APIResource {
    /**
     * Create a chat session
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.create({
     *   agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-chat', { body, ...options });
    }
    /**
     * Retrieve details of a specific chat
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(chatID, options) {
        return this._client.get(path `/get-chat/${chatID}`, options);
    }
    /**
     * Update metadata and sensitive data storage settings for an existing chat.
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.update(
     *   'chat_98c1a2157aa0559144d67bb0729',
     *   {
     *     data_storage_setting: 'everything',
     *     metadata: {
     *       customer_id: 'cust_123',
     *       notes: 'Follow-up required',
     *     },
     *     override_dynamic_variables: {
     *       additional_discount: '15%',
     *     },
     *   },
     * );
     * ```
     */
    update(chatID, body, options) {
        return this._client.patch(path `/update-chat/${chatID}`, { body, ...options });
    }
    /**
     * List all chats
     *
     * @example
     * ```ts
     * const chatResponses = await client.chat.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-chat', {
            query,
            timeout: this._client._options.timeout ?? 300000,
            ...options,
        });
    }
    /**
     * Create a chat completion message
     *
     * @example
     * ```ts
     * const response = await client.chat.createChatCompletion({
     *   chat_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *   content: 'hi how are you doing?',
     * });
     * ```
     */
    createChatCompletion(body, options) {
        return this._client.post('/create-chat-completion', {
            body,
            timeout: this._client._options.timeout ?? 300000,
            ...options,
        });
    }
    /**
     * Start an outbound SMS chat conversation with a phone number using the specified
     * agent. The agent must be configured for chat mode. The initial SMS message will
     * be automatically generated and sent based on the agent's configuration.
     *
     * @example
     * ```ts
     * const chatResponse = await client.chat.createSMSChat({
     *   from_number: '+12137771234',
     *   to_number: '+14155551234',
     * });
     * ```
     */
    createSMSChat(body, options) {
        return this._client.post('/create-sms-chat', { body, ...options });
    }
    /**
     * End an ongoing chat
     *
     * @example
     * ```ts
     * await client.chat.end('16b980523634a6dc504898cda492e939');
     * ```
     */
    end(chatID, options) {
        return this._client.patch(path `/end-chat/${chatID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
