// retell-sdk/resources/chat-agent.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class ChatAgent extends APIResource {
    /**
     * Create a new chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.create({
     *   response_engine: {
     *     llm_id: 'llm_234sdertfsdsfsdf',
     *     type: 'retell-llm',
     *   },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-chat-agent', { body, ...options });
    }
    /**
     * Retrieve details of a specific chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(agentID, query = {}, options) {
        return this._client.get(path `/get-chat-agent/${agentID}`, { query, ...options });
    }
    /**
     * Update an existing chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.update(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    update(agentID, params, options) {
        const { version, ...body } = params;
        return this._client.patch(path `/update-chat-agent/${agentID}`, { query: { version }, body, ...options });
    }
    /**
     * List all chat agents
     *
     * @example
     * ```ts
     * const chatAgentResponses = await client.chatAgent.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-chat-agents', { query, ...options });
    }
    /**
     * Delete an existing chat agent
     *
     * @example
     * ```ts
     * await client.chatAgent.delete(
     *   'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * );
     * ```
     */
    delete(agentID, options) {
        return this._client.delete(path `/delete-chat-agent/${agentID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get all versions of a chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponses =
     *   await client.chatAgent.getVersions(
     *     '16b980523634a6dc504898cda492e939',
     *   );
     * ```
     */
    getVersions(agentID, options) {
        return this._client.get(path `/get-chat-agent-versions/${agentID}`, options);
    }
    /**
     * Publish the latest version of the chat agent and create a new draft chat agent
     * with newer version.
     *
     * @example
     * ```ts
     * await client.chatAgent.publish(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    publish(agentID, options) {
        return this._client.post(path `/publish-chat-agent/${agentID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
