// retell-sdk/resources/conversation-flow-component.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationFlowComponent = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class ConversationFlowComponent extends resource_1.APIResource {
    /**
     * Create a new shared conversation flow component
     *
     * @example
     * ```ts
     * const conversationFlowComponentResponse =
     *   await client.conversationFlowComponent.create({
     *     name: 'Customer Information Collector',
     *     nodes: [
     *       {
     *         id: 'collect_info',
     *         instruction: {
     *           text: 'Ask the customer for their name and contact information.',
     *           type: 'prompt',
     *         },
     *         type: 'conversation',
     *       },
     *     ],
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-conversation-flow-component', { body, ...options });
    }
    /**
     * Get a shared conversation flow component
     *
     * @example
     * ```ts
     * const conversationFlowComponentResponse =
     *   await client.conversationFlowComponent.retrieve(
     *     'conversation_flow_component_id',
     *   );
     * ```
     */
    retrieve(conversationFlowComponentID, options) {
        return this._client.get((0, path_1.path) `/get-conversation-flow-component/${conversationFlowComponentID}`, options);
    }
    /**
     * Update an existing shared conversation flow component
     *
     * @example
     * ```ts
     * const conversationFlowComponentResponse =
     *   await client.conversationFlowComponent.update(
     *     'conversation_flow_component_id',
     *   );
     * ```
     */
    update(conversationFlowComponentID, body, options) {
        return this._client.patch((0, path_1.path) `/update-conversation-flow-component/${conversationFlowComponentID}`, {
            body,
            ...options,
        });
    }
    /**
     * List shared conversation flow components
     *
     * @example
     * ```ts
     * const conversationFlowComponentResponses =
     *   await client.conversationFlowComponent.list();
     * ```
     */
    list(options) {
        return this._client.get('/list-conversation-flow-components', options);
    }
    /**
     * Delete a shared conversation flow component. When deleting a shared component,
     * creates local copies for all linked conversation flows.
     *
     * @example
     * ```ts
     * await client.conversationFlowComponent.delete(
     *   'conversation_flow_component_id',
     * );
     * ```
     */
    delete(conversationFlowComponentID, options) {
        return this._client.delete((0, path_1.path) `/delete-conversation-flow-component/${conversationFlowComponentID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.ConversationFlowComponent = ConversationFlowComponent;
