// retell-sdk/resources/conversation-flow.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class ConversationFlow extends APIResource {
    /**
     * Create a new Conversation Flow that can be attached to an agent. This is used to
     * generate response output for the agent.
     *
     * @example
     * ```ts
     * const conversationFlowResponse =
     *   await client.conversationFlow.create({
     *     model_choice: { model: 'gpt-4.1', type: 'cascading' },
     *     nodes: [
     *       {
     *         id: 'start',
     *         instruction: {
     *           text: 'Greet the customer and ask how you can help them.',
     *           type: 'prompt',
     *         },
     *         type: 'conversation',
     *       },
     *     ],
     *     start_speaker: 'agent',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-conversation-flow', { body, ...options });
    }
    /**
     * Retrieve details of a specific Conversation Flow
     *
     * @example
     * ```ts
     * const conversationFlowResponse =
     *   await client.conversationFlow.retrieve(
     *     'conversation_flow_id',
     *   );
     * ```
     */
    retrieve(conversationFlowID, query = {}, options) {
        return this._client.get(path `/get-conversation-flow/${conversationFlowID}`, { query, ...options });
    }
    /**
     * Update an existing conversation flow
     *
     * @example
     * ```ts
     * const conversationFlowResponse =
     *   await client.conversationFlow.update(
     *     'conversation_flow_id',
     *   );
     * ```
     */
    update(conversationFlowID, params, options) {
        const { version, ...body } = params;
        return this._client.patch(path `/update-conversation-flow/${conversationFlowID}`, {
            query: { version },
            body,
            ...options,
        });
    }
    /**
     * List all conversation flows that can be attached to an agent.
     *
     * @example
     * ```ts
     * const conversationFlowResponses =
     *   await client.conversationFlow.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-conversation-flows', { query, ...options });
    }
    /**
     * Delete a conversation flow and all its versions
     *
     * @example
     * ```ts
     * await client.conversationFlow.delete(
     *   'conversation_flow_id',
     * );
     * ```
     */
    delete(conversationFlowID, options) {
        return this._client.delete(path `/delete-conversation-flow/${conversationFlowID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
