// retell-sdk/resources/knowledge-base.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnowledgeBase = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const uploads_1 = require("../internal/uploads.js");
const path_1 = require("../internal/utils/path.js");
class KnowledgeBase extends resource_1.APIResource {
    /**
     * Create a new knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.create({
     *     knowledge_base_name: 'Sample KB',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-knowledge-base', (0, uploads_1.multipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * Retrieve details of a specific knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.retrieve('kb_1234567890');
     * ```
     */
    retrieve(knowledgeBaseID, options) {
        return this._client.get((0, path_1.path) `/get-knowledge-base/${knowledgeBaseID}`, options);
    }
    /**
     * List all knowledge bases
     *
     * @example
     * ```ts
     * const knowledgeBaseResponses =
     *   await client.knowledgeBase.list();
     * ```
     */
    list(options) {
        return this._client.get('/list-knowledge-bases', options);
    }
    /**
     * Delete an existing knowledge base
     *
     * @example
     * ```ts
     * await client.knowledgeBase.delete('kb_1234567890');
     * ```
     */
    delete(knowledgeBaseID, options) {
        return this._client.delete((0, path_1.path) `/delete-knowledge-base/${knowledgeBaseID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Add sources to a knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.addSources('kb_1234567890');
     * ```
     */
    addSources(knowledgeBaseID, body, options) {
        return this._client.post((0, path_1.path) `/add-knowledge-base-sources/${knowledgeBaseID}`, (0, uploads_1.multipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * Delete an existing source from knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.deleteSource(
     *     'source_1234567890',
     *     { knowledge_base_id: 'kb_1234567890' },
     *   );
     * ```
     */
    deleteSource(sourceID, params, options) {
        const { knowledge_base_id } = params;
        return this._client.delete((0, path_1.path) `/delete-knowledge-base-source/${knowledge_base_id}/source/${sourceID}`, options);
    }
}
exports.KnowledgeBase = KnowledgeBase;
