// retell-sdk/resources/knowledge-base.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { multipartFormRequestOptions } from "../internal/uploads.mjs";
import { path } from "../internal/utils/path.mjs";
export class KnowledgeBase extends APIResource {
    /**
     * Create a new knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.create({
     *     knowledge_base_name: 'Sample KB',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-knowledge-base', multipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * Retrieve details of a specific knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.retrieve('kb_1234567890');
     * ```
     */
    retrieve(knowledgeBaseID, options) {
        return this._client.get(path `/get-knowledge-base/${knowledgeBaseID}`, options);
    }
    /**
     * List all knowledge bases
     *
     * @example
     * ```ts
     * const knowledgeBaseResponses =
     *   await client.knowledgeBase.list();
     * ```
     */
    list(options) {
        return this._client.get('/list-knowledge-bases', options);
    }
    /**
     * Delete an existing knowledge base
     *
     * @example
     * ```ts
     * await client.knowledgeBase.delete('kb_1234567890');
     * ```
     */
    delete(knowledgeBaseID, options) {
        return this._client.delete(path `/delete-knowledge-base/${knowledgeBaseID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Add sources to a knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.addSources('kb_1234567890');
     * ```
     */
    addSources(knowledgeBaseID, body, options) {
        return this._client.post(path `/add-knowledge-base-sources/${knowledgeBaseID}`, multipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * Delete an existing source from knowledge base
     *
     * @example
     * ```ts
     * const knowledgeBaseResponse =
     *   await client.knowledgeBase.deleteSource(
     *     'source_1234567890',
     *     { knowledge_base_id: 'kb_1234567890' },
     *   );
     * ```
     */
    deleteSource(sourceID, params, options) {
        const { knowledge_base_id } = params;
        return this._client.delete(path `/delete-knowledge-base-source/${knowledge_base_id}/source/${sourceID}`, options);
    }
}
