// retell-sdk/resources/llm.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Llm = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Llm extends resource_1.APIResource {
    /**
     * Create a new Retell LLM Response Engine that can be attached to an agent. This
     * is used to generate response output for the agent.
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/create-retell-llm', { body, ...options });
    }
    /**
     * Retrieve details of a specific Retell LLM Response Engine
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(llmID, query = {}, options) {
        return this._client.get((0, path_1.path) `/get-retell-llm/${llmID}`, { query, ...options });
    }
    /**
     * Update an existing Retell LLM Response Engine
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.update(
     *   '16b980523634a6dc504898cda492e939',
     *   {
     *     begin_message:
     *       'Hey I am a virtual assistant calling from Retell Hospital.',
     *   },
     * );
     * ```
     */
    update(llmID, params, options) {
        const { query_version, ...body } = params;
        return this._client.patch((0, path_1.path) `/update-retell-llm/${llmID}`, {
            query: { version: query_version },
            body,
            ...options,
        });
    }
    /**
     * List all Retell LLM Response Engines that can be attached to an agent.
     *
     * @example
     * ```ts
     * const llmResponses = await client.llm.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-retell-llms', { query, ...options });
    }
    /**
     * Delete an existing Retell LLM Response Engine
     *
     * @example
     * ```ts
     * await client.llm.delete('oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD');
     * ```
     */
    delete(llmID, options) {
        return this._client.delete((0, path_1.path) `/delete-retell-llm/${llmID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Llm = Llm;
