// retell-sdk/resources/llm.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Llm extends APIResource {
    /**
     * Create a new Retell LLM Response Engine that can be attached to an agent. This
     * is used to generate response output for the agent.
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/create-retell-llm', { body, ...options });
    }
    /**
     * Retrieve details of a specific Retell LLM Response Engine
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.retrieve(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    retrieve(llmID, query = {}, options) {
        return this._client.get(path `/get-retell-llm/${llmID}`, { query, ...options });
    }
    /**
     * Update an existing Retell LLM Response Engine
     *
     * @example
     * ```ts
     * const llmResponse = await client.llm.update(
     *   '16b980523634a6dc504898cda492e939',
     *   {
     *     begin_message:
     *       'Hey I am a virtual assistant calling from Retell Hospital.',
     *   },
     * );
     * ```
     */
    update(llmID, params, options) {
        const { query_version, ...body } = params;
        return this._client.patch(path `/update-retell-llm/${llmID}`, {
            query: { version: query_version },
            body,
            ...options,
        });
    }
    /**
     * List all Retell LLM Response Engines that can be attached to an agent.
     *
     * @example
     * ```ts
     * const llmResponses = await client.llm.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/list-retell-llms', { query, ...options });
    }
    /**
     * Delete an existing Retell LLM Response Engine
     *
     * @example
     * ```ts
     * await client.llm.delete('oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD');
     * ```
     */
    delete(llmID, options) {
        return this._client.delete(path `/delete-retell-llm/${llmID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
