// retell-sdk/resources/phone-number.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumber = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class PhoneNumber extends resource_1.APIResource {
    /**
     * Buy a new phone number & Bind agents
     *
     * @example
     * ```ts
     * const phoneNumberResponse =
     *   await client.phoneNumber.create();
     * ```
     */
    create(body, options) {
        return this._client.post('/create-phone-number', { body, ...options });
    }
    /**
     * Retrieve details of a specific phone number
     *
     * @example
     * ```ts
     * const phoneNumberResponse =
     *   await client.phoneNumber.retrieve('+14157774444');
     * ```
     */
    retrieve(phoneNumber, options) {
        return this._client.get((0, path_1.path) `/get-phone-number/${phoneNumber}`, options);
    }
    /**
     * Update agent bound to a purchased phone number
     *
     * @example
     * ```ts
     * const phoneNumberResponse = await client.phoneNumber.update(
     *   '+14157774444',
     *   {
     *     inbound_agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *     nickname: 'Frontdesk Number',
     *     outbound_agent_id: 'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     *   },
     * );
     * ```
     */
    update(phoneNumber, body, options) {
        return this._client.patch((0, path_1.path) `/update-phone-number/${phoneNumber}`, { body, ...options });
    }
    /**
     * List all phone numbers
     *
     * @example
     * ```ts
     * const phoneNumberResponses =
     *   await client.phoneNumber.list();
     * ```
     */
    list(options) {
        return this._client.get('/list-phone-numbers', options);
    }
    /**
     * Delete an existing phone number
     *
     * @example
     * ```ts
     * await client.phoneNumber.delete('+14157774444');
     * ```
     */
    delete(phoneNumber, options) {
        return this._client.delete((0, path_1.path) `/delete-phone-number/${phoneNumber}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Import a phone number from custom telephony & Bind agents
     *
     * @example
     * ```ts
     * const phoneNumberResponse = await client.phoneNumber.import(
     *   {
     *     phone_number: '+14157774444',
     *     termination_uri: 'someuri.pstn.twilio.com',
     *   },
     * );
     * ```
     */
    import(body, options) {
        return this._client.post('/import-phone-number', { body, ...options });
    }
}
exports.PhoneNumber = PhoneNumber;
