// retell-sdk/resources/tests.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tests = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Tests extends resource_1.APIResource {
    /**
     * Create a batch test to run multiple test cases
     */
    createBatchTest(body, options) {
        return this._client.post('/create-batch-test', { body, ...options });
    }
    /**
     * Create a new test case definition
     */
    createTestCaseDefinition(body, options) {
        return this._client.post('/create-test-case-definition', { body, ...options });
    }
    /**
     * Delete a test case definition
     */
    deleteTestCaseDefinition(testCaseDefinitionID, options) {
        return this._client.delete((0, path_1.path) `/delete-test-case-definition/${testCaseDefinitionID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a batch test job by ID
     */
    getBatchTest(testCaseBatchJobID, options) {
        return this._client.get((0, path_1.path) `/get-batch-test/${testCaseBatchJobID}`, options);
    }
    /**
     * Get a test case definition by ID
     */
    getTestCaseDefinition(testCaseDefinitionID, options) {
        return this._client.get((0, path_1.path) `/get-test-case-definition/${testCaseDefinitionID}`, options);
    }
    /**
     * Get a test case job (test run) by ID
     */
    getTestRun(testCaseJobID, options) {
        return this._client.get((0, path_1.path) `/get-test-run/${testCaseJobID}`, options);
    }
    /**
     * List batch test jobs for a response engine
     */
    listBatchTests(query, options) {
        return this._client.get('/list-batch-tests', { query, ...options });
    }
    /**
     * List test case definitions for a response engine
     */
    listTestCaseDefinitions(query, options) {
        return this._client.get('/list-test-case-definitions', { query, ...options });
    }
    /**
     * List all test case jobs (test runs) for a batch test job
     */
    listTestRuns(testCaseBatchJobID, options) {
        return this._client.get((0, path_1.path) `/list-test-runs/${testCaseBatchJobID}`, options);
    }
    /**
     * Update a test case definition
     */
    updateTestCaseDefinition(testCaseDefinitionID, body, options) {
        return this._client.put((0, path_1.path) `/update-test-case-definition/${testCaseDefinitionID}`, { body, ...options });
    }
}
exports.Tests = Tests;
