// retell-sdk/resources/tests.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Tests extends APIResource {
    /**
     * Create a batch test to run multiple test cases
     */
    createBatchTest(body, options) {
        return this._client.post('/create-batch-test', { body, ...options });
    }
    /**
     * Create a new test case definition
     */
    createTestCaseDefinition(body, options) {
        return this._client.post('/create-test-case-definition', { body, ...options });
    }
    /**
     * Delete a test case definition
     */
    deleteTestCaseDefinition(testCaseDefinitionID, options) {
        return this._client.delete(path `/delete-test-case-definition/${testCaseDefinitionID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get a batch test job by ID
     */
    getBatchTest(testCaseBatchJobID, options) {
        return this._client.get(path `/get-batch-test/${testCaseBatchJobID}`, options);
    }
    /**
     * Get a test case definition by ID
     */
    getTestCaseDefinition(testCaseDefinitionID, options) {
        return this._client.get(path `/get-test-case-definition/${testCaseDefinitionID}`, options);
    }
    /**
     * Get a test case job (test run) by ID
     */
    getTestRun(testCaseJobID, options) {
        return this._client.get(path `/get-test-run/${testCaseJobID}`, options);
    }
    /**
     * List batch test jobs for a response engine
     */
    listBatchTests(query, options) {
        return this._client.get('/list-batch-tests', { query, ...options });
    }
    /**
     * List test case definitions for a response engine
     */
    listTestCaseDefinitions(query, options) {
        return this._client.get('/list-test-case-definitions', { query, ...options });
    }
    /**
     * List all test case jobs (test runs) for a batch test job
     */
    listTestRuns(testCaseBatchJobID, options) {
        return this._client.get(path `/list-test-runs/${testCaseBatchJobID}`, options);
    }
    /**
     * Update a test case definition
     */
    updateTestCaseDefinition(testCaseDefinitionID, body, options) {
        return this._client.put(path `/update-test-case-definition/${testCaseDefinitionID}`, { body, ...options });
    }
}
