// retell-sdk/resources/voice.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { multipartFormRequestOptions } from "../internal/uploads.mjs";
import { path } from "../internal/utils/path.mjs";
export class Voice extends APIResource {
    /**
     * Retrieve details of a specific voice
     */
    retrieve(voiceID, options) {
        return this._client.get(path `/get-voice/${voiceID}`, options);
    }
    /**
     * List all voices available to the user
     */
    list(options) {
        return this._client.get('/list-voices', options);
    }
    /**
     * Add a community voice to the voice library
     */
    addResource(body, options) {
        return this._client.post('/add-community-voice', { body, ...options });
    }
    /**
     * Clone a voice from audio files
     */
    clone(body, options) {
        return this._client.post('/clone-voice', multipartFormRequestOptions({ body, ...options }, this._client));
    }
    /**
     * Search for community voices from voice providers
     */
    search(body, options) {
        return this._client.post('/search-community-voice', { body, ...options });
    }
}
