// retell-sdk/resources/voice.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Voice = void 0;
const resource_1 = require("../core/resource.js");
const uploads_1 = require("../internal/uploads.js");
const path_1 = require("../internal/utils/path.js");
class Voice extends resource_1.APIResource {
    /**
     * Retrieve details of a specific voice
     */
    retrieve(voiceID, options) {
        return this._client.get((0, path_1.path) `/get-voice/${voiceID}`, options);
    }
    /**
     * List all voices available to the user
     */
    list(options) {
        return this._client.get('/list-voices', options);
    }
    /**
     * Add a community voice to the voice library
     */
    addResource(body, options) {
        return this._client.post('/add-community-voice', { body, ...options });
    }
    /**
     * Clone a voice from audio files
     */
    clone(body, options) {
        return this._client.post('/clone-voice', (0, uploads_1.multipartFormRequestOptions)({ body, ...options }, this._client));
    }
    /**
     * Search for community voices from voice providers
     */
    search(body, options) {
        return this._client.post('/search-community-voice', { body, ...options });
    }
}
exports.Voice = Voice;
