// @polar-sh/sdk/dist/esm/models/errors/httpvalidationerror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { ValidationError$inboundSchema, ValidationError$outboundSchema, } from "../components/validationerror.js";
import { PolarError } from "./polarerror.js";
export class HTTPValidationError extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        if (err.detail != null)
            this.detail = err.detail;
        this.name = "HTTPValidationError";
    }
}
/** @internal */
export const HTTPValidationError$inboundSchema = z.pipe(z.object({
    detail: z.optional(z.array(ValidationError$inboundSchema)),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
}), z.transform((v) => {
    return new HTTPValidationError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
}));
/** @internal */
export const HTTPValidationError$outboundSchema = z.pipe(z.pipe(z.custom(x => x instanceof HTTPValidationError), z.transform(v => v.data$)), z.object({
    detail: z.optional(z.array(ValidationError$outboundSchema)),
}));
