// @polar-sh/sdk/dist/esm/models/errors/refundedalready.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { PolarError } from "./polarerror.js";
export class RefundedAlready extends PolarError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.error = err.error;
        this.detail = err.detail;
        this.name = "RefundedAlready";
    }
}
/** @internal */
export const RefundedAlready$inboundSchema = z.pipe(z.object({
    error: z.literal("RefundedAlready"),
    detail: z.string(),
    request$: z.custom(x => x instanceof Request),
    response$: z.custom(x => x instanceof Response),
    body$: z.string(),
}), z.transform((v) => {
    return new RefundedAlready(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
}));
