// @polar-sh/sdk/dist/commonjs/sdk/customerseats.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSeats = void 0;
const customerSeatsAssignSeat_js_1 = require("../funcs/customerSeatsAssignSeat.js");
const customerSeatsClaimSeat_js_1 = require("../funcs/customerSeatsClaimSeat.js");
const customerSeatsGetClaimInfo_js_1 = require("../funcs/customerSeatsGetClaimInfo.js");
const customerSeatsListSeats_js_1 = require("../funcs/customerSeatsListSeats.js");
const customerSeatsResendInvitation_js_1 = require("../funcs/customerSeatsResendInvitation.js");
const customerSeatsRevokeSeat_js_1 = require("../funcs/customerSeatsRevokeSeat.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class CustomerSeats extends sdks_js_1.ClientSDK {
    /**
     * List Seats
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async listSeats(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsListSeats_js_1.customerSeatsListSeats)(this, request, options));
    }
    /**
     * Assign Seat
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async assignSeat(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsAssignSeat_js_1.customerSeatsAssignSeat)(this, request, options));
    }
    /**
     * Revoke Seat
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async revokeSeat(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsRevokeSeat_js_1.customerSeatsRevokeSeat)(this, request, options));
    }
    /**
     * Resend Invitation
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async resendInvitation(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsResendInvitation_js_1.customerSeatsResendInvitation)(this, request, options));
    }
    /**
     * Get Claim Info
     */
    async getClaimInfo(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsGetClaimInfo_js_1.customerSeatsGetClaimInfo)(this, request, options));
    }
    /**
     * Claim Seat
     */
    async claimSeat(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerSeatsClaimSeat_js_1.customerSeatsClaimSeat)(this, request, options));
    }
}
exports.CustomerSeats = CustomerSeats;
