// @polar-sh/sdk/dist/commonjs/sdk/members.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Members = void 0;
const membersCreateMember_js_1 = require("../funcs/membersCreateMember.js");
const membersDeleteMember_js_1 = require("../funcs/membersDeleteMember.js");
const membersGetMember_js_1 = require("../funcs/membersGetMember.js");
const membersListMembers_js_1 = require("../funcs/membersListMembers.js");
const membersUpdateMember_js_1 = require("../funcs/membersUpdateMember.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class Members extends sdks_js_1.ClientSDK {
    /**
     * List Members
     *
     * @remarks
     * List members with optional customer ID filter.
     *
     * **Scopes**: `members:read` `members:write`
     */
    async listMembers(request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, membersListMembers_js_1.membersListMembers)(this, request, options));
    }
    /**
     * Create Member
     *
     * @remarks
     * Create a new member for a customer.
     *
     * Only B2B customers with the member management feature enabled can add members.
     * The authenticated user or organization must have access to the customer's organization.
     *
     * **Scopes**: `members:write`
     */
    async createMember(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, membersCreateMember_js_1.membersCreateMember)(this, request, options));
    }
    /**
     * Get Member
     *
     * @remarks
     * Get a member by ID.
     *
     * The authenticated user or organization must have access to the member's organization.
     *
     * **Scopes**: `members:read` `members:write`
     */
    async getMember(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, membersGetMember_js_1.membersGetMember)(this, request, options));
    }
    /**
     * Delete Member
     *
     * @remarks
     * Delete a member.
     *
     * The authenticated user or organization must have access to the member's organization.
     *
     * **Scopes**: `members:write`
     */
    async deleteMember(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, membersDeleteMember_js_1.membersDeleteMember)(this, request, options));
    }
    /**
     * Update Member
     *
     * @remarks
     * Update a member.
     *
     * Only name and role can be updated.
     * The authenticated user or organization must have access to the member's organization.
     *
     * **Scopes**: `members:write`
     */
    async updateMember(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, membersUpdateMember_js_1.membersUpdateMember)(this, request, options));
    }
}
exports.Members = Members;
