// @polar-sh/sdk/dist/commonjs/sdk/metrics.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Metrics = void 0;
const metricsCreateDashboard_js_1 = require("../funcs/metricsCreateDashboard.js");
const metricsDeleteDashboard_js_1 = require("../funcs/metricsDeleteDashboard.js");
const metricsGet_js_1 = require("../funcs/metricsGet.js");
const metricsGetDashboard_js_1 = require("../funcs/metricsGetDashboard.js");
const metricsLimits_js_1 = require("../funcs/metricsLimits.js");
const metricsListDashboards_js_1 = require("../funcs/metricsListDashboards.js");
const metricsUpdateDashboard_js_1 = require("../funcs/metricsUpdateDashboard.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class Metrics extends sdks_js_1.ClientSDK {
    /**
     * Get Metrics
     *
     * @remarks
     * Get metrics about your orders and subscriptions.
     *
     * Currency values are output in cents.
     *
     * **Scopes**: `metrics:read`
     */
    async get(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsGet_js_1.metricsGet)(this, request, options));
    }
    /**
     * Get Metrics Limits
     *
     * @remarks
     * Get the interval limits for the metrics endpoint.
     *
     * **Scopes**: `metrics:read`
     */
    async limits(options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsLimits_js_1.metricsLimits)(this, options));
    }
    /**
     * List Metric Dashboards
     *
     * @remarks
     * List user-defined metric dashboards.
     *
     * **Scopes**: `metrics:read`
     */
    async listDashboards(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsListDashboards_js_1.metricsListDashboards)(this, request, options));
    }
    /**
     * Create Metric Dashboard
     *
     * @remarks
     * Create a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async createDashboard(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsCreateDashboard_js_1.metricsCreateDashboard)(this, request, options));
    }
    /**
     * Get Metric Dashboard
     *
     * @remarks
     * Get a user-defined metric dashboard by ID.
     *
     * **Scopes**: `metrics:read`
     */
    async getDashboard(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsGetDashboard_js_1.metricsGetDashboard)(this, request, options));
    }
    /**
     * Delete Metric Dashboard
     *
     * @remarks
     * Delete a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async deleteDashboard(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsDeleteDashboard_js_1.metricsDeleteDashboard)(this, request, options));
    }
    /**
     * Update Metric Dashboard
     *
     * @remarks
     * Update a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async updateDashboard(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, metricsUpdateDashboard_js_1.metricsUpdateDashboard)(this, request, options));
    }
}
exports.Metrics = Metrics;
