// @polar-sh/sdk/dist/commonjs/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Polar = void 0;
const validateWebhook_js_1 = require("../funcs/validateWebhook.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const benefitgrants_js_1 = require("./benefitgrants.js");
const benefits_js_1 = require("./benefits.js");
const checkoutlinks_js_1 = require("./checkoutlinks.js");
const checkouts_js_1 = require("./checkouts.js");
const customermeters_js_1 = require("./customermeters.js");
const customerportal_js_1 = require("./customerportal.js");
const customers_js_1 = require("./customers.js");
const customerseats_js_1 = require("./customerseats.js");
const customersessions_js_1 = require("./customersessions.js");
const customfields_js_1 = require("./customfields.js");
const discounts_js_1 = require("./discounts.js");
const disputes_js_1 = require("./disputes.js");
const events_js_1 = require("./events.js");
const eventtypes_js_1 = require("./eventtypes.js");
const files_js_1 = require("./files.js");
const licensekeys_js_1 = require("./licensekeys.js");
const members_js_1 = require("./members.js");
const meters_js_1 = require("./meters.js");
const metrics_js_1 = require("./metrics.js");
const oauth2_js_1 = require("./oauth2.js");
const orders_js_1 = require("./orders.js");
const organizationaccesstokens_js_1 = require("./organizationaccesstokens.js");
const organizations_js_1 = require("./organizations.js");
const payments_js_1 = require("./payments.js");
const products_js_1 = require("./products.js");
const refunds_js_1 = require("./refunds.js");
const subscriptions_js_1 = require("./subscriptions.js");
const webhooks_js_1 = require("./webhooks.js");
class Polar extends sdks_js_1.ClientSDK {
    get organizations() {
        return (this._organizations ?? (this._organizations = new organizations_js_1.Organizations(this._options)));
    }
    get subscriptions() {
        return (this._subscriptions ?? (this._subscriptions = new subscriptions_js_1.Subscriptions(this._options)));
    }
    get oauth2() {
        return (this._oauth2 ?? (this._oauth2 = new oauth2_js_1.Oauth2(this._options)));
    }
    get benefits() {
        return (this._benefits ?? (this._benefits = new benefits_js_1.Benefits(this._options)));
    }
    get benefitGrants() {
        return (this._benefitGrants ?? (this._benefitGrants = new benefitgrants_js_1.BenefitGrants(this._options)));
    }
    get webhooks() {
        return (this._webhooks ?? (this._webhooks = new webhooks_js_1.Webhooks(this._options)));
    }
    get products() {
        return (this._products ?? (this._products = new products_js_1.Products(this._options)));
    }
    get orders() {
        return (this._orders ?? (this._orders = new orders_js_1.Orders(this._options)));
    }
    get refunds() {
        return (this._refunds ?? (this._refunds = new refunds_js_1.Refunds(this._options)));
    }
    get disputes() {
        return (this._disputes ?? (this._disputes = new disputes_js_1.Disputes(this._options)));
    }
    get checkouts() {
        return (this._checkouts ?? (this._checkouts = new checkouts_js_1.Checkouts(this._options)));
    }
    get files() {
        return (this._files ?? (this._files = new files_js_1.Files(this._options)));
    }
    get metrics() {
        return (this._metrics ?? (this._metrics = new metrics_js_1.Metrics(this._options)));
    }
    get licenseKeys() {
        return (this._licenseKeys ?? (this._licenseKeys = new licensekeys_js_1.LicenseKeys(this._options)));
    }
    get checkoutLinks() {
        return (this._checkoutLinks ?? (this._checkoutLinks = new checkoutlinks_js_1.CheckoutLinks(this._options)));
    }
    get customFields() {
        return (this._customFields ?? (this._customFields = new customfields_js_1.CustomFields(this._options)));
    }
    get discounts() {
        return (this._discounts ?? (this._discounts = new discounts_js_1.Discounts(this._options)));
    }
    get customers() {
        return (this._customers ?? (this._customers = new customers_js_1.Customers(this._options)));
    }
    get members() {
        return (this._members ?? (this._members = new members_js_1.Members(this._options)));
    }
    get customerPortal() {
        return (this._customerPortal ?? (this._customerPortal = new customerportal_js_1.CustomerPortal(this._options)));
    }
    get customerSeats() {
        return (this._customerSeats ?? (this._customerSeats = new customerseats_js_1.CustomerSeats(this._options)));
    }
    get customerSessions() {
        return (this._customerSessions ?? (this._customerSessions = new customersessions_js_1.CustomerSessions(this._options)));
    }
    get events() {
        return (this._events ?? (this._events = new events_js_1.Events(this._options)));
    }
    get eventTypes() {
        return (this._eventTypes ?? (this._eventTypes = new eventtypes_js_1.EventTypes(this._options)));
    }
    get meters() {
        return (this._meters ?? (this._meters = new meters_js_1.Meters(this._options)));
    }
    get organizationAccessTokens() {
        return (this._organizationAccessTokens ?? (this._organizationAccessTokens = new organizationaccesstokens_js_1.OrganizationAccessTokens(this._options)));
    }
    get customerMeters() {
        return (this._customerMeters ?? (this._customerMeters = new customermeters_js_1.CustomerMeters(this._options)));
    }
    get payments() {
        return (this._payments ?? (this._payments = new payments_js_1.Payments(this._options)));
    }
    async validateWebhook({ request, }) {
        return (0, fp_js_1.unwrapAsync)((0, validateWebhook_js_1.validateWebhook)(this, { request }));
    }
}
exports.Polar = Polar;
