// @polar-sh/sdk/dist/commonjs/sdk/seats.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Seats = void 0;
const customerPortalSeatsAssignSeat_js_1 = require("../funcs/customerPortalSeatsAssignSeat.js");
const customerPortalSeatsListClaimedSubscriptions_js_1 = require("../funcs/customerPortalSeatsListClaimedSubscriptions.js");
const customerPortalSeatsListSeats_js_1 = require("../funcs/customerPortalSeatsListSeats.js");
const customerPortalSeatsResendInvitation_js_1 = require("../funcs/customerPortalSeatsResendInvitation.js");
const customerPortalSeatsRevokeSeat_js_1 = require("../funcs/customerPortalSeatsRevokeSeat.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const operations_js_1 = require("../types/operations.js");
class Seats extends sdks_js_1.ClientSDK {
    /**
     * List Seats
     *
     * @remarks
     * **Scopes**: `customer_portal:read` `customer_portal:write`
     */
    async listSeats(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerPortalSeatsListSeats_js_1.customerPortalSeatsListSeats)(this, security, request, options));
    }
    /**
     * Assign Seat
     */
    async assignSeat(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerPortalSeatsAssignSeat_js_1.customerPortalSeatsAssignSeat)(this, security, request, options));
    }
    /**
     * Revoke Seat
     */
    async revokeSeat(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerPortalSeatsRevokeSeat_js_1.customerPortalSeatsRevokeSeat)(this, security, request, options));
    }
    /**
     * Resend Invitation
     */
    async resendInvitation(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, customerPortalSeatsResendInvitation_js_1.customerPortalSeatsResendInvitation)(this, security, request, options));
    }
    /**
     * List Claimed Subscriptions
     *
     * @remarks
     * List all subscriptions where the authenticated customer has claimed a seat.
     *
     * **Scopes**: `customer_portal:read` `customer_portal:write`
     */
    async listClaimedSubscriptions(security, request, options) {
        return (0, operations_js_1.unwrapResultIterator)((0, customerPortalSeatsListClaimedSubscriptions_js_1.customerPortalSeatsListClaimedSubscriptions)(this, security, request, options));
    }
}
exports.Seats = Seats;
