// @polar-sh/sdk/dist/commonjs/models/operations/benefitsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitsUpdateRequest$outboundSchema = exports.BenefitsUpdateBenefitUpdate$outboundSchema = void 0;
exports.benefitsUpdateBenefitUpdateToJSON = benefitsUpdateBenefitUpdateToJSON;
exports.benefitsUpdateRequestToJSON = benefitsUpdateRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const benefitcustomupdate_js_1 = require("../components/benefitcustomupdate.js");
const benefitdiscordupdate_js_1 = require("../components/benefitdiscordupdate.js");
const benefitdownloadablesupdate_js_1 = require("../components/benefitdownloadablesupdate.js");
const benefitfeatureflagupdate_js_1 = require("../components/benefitfeatureflagupdate.js");
const benefitgithubrepositoryupdate_js_1 = require("../components/benefitgithubrepositoryupdate.js");
const benefitlicensekeysupdate_js_1 = require("../components/benefitlicensekeysupdate.js");
const benefitmetercreditupdate_js_1 = require("../components/benefitmetercreditupdate.js");
/** @internal */
exports.BenefitsUpdateBenefitUpdate$outboundSchema = z.union([
    benefitcustomupdate_js_1.BenefitCustomUpdate$outboundSchema,
    benefitdiscordupdate_js_1.BenefitDiscordUpdate$outboundSchema,
    benefitgithubrepositoryupdate_js_1.BenefitGitHubRepositoryUpdate$outboundSchema,
    benefitdownloadablesupdate_js_1.BenefitDownloadablesUpdate$outboundSchema,
    benefitlicensekeysupdate_js_1.BenefitLicenseKeysUpdate$outboundSchema,
    benefitmetercreditupdate_js_1.BenefitMeterCreditUpdate$outboundSchema,
    benefitfeatureflagupdate_js_1.BenefitFeatureFlagUpdate$outboundSchema,
]);
function benefitsUpdateBenefitUpdateToJSON(benefitsUpdateBenefitUpdate) {
    return JSON.stringify(exports.BenefitsUpdateBenefitUpdate$outboundSchema.parse(benefitsUpdateBenefitUpdate));
}
/** @internal */
exports.BenefitsUpdateRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    requestBody: z.union([
        benefitcustomupdate_js_1.BenefitCustomUpdate$outboundSchema,
        benefitdiscordupdate_js_1.BenefitDiscordUpdate$outboundSchema,
        benefitgithubrepositoryupdate_js_1.BenefitGitHubRepositoryUpdate$outboundSchema,
        benefitdownloadablesupdate_js_1.BenefitDownloadablesUpdate$outboundSchema,
        benefitlicensekeysupdate_js_1.BenefitLicenseKeysUpdate$outboundSchema,
        benefitmetercreditupdate_js_1.BenefitMeterCreditUpdate$outboundSchema,
        benefitfeatureflagupdate_js_1.BenefitFeatureFlagUpdate$outboundSchema,
    ]),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        requestBody: "RequestBody",
    });
}));
function benefitsUpdateRequestToJSON(benefitsUpdateRequest) {
    return JSON.stringify(exports.BenefitsUpdateRequest$outboundSchema.parse(benefitsUpdateRequest));
}
