// @polar-sh/sdk/dist/commonjs/models/operations/customerportalbenefitgrantslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalBenefitGrantsListResponse$inboundSchema = exports.CustomerPortalBenefitGrantsListRequest$outboundSchema = exports.QueryParamMemberIDFilter$outboundSchema = exports.QueryParamSubscriptionIDFilter$outboundSchema = exports.CustomerPortalBenefitGrantsListQueryParamOrderIDFilter$outboundSchema = exports.QueryParamCheckoutIDFilter$outboundSchema = exports.CustomerPortalBenefitGrantsListQueryParamBenefitIDFilter$outboundSchema = exports.QueryParamBenefitTypeFilter$outboundSchema = exports.CustomerPortalBenefitGrantsListSecurity$outboundSchema = void 0;
exports.customerPortalBenefitGrantsListSecurityToJSON = customerPortalBenefitGrantsListSecurityToJSON;
exports.queryParamBenefitTypeFilterToJSON = queryParamBenefitTypeFilterToJSON;
exports.customerPortalBenefitGrantsListQueryParamBenefitIDFilterToJSON = customerPortalBenefitGrantsListQueryParamBenefitIDFilterToJSON;
exports.queryParamCheckoutIDFilterToJSON = queryParamCheckoutIDFilterToJSON;
exports.customerPortalBenefitGrantsListQueryParamOrderIDFilterToJSON = customerPortalBenefitGrantsListQueryParamOrderIDFilterToJSON;
exports.queryParamSubscriptionIDFilterToJSON = queryParamSubscriptionIDFilterToJSON;
exports.queryParamMemberIDFilterToJSON = queryParamMemberIDFilterToJSON;
exports.customerPortalBenefitGrantsListRequestToJSON = customerPortalBenefitGrantsListRequestToJSON;
exports.customerPortalBenefitGrantsListResponseFromJSON = customerPortalBenefitGrantsListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const benefittype_js_1 = require("../components/benefittype.js");
const customerbenefitgrantsortproperty_js_1 = require("../components/customerbenefitgrantsortproperty.js");
const listresourcecustomerbenefitgrant_js_1 = require("../components/listresourcecustomerbenefitgrant.js");
/** @internal */
exports.CustomerPortalBenefitGrantsListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
function customerPortalBenefitGrantsListSecurityToJSON(customerPortalBenefitGrantsListSecurity) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsListSecurity$outboundSchema.parse(customerPortalBenefitGrantsListSecurity));
}
/** @internal */
exports.QueryParamBenefitTypeFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    benefittype_js_1.BenefitType$outboundSchema,
    z.array(benefittype_js_1.BenefitType$outboundSchema),
]);
function queryParamBenefitTypeFilterToJSON(queryParamBenefitTypeFilter) {
    return JSON.stringify(exports.QueryParamBenefitTypeFilter$outboundSchema.parse(queryParamBenefitTypeFilter));
}
/** @internal */
exports.CustomerPortalBenefitGrantsListQueryParamBenefitIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function customerPortalBenefitGrantsListQueryParamBenefitIDFilterToJSON(customerPortalBenefitGrantsListQueryParamBenefitIDFilter) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsListQueryParamBenefitIDFilter$outboundSchema
        .parse(customerPortalBenefitGrantsListQueryParamBenefitIDFilter));
}
/** @internal */
exports.QueryParamCheckoutIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function queryParamCheckoutIDFilterToJSON(queryParamCheckoutIDFilter) {
    return JSON.stringify(exports.QueryParamCheckoutIDFilter$outboundSchema.parse(queryParamCheckoutIDFilter));
}
/** @internal */
exports.CustomerPortalBenefitGrantsListQueryParamOrderIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function customerPortalBenefitGrantsListQueryParamOrderIDFilterToJSON(customerPortalBenefitGrantsListQueryParamOrderIDFilter) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsListQueryParamOrderIDFilter$outboundSchema.parse(customerPortalBenefitGrantsListQueryParamOrderIDFilter));
}
/** @internal */
exports.QueryParamSubscriptionIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function queryParamSubscriptionIDFilterToJSON(queryParamSubscriptionIDFilter) {
    return JSON.stringify(exports.QueryParamSubscriptionIDFilter$outboundSchema.parse(queryParamSubscriptionIDFilter));
}
/** @internal */
exports.QueryParamMemberIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function queryParamMemberIDFilterToJSON(queryParamMemberIDFilter) {
    return JSON.stringify(exports.QueryParamMemberIDFilter$outboundSchema.parse(queryParamMemberIDFilter));
}
/** @internal */
exports.CustomerPortalBenefitGrantsListRequest$outboundSchema = z.pipe(z.object({
    query: z.optional(z.nullable(z.string())),
    typeFilter: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([
        benefittype_js_1.BenefitType$outboundSchema,
        z.array(benefittype_js_1.BenefitType$outboundSchema),
    ]))),
    benefitId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    checkoutId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    subscriptionId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    memberId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(customerbenefitgrantsortproperty_js_1.CustomerBenefitGrantSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        typeFilter: "type_filter",
        benefitId: "benefit_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        subscriptionId: "subscription_id",
        memberId: "member_id",
    });
}));
function customerPortalBenefitGrantsListRequestToJSON(customerPortalBenefitGrantsListRequest) {
    return JSON.stringify(exports.CustomerPortalBenefitGrantsListRequest$outboundSchema.parse(customerPortalBenefitGrantsListRequest));
}
/** @internal */
exports.CustomerPortalBenefitGrantsListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcecustomerbenefitgrant_js_1.ListResourceCustomerBenefitGrant$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function customerPortalBenefitGrantsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalBenefitGrantsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalBenefitGrantsListResponse' from JSON`);
}
