// @polar-sh/sdk/dist/commonjs/models/operations/customerportalwalletslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalWalletsListResponse$inboundSchema = exports.CustomerPortalWalletsListRequest$outboundSchema = exports.CustomerPortalWalletsListSecurity$outboundSchema = void 0;
exports.customerPortalWalletsListSecurityToJSON = customerPortalWalletsListSecurityToJSON;
exports.customerPortalWalletsListRequestToJSON = customerPortalWalletsListRequestToJSON;
exports.customerPortalWalletsListResponseFromJSON = customerPortalWalletsListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerwalletsortproperty_js_1 = require("../components/customerwalletsortproperty.js");
const listresourcecustomerwallet_js_1 = require("../components/listresourcecustomerwallet.js");
/** @internal */
exports.CustomerPortalWalletsListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
function customerPortalWalletsListSecurityToJSON(customerPortalWalletsListSecurity) {
    return JSON.stringify(exports.CustomerPortalWalletsListSecurity$outboundSchema.parse(customerPortalWalletsListSecurity));
}
/** @internal */
exports.CustomerPortalWalletsListRequest$outboundSchema = z.object({
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(customerwalletsortproperty_js_1.CustomerWalletSortProperty$outboundSchema))),
});
function customerPortalWalletsListRequestToJSON(customerPortalWalletsListRequest) {
    return JSON.stringify(exports.CustomerPortalWalletsListRequest$outboundSchema.parse(customerPortalWalletsListRequest));
}
/** @internal */
exports.CustomerPortalWalletsListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcecustomerwallet_js_1.ListResourceCustomerWallet$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function customerPortalWalletsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalWalletsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalWalletsListResponse' from JSON`);
}
