// @polar-sh/sdk/dist/commonjs/models/operations/disputeslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisputesListResponse$inboundSchema = exports.DisputesListRequest$outboundSchema = exports.StatusFilter$outboundSchema = exports.QueryParamOrderIDFilter$outboundSchema = exports.DisputesListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.disputesListQueryParamOrganizationIDFilterToJSON = disputesListQueryParamOrganizationIDFilterToJSON;
exports.queryParamOrderIDFilterToJSON = queryParamOrderIDFilterToJSON;
exports.statusFilterToJSON = statusFilterToJSON;
exports.disputesListRequestToJSON = disputesListRequestToJSON;
exports.disputesListResponseFromJSON = disputesListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const disputesortproperty_js_1 = require("../components/disputesortproperty.js");
const disputestatus_js_1 = require("../components/disputestatus.js");
const listresourcedispute_js_1 = require("../components/listresourcedispute.js");
/** @internal */
exports.DisputesListQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function disputesListQueryParamOrganizationIDFilterToJSON(disputesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.DisputesListQueryParamOrganizationIDFilter$outboundSchema.parse(disputesListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.QueryParamOrderIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function queryParamOrderIDFilterToJSON(queryParamOrderIDFilter) {
    return JSON.stringify(exports.QueryParamOrderIDFilter$outboundSchema.parse(queryParamOrderIDFilter));
}
/** @internal */
exports.StatusFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    disputestatus_js_1.DisputeStatus$outboundSchema,
    z.array(disputestatus_js_1.DisputeStatus$outboundSchema),
]);
function statusFilterToJSON(statusFilter) {
    return JSON.stringify(exports.StatusFilter$outboundSchema.parse(statusFilter));
}
/** @internal */
exports.DisputesListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    status: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([
        disputestatus_js_1.DisputeStatus$outboundSchema,
        z.array(disputestatus_js_1.DisputeStatus$outboundSchema),
    ]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(disputesortproperty_js_1.DisputeSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        orderId: "order_id",
    });
}));
function disputesListRequestToJSON(disputesListRequest) {
    return JSON.stringify(exports.DisputesListRequest$outboundSchema.parse(disputesListRequest));
}
/** @internal */
exports.DisputesListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcedispute_js_1.ListResourceDispute$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function disputesListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DisputesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DisputesListResponse' from JSON`);
}
