// @polar-sh/sdk/dist/commonjs/models/operations/eventtypeslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventTypesListResponse$inboundSchema = exports.EventTypesListRequest$outboundSchema = exports.EventTypesListQueryParamExternalCustomerIDFilter$outboundSchema = exports.EventTypesListQueryParamCustomerIDFilter$outboundSchema = exports.EventTypesListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.eventTypesListQueryParamOrganizationIDFilterToJSON = eventTypesListQueryParamOrganizationIDFilterToJSON;
exports.eventTypesListQueryParamCustomerIDFilterToJSON = eventTypesListQueryParamCustomerIDFilterToJSON;
exports.eventTypesListQueryParamExternalCustomerIDFilterToJSON = eventTypesListQueryParamExternalCustomerIDFilterToJSON;
exports.eventTypesListRequestToJSON = eventTypesListRequestToJSON;
exports.eventTypesListResponseFromJSON = eventTypesListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const eventsource_js_1 = require("../components/eventsource.js");
const eventtypessortproperty_js_1 = require("../components/eventtypessortproperty.js");
const listresourceeventtypewithstats_js_1 = require("../components/listresourceeventtypewithstats.js");
/** @internal */
exports.EventTypesListQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventTypesListQueryParamOrganizationIDFilterToJSON(eventTypesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.EventTypesListQueryParamOrganizationIDFilter$outboundSchema.parse(eventTypesListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.EventTypesListQueryParamCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventTypesListQueryParamCustomerIDFilterToJSON(eventTypesListQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.EventTypesListQueryParamCustomerIDFilter$outboundSchema.parse(eventTypesListQueryParamCustomerIDFilter));
}
/** @internal */
exports.EventTypesListQueryParamExternalCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventTypesListQueryParamExternalCustomerIDFilterToJSON(eventTypesListQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(exports.EventTypesListQueryParamExternalCustomerIDFilter$outboundSchema.parse(eventTypesListQueryParamExternalCustomerIDFilter));
}
/** @internal */
exports.EventTypesListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    query: z.optional(z.nullable(z.string())),
    rootEvents: z._default(z.boolean(), false),
    parentId: z.optional(z.nullable(z.string())),
    source: z.optional(z.nullable(eventsource_js_1.EventSource$outboundSchema)),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(eventtypessortproperty_js_1.EventTypesSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        rootEvents: "root_events",
        parentId: "parent_id",
    });
}));
function eventTypesListRequestToJSON(eventTypesListRequest) {
    return JSON.stringify(exports.EventTypesListRequest$outboundSchema.parse(eventTypesListRequest));
}
/** @internal */
exports.EventTypesListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourceeventtypewithstats_js_1.ListResourceEventTypeWithStats$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function eventTypesListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventTypesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventTypesListResponse' from JSON`);
}
