// @polar-sh/sdk/dist/commonjs/models/operations/eventslistnames.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsListNamesResponse$inboundSchema = exports.EventsListNamesRequest$outboundSchema = exports.QueryParamSourceFilter$outboundSchema = exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema = exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema = exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.eventsListNamesQueryParamOrganizationIDFilterToJSON = eventsListNamesQueryParamOrganizationIDFilterToJSON;
exports.eventsListNamesQueryParamCustomerIDFilterToJSON = eventsListNamesQueryParamCustomerIDFilterToJSON;
exports.eventsListNamesQueryParamExternalCustomerIDFilterToJSON = eventsListNamesQueryParamExternalCustomerIDFilterToJSON;
exports.queryParamSourceFilterToJSON = queryParamSourceFilterToJSON;
exports.eventsListNamesRequestToJSON = eventsListNamesRequestToJSON;
exports.eventsListNamesResponseFromJSON = eventsListNamesResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const eventnamessortproperty_js_1 = require("../components/eventnamessortproperty.js");
const eventsource_js_1 = require("../components/eventsource.js");
const listresourceeventname_js_1 = require("../components/listresourceeventname.js");
/** @internal */
exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventsListNamesQueryParamOrganizationIDFilterToJSON(eventsListNamesQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamOrganizationIDFilter$outboundSchema.parse(eventsListNamesQueryParamOrganizationIDFilter));
}
/** @internal */
exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventsListNamesQueryParamCustomerIDFilterToJSON(eventsListNamesQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamCustomerIDFilter$outboundSchema.parse(eventsListNamesQueryParamCustomerIDFilter));
}
/** @internal */
exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function eventsListNamesQueryParamExternalCustomerIDFilterToJSON(eventsListNamesQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(exports.EventsListNamesQueryParamExternalCustomerIDFilter$outboundSchema.parse(eventsListNamesQueryParamExternalCustomerIDFilter));
}
/** @internal */
exports.QueryParamSourceFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    eventsource_js_1.EventSource$outboundSchema,
    z.array(eventsource_js_1.EventSource$outboundSchema),
]);
function queryParamSourceFilterToJSON(queryParamSourceFilter) {
    return JSON.stringify(exports.QueryParamSourceFilter$outboundSchema.parse(queryParamSourceFilter));
}
/** @internal */
exports.EventsListNamesRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    source: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([
        eventsource_js_1.EventSource$outboundSchema,
        z.array(eventsource_js_1.EventSource$outboundSchema),
    ]))),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(eventnamessortproperty_js_1.EventNamesSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
}));
function eventsListNamesRequestToJSON(eventsListNamesRequest) {
    return JSON.stringify(exports.EventsListNamesRequest$outboundSchema.parse(eventsListNamesRequest));
}
/** @internal */
exports.EventsListNamesResponse$inboundSchema = z.pipe(z.object({
    Result: listresourceeventname_js_1.ListResourceEventName$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function eventsListNamesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsListNamesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListNamesResponse' from JSON`);
}
