// @polar-sh/sdk/dist/commonjs/models/operations/metersquantities.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetersQuantitiesRequest$outboundSchema = exports.MetersQuantitiesQueryParamExternalCustomerIDFilter$outboundSchema = exports.MetersQuantitiesQueryParamCustomerIDFilter$outboundSchema = void 0;
exports.metersQuantitiesQueryParamCustomerIDFilterToJSON = metersQuantitiesQueryParamCustomerIDFilterToJSON;
exports.metersQuantitiesQueryParamExternalCustomerIDFilterToJSON = metersQuantitiesQueryParamExternalCustomerIDFilterToJSON;
exports.metersQuantitiesRequestToJSON = metersQuantitiesRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const aggregationfunction_js_1 = require("../components/aggregationfunction.js");
const subscriptionslist_js_1 = require("../components/subscriptionslist.js");
const timeinterval_js_1 = require("../components/timeinterval.js");
/** @internal */
exports.MetersQuantitiesQueryParamCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function metersQuantitiesQueryParamCustomerIDFilterToJSON(metersQuantitiesQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.MetersQuantitiesQueryParamCustomerIDFilter$outboundSchema.parse(metersQuantitiesQueryParamCustomerIDFilter));
}
/** @internal */
exports.MetersQuantitiesQueryParamExternalCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function metersQuantitiesQueryParamExternalCustomerIDFilterToJSON(metersQuantitiesQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(exports.MetersQuantitiesQueryParamExternalCustomerIDFilter$outboundSchema.parse(metersQuantitiesQueryParamExternalCustomerIDFilter));
}
/** @internal */
exports.MetersQuantitiesRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    startTimestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    endTimestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    interval: timeinterval_js_1.TimeInterval$outboundSchema,
    timezone: z._default(z.string(), "UTC"),
    customerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    customerAggregationFunction: z.optional(z.nullable(aggregationfunction_js_1.AggregationFunction$outboundSchema)),
    metadata: z.optional(z.nullable(z.record(z.string(), subscriptionslist_js_1.MetadataQuery$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startTimestamp: "start_timestamp",
        endTimestamp: "end_timestamp",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        customerAggregationFunction: "customer_aggregation_function",
    });
}));
function metersQuantitiesRequestToJSON(metersQuantitiesRequest) {
    return JSON.stringify(exports.MetersQuantitiesRequest$outboundSchema.parse(metersQuantitiesRequest));
}
