// @polar-sh/sdk/dist/commonjs/models/operations/meterslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetersListResponse$inboundSchema = exports.MetersListRequest$outboundSchema = exports.MetersListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.metersListQueryParamOrganizationIDFilterToJSON = metersListQueryParamOrganizationIDFilterToJSON;
exports.metersListRequestToJSON = metersListRequestToJSON;
exports.metersListResponseFromJSON = metersListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const listresourcemeter_js_1 = require("../components/listresourcemeter.js");
const metersortproperty_js_1 = require("../components/metersortproperty.js");
const subscriptionslist_js_1 = require("../components/subscriptionslist.js");
/** @internal */
exports.MetersListQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function metersListQueryParamOrganizationIDFilterToJSON(metersListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.MetersListQueryParamOrganizationIDFilter$outboundSchema.parse(metersListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.MetersListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    query: z.optional(z.nullable(z.string())),
    isArchived: z.optional(z.nullable(z.boolean())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(metersortproperty_js_1.MeterSortProperty$outboundSchema))),
    metadata: z.optional(z.nullable(z.record(z.string(), subscriptionslist_js_1.MetadataQuery$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        isArchived: "is_archived",
    });
}));
function metersListRequestToJSON(metersListRequest) {
    return JSON.stringify(exports.MetersListRequest$outboundSchema.parse(metersListRequest));
}
/** @internal */
exports.MetersListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcemeter_js_1.ListResourceMeter$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function metersListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetersListResponse' from JSON`);
}
