// @polar-sh/sdk/dist/commonjs/models/operations/organizationaccesstokenslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationAccessTokensListResponse$inboundSchema = exports.OrganizationAccessTokensListRequest$outboundSchema = exports.OrganizationAccessTokensListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.organizationAccessTokensListQueryParamOrganizationIDFilterToJSON = organizationAccessTokensListQueryParamOrganizationIDFilterToJSON;
exports.organizationAccessTokensListRequestToJSON = organizationAccessTokensListRequestToJSON;
exports.organizationAccessTokensListResponseFromJSON = organizationAccessTokensListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const listresourceorganizationaccesstoken_js_1 = require("../components/listresourceorganizationaccesstoken.js");
const organizationaccesstokensortproperty_js_1 = require("../components/organizationaccesstokensortproperty.js");
/** @internal */
exports.OrganizationAccessTokensListQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function organizationAccessTokensListQueryParamOrganizationIDFilterToJSON(organizationAccessTokensListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.OrganizationAccessTokensListQueryParamOrganizationIDFilter$outboundSchema
        .parse(organizationAccessTokensListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.OrganizationAccessTokensListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(organizationaccesstokensortproperty_js_1.OrganizationAccessTokenSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
}));
function organizationAccessTokensListRequestToJSON(organizationAccessTokensListRequest) {
    return JSON.stringify(exports.OrganizationAccessTokensListRequest$outboundSchema.parse(organizationAccessTokensListRequest));
}
/** @internal */
exports.OrganizationAccessTokensListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourceorganizationaccesstoken_js_1.ListResourceOrganizationAccessToken$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function organizationAccessTokensListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationAccessTokensListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationAccessTokensListResponse' from JSON`);
}
