// @polar-sh/sdk/dist/commonjs/models/operations/paymentslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentsListResponse$inboundSchema = exports.PaymentsListRequest$outboundSchema = exports.CustomerEmailFilter$outboundSchema = exports.MethodFilter$outboundSchema = exports.PaymentsListQueryParamStatusFilter$outboundSchema = exports.PaymentsListQueryParamOrderIDFilter$outboundSchema = exports.PaymentsListQueryParamCheckoutIDFilter$outboundSchema = exports.PaymentsListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.paymentsListQueryParamOrganizationIDFilterToJSON = paymentsListQueryParamOrganizationIDFilterToJSON;
exports.paymentsListQueryParamCheckoutIDFilterToJSON = paymentsListQueryParamCheckoutIDFilterToJSON;
exports.paymentsListQueryParamOrderIDFilterToJSON = paymentsListQueryParamOrderIDFilterToJSON;
exports.paymentsListQueryParamStatusFilterToJSON = paymentsListQueryParamStatusFilterToJSON;
exports.methodFilterToJSON = methodFilterToJSON;
exports.customerEmailFilterToJSON = customerEmailFilterToJSON;
exports.paymentsListRequestToJSON = paymentsListRequestToJSON;
exports.paymentsListResponseFromJSON = paymentsListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const listresourcepayment_js_1 = require("../components/listresourcepayment.js");
const paymentsortproperty_js_1 = require("../components/paymentsortproperty.js");
const paymentstatus_js_1 = require("../components/paymentstatus.js");
/** @internal */
exports.PaymentsListQueryParamOrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function paymentsListQueryParamOrganizationIDFilterToJSON(paymentsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.PaymentsListQueryParamOrganizationIDFilter$outboundSchema.parse(paymentsListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.PaymentsListQueryParamCheckoutIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function paymentsListQueryParamCheckoutIDFilterToJSON(paymentsListQueryParamCheckoutIDFilter) {
    return JSON.stringify(exports.PaymentsListQueryParamCheckoutIDFilter$outboundSchema.parse(paymentsListQueryParamCheckoutIDFilter));
}
/** @internal */
exports.PaymentsListQueryParamOrderIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function paymentsListQueryParamOrderIDFilterToJSON(paymentsListQueryParamOrderIDFilter) {
    return JSON.stringify(exports.PaymentsListQueryParamOrderIDFilter$outboundSchema.parse(paymentsListQueryParamOrderIDFilter));
}
/** @internal */
exports.PaymentsListQueryParamStatusFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    paymentstatus_js_1.PaymentStatus$outboundSchema,
    z.array(paymentstatus_js_1.PaymentStatus$outboundSchema),
]);
function paymentsListQueryParamStatusFilterToJSON(paymentsListQueryParamStatusFilter) {
    return JSON.stringify(exports.PaymentsListQueryParamStatusFilter$outboundSchema.parse(paymentsListQueryParamStatusFilter));
}
/** @internal */
exports.MethodFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function methodFilterToJSON(methodFilter) {
    return JSON.stringify(exports.MethodFilter$outboundSchema.parse(methodFilter));
}
/** @internal */
exports.CustomerEmailFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function customerEmailFilterToJSON(customerEmailFilter) {
    return JSON.stringify(exports.CustomerEmailFilter$outboundSchema.parse(customerEmailFilter));
}
/** @internal */
exports.PaymentsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    checkoutId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    status: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([
        paymentstatus_js_1.PaymentStatus$outboundSchema,
        z.array(paymentstatus_js_1.PaymentStatus$outboundSchema),
    ]))),
    method: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    customerEmail: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(paymentsortproperty_js_1.PaymentSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        customerEmail: "customer_email",
    });
}));
function paymentsListRequestToJSON(paymentsListRequest) {
    return JSON.stringify(exports.PaymentsListRequest$outboundSchema.parse(paymentsListRequest));
}
/** @internal */
exports.PaymentsListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcepayment_js_1.ListResourcePayment$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function paymentsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PaymentsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentsListResponse' from JSON`);
}
