// @polar-sh/sdk/dist/commonjs/models/operations/subscriptionslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionsListResponse$inboundSchema = exports.SubscriptionsListRequest$outboundSchema = exports.CustomerCancellationReasonFilter$outboundSchema = exports.DiscountIDFilter$outboundSchema = exports.ExternalCustomerIDFilter$outboundSchema = exports.CustomerIDFilter$outboundSchema = exports.ProductIDFilter$outboundSchema = exports.OrganizationIDFilter$outboundSchema = void 0;
exports.organizationIDFilterToJSON = organizationIDFilterToJSON;
exports.productIDFilterToJSON = productIDFilterToJSON;
exports.customerIDFilterToJSON = customerIDFilterToJSON;
exports.externalCustomerIDFilterToJSON = externalCustomerIDFilterToJSON;
exports.discountIDFilterToJSON = discountIDFilterToJSON;
exports.customerCancellationReasonFilterToJSON = customerCancellationReasonFilterToJSON;
exports.subscriptionsListRequestToJSON = subscriptionsListRequestToJSON;
exports.subscriptionsListResponseFromJSON = subscriptionsListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const customercancellationreason_js_1 = require("../components/customercancellationreason.js");
const listresourcesubscription_js_1 = require("../components/listresourcesubscription.js");
const subscriptionslist_js_1 = require("../components/subscriptionslist.js");
const subscriptionsortproperty_js_1 = require("../components/subscriptionsortproperty.js");
/** @internal */
exports.OrganizationIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function organizationIDFilterToJSON(organizationIDFilter) {
    return JSON.stringify(exports.OrganizationIDFilter$outboundSchema.parse(organizationIDFilter));
}
/** @internal */
exports.ProductIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function productIDFilterToJSON(productIDFilter) {
    return JSON.stringify(exports.ProductIDFilter$outboundSchema.parse(productIDFilter));
}
/** @internal */
exports.CustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function customerIDFilterToJSON(customerIDFilter) {
    return JSON.stringify(exports.CustomerIDFilter$outboundSchema.parse(customerIDFilter));
}
/** @internal */
exports.ExternalCustomerIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function externalCustomerIDFilterToJSON(externalCustomerIDFilter) {
    return JSON.stringify(exports.ExternalCustomerIDFilter$outboundSchema.parse(externalCustomerIDFilter));
}
/** @internal */
exports.DiscountIDFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]);
function discountIDFilterToJSON(discountIDFilter) {
    return JSON.stringify(exports.DiscountIDFilter$outboundSchema.parse(discountIDFilter));
}
/** @internal */
exports.CustomerCancellationReasonFilter$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    customercancellationreason_js_1.CustomerCancellationReason$outboundSchema,
    z.array(customercancellationreason_js_1.CustomerCancellationReason$outboundSchema),
]);
function customerCancellationReasonFilterToJSON(customerCancellationReasonFilter) {
    return JSON.stringify(exports.CustomerCancellationReasonFilter$outboundSchema.parse(customerCancellationReasonFilter));
}
/** @internal */
exports.SubscriptionsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    productId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    discountId: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), z.array(z.string())]))),
    active: z.optional(z.nullable(z.boolean())),
    cancelAtPeriodEnd: z.optional(z.nullable(z.boolean())),
    customerCancellationReason: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([
        customercancellationreason_js_1.CustomerCancellationReason$outboundSchema,
        z.array(customercancellationreason_js_1.CustomerCancellationReason$outboundSchema),
    ]))),
    canceledAtAfter: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    canceledAtBefore: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(subscriptionsortproperty_js_1.SubscriptionSortProperty$outboundSchema))),
    metadata: z.optional(z.nullable(z.record(z.string(), z.lazy(() => subscriptionslist_js_1.MetadataQuery$outboundSchema)))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        discountId: "discount_id",
        cancelAtPeriodEnd: "cancel_at_period_end",
        customerCancellationReason: "customer_cancellation_reason",
        canceledAtAfter: "canceled_at_after",
        canceledAtBefore: "canceled_at_before",
    });
}));
function subscriptionsListRequestToJSON(subscriptionsListRequest) {
    return JSON.stringify(exports.SubscriptionsListRequest$outboundSchema.parse(subscriptionsListRequest));
}
/** @internal */
exports.SubscriptionsListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcesubscription_js_1.ListResourceSubscription$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function subscriptionsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionsListResponse' from JSON`);
}
