// @polar-sh/sdk/dist/esm/models/operations/benefitsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { BenefitCustomUpdate$outboundSchema, } from "../components/benefitcustomupdate.js";
import { BenefitDiscordUpdate$outboundSchema, } from "../components/benefitdiscordupdate.js";
import { BenefitDownloadablesUpdate$outboundSchema, } from "../components/benefitdownloadablesupdate.js";
import { BenefitFeatureFlagUpdate$outboundSchema, } from "../components/benefitfeatureflagupdate.js";
import { BenefitGitHubRepositoryUpdate$outboundSchema, } from "../components/benefitgithubrepositoryupdate.js";
import { BenefitLicenseKeysUpdate$outboundSchema, } from "../components/benefitlicensekeysupdate.js";
import { BenefitMeterCreditUpdate$outboundSchema, } from "../components/benefitmetercreditupdate.js";
/** @internal */
export const BenefitsUpdateBenefitUpdate$outboundSchema = z.union([
    BenefitCustomUpdate$outboundSchema,
    BenefitDiscordUpdate$outboundSchema,
    BenefitGitHubRepositoryUpdate$outboundSchema,
    BenefitDownloadablesUpdate$outboundSchema,
    BenefitLicenseKeysUpdate$outboundSchema,
    BenefitMeterCreditUpdate$outboundSchema,
    BenefitFeatureFlagUpdate$outboundSchema,
]);
export function benefitsUpdateBenefitUpdateToJSON(benefitsUpdateBenefitUpdate) {
    return JSON.stringify(BenefitsUpdateBenefitUpdate$outboundSchema.parse(benefitsUpdateBenefitUpdate));
}
/** @internal */
export const BenefitsUpdateRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    requestBody: z.union([
        BenefitCustomUpdate$outboundSchema,
        BenefitDiscordUpdate$outboundSchema,
        BenefitGitHubRepositoryUpdate$outboundSchema,
        BenefitDownloadablesUpdate$outboundSchema,
        BenefitLicenseKeysUpdate$outboundSchema,
        BenefitMeterCreditUpdate$outboundSchema,
        BenefitFeatureFlagUpdate$outboundSchema,
    ]),
}), z.transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
}));
export function benefitsUpdateRequestToJSON(benefitsUpdateRequest) {
    return JSON.stringify(BenefitsUpdateRequest$outboundSchema.parse(benefitsUpdateRequest));
}
