// @polar-sh/sdk/dist/esm/models/operations/benefitgrantslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitGrantSortProperty$outboundSchema, } from "../components/benefitgrantsortproperty.js";
import { ListResourceBenefitGrant$inboundSchema, } from "../components/listresourcebenefitgrant.js";
/** @internal */
export const BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function benefitGrantsListQueryParamOrganizationIDFilterToJSON(benefitGrantsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(BenefitGrantsListQueryParamOrganizationIDFilter$outboundSchema.parse(benefitGrantsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function benefitGrantsListQueryParamCustomerIDFilterToJSON(benefitGrantsListQueryParamCustomerIDFilter) {
    return JSON.stringify(BenefitGrantsListQueryParamCustomerIDFilter$outboundSchema.parse(benefitGrantsListQueryParamCustomerIDFilter));
}
/** @internal */
export const QueryParamExternalCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function queryParamExternalCustomerIDFilterToJSON(queryParamExternalCustomerIDFilter) {
    return JSON.stringify(QueryParamExternalCustomerIDFilter$outboundSchema.parse(queryParamExternalCustomerIDFilter));
}
/** @internal */
export const BenefitGrantsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    isGranted: z.optional(z.nullable(z.boolean())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(BenefitGrantSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        isGranted: "is_granted",
    });
}));
export function benefitGrantsListRequestToJSON(benefitGrantsListRequest) {
    return JSON.stringify(BenefitGrantsListRequest$outboundSchema.parse(benefitGrantsListRequest));
}
/** @internal */
export const BenefitGrantsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceBenefitGrant$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function benefitGrantsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantsListResponse' from JSON`);
}
