// @polar-sh/sdk/dist/esm/models/operations/checkoutslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CheckoutSortProperty$outboundSchema, } from "../components/checkoutsortproperty.js";
import { CheckoutStatus$outboundSchema, } from "../components/checkoutstatus.js";
import { ListResourceCheckout$inboundSchema, } from "../components/listresourcecheckout.js";
/** @internal */
export const CheckoutsListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function checkoutsListQueryParamOrganizationIDFilterToJSON(checkoutsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(CheckoutsListQueryParamOrganizationIDFilter$outboundSchema.parse(checkoutsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const CheckoutsListQueryParamProductIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function checkoutsListQueryParamProductIDFilterToJSON(checkoutsListQueryParamProductIDFilter) {
    return JSON.stringify(CheckoutsListQueryParamProductIDFilter$outboundSchema.parse(checkoutsListQueryParamProductIDFilter));
}
/** @internal */
export const CheckoutsListQueryParamCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function checkoutsListQueryParamCustomerIDFilterToJSON(checkoutsListQueryParamCustomerIDFilter) {
    return JSON.stringify(CheckoutsListQueryParamCustomerIDFilter$outboundSchema.parse(checkoutsListQueryParamCustomerIDFilter));
}
/** @internal */
export const CheckoutsListQueryParamExternalCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function checkoutsListQueryParamExternalCustomerIDFilterToJSON(checkoutsListQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(CheckoutsListQueryParamExternalCustomerIDFilter$outboundSchema.parse(checkoutsListQueryParamExternalCustomerIDFilter));
}
/** @internal */
export const QueryParamStatusFilter$outboundSchema = smartUnion([
    CheckoutStatus$outboundSchema,
    z.array(CheckoutStatus$outboundSchema),
]);
export function queryParamStatusFilterToJSON(queryParamStatusFilter) {
    return JSON.stringify(QueryParamStatusFilter$outboundSchema.parse(queryParamStatusFilter));
}
/** @internal */
export const CheckoutsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    productId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    status: z.optional(z.nullable(smartUnion([
        CheckoutStatus$outboundSchema,
        z.array(CheckoutStatus$outboundSchema),
    ]))),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CheckoutSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        productId: "product_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
}));
export function checkoutsListRequestToJSON(checkoutsListRequest) {
    return JSON.stringify(CheckoutsListRequest$outboundSchema.parse(checkoutsListRequest));
}
/** @internal */
export const CheckoutsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCheckout$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function checkoutsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsListResponse' from JSON`);
}
