// @polar-sh/sdk/dist/esm/models/operations/customerportalordersinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalOrdersInvoiceSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersInvoiceSecurityToJSON(customerPortalOrdersInvoiceSecurity) {
    return JSON.stringify(CustomerPortalOrdersInvoiceSecurity$outboundSchema.parse(customerPortalOrdersInvoiceSecurity));
}
/** @internal */
export const CustomerPortalOrdersInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalOrdersInvoiceRequestToJSON(customerPortalOrdersInvoiceRequest) {
    return JSON.stringify(CustomerPortalOrdersInvoiceRequest$outboundSchema.parse(customerPortalOrdersInvoiceRequest));
}
