// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomermetersget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalCustomerMetersGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalCustomerMetersGetSecurityToJSON(customerPortalCustomerMetersGetSecurity) {
    return JSON.stringify(CustomerPortalCustomerMetersGetSecurity$outboundSchema.parse(customerPortalCustomerMetersGetSecurity));
}
/** @internal */
export const CustomerPortalCustomerMetersGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalCustomerMetersGetRequestToJSON(customerPortalCustomerMetersGetRequest) {
    return JSON.stringify(CustomerPortalCustomerMetersGetRequest$outboundSchema.parse(customerPortalCustomerMetersGetRequest));
}
