// @polar-sh/sdk/dist/esm/models/operations/customerportalordersgetpaymentstatus.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersGetPaymentStatusSecurityToJSON(customerPortalOrdersGetPaymentStatusSecurity) {
    return JSON.stringify(CustomerPortalOrdersGetPaymentStatusSecurity$outboundSchema.parse(customerPortalOrdersGetPaymentStatusSecurity));
}
/** @internal */
export const CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalOrdersGetPaymentStatusRequestToJSON(customerPortalOrdersGetPaymentStatusRequest) {
    return JSON.stringify(CustomerPortalOrdersGetPaymentStatusRequest$outboundSchema.parse(customerPortalOrdersGetPaymentStatusRequest));
}
