// @polar-sh/sdk/dist/esm/models/operations/customerportalwalletslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerWalletSortProperty$outboundSchema, } from "../components/customerwalletsortproperty.js";
import { ListResourceCustomerWallet$inboundSchema, } from "../components/listresourcecustomerwallet.js";
/** @internal */
export const CustomerPortalWalletsListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalWalletsListSecurityToJSON(customerPortalWalletsListSecurity) {
    return JSON.stringify(CustomerPortalWalletsListSecurity$outboundSchema.parse(customerPortalWalletsListSecurity));
}
/** @internal */
export const CustomerPortalWalletsListRequest$outboundSchema = z.object({
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerWalletSortProperty$outboundSchema))),
});
export function customerPortalWalletsListRequestToJSON(customerPortalWalletsListRequest) {
    return JSON.stringify(CustomerPortalWalletsListRequest$outboundSchema.parse(customerPortalWalletsListRequest));
}
/** @internal */
export const CustomerPortalWalletsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerWallet$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalWalletsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalWalletsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalWalletsListResponse' from JSON`);
}
