// @polar-sh/sdk/dist/esm/models/operations/customerportalwalletsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalWalletsGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalWalletsGetSecurityToJSON(customerPortalWalletsGetSecurity) {
    return JSON.stringify(CustomerPortalWalletsGetSecurity$outboundSchema.parse(customerPortalWalletsGetSecurity));
}
/** @internal */
export const CustomerPortalWalletsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalWalletsGetRequestToJSON(customerPortalWalletsGetRequest) {
    return JSON.stringify(CustomerPortalWalletsGetRequest$outboundSchema.parse(customerPortalWalletsGetRequest));
}
