// @polar-sh/sdk/dist/esm/models/operations/customerportalordersgenerateinvoice.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersGenerateInvoiceSecurityToJSON(customerPortalOrdersGenerateInvoiceSecurity) {
    return JSON.stringify(CustomerPortalOrdersGenerateInvoiceSecurity$outboundSchema.parse(customerPortalOrdersGenerateInvoiceSecurity));
}
/** @internal */
export const CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalOrdersGenerateInvoiceRequestToJSON(customerPortalOrdersGenerateInvoiceRequest) {
    return JSON.stringify(CustomerPortalOrdersGenerateInvoiceRequest$outboundSchema.parse(customerPortalOrdersGenerateInvoiceRequest));
}
