// @polar-sh/sdk/dist/esm/models/operations/customerportalsubscriptionscancel.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalSubscriptionsCancelSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSubscriptionsCancelSecurityToJSON(customerPortalSubscriptionsCancelSecurity) {
    return JSON.stringify(CustomerPortalSubscriptionsCancelSecurity$outboundSchema.parse(customerPortalSubscriptionsCancelSecurity));
}
/** @internal */
export const CustomerPortalSubscriptionsCancelRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalSubscriptionsCancelRequestToJSON(customerPortalSubscriptionsCancelRequest) {
    return JSON.stringify(CustomerPortalSubscriptionsCancelRequest$outboundSchema.parse(customerPortalSubscriptionsCancelRequest));
}
