// @polar-sh/sdk/dist/esm/models/operations/customerportalordersconfirmretrypayment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomerOrderConfirmPayment$outboundSchema, } from "../components/customerorderconfirmpayment.js";
/** @internal */
export const CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersConfirmRetryPaymentSecurityToJSON(customerPortalOrdersConfirmRetryPaymentSecurity) {
    return JSON.stringify(CustomerPortalOrdersConfirmRetryPaymentSecurity$outboundSchema.parse(customerPortalOrdersConfirmRetryPaymentSecurity));
}
/** @internal */
export const CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    customerOrderConfirmPayment: CustomerOrderConfirmPayment$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        customerOrderConfirmPayment: "CustomerOrderConfirmPayment",
    });
}));
export function customerPortalOrdersConfirmRetryPaymentRequestToJSON(customerPortalOrdersConfirmRetryPaymentRequest) {
    return JSON.stringify(CustomerPortalOrdersConfirmRetryPaymentRequest$outboundSchema.parse(customerPortalOrdersConfirmRetryPaymentRequest));
}
