// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomersdeletepaymentmethod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalCustomersDeletePaymentMethodSecurityToJSON(customerPortalCustomersDeletePaymentMethodSecurity) {
    return JSON.stringify(CustomerPortalCustomersDeletePaymentMethodSecurity$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodSecurity));
}
/** @internal */
export const CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalCustomersDeletePaymentMethodRequestToJSON(customerPortalCustomersDeletePaymentMethodRequest) {
    return JSON.stringify(CustomerPortalCustomersDeletePaymentMethodRequest$outboundSchema.parse(customerPortalCustomersDeletePaymentMethodRequest));
}
