// @polar-sh/sdk/dist/esm/models/operations/customerslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerSortProperty$outboundSchema, } from "../components/customersortproperty.js";
import { ListResourceCustomer$inboundSchema, } from "../components/listresourcecustomer.js";
import { MetadataQuery$outboundSchema, } from "../components/subscriptionslist.js";
/** @internal */
export const CustomersListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customersListQueryParamOrganizationIDFilterToJSON(customersListQueryParamOrganizationIDFilter) {
    return JSON.stringify(CustomersListQueryParamOrganizationIDFilter$outboundSchema.parse(customersListQueryParamOrganizationIDFilter));
}
/** @internal */
export const CustomersListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    email: z.optional(z.nullable(z.string())),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerSortProperty$outboundSchema))),
    metadata: z.optional(z.nullable(z.record(z.string(), MetadataQuery$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customersListRequestToJSON(customersListRequest) {
    return JSON.stringify(CustomersListRequest$outboundSchema.parse(customersListRequest));
}
/** @internal */
export const CustomersListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomer$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customersListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersListResponse' from JSON`);
}
