// @polar-sh/sdk/dist/esm/models/operations/customerportalordersupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomerOrderUpdate$outboundSchema, } from "../components/customerorderupdate.js";
/** @internal */
export const CustomerPortalOrdersUpdateSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersUpdateSecurityToJSON(customerPortalOrdersUpdateSecurity) {
    return JSON.stringify(CustomerPortalOrdersUpdateSecurity$outboundSchema.parse(customerPortalOrdersUpdateSecurity));
}
/** @internal */
export const CustomerPortalOrdersUpdateRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    customerOrderUpdate: CustomerOrderUpdate$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        customerOrderUpdate: "CustomerOrderUpdate",
    });
}));
export function customerPortalOrdersUpdateRequestToJSON(customerPortalOrdersUpdateRequest) {
    return JSON.stringify(CustomerPortalOrdersUpdateRequest$outboundSchema.parse(customerPortalOrdersUpdateRequest));
}
