// @polar-sh/sdk/dist/esm/models/operations/customersexport.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const CustomersExportQueryParamOrganizationId$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customersExportQueryParamOrganizationIdToJSON(customersExportQueryParamOrganizationId) {
    return JSON.stringify(CustomersExportQueryParamOrganizationId$outboundSchema.parse(customersExportQueryParamOrganizationId));
}
/** @internal */
export const CustomersExportRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customersExportRequestToJSON(customersExportRequest) {
    return JSON.stringify(CustomersExportRequest$outboundSchema.parse(customersExportRequest));
}
/** @internal */
export const CustomersExportResponse$inboundSchema = smartUnion([z.any(), z.string()]);
export function customersExportResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomersExportResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomersExportResponse' from JSON`);
}
