// @polar-sh/sdk/dist/esm/models/operations/customerportalseatsresendinvitation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalSeatsResendInvitationSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSeatsResendInvitationSecurityToJSON(customerPortalSeatsResendInvitationSecurity) {
    return JSON.stringify(CustomerPortalSeatsResendInvitationSecurity$outboundSchema.parse(customerPortalSeatsResendInvitationSecurity));
}
/** @internal */
export const CustomerPortalSeatsResendInvitationRequest$outboundSchema = z.pipe(z.object({
    seatId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
}));
export function customerPortalSeatsResendInvitationRequestToJSON(customerPortalSeatsResendInvitationRequest) {
    return JSON.stringify(CustomerPortalSeatsResendInvitationRequest$outboundSchema.parse(customerPortalSeatsResendInvitationRequest));
}
