// @polar-sh/sdk/dist/esm/models/operations/customerportalsubscriptionsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalSubscriptionsGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSubscriptionsGetSecurityToJSON(customerPortalSubscriptionsGetSecurity) {
    return JSON.stringify(CustomerPortalSubscriptionsGetSecurity$outboundSchema.parse(customerPortalSubscriptionsGetSecurity));
}
/** @internal */
export const CustomerPortalSubscriptionsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalSubscriptionsGetRequestToJSON(customerPortalSubscriptionsGetRequest) {
    return JSON.stringify(CustomerPortalSubscriptionsGetRequest$outboundSchema.parse(customerPortalSubscriptionsGetRequest));
}
