// @polar-sh/sdk/dist/esm/models/operations/customerportalseatsrevokeseat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalSeatsRevokeSeatSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSeatsRevokeSeatSecurityToJSON(customerPortalSeatsRevokeSeatSecurity) {
    return JSON.stringify(CustomerPortalSeatsRevokeSeatSecurity$outboundSchema.parse(customerPortalSeatsRevokeSeatSecurity));
}
/** @internal */
export const CustomerPortalSeatsRevokeSeatRequest$outboundSchema = z.pipe(z.object({
    seatId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
}));
export function customerPortalSeatsRevokeSeatRequestToJSON(customerPortalSeatsRevokeSeatRequest) {
    return JSON.stringify(CustomerPortalSeatsRevokeSeatRequest$outboundSchema.parse(customerPortalSeatsRevokeSeatRequest));
}
