// @polar-sh/sdk/dist/esm/models/operations/customerportallicensekeyslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceLicenseKeyRead$inboundSchema, } from "../components/listresourcelicensekeyread.js";
/** @internal */
export const CustomerPortalLicenseKeysListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalLicenseKeysListSecurityToJSON(customerPortalLicenseKeysListSecurity) {
    return JSON.stringify(CustomerPortalLicenseKeysListSecurity$outboundSchema.parse(customerPortalLicenseKeysListSecurity));
}
/** @internal */
export const CustomerPortalLicenseKeysListRequest$outboundSchema = z.pipe(z.object({
    benefitId: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
}), z.transform((v) => {
    return remap$(v, {
        benefitId: "benefit_id",
    });
}));
export function customerPortalLicenseKeysListRequestToJSON(customerPortalLicenseKeysListRequest) {
    return JSON.stringify(CustomerPortalLicenseKeysListRequest$outboundSchema.parse(customerPortalLicenseKeysListRequest));
}
/** @internal */
export const CustomerPortalLicenseKeysListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceLicenseKeyRead$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalLicenseKeysListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalLicenseKeysListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalLicenseKeysListResponse' from JSON`);
}
