// @polar-sh/sdk/dist/esm/models/operations/customerportalseatslistclaimedsubscriptions.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceCustomerSubscription$inboundSchema, } from "../components/listresourcecustomersubscription.js";
/** @internal */
export const CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSeatsListClaimedSubscriptionsSecurityToJSON(customerPortalSeatsListClaimedSubscriptionsSecurity) {
    return JSON.stringify(CustomerPortalSeatsListClaimedSubscriptionsSecurity$outboundSchema.parse(customerPortalSeatsListClaimedSubscriptionsSecurity));
}
/** @internal */
export const CustomerPortalSeatsListClaimedSubscriptionsRequest$outboundSchema = z.object({
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
});
export function customerPortalSeatsListClaimedSubscriptionsRequestToJSON(customerPortalSeatsListClaimedSubscriptionsRequest) {
    return JSON.stringify(CustomerPortalSeatsListClaimedSubscriptionsRequest$outboundSchema.parse(customerPortalSeatsListClaimedSubscriptionsRequest));
}
/** @internal */
export const CustomerPortalSeatsListClaimedSubscriptionsResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerSubscription$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalSeatsListClaimedSubscriptionsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSeatsListClaimedSubscriptionsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSeatsListClaimedSubscriptionsResponse' from JSON`);
}
