// @polar-sh/sdk/dist/esm/models/operations/customerportalbenefitgrantsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomerBenefitGrantUpdate$outboundSchema, } from "../components/customerbenefitgrantupdate.js";
/** @internal */
export const CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalBenefitGrantsUpdateSecurityToJSON(customerPortalBenefitGrantsUpdateSecurity) {
    return JSON.stringify(CustomerPortalBenefitGrantsUpdateSecurity$outboundSchema.parse(customerPortalBenefitGrantsUpdateSecurity));
}
/** @internal */
export const CustomerPortalBenefitGrantsUpdateRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    customerBenefitGrantUpdate: CustomerBenefitGrantUpdate$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        customerBenefitGrantUpdate: "CustomerBenefitGrantUpdate",
    });
}));
export function customerPortalBenefitGrantsUpdateRequestToJSON(customerPortalBenefitGrantsUpdateRequest) {
    return JSON.stringify(CustomerPortalBenefitGrantsUpdateRequest$outboundSchema.parse(customerPortalBenefitGrantsUpdateRequest));
}
