// @polar-sh/sdk/dist/esm/models/operations/customerportalseatslistseats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalSeatsListSeatsSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSeatsListSeatsSecurityToJSON(customerPortalSeatsListSeatsSecurity) {
    return JSON.stringify(CustomerPortalSeatsListSeatsSecurity$outboundSchema.parse(customerPortalSeatsListSeatsSecurity));
}
/** @internal */
export const CustomerPortalSeatsListSeatsRequest$outboundSchema = z.pipe(z.object({
    subscriptionId: z.optional(z.nullable(z.string())),
    orderId: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
        orderId: "order_id",
    });
}));
export function customerPortalSeatsListSeatsRequestToJSON(customerPortalSeatsListSeatsRequest) {
    return JSON.stringify(CustomerPortalSeatsListSeatsRequest$outboundSchema.parse(customerPortalSeatsListSeatsRequest));
}
