// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomerslistpaymentmethods.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceCustomerPaymentMethod$inboundSchema, } from "../components/listresourcecustomerpaymentmethod.js";
/** @internal */
export const CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalCustomersListPaymentMethodsSecurityToJSON(customerPortalCustomersListPaymentMethodsSecurity) {
    return JSON.stringify(CustomerPortalCustomersListPaymentMethodsSecurity$outboundSchema.parse(customerPortalCustomersListPaymentMethodsSecurity));
}
/** @internal */
export const CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema = z.object({
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
});
export function customerPortalCustomersListPaymentMethodsRequestToJSON(customerPortalCustomersListPaymentMethodsRequest) {
    return JSON.stringify(CustomerPortalCustomersListPaymentMethodsRequest$outboundSchema.parse(customerPortalCustomersListPaymentMethodsRequest));
}
/** @internal */
export const CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema = z
    .pipe(z.object({
    Result: ListResourceCustomerPaymentMethod$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalCustomersListPaymentMethodsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomersListPaymentMethodsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomersListPaymentMethodsResponse' from JSON`);
}
