// @polar-sh/sdk/dist/esm/models/operations/customerportalordersget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalOrdersGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersGetSecurityToJSON(customerPortalOrdersGetSecurity) {
    return JSON.stringify(CustomerPortalOrdersGetSecurity$outboundSchema.parse(customerPortalOrdersGetSecurity));
}
/** @internal */
export const CustomerPortalOrdersGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalOrdersGetRequestToJSON(customerPortalOrdersGetRequest) {
    return JSON.stringify(CustomerPortalOrdersGetRequest$outboundSchema.parse(customerPortalOrdersGetRequest));
}
