// @polar-sh/sdk/dist/esm/models/operations/customerportalsubscriptionsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomerSubscriptionUpdate$outboundSchema, } from "../components/customersubscriptionupdate.js";
/** @internal */
export const CustomerPortalSubscriptionsUpdateSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSubscriptionsUpdateSecurityToJSON(customerPortalSubscriptionsUpdateSecurity) {
    return JSON.stringify(CustomerPortalSubscriptionsUpdateSecurity$outboundSchema.parse(customerPortalSubscriptionsUpdateSecurity));
}
/** @internal */
export const CustomerPortalSubscriptionsUpdateRequest$outboundSchema = z.pipe(z.object({
    id: z.string(),
    customerSubscriptionUpdate: CustomerSubscriptionUpdate$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        customerSubscriptionUpdate: "CustomerSubscriptionUpdate",
    });
}));
export function customerPortalSubscriptionsUpdateRequestToJSON(customerPortalSubscriptionsUpdateRequest) {
    return JSON.stringify(CustomerPortalSubscriptionsUpdateRequest$outboundSchema.parse(customerPortalSubscriptionsUpdateRequest));
}
