// @polar-sh/sdk/dist/esm/models/operations/customerportalmemberslistmembers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceCustomerPortalMember$inboundSchema, } from "../components/listresourcecustomerportalmember.js";
/** @internal */
export const CustomerPortalMembersListMembersRequest$outboundSchema = z.object({
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
});
export function customerPortalMembersListMembersRequestToJSON(customerPortalMembersListMembersRequest) {
    return JSON.stringify(CustomerPortalMembersListMembersRequest$outboundSchema.parse(customerPortalMembersListMembersRequest));
}
/** @internal */
export const CustomerPortalMembersListMembersResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerPortalMember$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalMembersListMembersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalMembersListMembersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalMembersListMembersResponse' from JSON`);
}
