// @polar-sh/sdk/dist/esm/models/operations/customerportalbenefitgrantsget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalBenefitGrantsGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalBenefitGrantsGetSecurityToJSON(customerPortalBenefitGrantsGetSecurity) {
    return JSON.stringify(CustomerPortalBenefitGrantsGetSecurity$outboundSchema.parse(customerPortalBenefitGrantsGetSecurity));
}
/** @internal */
export const CustomerPortalBenefitGrantsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalBenefitGrantsGetRequestToJSON(customerPortalBenefitGrantsGetRequest) {
    return JSON.stringify(CustomerPortalBenefitGrantsGetRequest$outboundSchema.parse(customerPortalBenefitGrantsGetRequest));
}
