// @polar-sh/sdk/dist/esm/models/operations/customerportallicensekeysget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const CustomerPortalLicenseKeysGetSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalLicenseKeysGetSecurityToJSON(customerPortalLicenseKeysGetSecurity) {
    return JSON.stringify(CustomerPortalLicenseKeysGetSecurity$outboundSchema.parse(customerPortalLicenseKeysGetSecurity));
}
/** @internal */
export const CustomerPortalLicenseKeysGetRequest$outboundSchema = z.object({
    id: z.string(),
});
export function customerPortalLicenseKeysGetRequestToJSON(customerPortalLicenseKeysGetRequest) {
    return JSON.stringify(CustomerPortalLicenseKeysGetRequest$outboundSchema.parse(customerPortalLicenseKeysGetRequest));
}
