// @polar-sh/sdk/dist/esm/models/operations/customermeterslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerMeterSortProperty$outboundSchema, } from "../components/customermetersortproperty.js";
import { ListResourceCustomerMeter$inboundSchema, } from "../components/listresourcecustomermeter.js";
/** @internal */
export const CustomerMetersListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerMetersListQueryParamOrganizationIDFilterToJSON(customerMetersListQueryParamOrganizationIDFilter) {
    return JSON.stringify(CustomerMetersListQueryParamOrganizationIDFilter$outboundSchema.parse(customerMetersListQueryParamOrganizationIDFilter));
}
/** @internal */
export const CustomerMetersListQueryParamCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerMetersListQueryParamCustomerIDFilterToJSON(customerMetersListQueryParamCustomerIDFilter) {
    return JSON.stringify(CustomerMetersListQueryParamCustomerIDFilter$outboundSchema.parse(customerMetersListQueryParamCustomerIDFilter));
}
/** @internal */
export const CustomerMetersListQueryParamExternalCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerMetersListQueryParamExternalCustomerIDFilterToJSON(customerMetersListQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(CustomerMetersListQueryParamExternalCustomerIDFilter$outboundSchema.parse(customerMetersListQueryParamExternalCustomerIDFilter));
}
/** @internal */
export const QueryParamMeterIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function queryParamMeterIDFilterToJSON(queryParamMeterIDFilter) {
    return JSON.stringify(QueryParamMeterIDFilter$outboundSchema.parse(queryParamMeterIDFilter));
}
/** @internal */
export const CustomerMetersListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    meterId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerMeterSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        meterId: "meter_id",
    });
}));
export function customerMetersListRequestToJSON(customerMetersListRequest) {
    return JSON.stringify(CustomerMetersListRequest$outboundSchema.parse(customerMetersListRequest));
}
/** @internal */
export const CustomerMetersListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerMeter$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerMetersListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerMetersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerMetersListResponse' from JSON`);
}
