// @polar-sh/sdk/dist/esm/models/operations/customerportalsubscriptionslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerSubscriptionSortProperty$outboundSchema, } from "../components/customersubscriptionsortproperty.js";
import { ListResourceCustomerSubscription$inboundSchema, } from "../components/listresourcecustomersubscription.js";
/** @internal */
export const CustomerPortalSubscriptionsListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalSubscriptionsListSecurityToJSON(customerPortalSubscriptionsListSecurity) {
    return JSON.stringify(CustomerPortalSubscriptionsListSecurity$outboundSchema.parse(customerPortalSubscriptionsListSecurity));
}
/** @internal */
export const CustomerPortalSubscriptionsListQueryParamProductIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerPortalSubscriptionsListQueryParamProductIDFilterToJSON(customerPortalSubscriptionsListQueryParamProductIDFilter) {
    return JSON.stringify(CustomerPortalSubscriptionsListQueryParamProductIDFilter$outboundSchema
        .parse(customerPortalSubscriptionsListQueryParamProductIDFilter));
}
/** @internal */
export const CustomerPortalSubscriptionsListRequest$outboundSchema = z.pipe(z.object({
    productId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    active: z.optional(z.nullable(z.boolean())),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerSubscriptionSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        productId: "product_id",
    });
}));
export function customerPortalSubscriptionsListRequestToJSON(customerPortalSubscriptionsListRequest) {
    return JSON.stringify(CustomerPortalSubscriptionsListRequest$outboundSchema.parse(customerPortalSubscriptionsListRequest));
}
/** @internal */
export const CustomerPortalSubscriptionsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerSubscription$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalSubscriptionsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSubscriptionsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionsListResponse' from JSON`);
}
