// @polar-sh/sdk/dist/esm/models/operations/customerportalorderslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerOrderSortProperty$outboundSchema, } from "../components/customerordersortproperty.js";
import { ListResourceCustomerOrder$inboundSchema, } from "../components/listresourcecustomerorder.js";
import { ProductBillingType$outboundSchema, } from "../components/productbillingtype.js";
/** @internal */
export const CustomerPortalOrdersListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalOrdersListSecurityToJSON(customerPortalOrdersListSecurity) {
    return JSON.stringify(CustomerPortalOrdersListSecurity$outboundSchema.parse(customerPortalOrdersListSecurity));
}
/** @internal */
export const CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerPortalOrdersListQueryParamProductIDFilterToJSON(customerPortalOrdersListQueryParamProductIDFilter) {
    return JSON.stringify(CustomerPortalOrdersListQueryParamProductIDFilter$outboundSchema.parse(customerPortalOrdersListQueryParamProductIDFilter));
}
/** @internal */
export const CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema = smartUnion([
    ProductBillingType$outboundSchema,
    z.array(ProductBillingType$outboundSchema),
]);
export function customerPortalOrdersListQueryParamProductBillingTypeFilterToJSON(customerPortalOrdersListQueryParamProductBillingTypeFilter) {
    return JSON.stringify(CustomerPortalOrdersListQueryParamProductBillingTypeFilter$outboundSchema
        .parse(customerPortalOrdersListQueryParamProductBillingTypeFilter));
}
/** @internal */
export const CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerPortalOrdersListQueryParamSubscriptionIDFilterToJSON(customerPortalOrdersListQueryParamSubscriptionIDFilter) {
    return JSON.stringify(CustomerPortalOrdersListQueryParamSubscriptionIDFilter$outboundSchema.parse(customerPortalOrdersListQueryParamSubscriptionIDFilter));
}
/** @internal */
export const CustomerPortalOrdersListRequest$outboundSchema = z.pipe(z.object({
    productId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    productBillingType: z.optional(z.nullable(smartUnion([
        ProductBillingType$outboundSchema,
        z.array(ProductBillingType$outboundSchema),
    ]))),
    subscriptionId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerOrderSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        productId: "product_id",
        productBillingType: "product_billing_type",
        subscriptionId: "subscription_id",
    });
}));
export function customerPortalOrdersListRequestToJSON(customerPortalOrdersListRequest) {
    return JSON.stringify(CustomerPortalOrdersListRequest$outboundSchema.parse(customerPortalOrdersListRequest));
}
/** @internal */
export const CustomerPortalOrdersListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerOrder$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalOrdersListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersListResponse' from JSON`);
}
