// @polar-sh/sdk/dist/esm/models/operations/customerportalcustomermeterslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerCustomerMeterSortProperty$outboundSchema, } from "../components/customercustomermetersortproperty.js";
import { ListResourceCustomerCustomerMeter$inboundSchema, } from "../components/listresourcecustomercustomermeter.js";
/** @internal */
export const CustomerPortalCustomerMetersListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.optional(z.string()),
    memberSession: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
        memberSession: "member_session",
    });
}));
export function customerPortalCustomerMetersListSecurityToJSON(customerPortalCustomerMetersListSecurity) {
    return JSON.stringify(CustomerPortalCustomerMetersListSecurity$outboundSchema.parse(customerPortalCustomerMetersListSecurity));
}
/** @internal */
export const MeterIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function meterIDFilterToJSON(meterIDFilter) {
    return JSON.stringify(MeterIDFilter$outboundSchema.parse(meterIDFilter));
}
/** @internal */
export const CustomerPortalCustomerMetersListRequest$outboundSchema = z.pipe(z.object({
    meterId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    query: z.optional(z.nullable(z.string())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(CustomerCustomerMeterSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
}));
export function customerPortalCustomerMetersListRequestToJSON(customerPortalCustomerMetersListRequest) {
    return JSON.stringify(CustomerPortalCustomerMetersListRequest$outboundSchema.parse(customerPortalCustomerMetersListRequest));
}
/** @internal */
export const CustomerPortalCustomerMetersListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceCustomerCustomerMeter$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalCustomerMetersListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalCustomerMetersListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalCustomerMetersListResponse' from JSON`);
}
