// @polar-sh/sdk/dist/esm/models/operations/disputeslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { DisputeSortProperty$outboundSchema, } from "../components/disputesortproperty.js";
import { DisputeStatus$outboundSchema, } from "../components/disputestatus.js";
import { ListResourceDispute$inboundSchema, } from "../components/listresourcedispute.js";
/** @internal */
export const DisputesListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function disputesListQueryParamOrganizationIDFilterToJSON(disputesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(DisputesListQueryParamOrganizationIDFilter$outboundSchema.parse(disputesListQueryParamOrganizationIDFilter));
}
/** @internal */
export const QueryParamOrderIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function queryParamOrderIDFilterToJSON(queryParamOrderIDFilter) {
    return JSON.stringify(QueryParamOrderIDFilter$outboundSchema.parse(queryParamOrderIDFilter));
}
/** @internal */
export const StatusFilter$outboundSchema = smartUnion([
    DisputeStatus$outboundSchema,
    z.array(DisputeStatus$outboundSchema),
]);
export function statusFilterToJSON(statusFilter) {
    return JSON.stringify(StatusFilter$outboundSchema.parse(statusFilter));
}
/** @internal */
export const DisputesListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    status: z.optional(z.nullable(smartUnion([
        DisputeStatus$outboundSchema,
        z.array(DisputeStatus$outboundSchema),
    ]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(DisputeSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        orderId: "order_id",
    });
}));
export function disputesListRequestToJSON(disputesListRequest) {
    return JSON.stringify(DisputesListRequest$outboundSchema.parse(disputesListRequest));
}
/** @internal */
export const DisputesListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceDispute$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function disputesListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DisputesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DisputesListResponse' from JSON`);
}
