// @polar-sh/sdk/dist/esm/models/operations/eventtypeslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { EventSource$outboundSchema, } from "../components/eventsource.js";
import { EventTypesSortProperty$outboundSchema, } from "../components/eventtypessortproperty.js";
import { ListResourceEventTypeWithStats$inboundSchema, } from "../components/listresourceeventtypewithstats.js";
/** @internal */
export const EventTypesListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function eventTypesListQueryParamOrganizationIDFilterToJSON(eventTypesListQueryParamOrganizationIDFilter) {
    return JSON.stringify(EventTypesListQueryParamOrganizationIDFilter$outboundSchema.parse(eventTypesListQueryParamOrganizationIDFilter));
}
/** @internal */
export const EventTypesListQueryParamCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function eventTypesListQueryParamCustomerIDFilterToJSON(eventTypesListQueryParamCustomerIDFilter) {
    return JSON.stringify(EventTypesListQueryParamCustomerIDFilter$outboundSchema.parse(eventTypesListQueryParamCustomerIDFilter));
}
/** @internal */
export const EventTypesListQueryParamExternalCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function eventTypesListQueryParamExternalCustomerIDFilterToJSON(eventTypesListQueryParamExternalCustomerIDFilter) {
    return JSON.stringify(EventTypesListQueryParamExternalCustomerIDFilter$outboundSchema.parse(eventTypesListQueryParamExternalCustomerIDFilter));
}
/** @internal */
export const EventTypesListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    query: z.optional(z.nullable(z.string())),
    rootEvents: z._default(z.boolean(), false),
    parentId: z.optional(z.nullable(z.string())),
    source: z.optional(z.nullable(EventSource$outboundSchema)),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(EventTypesSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        rootEvents: "root_events",
        parentId: "parent_id",
    });
}));
export function eventTypesListRequestToJSON(eventTypesListRequest) {
    return JSON.stringify(EventTypesListRequest$outboundSchema.parse(eventTypesListRequest));
}
/** @internal */
export const EventTypesListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceEventTypeWithStats$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function eventTypesListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventTypesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventTypesListResponse' from JSON`);
}
