// @polar-sh/sdk/dist/esm/models/operations/memberslistmembers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceMember$inboundSchema, } from "../components/listresourcemember.js";
import { MemberRole$outboundSchema, } from "../components/memberrole.js";
import { MemberSortProperty$outboundSchema, } from "../components/membersortproperty.js";
/** @internal */
export const MembersListMembersRequest$outboundSchema = z.pipe(z.object({
    customerId: z.optional(z.nullable(z.string())),
    externalCustomerId: z.optional(z.nullable(z.string())),
    role: z.optional(z.nullable(MemberRole$outboundSchema)),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(MemberSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
}));
export function membersListMembersRequestToJSON(membersListMembersRequest) {
    return JSON.stringify(MembersListMembersRequest$outboundSchema.parse(membersListMembersRequest));
}
/** @internal */
export const MembersListMembersResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceMember$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function membersListMembersResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MembersListMembersResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MembersListMembersResponse' from JSON`);
}
