// @polar-sh/sdk/dist/esm/models/operations/ordersexport.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const OrdersExportQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function ordersExportQueryParamOrganizationIDFilterToJSON(ordersExportQueryParamOrganizationIDFilter) {
    return JSON.stringify(OrdersExportQueryParamOrganizationIDFilter$outboundSchema.parse(ordersExportQueryParamOrganizationIDFilter));
}
/** @internal */
export const OrdersExportQueryParamProductIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function ordersExportQueryParamProductIDFilterToJSON(ordersExportQueryParamProductIDFilter) {
    return JSON.stringify(OrdersExportQueryParamProductIDFilter$outboundSchema.parse(ordersExportQueryParamProductIDFilter));
}
/** @internal */
export const OrdersExportRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    productId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
}));
export function ordersExportRequestToJSON(ordersExportRequest) {
    return JSON.stringify(OrdersExportRequest$outboundSchema.parse(ordersExportRequest));
}
/** @internal */
export const OrdersExportResponse$inboundSchema = smartUnion([z.any(), z.string()]);
export function ordersExportResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersExportResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportResponse' from JSON`);
}
