// @polar-sh/sdk/dist/esm/models/operations/organizationaccesstokenslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { ListResourceOrganizationAccessToken$inboundSchema, } from "../components/listresourceorganizationaccesstoken.js";
import { OrganizationAccessTokenSortProperty$outboundSchema, } from "../components/organizationaccesstokensortproperty.js";
/** @internal */
export const OrganizationAccessTokensListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function organizationAccessTokensListQueryParamOrganizationIDFilterToJSON(organizationAccessTokensListQueryParamOrganizationIDFilter) {
    return JSON.stringify(OrganizationAccessTokensListQueryParamOrganizationIDFilter$outboundSchema
        .parse(organizationAccessTokensListQueryParamOrganizationIDFilter));
}
/** @internal */
export const OrganizationAccessTokensListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(OrganizationAccessTokenSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function organizationAccessTokensListRequestToJSON(organizationAccessTokensListRequest) {
    return JSON.stringify(OrganizationAccessTokensListRequest$outboundSchema.parse(organizationAccessTokensListRequest));
}
/** @internal */
export const OrganizationAccessTokensListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceOrganizationAccessToken$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function organizationAccessTokensListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationAccessTokensListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationAccessTokensListResponse' from JSON`);
}
