// @polar-sh/sdk/dist/esm/models/operations/paymentslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { ListResourcePayment$inboundSchema, } from "../components/listresourcepayment.js";
import { PaymentSortProperty$outboundSchema, } from "../components/paymentsortproperty.js";
import { PaymentStatus$outboundSchema, } from "../components/paymentstatus.js";
/** @internal */
export const PaymentsListQueryParamOrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function paymentsListQueryParamOrganizationIDFilterToJSON(paymentsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(PaymentsListQueryParamOrganizationIDFilter$outboundSchema.parse(paymentsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const PaymentsListQueryParamCheckoutIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function paymentsListQueryParamCheckoutIDFilterToJSON(paymentsListQueryParamCheckoutIDFilter) {
    return JSON.stringify(PaymentsListQueryParamCheckoutIDFilter$outboundSchema.parse(paymentsListQueryParamCheckoutIDFilter));
}
/** @internal */
export const PaymentsListQueryParamOrderIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function paymentsListQueryParamOrderIDFilterToJSON(paymentsListQueryParamOrderIDFilter) {
    return JSON.stringify(PaymentsListQueryParamOrderIDFilter$outboundSchema.parse(paymentsListQueryParamOrderIDFilter));
}
/** @internal */
export const PaymentsListQueryParamStatusFilter$outboundSchema = smartUnion([
    PaymentStatus$outboundSchema,
    z.array(PaymentStatus$outboundSchema),
]);
export function paymentsListQueryParamStatusFilterToJSON(paymentsListQueryParamStatusFilter) {
    return JSON.stringify(PaymentsListQueryParamStatusFilter$outboundSchema.parse(paymentsListQueryParamStatusFilter));
}
/** @internal */
export const MethodFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function methodFilterToJSON(methodFilter) {
    return JSON.stringify(MethodFilter$outboundSchema.parse(methodFilter));
}
/** @internal */
export const CustomerEmailFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerEmailFilterToJSON(customerEmailFilter) {
    return JSON.stringify(CustomerEmailFilter$outboundSchema.parse(customerEmailFilter));
}
/** @internal */
export const PaymentsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    checkoutId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    status: z.optional(z.nullable(smartUnion([
        PaymentStatus$outboundSchema,
        z.array(PaymentStatus$outboundSchema),
    ]))),
    method: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerEmail: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(PaymentSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        customerEmail: "customer_email",
    });
}));
export function paymentsListRequestToJSON(paymentsListRequest) {
    return JSON.stringify(PaymentsListRequest$outboundSchema.parse(paymentsListRequest));
}
/** @internal */
export const PaymentsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourcePayment$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function paymentsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PaymentsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentsListResponse' from JSON`);
}
