// @polar-sh/sdk/dist/esm/models/operations/subscriptionsexport.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const OrganizationId$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function organizationIdToJSON(organizationId) {
    return JSON.stringify(OrganizationId$outboundSchema.parse(organizationId));
}
/** @internal */
export const SubscriptionsExportRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function subscriptionsExportRequestToJSON(subscriptionsExportRequest) {
    return JSON.stringify(SubscriptionsExportRequest$outboundSchema.parse(subscriptionsExportRequest));
}
/** @internal */
export const SubscriptionsExportResponse$inboundSchema = smartUnion([z.any(), z.string()]);
export function subscriptionsExportResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionsExportResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionsExportResponse' from JSON`);
}
