// @polar-sh/sdk/dist/esm/models/operations/subscriptionslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerCancellationReason$outboundSchema, } from "../components/customercancellationreason.js";
import { ListResourceSubscription$inboundSchema, } from "../components/listresourcesubscription.js";
import { MetadataQuery$outboundSchema, } from "../components/subscriptionslist.js";
import { SubscriptionSortProperty$outboundSchema, } from "../components/subscriptionsortproperty.js";
/** @internal */
export const OrganizationIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function organizationIDFilterToJSON(organizationIDFilter) {
    return JSON.stringify(OrganizationIDFilter$outboundSchema.parse(organizationIDFilter));
}
/** @internal */
export const ProductIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function productIDFilterToJSON(productIDFilter) {
    return JSON.stringify(ProductIDFilter$outboundSchema.parse(productIDFilter));
}
/** @internal */
export const CustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function customerIDFilterToJSON(customerIDFilter) {
    return JSON.stringify(CustomerIDFilter$outboundSchema.parse(customerIDFilter));
}
/** @internal */
export const ExternalCustomerIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function externalCustomerIDFilterToJSON(externalCustomerIDFilter) {
    return JSON.stringify(ExternalCustomerIDFilter$outboundSchema.parse(externalCustomerIDFilter));
}
/** @internal */
export const DiscountIDFilter$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function discountIDFilterToJSON(discountIDFilter) {
    return JSON.stringify(DiscountIDFilter$outboundSchema.parse(discountIDFilter));
}
/** @internal */
export const CustomerCancellationReasonFilter$outboundSchema = smartUnion([
    CustomerCancellationReason$outboundSchema,
    z.array(CustomerCancellationReason$outboundSchema),
]);
export function customerCancellationReasonFilterToJSON(customerCancellationReasonFilter) {
    return JSON.stringify(CustomerCancellationReasonFilter$outboundSchema.parse(customerCancellationReasonFilter));
}
/** @internal */
export const SubscriptionsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    productId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    discountId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    active: z.optional(z.nullable(z.boolean())),
    cancelAtPeriodEnd: z.optional(z.nullable(z.boolean())),
    customerCancellationReason: z.optional(z.nullable(smartUnion([
        CustomerCancellationReason$outboundSchema,
        z.array(CustomerCancellationReason$outboundSchema),
    ]))),
    canceledAtAfter: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    canceledAtBefore: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(SubscriptionSortProperty$outboundSchema))),
    metadata: z.optional(z.nullable(z.record(z.string(), z.lazy(() => MetadataQuery$outboundSchema)))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        productId: "product_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        discountId: "discount_id",
        cancelAtPeriodEnd: "cancel_at_period_end",
        customerCancellationReason: "customer_cancellation_reason",
        canceledAtAfter: "canceled_at_after",
        canceledAtBefore: "canceled_at_before",
    });
}));
export function subscriptionsListRequestToJSON(subscriptionsListRequest) {
    return JSON.stringify(SubscriptionsListRequest$outboundSchema.parse(subscriptionsListRequest));
}
/** @internal */
export const SubscriptionsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceSubscription$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function subscriptionsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionsListResponse' from JSON`);
}
