// @polar-sh/sdk/dist/esm/models/operations/webhookslistwebhookendpoints.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { ListResourceWebhookEndpoint$inboundSchema, } from "../components/listresourcewebhookendpoint.js";
/** @internal */
export const QueryParamOrganizationId$outboundSchema = smartUnion([z.string(), z.array(z.string())]);
export function queryParamOrganizationIdToJSON(queryParamOrganizationId) {
    return JSON.stringify(QueryParamOrganizationId$outboundSchema.parse(queryParamOrganizationId));
}
/** @internal */
export const WebhooksListWebhookEndpointsRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(smartUnion([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function webhooksListWebhookEndpointsRequestToJSON(webhooksListWebhookEndpointsRequest) {
    return JSON.stringify(WebhooksListWebhookEndpointsRequest$outboundSchema.parse(webhooksListWebhookEndpointsRequest));
}
/** @internal */
export const WebhooksListWebhookEndpointsResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceWebhookEndpoint$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function webhooksListWebhookEndpointsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksListWebhookEndpointsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookEndpointsResponse' from JSON`);
}
