// @polar-sh/sdk/dist/commonjs/models/components/balancerefundmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceRefundMetadata$inboundSchema = void 0;
exports.balanceRefundMetadataFromJSON = balanceRefundMetadataFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BalanceRefundMetadata$inboundSchema = z.pipe(z.object({
    transaction_id: z.string(),
    refund_id: z.string(),
    order_id: z.optional(z.string()),
    order_created_at: z.optional(z.string()),
    product_id: z.optional(z.string()),
    subscription_id: z.optional(z.string()),
    amount: z.int(),
    currency: z.string(),
    presentment_amount: z.int(),
    presentment_currency: z.string(),
    refundable_amount: z.optional(z.int()),
    tax_amount: z.int(),
    tax_state: z.optional(z.nullable(z.string())),
    tax_country: z.optional(z.nullable(z.string())),
    fee: z.int(),
    exchange_rate: z.optional(z.number()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "transaction_id": "transactionId",
        "refund_id": "refundId",
        "order_id": "orderId",
        "order_created_at": "orderCreatedAt",
        "product_id": "productId",
        "subscription_id": "subscriptionId",
        "presentment_amount": "presentmentAmount",
        "presentment_currency": "presentmentCurrency",
        "refundable_amount": "refundableAmount",
        "tax_amount": "taxAmount",
        "tax_state": "taxState",
        "tax_country": "taxCountry",
        "exchange_rate": "exchangeRate",
    });
}));
function balanceRefundMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BalanceRefundMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BalanceRefundMetadata' from JSON`);
}
