// @polar-sh/sdk/dist/commonjs/models/components/benefitfeatureflagupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitFeatureFlagUpdate$outboundSchema = exports.BenefitFeatureFlagUpdateMetadata$outboundSchema = void 0;
exports.benefitFeatureFlagUpdateMetadataToJSON = benefitFeatureFlagUpdateMetadataToJSON;
exports.benefitFeatureFlagUpdateToJSON = benefitFeatureFlagUpdateToJSON;
const z = __importStar(require("zod/v4-mini"));
const smartUnion_js_1 = require("../../types/smartUnion.js");
const benefitfeatureflagproperties_js_1 = require("./benefitfeatureflagproperties.js");
/** @internal */
exports.BenefitFeatureFlagUpdateMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function benefitFeatureFlagUpdateMetadataToJSON(benefitFeatureFlagUpdateMetadata) {
    return JSON.stringify(exports.BenefitFeatureFlagUpdateMetadata$outboundSchema.parse(benefitFeatureFlagUpdateMetadata));
}
/** @internal */
exports.BenefitFeatureFlagUpdate$outboundSchema = z.object({
    metadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    description: z.optional(z.nullable(z.string())),
    type: z.literal("feature_flag"),
    properties: z.optional(z.nullable(benefitfeatureflagproperties_js_1.BenefitFeatureFlagProperties$outboundSchema)),
});
function benefitFeatureFlagUpdateToJSON(benefitFeatureFlagUpdate) {
    return JSON.stringify(exports.BenefitFeatureFlagUpdate$outboundSchema.parse(benefitFeatureFlagUpdate));
}
