// @polar-sh/sdk/dist/commonjs/models/components/benefitgrantgithubrepositoryproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantGitHubRepositoryProperties$outboundSchema = exports.BenefitGrantGitHubRepositoryProperties$inboundSchema = exports.BenefitGrantGitHubRepositoryPropertiesPermission$outboundSchema = exports.BenefitGrantGitHubRepositoryPropertiesPermission$inboundSchema = exports.BenefitGrantGitHubRepositoryPropertiesPermission = void 0;
exports.benefitGrantGitHubRepositoryPropertiesToJSON = benefitGrantGitHubRepositoryPropertiesToJSON;
exports.benefitGrantGitHubRepositoryPropertiesFromJSON = benefitGrantGitHubRepositoryPropertiesFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const openEnums = __importStar(require("../../types/enums.js"));
exports.BenefitGrantGitHubRepositoryPropertiesPermission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
exports.BenefitGrantGitHubRepositoryPropertiesPermission$inboundSchema = openEnums.inboundSchema(exports.BenefitGrantGitHubRepositoryPropertiesPermission);
/** @internal */
exports.BenefitGrantGitHubRepositoryPropertiesPermission$outboundSchema = openEnums.outboundSchema(exports.BenefitGrantGitHubRepositoryPropertiesPermission);
/** @internal */
exports.BenefitGrantGitHubRepositoryProperties$inboundSchema = z.pipe(z.object({
    account_id: z.optional(z.nullable(z.string())),
    repository_owner: z.optional(z.string()),
    repository_name: z.optional(z.string()),
    permission: z.optional(exports.BenefitGrantGitHubRepositoryPropertiesPermission$inboundSchema),
    granted_account_id: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
        "granted_account_id": "grantedAccountId",
    });
}));
/** @internal */
exports.BenefitGrantGitHubRepositoryProperties$outboundSchema = z.pipe(z.object({
    accountId: z.optional(z.nullable(z.string())),
    repositoryOwner: z.optional(z.string()),
    repositoryName: z.optional(z.string()),
    permission: z.optional(exports.BenefitGrantGitHubRepositoryPropertiesPermission$outboundSchema),
    grantedAccountId: z.optional(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
        grantedAccountId: "granted_account_id",
    });
}));
function benefitGrantGitHubRepositoryPropertiesToJSON(benefitGrantGitHubRepositoryProperties) {
    return JSON.stringify(exports.BenefitGrantGitHubRepositoryProperties$outboundSchema.parse(benefitGrantGitHubRepositoryProperties));
}
function benefitGrantGitHubRepositoryPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantGitHubRepositoryProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantGitHubRepositoryProperties' from JSON`);
}
