// @polar-sh/sdk/dist/commonjs/models/components/benefitgrantcustomwebhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantCustomWebhook$outboundSchema = exports.BenefitGrantCustomWebhook$inboundSchema = void 0;
exports.benefitGrantCustomWebhookToJSON = benefitGrantCustomWebhookToJSON;
exports.benefitGrantCustomWebhookFromJSON = benefitGrantCustomWebhookFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitcustom_js_1 = require("./benefitcustom.js");
const benefitgrantcustomproperties_js_1 = require("./benefitgrantcustomproperties.js");
const benefitgranterror_js_1 = require("./benefitgranterror.js");
const customer_js_1 = require("./customer.js");
const member_js_1 = require("./member.js");
/** @internal */
exports.BenefitGrantCustomWebhook$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    granted_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_granted: z.boolean(),
    revoked_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    customer_id: z.string(),
    member_id: z.optional(z.nullable(z.string())),
    benefit_id: z.string(),
    error: z.optional(z.nullable(benefitgranterror_js_1.BenefitGrantError$inboundSchema)),
    customer: customer_js_1.Customer$inboundSchema,
    member: z.optional(z.nullable(member_js_1.Member$inboundSchema)),
    benefit: benefitcustom_js_1.BenefitCustom$inboundSchema,
    properties: benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
    previous_properties: z.optional(z.nullable(benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema)),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "member_id": "memberId",
        "benefit_id": "benefitId",
        "previous_properties": "previousProperties",
    });
}));
/** @internal */
exports.BenefitGrantCustomWebhook$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    grantedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    isGranted: z.boolean(),
    revokedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    isRevoked: z.boolean(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    customerId: z.string(),
    memberId: z.optional(z.nullable(z.string())),
    benefitId: z.string(),
    error: z.optional(z.nullable(benefitgranterror_js_1.BenefitGrantError$outboundSchema)),
    customer: customer_js_1.Customer$outboundSchema,
    member: z.optional(z.nullable(member_js_1.Member$outboundSchema)),
    benefit: benefitcustom_js_1.BenefitCustom$outboundSchema,
    properties: benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
    previousProperties: z.optional(z.nullable(benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema)),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        isGranted: "is_granted",
        revokedAt: "revoked_at",
        isRevoked: "is_revoked",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        customerId: "customer_id",
        memberId: "member_id",
        benefitId: "benefit_id",
        previousProperties: "previous_properties",
    });
}));
function benefitGrantCustomWebhookToJSON(benefitGrantCustomWebhook) {
    return JSON.stringify(exports.BenefitGrantCustomWebhook$outboundSchema.parse(benefitGrantCustomWebhook));
}
function benefitGrantCustomWebhookFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantCustomWebhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantCustomWebhook' from JSON`);
}
