// @polar-sh/sdk/dist/commonjs/models/components/cardpayment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardPayment$inboundSchema = void 0;
exports.cardPaymentFromJSON = cardPaymentFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const cardpaymentmetadata_js_1 = require("./cardpaymentmetadata.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
const paymentstatus_js_1 = require("./paymentstatus.js");
/** @internal */
exports.CardPayment$inboundSchema = z
    .pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema,
    status: paymentstatus_js_1.PaymentStatus$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    method: z.literal("card"),
    decline_reason: z.nullable(z.string()),
    decline_message: z.nullable(z.string()),
    organization_id: z.string(),
    checkout_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    processor_metadata: z.optional(z.record(z.string(), z.any())),
    method_metadata: cardpaymentmetadata_js_1.CardPaymentMetadata$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "decline_reason": "declineReason",
        "decline_message": "declineMessage",
        "organization_id": "organizationId",
        "checkout_id": "checkoutId",
        "order_id": "orderId",
        "processor_metadata": "processorMetadata",
        "method_metadata": "methodMetadata",
    });
}));
function cardPaymentFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CardPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CardPayment' from JSON`);
}
