// @polar-sh/sdk/dist/commonjs/models/components/checkoutproduct.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutProduct$outboundSchema = exports.CheckoutProduct$inboundSchema = exports.CheckoutProductPrices$outboundSchema = exports.CheckoutProductPrices$inboundSchema = void 0;
exports.checkoutProductPricesToJSON = checkoutProductPricesToJSON;
exports.checkoutProductPricesFromJSON = checkoutProductPricesFromJSON;
exports.checkoutProductToJSON = checkoutProductToJSON;
exports.checkoutProductFromJSON = checkoutProductFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const benefitpublic_js_1 = require("./benefitpublic.js");
const legacyrecurringproductprice_js_1 = require("./legacyrecurringproductprice.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
const productprice_js_1 = require("./productprice.js");
const productvisibility_js_1 = require("./productvisibility.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.CheckoutProductPrices$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
/** @internal */
exports.CheckoutProductPrices$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
    productprice_js_1.ProductPrice$outboundSchema,
]);
function checkoutProductPricesToJSON(checkoutProductPrices) {
    return JSON.stringify(exports.CheckoutProductPrices$outboundSchema.parse(checkoutProductPrices));
}
function checkoutProductPricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductPrices' from JSON`);
}
/** @internal */
exports.CheckoutProduct$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.int()),
    name: z.string(),
    description: z.nullable(z.string()),
    visibility: productvisibility_js_1.ProductVisibility$inboundSchema,
    recurring_interval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema),
    recurring_interval_count: z.nullable(z.int()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ])),
    benefits: z.array(benefitpublic_js_1.BenefitPublic$inboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
}));
/** @internal */
exports.CheckoutProduct$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    trialInterval: z.nullable(trialinterval_js_1.TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.int()),
    name: z.string(),
    description: z.nullable(z.string()),
    visibility: productvisibility_js_1.ProductVisibility$outboundSchema,
    recurringInterval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema),
    recurringIntervalCount: z.nullable(z.int()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
        productprice_js_1.ProductPrice$outboundSchema,
    ])),
    benefits: z.array(benefitpublic_js_1.BenefitPublic$outboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        recurringIntervalCount: "recurring_interval_count",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
}));
function checkoutProductToJSON(checkoutProduct) {
    return JSON.stringify(exports.CheckoutProduct$outboundSchema.parse(checkoutProduct));
}
function checkoutProductFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProduct' from JSON`);
}
