// @polar-sh/sdk/dist/commonjs/models/components/checkoutpublicconfirmed.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutPublicConfirmed$inboundSchema = exports.CheckoutPublicConfirmedDiscount$inboundSchema = exports.CheckoutPublicConfirmedPrices$inboundSchema = exports.CheckoutPublicConfirmedProductPrice$inboundSchema = exports.CheckoutPublicConfirmedCustomFieldData$inboundSchema = void 0;
exports.checkoutPublicConfirmedCustomFieldDataFromJSON = checkoutPublicConfirmedCustomFieldDataFromJSON;
exports.checkoutPublicConfirmedProductPriceFromJSON = checkoutPublicConfirmedProductPriceFromJSON;
exports.checkoutPublicConfirmedPricesFromJSON = checkoutPublicConfirmedPricesFromJSON;
exports.checkoutPublicConfirmedDiscountFromJSON = checkoutPublicConfirmedDiscountFromJSON;
exports.checkoutPublicConfirmedFromJSON = checkoutPublicConfirmedFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const address_js_1 = require("./address.js");
const attachedcustomfield_js_1 = require("./attachedcustomfield.js");
const checkoutbillingaddressfields_js_1 = require("./checkoutbillingaddressfields.js");
const checkoutdiscountfixedonceforeverduration_js_1 = require("./checkoutdiscountfixedonceforeverduration.js");
const checkoutdiscountfixedrepeatduration_js_1 = require("./checkoutdiscountfixedrepeatduration.js");
const checkoutdiscountpercentageonceforeverduration_js_1 = require("./checkoutdiscountpercentageonceforeverduration.js");
const checkoutdiscountpercentagerepeatduration_js_1 = require("./checkoutdiscountpercentagerepeatduration.js");
const checkoutorganization_js_1 = require("./checkoutorganization.js");
const checkoutproduct_js_1 = require("./checkoutproduct.js");
const legacyrecurringproductprice_js_1 = require("./legacyrecurringproductprice.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
const productprice_js_1 = require("./productprice.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.CheckoutPublicConfirmedCustomFieldData$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
function checkoutPublicConfirmedCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPublicConfirmedCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicConfirmedCustomFieldData' from JSON`);
}
/** @internal */
exports.CheckoutPublicConfirmedProductPrice$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
function checkoutPublicConfirmedProductPriceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPublicConfirmedProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicConfirmedProductPrice' from JSON`);
}
/** @internal */
exports.CheckoutPublicConfirmedPrices$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
function checkoutPublicConfirmedPricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPublicConfirmedPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicConfirmedPrices' from JSON`);
}
/** @internal */
exports.CheckoutPublicConfirmedDiscount$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
    checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
    checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
    checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
]);
function checkoutPublicConfirmedDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPublicConfirmedDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicConfirmedDiscount' from JSON`);
}
/** @internal */
exports.CheckoutPublicConfirmed$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    payment_processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema,
    status: z.literal("confirmed"),
    client_secret: z.string(),
    url: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    success_url: z.string(),
    return_url: z.nullable(z.string()),
    embed_origin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    min_seats: z.optional(z.nullable(z.int())),
    max_seats: z.optional(z.nullable(z.int())),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.nullable(z.int()),
    total_amount: z.int(),
    currency: z.string(),
    allow_trial: z.nullable(z.boolean()),
    active_trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema),
    active_trial_interval_count: z.nullable(z.int()),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    organization_id: z.string(),
    product_id: z.nullable(z.string()),
    product_price_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    require_billing_address: z.boolean(),
    is_discount_applicable: z.boolean(),
    is_free_product_price: z.boolean(),
    is_payment_required: z.boolean(),
    is_payment_setup_required: z.boolean(),
    is_payment_form_required: z.boolean(),
    customer_id: z.nullable(z.string()),
    is_business_customer: z.boolean(),
    customer_name: z.nullable(z.string()),
    customer_email: z.nullable(z.string()),
    customer_ip_address: z.nullable(z.string()),
    customer_billing_name: z.nullable(z.string()),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema),
    customer_tax_id: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    payment_processor_metadata: z.record(z.string(), z.string()),
    billing_address_fields: checkoutbillingaddressfields_js_1.CheckoutBillingAddressFields$inboundSchema,
    products: z.array(checkoutproduct_js_1.CheckoutProduct$inboundSchema),
    product: z.nullable(checkoutproduct_js_1.CheckoutProduct$inboundSchema),
    product_price: z.nullable((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ])))),
    discount: z.nullable((0, smartUnion_js_1.smartUnion)([
        checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
        checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
        checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
        checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
    ])),
    organization: checkoutorganization_js_1.CheckoutOrganization$inboundSchema,
    attached_custom_fields: z.nullable(z.array(attachedcustomfield_js_1.AttachedCustomField$inboundSchema)),
    customer_session_token: z.nullable(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "expires_at": "expiresAt",
        "success_url": "successUrl",
        "return_url": "returnUrl",
        "embed_origin": "embedOrigin",
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "allow_trial": "allowTrial",
        "active_trial_interval": "activeTrialInterval",
        "active_trial_interval_count": "activeTrialIntervalCount",
        "trial_end": "trialEnd",
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "is_discount_applicable": "isDiscountApplicable",
        "is_free_product_price": "isFreeProductPrice",
        "is_payment_required": "isPaymentRequired",
        "is_payment_setup_required": "isPaymentSetupRequired",
        "is_payment_form_required": "isPaymentFormRequired",
        "customer_id": "customerId",
        "is_business_customer": "isBusinessCustomer",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_name": "customerBillingName",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "payment_processor_metadata": "paymentProcessorMetadata",
        "billing_address_fields": "billingAddressFields",
        "product_price": "productPrice",
        "attached_custom_fields": "attachedCustomFields",
        "customer_session_token": "customerSessionToken",
    });
}));
function checkoutPublicConfirmedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPublicConfirmed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicConfirmed' from JSON`);
}
