// @polar-sh/sdk/dist/commonjs/models/components/checkoutdiscountfixedrepeatduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutDiscountFixedRepeatDuration$outboundSchema = exports.CheckoutDiscountFixedRepeatDuration$inboundSchema = void 0;
exports.checkoutDiscountFixedRepeatDurationToJSON = checkoutDiscountFixedRepeatDurationToJSON;
exports.checkoutDiscountFixedRepeatDurationFromJSON = checkoutDiscountFixedRepeatDurationFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discounttype_js_1 = require("./discounttype.js");
/** @internal */
exports.CheckoutDiscountFixedRepeatDuration$inboundSchema = z.pipe(z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
    });
}));
/** @internal */
exports.CheckoutDiscountFixedRepeatDuration$outboundSchema = z.pipe(z.object({
    duration: discountduration_js_1.DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: discounttype_js_1.DiscountType$outboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        durationInMonths: "duration_in_months",
    });
}));
function checkoutDiscountFixedRepeatDurationToJSON(checkoutDiscountFixedRepeatDuration) {
    return JSON.stringify(exports.CheckoutDiscountFixedRepeatDuration$outboundSchema.parse(checkoutDiscountFixedRepeatDuration));
}
function checkoutDiscountFixedRepeatDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutDiscountFixedRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedRepeatDuration' from JSON`);
}
