// @polar-sh/sdk/dist/commonjs/models/components/checkout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Checkout$outboundSchema = exports.Checkout$inboundSchema = exports.CustomerMetadata$outboundSchema = exports.CustomerMetadata$inboundSchema = exports.CheckoutDiscount$outboundSchema = exports.CheckoutDiscount$inboundSchema = exports.CheckoutPrices$outboundSchema = exports.CheckoutPrices$inboundSchema = exports.CheckoutProductPrice$outboundSchema = exports.CheckoutProductPrice$inboundSchema = exports.CheckoutCustomFieldData$outboundSchema = exports.CheckoutCustomFieldData$inboundSchema = void 0;
exports.checkoutCustomFieldDataToJSON = checkoutCustomFieldDataToJSON;
exports.checkoutCustomFieldDataFromJSON = checkoutCustomFieldDataFromJSON;
exports.checkoutProductPriceToJSON = checkoutProductPriceToJSON;
exports.checkoutProductPriceFromJSON = checkoutProductPriceFromJSON;
exports.checkoutPricesToJSON = checkoutPricesToJSON;
exports.checkoutPricesFromJSON = checkoutPricesFromJSON;
exports.checkoutDiscountToJSON = checkoutDiscountToJSON;
exports.checkoutDiscountFromJSON = checkoutDiscountFromJSON;
exports.customerMetadataToJSON = customerMetadataToJSON;
exports.customerMetadataFromJSON = customerMetadataFromJSON;
exports.checkoutToJSON = checkoutToJSON;
exports.checkoutFromJSON = checkoutFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const address_js_1 = require("./address.js");
const attachedcustomfield_js_1 = require("./attachedcustomfield.js");
const checkoutbillingaddressfields_js_1 = require("./checkoutbillingaddressfields.js");
const checkoutdiscountfixedonceforeverduration_js_1 = require("./checkoutdiscountfixedonceforeverduration.js");
const checkoutdiscountfixedrepeatduration_js_1 = require("./checkoutdiscountfixedrepeatduration.js");
const checkoutdiscountpercentageonceforeverduration_js_1 = require("./checkoutdiscountpercentageonceforeverduration.js");
const checkoutdiscountpercentagerepeatduration_js_1 = require("./checkoutdiscountpercentagerepeatduration.js");
const checkoutproduct_js_1 = require("./checkoutproduct.js");
const checkoutstatus_js_1 = require("./checkoutstatus.js");
const legacyrecurringproductprice_js_1 = require("./legacyrecurringproductprice.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
const paymentprocessor_js_1 = require("./paymentprocessor.js");
const productprice_js_1 = require("./productprice.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.CheckoutCustomFieldData$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
exports.CheckoutCustomFieldData$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
function checkoutCustomFieldDataToJSON(checkoutCustomFieldData) {
    return JSON.stringify(exports.CheckoutCustomFieldData$outboundSchema.parse(checkoutCustomFieldData));
}
function checkoutCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutCustomFieldData' from JSON`);
}
/** @internal */
exports.CheckoutProductPrice$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
/** @internal */
exports.CheckoutProductPrice$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
    productprice_js_1.ProductPrice$outboundSchema,
]);
function checkoutProductPriceToJSON(checkoutProductPrice) {
    return JSON.stringify(exports.CheckoutProductPrice$outboundSchema.parse(checkoutProductPrice));
}
function checkoutProductPriceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductPrice' from JSON`);
}
/** @internal */
exports.CheckoutPrices$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
/** @internal */
exports.CheckoutPrices$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
    productprice_js_1.ProductPrice$outboundSchema,
]);
function checkoutPricesToJSON(checkoutPrices) {
    return JSON.stringify(exports.CheckoutPrices$outboundSchema.parse(checkoutPrices));
}
function checkoutPricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPrices' from JSON`);
}
/** @internal */
exports.CheckoutDiscount$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
    checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
    checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
    checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
]);
/** @internal */
exports.CheckoutDiscount$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$outboundSchema,
    checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
    checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$outboundSchema,
    checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
]);
function checkoutDiscountToJSON(checkoutDiscount) {
    return JSON.stringify(exports.CheckoutDiscount$outboundSchema.parse(checkoutDiscount));
}
function checkoutDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscount' from JSON`);
}
/** @internal */
exports.CustomerMetadata$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()]);
/** @internal */
exports.CustomerMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()]);
function customerMetadataToJSON(customerMetadata) {
    return JSON.stringify(exports.CustomerMetadata$outboundSchema.parse(customerMetadata));
}
function customerMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerMetadata' from JSON`);
}
/** @internal */
exports.Checkout$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    payment_processor: paymentprocessor_js_1.PaymentProcessor$inboundSchema,
    status: checkoutstatus_js_1.CheckoutStatus$inboundSchema,
    client_secret: z.string(),
    url: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    success_url: z.string(),
    return_url: z.nullable(z.string()),
    embed_origin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    min_seats: z.optional(z.nullable(z.int())),
    max_seats: z.optional(z.nullable(z.int())),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.nullable(z.int()),
    total_amount: z.int(),
    currency: z.string(),
    allow_trial: z.nullable(z.boolean()),
    active_trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema),
    active_trial_interval_count: z.nullable(z.int()),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    organization_id: z.string(),
    product_id: z.nullable(z.string()),
    product_price_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    require_billing_address: z.boolean(),
    is_discount_applicable: z.boolean(),
    is_free_product_price: z.boolean(),
    is_payment_required: z.boolean(),
    is_payment_setup_required: z.boolean(),
    is_payment_form_required: z.boolean(),
    customer_id: z.nullable(z.string()),
    is_business_customer: z.boolean(),
    customer_name: z.nullable(z.string()),
    customer_email: z.nullable(z.string()),
    customer_ip_address: z.nullable(z.string()),
    customer_billing_name: z.nullable(z.string()),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema),
    customer_tax_id: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    payment_processor_metadata: z.record(z.string(), z.string()),
    billing_address_fields: checkoutbillingaddressfields_js_1.CheckoutBillingAddressFields$inboundSchema,
    trial_interval: z.nullable(trialinterval_js_1.TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.int()),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    products: z.array(checkoutproduct_js_1.CheckoutProduct$inboundSchema),
    product: z.nullable(checkoutproduct_js_1.CheckoutProduct$inboundSchema),
    product_price: z.nullable((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ])))),
    discount: z.nullable((0, smartUnion_js_1.smartUnion)([
        checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
        checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
        checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
        checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
    ])),
    subscription_id: z.nullable(z.string()),
    attached_custom_fields: z.nullable(z.array(attachedcustomfield_js_1.AttachedCustomField$inboundSchema)),
    customer_metadata: z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()])),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "expires_at": "expiresAt",
        "success_url": "successUrl",
        "return_url": "returnUrl",
        "embed_origin": "embedOrigin",
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "allow_trial": "allowTrial",
        "active_trial_interval": "activeTrialInterval",
        "active_trial_interval_count": "activeTrialIntervalCount",
        "trial_end": "trialEnd",
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "is_discount_applicable": "isDiscountApplicable",
        "is_free_product_price": "isFreeProductPrice",
        "is_payment_required": "isPaymentRequired",
        "is_payment_setup_required": "isPaymentSetupRequired",
        "is_payment_form_required": "isPaymentFormRequired",
        "customer_id": "customerId",
        "is_business_customer": "isBusinessCustomer",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_name": "customerBillingName",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "payment_processor_metadata": "paymentProcessorMetadata",
        "billing_address_fields": "billingAddressFields",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "external_customer_id": "externalCustomerId",
        "product_price": "productPrice",
        "subscription_id": "subscriptionId",
        "attached_custom_fields": "attachedCustomFields",
        "customer_metadata": "customerMetadata",
    });
}));
/** @internal */
exports.Checkout$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    customFieldData: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    paymentProcessor: paymentprocessor_js_1.PaymentProcessor$outboundSchema,
    status: checkoutstatus_js_1.CheckoutStatus$outboundSchema,
    clientSecret: z.string(),
    url: z.string(),
    expiresAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    successUrl: z.string(),
    returnUrl: z.nullable(z.string()),
    embedOrigin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    minSeats: z.optional(z.nullable(z.int())),
    maxSeats: z.optional(z.nullable(z.int())),
    discountAmount: z.int(),
    netAmount: z.int(),
    taxAmount: z.nullable(z.int()),
    totalAmount: z.int(),
    currency: z.string(),
    allowTrial: z.nullable(z.boolean()),
    activeTrialInterval: z.nullable(trialinterval_js_1.TrialInterval$outboundSchema),
    activeTrialIntervalCount: z.nullable(z.int()),
    trialEnd: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    organizationId: z.string(),
    productId: z.nullable(z.string()),
    productPriceId: z.nullable(z.string()),
    discountId: z.nullable(z.string()),
    allowDiscountCodes: z.boolean(),
    requireBillingAddress: z.boolean(),
    isDiscountApplicable: z.boolean(),
    isFreeProductPrice: z.boolean(),
    isPaymentRequired: z.boolean(),
    isPaymentSetupRequired: z.boolean(),
    isPaymentFormRequired: z.boolean(),
    customerId: z.nullable(z.string()),
    isBusinessCustomer: z.boolean(),
    customerName: z.nullable(z.string()),
    customerEmail: z.nullable(z.string()),
    customerIpAddress: z.nullable(z.string()),
    customerBillingName: z.nullable(z.string()),
    customerBillingAddress: z.nullable(address_js_1.Address$outboundSchema),
    customerTaxId: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    paymentProcessorMetadata: z.record(z.string(), z.string()),
    billingAddressFields: checkoutbillingaddressfields_js_1.CheckoutBillingAddressFields$outboundSchema,
    trialInterval: z.nullable(trialinterval_js_1.TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.int()),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    products: z.array(checkoutproduct_js_1.CheckoutProduct$outboundSchema),
    product: z.nullable(checkoutproduct_js_1.CheckoutProduct$outboundSchema),
    productPrice: z.nullable((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
        productprice_js_1.ProductPrice$outboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array((0, smartUnion_js_1.smartUnion)([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
        productprice_js_1.ProductPrice$outboundSchema,
    ])))),
    discount: z.nullable((0, smartUnion_js_1.smartUnion)([
        checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$outboundSchema,
        checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
        checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$outboundSchema,
        checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
    ])),
    subscriptionId: z.nullable(z.string()),
    attachedCustomFields: z.nullable(z.array(attachedcustomfield_js_1.AttachedCustomField$outboundSchema)),
    customerMetadata: z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()])),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customFieldData: "custom_field_data",
        paymentProcessor: "payment_processor",
        clientSecret: "client_secret",
        expiresAt: "expires_at",
        successUrl: "success_url",
        returnUrl: "return_url",
        embedOrigin: "embed_origin",
        minSeats: "min_seats",
        maxSeats: "max_seats",
        discountAmount: "discount_amount",
        netAmount: "net_amount",
        taxAmount: "tax_amount",
        totalAmount: "total_amount",
        allowTrial: "allow_trial",
        activeTrialInterval: "active_trial_interval",
        activeTrialIntervalCount: "active_trial_interval_count",
        trialEnd: "trial_end",
        organizationId: "organization_id",
        productId: "product_id",
        productPriceId: "product_price_id",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        isDiscountApplicable: "is_discount_applicable",
        isFreeProductPrice: "is_free_product_price",
        isPaymentRequired: "is_payment_required",
        isPaymentSetupRequired: "is_payment_setup_required",
        isPaymentFormRequired: "is_payment_form_required",
        customerId: "customer_id",
        isBusinessCustomer: "is_business_customer",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingName: "customer_billing_name",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        paymentProcessorMetadata: "payment_processor_metadata",
        billingAddressFields: "billing_address_fields",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        externalCustomerId: "external_customer_id",
        productPrice: "product_price",
        subscriptionId: "subscription_id",
        attachedCustomFields: "attached_custom_fields",
        customerMetadata: "customer_metadata",
    });
}));
function checkoutToJSON(checkout) {
    return JSON.stringify(exports.Checkout$outboundSchema.parse(checkout));
}
function checkoutFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Checkout$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Checkout' from JSON`);
}
