// @polar-sh/sdk/dist/commonjs/models/components/checkoutcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutCreate$outboundSchema = exports.CheckoutCreatePrices$outboundSchema = exports.CheckoutCreateCustomerMetadata$outboundSchema = exports.CheckoutCreateCustomFieldData$outboundSchema = exports.CheckoutCreateMetadata$outboundSchema = void 0;
exports.checkoutCreateMetadataToJSON = checkoutCreateMetadataToJSON;
exports.checkoutCreateCustomFieldDataToJSON = checkoutCreateCustomFieldDataToJSON;
exports.checkoutCreateCustomerMetadataToJSON = checkoutCreateCustomerMetadataToJSON;
exports.checkoutCreatePricesToJSON = checkoutCreatePricesToJSON;
exports.checkoutCreateToJSON = checkoutCreateToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const addressinput_js_1 = require("./addressinput.js");
const presentmentcurrency_js_1 = require("./presentmentcurrency.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const productpriceseatbasedcreate_js_1 = require("./productpriceseatbasedcreate.js");
const trialinterval_js_1 = require("./trialinterval.js");
/** @internal */
exports.CheckoutCreateMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function checkoutCreateMetadataToJSON(checkoutCreateMetadata) {
    return JSON.stringify(exports.CheckoutCreateMetadata$outboundSchema.parse(checkoutCreateMetadata));
}
/** @internal */
exports.CheckoutCreateCustomFieldData$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
function checkoutCreateCustomFieldDataToJSON(checkoutCreateCustomFieldData) {
    return JSON.stringify(exports.CheckoutCreateCustomFieldData$outboundSchema.parse(checkoutCreateCustomFieldData));
}
/** @internal */
exports.CheckoutCreateCustomerMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function checkoutCreateCustomerMetadataToJSON(checkoutCreateCustomerMetadata) {
    return JSON.stringify(exports.CheckoutCreateCustomerMetadata$outboundSchema.parse(checkoutCreateCustomerMetadata));
}
/** @internal */
exports.CheckoutCreatePrices$outboundSchema = z.union([
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema,
]);
function checkoutCreatePricesToJSON(checkoutCreatePrices) {
    return JSON.stringify(exports.CheckoutCreatePrices$outboundSchema.parse(checkoutCreatePrices));
}
/** @internal */
exports.CheckoutCreate$outboundSchema = z.pipe(z.object({
    trialInterval: z.optional(z.nullable(trialinterval_js_1.TrialInterval$outboundSchema)),
    trialIntervalCount: z.optional(z.nullable(z.int())),
    metadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    customFieldData: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    discountId: z.optional(z.nullable(z.string())),
    allowDiscountCodes: z._default(z.boolean(), true),
    requireBillingAddress: z._default(z.boolean(), false),
    amount: z.optional(z.nullable(z.int())),
    seats: z.optional(z.nullable(z.int())),
    minSeats: z.optional(z.nullable(z.int())),
    maxSeats: z.optional(z.nullable(z.int())),
    allowTrial: z._default(z.boolean(), true),
    customerId: z.optional(z.nullable(z.string())),
    isBusinessCustomer: z._default(z.boolean(), false),
    externalCustomerId: z.optional(z.nullable(z.string())),
    customerName: z.optional(z.nullable(z.string())),
    customerEmail: z.optional(z.nullable(z.string())),
    customerIpAddress: z.optional(z.nullable(z.string())),
    customerBillingName: z.optional(z.nullable(z.string())),
    customerBillingAddress: z.optional(z.nullable(addressinput_js_1.AddressInput$outboundSchema)),
    customerTaxId: z.optional(z.nullable(z.string())),
    customerMetadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    subscriptionId: z.optional(z.nullable(z.string())),
    successUrl: z.optional(z.nullable(z.string())),
    returnUrl: z.optional(z.nullable(z.string())),
    embedOrigin: z.optional(z.nullable(z.string())),
    locale: z.optional(z.nullable(z.string())),
    currency: z.optional(z.nullable(presentmentcurrency_js_1.PresentmentCurrency$outboundSchema)),
    products: z.array(z.string()),
    prices: z.optional(z.nullable(z.record(z.string(), z.array(z.union([
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema,
    ]))))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        customFieldData: "custom_field_data",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        minSeats: "min_seats",
        maxSeats: "max_seats",
        allowTrial: "allow_trial",
        customerId: "customer_id",
        isBusinessCustomer: "is_business_customer",
        externalCustomerId: "external_customer_id",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingName: "customer_billing_name",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        customerMetadata: "customer_metadata",
        subscriptionId: "subscription_id",
        successUrl: "success_url",
        returnUrl: "return_url",
        embedOrigin: "embed_origin",
    });
}));
function checkoutCreateToJSON(checkoutCreate) {
    return JSON.stringify(exports.CheckoutCreate$outboundSchema.parse(checkoutCreate));
}
