// @polar-sh/sdk/dist/commonjs/models/components/customerstatesubscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerStateSubscription$outboundSchema = exports.CustomerStateSubscription$inboundSchema = exports.Status$outboundSchema = exports.Status$inboundSchema = exports.CustomerStateSubscriptionCustomFieldData$outboundSchema = exports.CustomerStateSubscriptionCustomFieldData$inboundSchema = exports.Status = void 0;
exports.customerStateSubscriptionCustomFieldDataToJSON = customerStateSubscriptionCustomFieldDataToJSON;
exports.customerStateSubscriptionCustomFieldDataFromJSON = customerStateSubscriptionCustomFieldDataFromJSON;
exports.customerStateSubscriptionToJSON = customerStateSubscriptionToJSON;
exports.customerStateSubscriptionFromJSON = customerStateSubscriptionFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const openEnums = __importStar(require("../../types/enums.js"));
const smartUnion_js_1 = require("../../types/smartUnion.js");
const customerstatesubscriptionmeter_js_1 = require("./customerstatesubscriptionmeter.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
exports.Status = {
    Active: "active",
    Trialing: "trialing",
};
/** @internal */
exports.CustomerStateSubscriptionCustomFieldData$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
exports.CustomerStateSubscriptionCustomFieldData$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
function customerStateSubscriptionCustomFieldDataToJSON(customerStateSubscriptionCustomFieldData) {
    return JSON.stringify(exports.CustomerStateSubscriptionCustomFieldData$outboundSchema.parse(customerStateSubscriptionCustomFieldData));
}
function customerStateSubscriptionCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerStateSubscriptionCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateSubscriptionCustomFieldData' from JSON`);
}
/** @internal */
exports.Status$inboundSchema = openEnums
    .inboundSchema(exports.Status);
/** @internal */
exports.Status$outboundSchema = openEnums
    .outboundSchema(exports.Status);
/** @internal */
exports.CustomerStateSubscription$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    status: exports.Status$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
    current_period_start: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    current_period_end: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    trial_start: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    cancel_at_period_end: z.boolean(),
    canceled_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    started_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    product_id: z.string(),
    discount_id: z.nullable(z.string()),
    meters: z.array(customerstatesubscriptionmeter_js_1.CustomerStateSubscriptionMeter$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "recurring_interval": "recurringInterval",
        "current_period_start": "currentPeriodStart",
        "current_period_end": "currentPeriodEnd",
        "trial_start": "trialStart",
        "trial_end": "trialEnd",
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "canceled_at": "canceledAt",
        "started_at": "startedAt",
        "ends_at": "endsAt",
        "product_id": "productId",
        "discount_id": "discountId",
    });
}));
/** @internal */
exports.CustomerStateSubscription$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    customFieldData: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$outboundSchema),
    status: exports.Status$outboundSchema,
    amount: z.int(),
    currency: z.string(),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
    currentPeriodStart: z.pipe(z.date(), z.transform(v => v.toISOString())),
    currentPeriodEnd: z.pipe(z.date(), z.transform(v => v.toISOString())),
    trialStart: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    trialEnd: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    cancelAtPeriodEnd: z.boolean(),
    canceledAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    startedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    endsAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    productId: z.string(),
    discountId: z.nullable(z.string()),
    meters: z.array(customerstatesubscriptionmeter_js_1.CustomerStateSubscriptionMeter$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customFieldData: "custom_field_data",
        recurringInterval: "recurring_interval",
        currentPeriodStart: "current_period_start",
        currentPeriodEnd: "current_period_end",
        trialStart: "trial_start",
        trialEnd: "trial_end",
        cancelAtPeriodEnd: "cancel_at_period_end",
        canceledAt: "canceled_at",
        startedAt: "started_at",
        endsAt: "ends_at",
        productId: "product_id",
        discountId: "discount_id",
    });
}));
function customerStateSubscriptionToJSON(customerStateSubscription) {
    return JSON.stringify(exports.CustomerStateSubscription$outboundSchema.parse(customerStateSubscription));
}
function customerStateSubscriptionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerStateSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateSubscription' from JSON`);
}
