// @polar-sh/sdk/dist/commonjs/models/components/customfielddateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldDateProperties$outboundSchema = exports.CustomFieldDateProperties$inboundSchema = void 0;
exports.customFieldDatePropertiesToJSON = customFieldDatePropertiesToJSON;
exports.customFieldDatePropertiesFromJSON = customFieldDatePropertiesFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomFieldDateProperties$inboundSchema = z.pipe(z.object({
    form_label: z.optional(z.string()),
    form_help_text: z.optional(z.string()),
    form_placeholder: z.optional(z.string()),
    ge: z.optional(z.int()),
    le: z.optional(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
}));
/** @internal */
exports.CustomFieldDateProperties$outboundSchema = z.pipe(z.object({
    formLabel: z.optional(z.string()),
    formHelpText: z.optional(z.string()),
    formPlaceholder: z.optional(z.string()),
    ge: z.optional(z.int()),
    le: z.optional(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
}));
function customFieldDatePropertiesToJSON(customFieldDateProperties) {
    return JSON.stringify(exports.CustomFieldDateProperties$outboundSchema.parse(customFieldDateProperties));
}
function customFieldDatePropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldDateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldDateProperties' from JSON`);
}
