// @polar-sh/sdk/dist/commonjs/models/components/customerupdatedmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerUpdatedMetadata$inboundSchema = void 0;
exports.customerUpdatedMetadataFromJSON = customerUpdatedMetadataFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerupdatedfields_js_1 = require("./customerupdatedfields.js");
/** @internal */
exports.CustomerUpdatedMetadata$inboundSchema = z.pipe(z.object({
    customer_id: z.string(),
    customer_email: z.nullable(z.string()),
    customer_name: z.nullable(z.string()),
    customer_external_id: z.nullable(z.string()),
    updated_fields: customerupdatedfields_js_1.CustomerUpdatedFields$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_id": "customerId",
        "customer_email": "customerEmail",
        "customer_name": "customerName",
        "customer_external_id": "customerExternalId",
        "updated_fields": "updatedFields",
    });
}));
function customerUpdatedMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerUpdatedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerUpdatedMetadata' from JSON`);
}
