// @polar-sh/sdk/dist/commonjs/models/components/customerseat.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSeat$outboundSchema = exports.CustomerSeat$inboundSchema = void 0;
exports.customerSeatToJSON = customerSeatToJSON;
exports.customerSeatFromJSON = customerSeatFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const member_js_1 = require("./member.js");
const seatstatus_js_1 = require("./seatstatus.js");
/** @internal */
exports.CustomerSeat$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    subscription_id: z.optional(z.nullable(z.string())),
    order_id: z.optional(z.nullable(z.string())),
    status: seatstatus_js_1.SeatStatus$inboundSchema,
    customer_id: z.optional(z.nullable(z.string())),
    member_id: z.optional(z.nullable(z.string())),
    member: z.optional(z.nullable(member_js_1.Member$inboundSchema)),
    email: z.optional(z.nullable(z.string())),
    customer_email: z.optional(z.nullable(z.string())),
    invitation_token_expires_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    claimed_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    revoked_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    seat_metadata: z.optional(z.nullable(z.record(z.string(), z.any()))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "member_id": "memberId",
        "customer_email": "customerEmail",
        "invitation_token_expires_at": "invitationTokenExpiresAt",
        "claimed_at": "claimedAt",
        "revoked_at": "revokedAt",
        "seat_metadata": "seatMetadata",
    });
}));
/** @internal */
exports.CustomerSeat$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    subscriptionId: z.optional(z.nullable(z.string())),
    orderId: z.optional(z.nullable(z.string())),
    status: seatstatus_js_1.SeatStatus$outboundSchema,
    customerId: z.optional(z.nullable(z.string())),
    memberId: z.optional(z.nullable(z.string())),
    member: z.optional(z.nullable(member_js_1.Member$outboundSchema)),
    email: z.optional(z.nullable(z.string())),
    customerEmail: z.optional(z.nullable(z.string())),
    invitationTokenExpiresAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    claimedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    revokedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    seatMetadata: z.optional(z.nullable(z.record(z.string(), z.any()))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        customerId: "customer_id",
        memberId: "member_id",
        customerEmail: "customer_email",
        invitationTokenExpiresAt: "invitation_token_expires_at",
        claimedAt: "claimed_at",
        revokedAt: "revoked_at",
        seatMetadata: "seat_metadata",
    });
}));
function customerSeatToJSON(customerSeat) {
    return JSON.stringify(exports.CustomerSeat$outboundSchema.parse(customerSeat));
}
function customerSeatFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerSeat$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeat' from JSON`);
}
