// @polar-sh/sdk/dist/commonjs/models/components/customerstatebenefitgrant.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerStateBenefitGrant$outboundSchema = exports.CustomerStateBenefitGrant$inboundSchema = exports.CustomerStateBenefitGrantProperties$outboundSchema = exports.CustomerStateBenefitGrantProperties$inboundSchema = void 0;
exports.customerStateBenefitGrantPropertiesToJSON = customerStateBenefitGrantPropertiesToJSON;
exports.customerStateBenefitGrantPropertiesFromJSON = customerStateBenefitGrantPropertiesFromJSON;
exports.customerStateBenefitGrantToJSON = customerStateBenefitGrantToJSON;
exports.customerStateBenefitGrantFromJSON = customerStateBenefitGrantFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const benefitgrantcustomproperties_js_1 = require("./benefitgrantcustomproperties.js");
const benefitgrantdiscordproperties_js_1 = require("./benefitgrantdiscordproperties.js");
const benefitgrantdownloadablesproperties_js_1 = require("./benefitgrantdownloadablesproperties.js");
const benefitgrantfeatureflagproperties_js_1 = require("./benefitgrantfeatureflagproperties.js");
const benefitgrantgithubrepositoryproperties_js_1 = require("./benefitgrantgithubrepositoryproperties.js");
const benefitgrantlicensekeysproperties_js_1 = require("./benefitgrantlicensekeysproperties.js");
const benefittype_js_1 = require("./benefittype.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
/** @internal */
exports.CustomerStateBenefitGrantProperties$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
    benefitgrantfeatureflagproperties_js_1.BenefitGrantFeatureFlagProperties$inboundSchema,
]);
/** @internal */
exports.CustomerStateBenefitGrantProperties$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
    benefitgrantfeatureflagproperties_js_1.BenefitGrantFeatureFlagProperties$outboundSchema,
]);
function customerStateBenefitGrantPropertiesToJSON(customerStateBenefitGrantProperties) {
    return JSON.stringify(exports.CustomerStateBenefitGrantProperties$outboundSchema.parse(customerStateBenefitGrantProperties));
}
function customerStateBenefitGrantPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerStateBenefitGrantProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateBenefitGrantProperties' from JSON`);
}
/** @internal */
exports.CustomerStateBenefitGrant$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    granted_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    benefit_id: z.string(),
    benefit_type: benefittype_js_1.BenefitType$inboundSchema,
    benefit_metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    properties: (0, smartUnion_js_1.smartUnion)([
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
        benefitgrantfeatureflagproperties_js_1.BenefitGrantFeatureFlagProperties$inboundSchema,
    ]),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "benefit_id": "benefitId",
        "benefit_type": "benefitType",
        "benefit_metadata": "benefitMetadata",
    });
}));
/** @internal */
exports.CustomerStateBenefitGrant$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    grantedAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    benefitId: z.string(),
    benefitType: benefittype_js_1.BenefitType$outboundSchema,
    benefitMetadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$outboundSchema),
    properties: (0, smartUnion_js_1.smartUnion)([
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
        benefitgrantfeatureflagproperties_js_1.BenefitGrantFeatureFlagProperties$outboundSchema,
    ]),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        benefitId: "benefit_id",
        benefitType: "benefit_type",
        benefitMetadata: "benefit_metadata",
    });
}));
function customerStateBenefitGrantToJSON(customerStateBenefitGrant) {
    return JSON.stringify(exports.CustomerStateBenefitGrant$outboundSchema.parse(customerStateBenefitGrant));
}
function customerStateBenefitGrantFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerStateBenefitGrant$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateBenefitGrant' from JSON`);
}
