// @polar-sh/sdk/dist/commonjs/models/components/customerindividual.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerIndividual$outboundSchema = exports.CustomerIndividual$inboundSchema = exports.CustomerIndividualTaxId$outboundSchema = exports.CustomerIndividualTaxId$inboundSchema = void 0;
exports.customerIndividualTaxIdToJSON = customerIndividualTaxIdToJSON;
exports.customerIndividualTaxIdFromJSON = customerIndividualTaxIdFromJSON;
exports.customerIndividualToJSON = customerIndividualToJSON;
exports.customerIndividualFromJSON = customerIndividualFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const address_js_1 = require("./address.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
const taxidformat_js_1 = require("./taxidformat.js");
/** @internal */
exports.CustomerIndividualTaxId$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema]);
/** @internal */
exports.CustomerIndividualTaxId$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema]);
function customerIndividualTaxIdToJSON(customerIndividualTaxId) {
    return JSON.stringify(exports.CustomerIndividualTaxId$outboundSchema.parse(customerIndividualTaxId));
}
function customerIndividualTaxIdFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerIndividualTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerIndividualTaxId' from JSON`);
}
/** @internal */
exports.CustomerIndividual$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    external_id: z.optional(z.nullable(z.string())),
    email: z.string(),
    email_verified: z.boolean(),
    type: z.literal("individual"),
    name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), taxidformat_js_1.TaxIDFormat$inboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
    deleted_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    avatar_url: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "deleted_at": "deletedAt",
        "avatar_url": "avatarUrl",
    });
}));
/** @internal */
exports.CustomerIndividual$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$outboundSchema),
    externalId: z.optional(z.nullable(z.string())),
    email: z.string(),
    emailVerified: z.boolean(),
    type: z.literal("individual"),
    name: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable((0, smartUnion_js_1.smartUnion)([z.string(), taxidformat_js_1.TaxIDFormat$outboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organizationId: z.string(),
    deletedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    avatarUrl: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        externalId: "external_id",
        emailVerified: "email_verified",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
        deletedAt: "deleted_at",
        avatarUrl: "avatar_url",
    });
}));
function customerIndividualToJSON(customerIndividual) {
    return JSON.stringify(exports.CustomerIndividual$outboundSchema.parse(customerIndividual));
}
function customerIndividualFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerIndividual$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerIndividual' from JSON`);
}
