// @polar-sh/sdk/dist/commonjs/models/components/customerorder.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrder$inboundSchema = void 0;
exports.customerOrderFromJSON = customerOrderFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const customerorderproduct_js_1 = require("./customerorderproduct.js");
const customerordersubscription_js_1 = require("./customerordersubscription.js");
const orderbillingreason_js_1 = require("./orderbillingreason.js");
const orderitemschema_js_1 = require("./orderitemschema.js");
const orderstatus_js_1 = require("./orderstatus.js");
/** @internal */
exports.CustomerOrder$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    status: orderstatus_js_1.OrderStatus$inboundSchema,
    paid: z.boolean(),
    subtotal_amount: z.int(),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.int(),
    total_amount: z.int(),
    applied_balance_amount: z.int(),
    due_amount: z.int(),
    refunded_amount: z.int(),
    refunded_tax_amount: z.int(),
    currency: z.string(),
    billing_reason: orderbillingreason_js_1.OrderBillingReason$inboundSchema,
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    invoice_number: z.string(),
    is_invoice_generated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customer_id: z.string(),
    product_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    subscription_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    product: z.nullable(customerorderproduct_js_1.CustomerOrderProduct$inboundSchema),
    subscription: z.nullable(customerordersubscription_js_1.CustomerOrderSubscription$inboundSchema),
    items: z.array(orderitemschema_js_1.OrderItemSchema$inboundSchema),
    description: z.string(),
    next_payment_attempt_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subtotal_amount": "subtotalAmount",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "applied_balance_amount": "appliedBalanceAmount",
        "due_amount": "dueAmount",
        "refunded_amount": "refundedAmount",
        "refunded_tax_amount": "refundedTaxAmount",
        "billing_reason": "billingReason",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "invoice_number": "invoiceNumber",
        "is_invoice_generated": "isInvoiceGenerated",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "next_payment_attempt_at": "nextPaymentAttemptAt",
    });
}));
function customerOrderFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerOrder$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrder' from JSON`);
}
