// @polar-sh/sdk/dist/commonjs/models/components/customerorganization.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrganization$inboundSchema = void 0;
exports.customerOrganizationFromJSON = customerOrganizationFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customerorganizationfeaturesettings_js_1 = require("./customerorganizationfeaturesettings.js");
const organizationcustomerportalsettings_js_1 = require("./organizationcustomerportalsettings.js");
const subscriptionprorationbehavior_js_1 = require("./subscriptionprorationbehavior.js");
/** @internal */
exports.CustomerOrganization$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()),
    proration_behavior: subscriptionprorationbehavior_js_1.SubscriptionProrationBehavior$inboundSchema,
    allow_customer_updates: z.boolean(),
    customer_portal_settings: organizationcustomerportalsettings_js_1.OrganizationCustomerPortalSettings$inboundSchema,
    organization_features: z.optional(customerorganizationfeaturesettings_js_1.CustomerOrganizationFeatureSettings$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "avatar_url": "avatarUrl",
        "proration_behavior": "prorationBehavior",
        "allow_customer_updates": "allowCustomerUpdates",
        "customer_portal_settings": "customerPortalSettings",
        "organization_features": "organizationFeatures",
    });
}));
function customerOrganizationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrganization' from JSON`);
}
