// @polar-sh/sdk/dist/commonjs/models/components/customerupdatedfields.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerUpdatedFields$inboundSchema = exports.CustomerUpdatedFieldsMetadata$inboundSchema = void 0;
exports.customerUpdatedFieldsMetadataFromJSON = customerUpdatedFieldsMetadataFromJSON;
exports.customerUpdatedFieldsFromJSON = customerUpdatedFieldsFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const addressdict_js_1 = require("./addressdict.js");
/** @internal */
exports.CustomerUpdatedFieldsMetadata$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()]);
function customerUpdatedFieldsMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerUpdatedFieldsMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerUpdatedFieldsMetadata' from JSON`);
}
/** @internal */
exports.CustomerUpdatedFields$inboundSchema = z.pipe(z.object({
    name: z.optional(z.nullable(z.string())),
    email: z.optional(z.nullable(z.string())),
    billing_address: z.optional(z.nullable(addressdict_js_1.AddressDict$inboundSchema)),
    tax_id: z.optional(z.nullable(z.string())),
    metadata: z.optional(z.nullable(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.boolean()])))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "billing_address": "billingAddress",
        "tax_id": "taxId",
    });
}));
function customerUpdatedFieldsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerUpdatedFields$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerUpdatedFields' from JSON`);
}
