// @polar-sh/sdk/dist/commonjs/models/components/customerindividualcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerIndividualCreate$outboundSchema = exports.CustomerIndividualCreateMetadata$outboundSchema = void 0;
exports.customerIndividualCreateMetadataToJSON = customerIndividualCreateMetadataToJSON;
exports.customerIndividualCreateToJSON = customerIndividualCreateToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const addressinput_js_1 = require("./addressinput.js");
const memberownercreate_js_1 = require("./memberownercreate.js");
/** @internal */
exports.CustomerIndividualCreateMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function customerIndividualCreateMetadataToJSON(customerIndividualCreateMetadata) {
    return JSON.stringify(exports.CustomerIndividualCreateMetadata$outboundSchema.parse(customerIndividualCreateMetadata));
}
/** @internal */
exports.CustomerIndividualCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    externalId: z.optional(z.nullable(z.string())),
    name: z.optional(z.nullable(z.string())),
    billingAddress: z.optional(z.nullable(addressinput_js_1.AddressInput$outboundSchema)),
    taxId: z.optional(z.nullable(z.string())),
    locale: z.optional(z.nullable(z.string())),
    organizationId: z.optional(z.nullable(z.string())),
    owner: z.optional(z.nullable(memberownercreate_js_1.MemberOwnerCreate$outboundSchema)),
    type: z._default(z.literal("individual"), "individual"),
    email: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        externalId: "external_id",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
    });
}));
function customerIndividualCreateToJSON(customerIndividualCreate) {
    return JSON.stringify(exports.CustomerIndividualCreate$outboundSchema.parse(customerIndividualCreate));
}
