// @polar-sh/sdk/dist/commonjs/models/components/discountpercentagerepeatduration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountPercentageRepeatDuration$inboundSchema = void 0;
exports.discountPercentageRepeatDurationFromJSON = discountPercentageRepeatDurationFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const discountduration_js_1 = require("./discountduration.js");
const discountproduct_js_1 = require("./discountproduct.js");
const discounttype_js_1 = require("./discounttype.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
/** @internal */
exports.DiscountPercentageRepeatDuration$inboundSchema = z.pipe(z.object({
    duration: discountduration_js_1.DiscountDuration$inboundSchema,
    duration_in_months: z.int(),
    type: discounttype_js_1.DiscountType$inboundSchema,
    basis_points: z.int(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    max_redemptions: z.nullable(z.int()),
    redemptions_count: z.int(),
    organization_id: z.string(),
    products: z.array(discountproduct_js_1.DiscountProduct$inboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
}));
function discountPercentageRepeatDurationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountPercentageRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageRepeatDuration' from JSON`);
}
