// @polar-sh/sdk/dist/commonjs/models/components/filedownload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileDownload$inboundSchema = void 0;
exports.fileDownloadFromJSON = fileDownloadFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const fileservicetypes_js_1 = require("./fileservicetypes.js");
const s3downloadurl_js_1 = require("./s3downloadurl.js");
/** @internal */
exports.FileDownload$inboundSchema = z.pipe(z.object({
    id: z.string(),
    organization_id: z.string(),
    name: z.string(),
    path: z.string(),
    mime_type: z.string(),
    size: z.int(),
    storage_version: z.nullable(z.string()),
    checksum_etag: z.nullable(z.string()),
    checksum_sha256_base64: z.nullable(z.string()),
    checksum_sha256_hex: z.nullable(z.string()),
    last_modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    download: s3downloadurl_js_1.S3DownloadURL$inboundSchema,
    version: z.nullable(z.string()),
    is_uploaded: z.boolean(),
    service: fileservicetypes_js_1.FileServiceTypes$inboundSchema,
    size_readable: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "storage_version": "storageVersion",
        "checksum_etag": "checksumEtag",
        "checksum_sha256_base64": "checksumSha256Base64",
        "checksum_sha256_hex": "checksumSha256Hex",
        "last_modified_at": "lastModifiedAt",
        "is_uploaded": "isUploaded",
        "size_readable": "sizeReadable",
    });
}));
function fileDownloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FileDownload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FileDownload' from JSON`);
}
