// @polar-sh/sdk/dist/commonjs/models/components/introspecttokenresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntrospectTokenResponse$inboundSchema = exports.TokenType$inboundSchema = exports.TokenType = void 0;
exports.introspectTokenResponseFromJSON = introspectTokenResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const openEnums = __importStar(require("../../types/enums.js"));
const subtype_js_1 = require("./subtype.js");
exports.TokenType = {
    AccessToken: "access_token",
    RefreshToken: "refresh_token",
};
/** @internal */
exports.TokenType$inboundSchema = openEnums.inboundSchema(exports.TokenType);
/** @internal */
exports.IntrospectTokenResponse$inboundSchema = z.pipe(z.object({
    active: z.boolean(),
    client_id: z.string(),
    token_type: exports.TokenType$inboundSchema,
    scope: z.string(),
    sub_type: subtype_js_1.SubType$inboundSchema,
    sub: z.string(),
    aud: z.string(),
    iss: z.string(),
    exp: z.int(),
    iat: z.int(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_id": "clientId",
        "token_type": "tokenType",
        "sub_type": "subType",
    });
}));
function introspectTokenResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.IntrospectTokenResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'IntrospectTokenResponse' from JSON`);
}
