// @polar-sh/sdk/dist/commonjs/models/components/llmmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMMetadata$outboundSchema = exports.LLMMetadata$inboundSchema = void 0;
exports.llmMetadataToJSON = llmMetadataToJSON;
exports.llmMetadataFromJSON = llmMetadataFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.LLMMetadata$inboundSchema = z
    .pipe(z.object({
    vendor: z.string(),
    model: z.string(),
    prompt: z.optional(z.nullable(z.string())),
    response: z.optional(z.nullable(z.string())),
    input_tokens: z.int(),
    cached_input_tokens: z.optional(z.int()),
    output_tokens: z.int(),
    total_tokens: z.int(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "input_tokens": "inputTokens",
        "cached_input_tokens": "cachedInputTokens",
        "output_tokens": "outputTokens",
        "total_tokens": "totalTokens",
    });
}));
/** @internal */
exports.LLMMetadata$outboundSchema = z.pipe(z.object({
    vendor: z.string(),
    model: z.string(),
    prompt: z.optional(z.nullable(z.string())),
    response: z.optional(z.nullable(z.string())),
    inputTokens: z.int(),
    cachedInputTokens: z.optional(z.int()),
    outputTokens: z.int(),
    totalTokens: z.int(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        inputTokens: "input_tokens",
        cachedInputTokens: "cached_input_tokens",
        outputTokens: "output_tokens",
        totalTokens: "total_tokens",
    });
}));
function llmMetadataToJSON(llmMetadata) {
    return JSON.stringify(exports.LLMMetadata$outboundSchema.parse(llmMetadata));
}
function llmMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LLMMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LLMMetadata' from JSON`);
}
