// @polar-sh/sdk/dist/commonjs/models/components/meterupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeterUpdate$outboundSchema = exports.Aggregation$outboundSchema = exports.MeterUpdateMetadata$outboundSchema = void 0;
exports.meterUpdateMetadataToJSON = meterUpdateMetadataToJSON;
exports.aggregationToJSON = aggregationToJSON;
exports.meterUpdateToJSON = meterUpdateToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const countaggregation_js_1 = require("./countaggregation.js");
const filter_js_1 = require("./filter.js");
const propertyaggregation_js_1 = require("./propertyaggregation.js");
const uniqueaggregation_js_1 = require("./uniqueaggregation.js");
/** @internal */
exports.MeterUpdateMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function meterUpdateMetadataToJSON(meterUpdateMetadata) {
    return JSON.stringify(exports.MeterUpdateMetadata$outboundSchema.parse(meterUpdateMetadata));
}
/** @internal */
exports.Aggregation$outboundSchema = z.union([
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
    countaggregation_js_1.CountAggregation$outboundSchema,
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
    uniqueaggregation_js_1.UniqueAggregation$outboundSchema,
]);
function aggregationToJSON(aggregation) {
    return JSON.stringify(exports.Aggregation$outboundSchema.parse(aggregation));
}
/** @internal */
exports.MeterUpdate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.optional(z.nullable(z.string())),
    filter: z.optional(z.nullable(filter_js_1.Filter$outboundSchema)),
    aggregation: z.optional(z.nullable(z.union([
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
        countaggregation_js_1.CountAggregation$outboundSchema,
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
        uniqueaggregation_js_1.UniqueAggregation$outboundSchema,
    ]))),
    isArchived: z.optional(z.nullable(z.boolean())),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isArchived: "is_archived",
    });
}));
function meterUpdateToJSON(meterUpdate) {
    return JSON.stringify(exports.MeterUpdate$outboundSchema.parse(meterUpdate));
}
