// @polar-sh/sdk/dist/commonjs/models/components/metricperiod.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricPeriod$inboundSchema = exports.Cashflow$inboundSchema = exports.GrossMarginPercentage$inboundSchema = exports.GrossMargin$inboundSchema = exports.Ltv$inboundSchema = exports.CheckoutsConversion$inboundSchema = exports.CanceledSubscriptionsOther$inboundSchema = exports.CanceledSubscriptionsUnused$inboundSchema = exports.CanceledSubscriptionsTooExpensive$inboundSchema = exports.CanceledSubscriptionsTooComplex$inboundSchema = exports.CanceledSubscriptionsSwitchedService$inboundSchema = exports.CanceledSubscriptionsMissingFeatures$inboundSchema = exports.CanceledSubscriptionsLowQuality$inboundSchema = exports.CanceledSubscriptionsCustomerService$inboundSchema = exports.CanceledSubscriptions$inboundSchema = exports.RenewedSubscriptionsNetRevenue$inboundSchema = exports.RenewedSubscriptionsRevenue$inboundSchema = exports.RenewedSubscriptions$inboundSchema = exports.NewSubscriptionsNetRevenue$inboundSchema = exports.NewSubscriptionsRevenue$inboundSchema = exports.NewSubscriptions$inboundSchema = exports.OneTimeProductsNetRevenue$inboundSchema = exports.OneTimeProductsRevenue$inboundSchema = exports.OneTimeProducts$inboundSchema = exports.ActiveUserByEvent$inboundSchema = exports.CostPerUser$inboundSchema = exports.NetAverageOrderValue$inboundSchema = exports.AverageOrderValue$inboundSchema = exports.CumulativeCosts$inboundSchema = exports.Costs$inboundSchema = exports.NetCumulativeRevenue$inboundSchema = exports.CumulativeRevenue$inboundSchema = exports.NetRevenue$inboundSchema = exports.Revenue$inboundSchema = exports.Orders$inboundSchema = exports.ChurnRate$inboundSchema = exports.ChurnedSubscriptions$inboundSchema = exports.SucceededCheckouts$inboundSchema = exports.Checkouts$inboundSchema = exports.AverageRevenuePerUser$inboundSchema = exports.CommittedMonthlyRecurringRevenue$inboundSchema = exports.MonthlyRecurringRevenue$inboundSchema = exports.CommittedSubscriptions$inboundSchema = exports.ActiveSubscriptions$inboundSchema = void 0;
exports.activeSubscriptionsFromJSON = activeSubscriptionsFromJSON;
exports.committedSubscriptionsFromJSON = committedSubscriptionsFromJSON;
exports.monthlyRecurringRevenueFromJSON = monthlyRecurringRevenueFromJSON;
exports.committedMonthlyRecurringRevenueFromJSON = committedMonthlyRecurringRevenueFromJSON;
exports.averageRevenuePerUserFromJSON = averageRevenuePerUserFromJSON;
exports.checkoutsFromJSON = checkoutsFromJSON;
exports.succeededCheckoutsFromJSON = succeededCheckoutsFromJSON;
exports.churnedSubscriptionsFromJSON = churnedSubscriptionsFromJSON;
exports.churnRateFromJSON = churnRateFromJSON;
exports.ordersFromJSON = ordersFromJSON;
exports.revenueFromJSON = revenueFromJSON;
exports.netRevenueFromJSON = netRevenueFromJSON;
exports.cumulativeRevenueFromJSON = cumulativeRevenueFromJSON;
exports.netCumulativeRevenueFromJSON = netCumulativeRevenueFromJSON;
exports.costsFromJSON = costsFromJSON;
exports.cumulativeCostsFromJSON = cumulativeCostsFromJSON;
exports.averageOrderValueFromJSON = averageOrderValueFromJSON;
exports.netAverageOrderValueFromJSON = netAverageOrderValueFromJSON;
exports.costPerUserFromJSON = costPerUserFromJSON;
exports.activeUserByEventFromJSON = activeUserByEventFromJSON;
exports.oneTimeProductsFromJSON = oneTimeProductsFromJSON;
exports.oneTimeProductsRevenueFromJSON = oneTimeProductsRevenueFromJSON;
exports.oneTimeProductsNetRevenueFromJSON = oneTimeProductsNetRevenueFromJSON;
exports.newSubscriptionsFromJSON = newSubscriptionsFromJSON;
exports.newSubscriptionsRevenueFromJSON = newSubscriptionsRevenueFromJSON;
exports.newSubscriptionsNetRevenueFromJSON = newSubscriptionsNetRevenueFromJSON;
exports.renewedSubscriptionsFromJSON = renewedSubscriptionsFromJSON;
exports.renewedSubscriptionsRevenueFromJSON = renewedSubscriptionsRevenueFromJSON;
exports.renewedSubscriptionsNetRevenueFromJSON = renewedSubscriptionsNetRevenueFromJSON;
exports.canceledSubscriptionsFromJSON = canceledSubscriptionsFromJSON;
exports.canceledSubscriptionsCustomerServiceFromJSON = canceledSubscriptionsCustomerServiceFromJSON;
exports.canceledSubscriptionsLowQualityFromJSON = canceledSubscriptionsLowQualityFromJSON;
exports.canceledSubscriptionsMissingFeaturesFromJSON = canceledSubscriptionsMissingFeaturesFromJSON;
exports.canceledSubscriptionsSwitchedServiceFromJSON = canceledSubscriptionsSwitchedServiceFromJSON;
exports.canceledSubscriptionsTooComplexFromJSON = canceledSubscriptionsTooComplexFromJSON;
exports.canceledSubscriptionsTooExpensiveFromJSON = canceledSubscriptionsTooExpensiveFromJSON;
exports.canceledSubscriptionsUnusedFromJSON = canceledSubscriptionsUnusedFromJSON;
exports.canceledSubscriptionsOtherFromJSON = canceledSubscriptionsOtherFromJSON;
exports.checkoutsConversionFromJSON = checkoutsConversionFromJSON;
exports.ltvFromJSON = ltvFromJSON;
exports.grossMarginFromJSON = grossMarginFromJSON;
exports.grossMarginPercentageFromJSON = grossMarginPercentageFromJSON;
exports.cashflowFromJSON = cashflowFromJSON;
exports.metricPeriodFromJSON = metricPeriodFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
/** @internal */
exports.ActiveSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function activeSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ActiveSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActiveSubscriptions' from JSON`);
}
/** @internal */
exports.CommittedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function committedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CommittedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CommittedSubscriptions' from JSON`);
}
/** @internal */
exports.MonthlyRecurringRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function monthlyRecurringRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MonthlyRecurringRevenue' from JSON`);
}
/** @internal */
exports.CommittedMonthlyRecurringRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function committedMonthlyRecurringRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CommittedMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CommittedMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
exports.AverageRevenuePerUser$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function averageRevenuePerUserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AverageRevenuePerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AverageRevenuePerUser' from JSON`);
}
/** @internal */
exports.Checkouts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function checkoutsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Checkouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Checkouts' from JSON`);
}
/** @internal */
exports.SucceededCheckouts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function succeededCheckoutsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SucceededCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SucceededCheckouts' from JSON`);
}
/** @internal */
exports.ChurnedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function churnedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ChurnedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChurnedSubscriptions' from JSON`);
}
/** @internal */
exports.ChurnRate$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function churnRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ChurnRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChurnRate' from JSON`);
}
/** @internal */
exports.Orders$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function ordersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Orders$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Orders' from JSON`);
}
/** @internal */
exports.Revenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function revenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Revenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Revenue' from JSON`);
}
/** @internal */
exports.NetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function netRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetRevenue' from JSON`);
}
/** @internal */
exports.CumulativeRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function cumulativeRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CumulativeRevenue' from JSON`);
}
/** @internal */
exports.NetCumulativeRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function netCumulativeRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NetCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetCumulativeRevenue' from JSON`);
}
/** @internal */
exports.Costs$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function costsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Costs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Costs' from JSON`);
}
/** @internal */
exports.CumulativeCosts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function cumulativeCostsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CumulativeCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CumulativeCosts' from JSON`);
}
/** @internal */
exports.AverageOrderValue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function averageOrderValueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AverageOrderValue' from JSON`);
}
/** @internal */
exports.NetAverageOrderValue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function netAverageOrderValueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NetAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetAverageOrderValue' from JSON`);
}
/** @internal */
exports.CostPerUser$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function costPerUserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CostPerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CostPerUser' from JSON`);
}
/** @internal */
exports.ActiveUserByEvent$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function activeUserByEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ActiveUserByEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActiveUserByEvent' from JSON`);
}
/** @internal */
exports.OneTimeProducts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function oneTimeProductsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OneTimeProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProducts' from JSON`);
}
/** @internal */
exports.OneTimeProductsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function oneTimeProductsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OneTimeProductsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProductsRevenue' from JSON`);
}
/** @internal */
exports.OneTimeProductsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function oneTimeProductsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OneTimeProductsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProductsNetRevenue' from JSON`);
}
/** @internal */
exports.NewSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function newSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptions' from JSON`);
}
/** @internal */
exports.NewSubscriptionsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function newSubscriptionsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptionsRevenue' from JSON`);
}
/** @internal */
exports.NewSubscriptionsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function newSubscriptionsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NewSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
exports.RenewedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function renewedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RenewedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptions' from JSON`);
}
/** @internal */
exports.RenewedSubscriptionsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function renewedSubscriptionsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RenewedSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptionsRevenue' from JSON`);
}
/** @internal */
exports.RenewedSubscriptionsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function renewedSubscriptionsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RenewedSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
exports.CanceledSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptions' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsCustomerService$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsCustomerServiceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsCustomerService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsCustomerService' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsLowQuality$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsLowQualityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsLowQuality$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsLowQuality' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsMissingFeatures$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsMissingFeaturesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsMissingFeatures$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsMissingFeatures' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsSwitchedService$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsSwitchedServiceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsSwitchedService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsSwitchedService' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsTooComplex$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsTooComplexFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsTooComplex$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsTooComplex' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsTooExpensive$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsTooExpensiveFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsTooExpensive$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsTooExpensive' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsUnused$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsUnusedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsUnused$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsUnused' from JSON`);
}
/** @internal */
exports.CanceledSubscriptionsOther$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function canceledSubscriptionsOtherFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CanceledSubscriptionsOther$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsOther' from JSON`);
}
/** @internal */
exports.CheckoutsConversion$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function checkoutsConversionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutsConversion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsConversion' from JSON`);
}
/** @internal */
exports.Ltv$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function ltvFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Ltv$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ltv' from JSON`);
}
/** @internal */
exports.GrossMargin$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function grossMarginFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GrossMargin$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrossMargin' from JSON`);
}
/** @internal */
exports.GrossMarginPercentage$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function grossMarginPercentageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GrossMarginPercentage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrossMarginPercentage' from JSON`);
}
/** @internal */
exports.Cashflow$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function cashflowFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Cashflow$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Cashflow' from JSON`);
}
/** @internal */
exports.MetricPeriod$inboundSchema = z.pipe(z.object({
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    active_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    committed_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    monthly_recurring_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    committed_monthly_recurring_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    average_revenue_per_user: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    checkouts: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    succeeded_checkouts: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    churned_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    churn_rate: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    orders: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cumulative_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_cumulative_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    costs: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cumulative_costs: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    average_order_value: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_average_order_value: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cost_per_user: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    active_user_by_event: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_customer_service: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_low_quality: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_missing_features: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_switched_service: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_too_complex: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_too_expensive: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_unused: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_other: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    checkouts_conversion: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    ltv: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    gross_margin: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    gross_margin_percentage: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cashflow: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "active_subscriptions": "activeSubscriptions",
        "committed_subscriptions": "committedSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "succeeded_checkouts": "succeededCheckouts",
        "churned_subscriptions": "churnedSubscriptions",
        "churn_rate": "churnRate",
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "checkouts_conversion": "checkoutsConversion",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
function metricPeriodFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricPeriod$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricPeriod' from JSON`);
}
