// @polar-sh/sdk/dist/commonjs/models/components/metricstotals.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricsTotals$inboundSchema = exports.MetricsTotalsCashflow$inboundSchema = exports.MetricsTotalsGrossMarginPercentage$inboundSchema = exports.MetricsTotalsGrossMargin$inboundSchema = exports.MetricsTotalsLtv$inboundSchema = exports.MetricsTotalsCheckoutsConversion$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsOther$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsUnused$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsTooExpensive$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsTooComplex$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsSwitchedService$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsMissingFeatures$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsLowQuality$inboundSchema = exports.MetricsTotalsCanceledSubscriptionsCustomerService$inboundSchema = exports.MetricsTotalsCanceledSubscriptions$inboundSchema = exports.MetricsTotalsRenewedSubscriptionsNetRevenue$inboundSchema = exports.MetricsTotalsRenewedSubscriptionsRevenue$inboundSchema = exports.MetricsTotalsRenewedSubscriptions$inboundSchema = exports.MetricsTotalsNewSubscriptionsNetRevenue$inboundSchema = exports.MetricsTotalsNewSubscriptionsRevenue$inboundSchema = exports.MetricsTotalsNewSubscriptions$inboundSchema = exports.MetricsTotalsOneTimeProductsNetRevenue$inboundSchema = exports.MetricsTotalsOneTimeProductsRevenue$inboundSchema = exports.MetricsTotalsOneTimeProducts$inboundSchema = exports.MetricsTotalsActiveUserByEvent$inboundSchema = exports.MetricsTotalsCostPerUser$inboundSchema = exports.MetricsTotalsNetAverageOrderValue$inboundSchema = exports.MetricsTotalsAverageOrderValue$inboundSchema = exports.MetricsTotalsCumulativeCosts$inboundSchema = exports.MetricsTotalsCosts$inboundSchema = exports.MetricsTotalsNetCumulativeRevenue$inboundSchema = exports.MetricsTotalsCumulativeRevenue$inboundSchema = exports.MetricsTotalsNetRevenue$inboundSchema = exports.MetricsTotalsRevenue$inboundSchema = exports.MetricsTotalsOrders$inboundSchema = exports.MetricsTotalsChurnRate$inboundSchema = exports.MetricsTotalsChurnedSubscriptions$inboundSchema = exports.MetricsTotalsSucceededCheckouts$inboundSchema = exports.MetricsTotalsCheckouts$inboundSchema = exports.MetricsTotalsAverageRevenuePerUser$inboundSchema = exports.MetricsTotalsCommittedMonthlyRecurringRevenue$inboundSchema = exports.MetricsTotalsMonthlyRecurringRevenue$inboundSchema = exports.MetricsTotalsCommittedSubscriptions$inboundSchema = exports.MetricsTotalsActiveSubscriptions$inboundSchema = void 0;
exports.metricsTotalsActiveSubscriptionsFromJSON = metricsTotalsActiveSubscriptionsFromJSON;
exports.metricsTotalsCommittedSubscriptionsFromJSON = metricsTotalsCommittedSubscriptionsFromJSON;
exports.metricsTotalsMonthlyRecurringRevenueFromJSON = metricsTotalsMonthlyRecurringRevenueFromJSON;
exports.metricsTotalsCommittedMonthlyRecurringRevenueFromJSON = metricsTotalsCommittedMonthlyRecurringRevenueFromJSON;
exports.metricsTotalsAverageRevenuePerUserFromJSON = metricsTotalsAverageRevenuePerUserFromJSON;
exports.metricsTotalsCheckoutsFromJSON = metricsTotalsCheckoutsFromJSON;
exports.metricsTotalsSucceededCheckoutsFromJSON = metricsTotalsSucceededCheckoutsFromJSON;
exports.metricsTotalsChurnedSubscriptionsFromJSON = metricsTotalsChurnedSubscriptionsFromJSON;
exports.metricsTotalsChurnRateFromJSON = metricsTotalsChurnRateFromJSON;
exports.metricsTotalsOrdersFromJSON = metricsTotalsOrdersFromJSON;
exports.metricsTotalsRevenueFromJSON = metricsTotalsRevenueFromJSON;
exports.metricsTotalsNetRevenueFromJSON = metricsTotalsNetRevenueFromJSON;
exports.metricsTotalsCumulativeRevenueFromJSON = metricsTotalsCumulativeRevenueFromJSON;
exports.metricsTotalsNetCumulativeRevenueFromJSON = metricsTotalsNetCumulativeRevenueFromJSON;
exports.metricsTotalsCostsFromJSON = metricsTotalsCostsFromJSON;
exports.metricsTotalsCumulativeCostsFromJSON = metricsTotalsCumulativeCostsFromJSON;
exports.metricsTotalsAverageOrderValueFromJSON = metricsTotalsAverageOrderValueFromJSON;
exports.metricsTotalsNetAverageOrderValueFromJSON = metricsTotalsNetAverageOrderValueFromJSON;
exports.metricsTotalsCostPerUserFromJSON = metricsTotalsCostPerUserFromJSON;
exports.metricsTotalsActiveUserByEventFromJSON = metricsTotalsActiveUserByEventFromJSON;
exports.metricsTotalsOneTimeProductsFromJSON = metricsTotalsOneTimeProductsFromJSON;
exports.metricsTotalsOneTimeProductsRevenueFromJSON = metricsTotalsOneTimeProductsRevenueFromJSON;
exports.metricsTotalsOneTimeProductsNetRevenueFromJSON = metricsTotalsOneTimeProductsNetRevenueFromJSON;
exports.metricsTotalsNewSubscriptionsFromJSON = metricsTotalsNewSubscriptionsFromJSON;
exports.metricsTotalsNewSubscriptionsRevenueFromJSON = metricsTotalsNewSubscriptionsRevenueFromJSON;
exports.metricsTotalsNewSubscriptionsNetRevenueFromJSON = metricsTotalsNewSubscriptionsNetRevenueFromJSON;
exports.metricsTotalsRenewedSubscriptionsFromJSON = metricsTotalsRenewedSubscriptionsFromJSON;
exports.metricsTotalsRenewedSubscriptionsRevenueFromJSON = metricsTotalsRenewedSubscriptionsRevenueFromJSON;
exports.metricsTotalsRenewedSubscriptionsNetRevenueFromJSON = metricsTotalsRenewedSubscriptionsNetRevenueFromJSON;
exports.metricsTotalsCanceledSubscriptionsFromJSON = metricsTotalsCanceledSubscriptionsFromJSON;
exports.metricsTotalsCanceledSubscriptionsCustomerServiceFromJSON = metricsTotalsCanceledSubscriptionsCustomerServiceFromJSON;
exports.metricsTotalsCanceledSubscriptionsLowQualityFromJSON = metricsTotalsCanceledSubscriptionsLowQualityFromJSON;
exports.metricsTotalsCanceledSubscriptionsMissingFeaturesFromJSON = metricsTotalsCanceledSubscriptionsMissingFeaturesFromJSON;
exports.metricsTotalsCanceledSubscriptionsSwitchedServiceFromJSON = metricsTotalsCanceledSubscriptionsSwitchedServiceFromJSON;
exports.metricsTotalsCanceledSubscriptionsTooComplexFromJSON = metricsTotalsCanceledSubscriptionsTooComplexFromJSON;
exports.metricsTotalsCanceledSubscriptionsTooExpensiveFromJSON = metricsTotalsCanceledSubscriptionsTooExpensiveFromJSON;
exports.metricsTotalsCanceledSubscriptionsUnusedFromJSON = metricsTotalsCanceledSubscriptionsUnusedFromJSON;
exports.metricsTotalsCanceledSubscriptionsOtherFromJSON = metricsTotalsCanceledSubscriptionsOtherFromJSON;
exports.metricsTotalsCheckoutsConversionFromJSON = metricsTotalsCheckoutsConversionFromJSON;
exports.metricsTotalsLtvFromJSON = metricsTotalsLtvFromJSON;
exports.metricsTotalsGrossMarginFromJSON = metricsTotalsGrossMarginFromJSON;
exports.metricsTotalsGrossMarginPercentageFromJSON = metricsTotalsGrossMarginPercentageFromJSON;
exports.metricsTotalsCashflowFromJSON = metricsTotalsCashflowFromJSON;
exports.metricsTotalsFromJSON = metricsTotalsFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
/** @internal */
exports.MetricsTotalsActiveSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsActiveSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsActiveSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsActiveSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsCommittedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCommittedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCommittedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCommittedSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsMonthlyRecurringRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsMonthlyRecurringRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsCommittedMonthlyRecurringRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCommittedMonthlyRecurringRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCommittedMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCommittedMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsAverageRevenuePerUser$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsAverageRevenuePerUserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsAverageRevenuePerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsAverageRevenuePerUser' from JSON`);
}
/** @internal */
exports.MetricsTotalsCheckouts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCheckoutsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCheckouts' from JSON`);
}
/** @internal */
exports.MetricsTotalsSucceededCheckouts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsSucceededCheckoutsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsSucceededCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsSucceededCheckouts' from JSON`);
}
/** @internal */
exports.MetricsTotalsChurnedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsChurnedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsChurnedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsChurnedSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsChurnRate$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsChurnRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsChurnRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsChurnRate' from JSON`);
}
/** @internal */
exports.MetricsTotalsOrders$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsOrdersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsOrders$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOrders' from JSON`);
}
/** @internal */
exports.MetricsTotalsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsCumulativeRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCumulativeRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCumulativeRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsNetCumulativeRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsNetCumulativeRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNetCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetCumulativeRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsCosts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCostsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCosts' from JSON`);
}
/** @internal */
exports.MetricsTotalsCumulativeCosts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCumulativeCostsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCumulativeCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCumulativeCosts' from JSON`);
}
/** @internal */
exports.MetricsTotalsAverageOrderValue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsAverageOrderValueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsAverageOrderValue' from JSON`);
}
/** @internal */
exports.MetricsTotalsNetAverageOrderValue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsNetAverageOrderValueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNetAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetAverageOrderValue' from JSON`);
}
/** @internal */
exports.MetricsTotalsCostPerUser$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCostPerUserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCostPerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCostPerUser' from JSON`);
}
/** @internal */
exports.MetricsTotalsActiveUserByEvent$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsActiveUserByEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsActiveUserByEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsActiveUserByEvent' from JSON`);
}
/** @internal */
exports.MetricsTotalsOneTimeProducts$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsOneTimeProductsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsOneTimeProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProducts' from JSON`);
}
/** @internal */
exports.MetricsTotalsOneTimeProductsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsOneTimeProductsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsOneTimeProductsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProductsRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsOneTimeProductsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function metricsTotalsOneTimeProductsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsOneTimeProductsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProductsNetRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsNewSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsNewSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNewSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsNewSubscriptionsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsNewSubscriptionsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNewSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptionsRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsNewSubscriptionsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function metricsTotalsNewSubscriptionsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsNewSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsRenewedSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsRenewedSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsRenewedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsRenewedSubscriptionsRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsRenewedSubscriptionsRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsRenewedSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptionsRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsRenewedSubscriptionsNetRevenue$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsRenewedSubscriptionsNetRevenueFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsRenewedSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptions$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptions' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsCustomerService$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsCustomerServiceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsCustomerService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsCustomerService' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsLowQuality$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsLowQualityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsLowQuality$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsLowQuality' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsMissingFeatures$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsMissingFeaturesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsMissingFeatures$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsMissingFeatures' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsSwitchedService$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsSwitchedServiceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsSwitchedService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsSwitchedService' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsTooComplex$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsTooComplexFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsTooComplex$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsTooComplex' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsTooExpensive$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsTooExpensiveFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsTooExpensive$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsTooExpensive' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsUnused$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCanceledSubscriptionsUnusedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsUnused$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsUnused' from JSON`);
}
/** @internal */
exports.MetricsTotalsCanceledSubscriptionsOther$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.int(),
    z.number(),
]);
function metricsTotalsCanceledSubscriptionsOtherFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCanceledSubscriptionsOther$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsOther' from JSON`);
}
/** @internal */
exports.MetricsTotalsCheckoutsConversion$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCheckoutsConversionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCheckoutsConversion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCheckoutsConversion' from JSON`);
}
/** @internal */
exports.MetricsTotalsLtv$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsLtvFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsLtv$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsLtv' from JSON`);
}
/** @internal */
exports.MetricsTotalsGrossMargin$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsGrossMarginFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsGrossMargin$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsGrossMargin' from JSON`);
}
/** @internal */
exports.MetricsTotalsGrossMarginPercentage$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsGrossMarginPercentageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsGrossMarginPercentage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsGrossMarginPercentage' from JSON`);
}
/** @internal */
exports.MetricsTotalsCashflow$inboundSchema = (0, smartUnion_js_1.smartUnion)([z.int(), z.number()]);
function metricsTotalsCashflowFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotalsCashflow$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCashflow' from JSON`);
}
/** @internal */
exports.MetricsTotals$inboundSchema = z.pipe(z.object({
    active_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    committed_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    monthly_recurring_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    committed_monthly_recurring_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    average_revenue_per_user: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    checkouts: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    succeeded_checkouts: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    churned_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    churn_rate: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    orders: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cumulative_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_cumulative_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    costs: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cumulative_costs: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    average_order_value: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    net_average_order_value: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cost_per_user: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    active_user_by_event: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    one_time_products_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    new_subscriptions_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    renewed_subscriptions_net_revenue: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_customer_service: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_low_quality: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_missing_features: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_switched_service: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_too_complex: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_too_expensive: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_unused: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    canceled_subscriptions_other: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    checkouts_conversion: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    ltv: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    gross_margin: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    gross_margin_percentage: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
    cashflow: z.optional(z.nullable((0, smartUnion_js_1.smartUnion)([z.int(), z.number()]))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "active_subscriptions": "activeSubscriptions",
        "committed_subscriptions": "committedSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "succeeded_checkouts": "succeededCheckouts",
        "churned_subscriptions": "churnedSubscriptions",
        "churn_rate": "churnRate",
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "checkouts_conversion": "checkoutsConversion",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
function metricsTotalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MetricsTotals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotals' from JSON`);
}
