// @polar-sh/sdk/dist/commonjs/models/components/order.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order$outboundSchema = exports.Order$inboundSchema = exports.OrderDiscount$outboundSchema = exports.OrderDiscount$inboundSchema = exports.OrderCustomFieldData$outboundSchema = exports.OrderCustomFieldData$inboundSchema = void 0;
exports.orderCustomFieldDataToJSON = orderCustomFieldDataToJSON;
exports.orderCustomFieldDataFromJSON = orderCustomFieldDataFromJSON;
exports.orderDiscountToJSON = orderDiscountToJSON;
exports.orderDiscountFromJSON = orderDiscountFromJSON;
exports.orderToJSON = orderToJSON;
exports.orderFromJSON = orderFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const address_js_1 = require("./address.js");
const discountfixedonceforeverdurationbase_js_1 = require("./discountfixedonceforeverdurationbase.js");
const discountfixedrepeatdurationbase_js_1 = require("./discountfixedrepeatdurationbase.js");
const discountpercentageonceforeverdurationbase_js_1 = require("./discountpercentageonceforeverdurationbase.js");
const discountpercentagerepeatdurationbase_js_1 = require("./discountpercentagerepeatdurationbase.js");
const metadataoutputtype_js_1 = require("./metadataoutputtype.js");
const orderbillingreason_js_1 = require("./orderbillingreason.js");
const ordercustomer_js_1 = require("./ordercustomer.js");
const orderitemschema_js_1 = require("./orderitemschema.js");
const orderproduct_js_1 = require("./orderproduct.js");
const orderstatus_js_1 = require("./orderstatus.js");
const ordersubscription_js_1 = require("./ordersubscription.js");
/** @internal */
exports.OrderCustomFieldData$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
exports.OrderCustomFieldData$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
function orderCustomFieldDataToJSON(orderCustomFieldData) {
    return JSON.stringify(exports.OrderCustomFieldData$outboundSchema.parse(orderCustomFieldData));
}
function orderCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderCustomFieldData' from JSON`);
}
/** @internal */
exports.OrderDiscount$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
]);
/** @internal */
exports.OrderDiscount$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
]);
function orderDiscountToJSON(orderDiscount) {
    return JSON.stringify(exports.OrderDiscount$outboundSchema.parse(orderDiscount));
}
function orderDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderDiscount' from JSON`);
}
/** @internal */
exports.Order$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    status: orderstatus_js_1.OrderStatus$inboundSchema,
    paid: z.boolean(),
    subtotal_amount: z.int(),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.int(),
    total_amount: z.int(),
    applied_balance_amount: z.int(),
    due_amount: z.int(),
    refunded_amount: z.int(),
    refunded_tax_amount: z.int(),
    currency: z.string(),
    billing_reason: orderbillingreason_js_1.OrderBillingReason$inboundSchema,
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(address_js_1.Address$inboundSchema),
    invoice_number: z.string(),
    is_invoice_generated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customer_id: z.string(),
    product_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    subscription_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$inboundSchema),
    custom_field_data: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    platform_fee_amount: z.int(),
    platform_fee_currency: z.nullable(z.string()),
    customer: ordercustomer_js_1.OrderCustomer$inboundSchema,
    product: z.nullable(orderproduct_js_1.OrderProduct$inboundSchema),
    discount: z.nullable((0, smartUnion_js_1.smartUnion)([
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
    ])),
    subscription: z.nullable(ordersubscription_js_1.OrderSubscription$inboundSchema),
    items: z.array(orderitemschema_js_1.OrderItemSchema$inboundSchema),
    description: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subtotal_amount": "subtotalAmount",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "applied_balance_amount": "appliedBalanceAmount",
        "due_amount": "dueAmount",
        "refunded_amount": "refundedAmount",
        "refunded_tax_amount": "refundedTaxAmount",
        "billing_reason": "billingReason",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "invoice_number": "invoiceNumber",
        "is_invoice_generated": "isInvoiceGenerated",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "custom_field_data": "customFieldData",
        "platform_fee_amount": "platformFeeAmount",
        "platform_fee_currency": "platformFeeCurrency",
    });
}));
/** @internal */
exports.Order$outboundSchema = z
    .pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    status: orderstatus_js_1.OrderStatus$outboundSchema,
    paid: z.boolean(),
    subtotalAmount: z.int(),
    discountAmount: z.int(),
    netAmount: z.int(),
    taxAmount: z.int(),
    totalAmount: z.int(),
    appliedBalanceAmount: z.int(),
    dueAmount: z.int(),
    refundedAmount: z.int(),
    refundedTaxAmount: z.int(),
    currency: z.string(),
    billingReason: orderbillingreason_js_1.OrderBillingReason$outboundSchema,
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(address_js_1.Address$outboundSchema),
    invoiceNumber: z.string(),
    isInvoiceGenerated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customerId: z.string(),
    productId: z.nullable(z.string()),
    discountId: z.nullable(z.string()),
    subscriptionId: z.nullable(z.string()),
    checkoutId: z.nullable(z.string()),
    metadata: z.record(z.string(), metadataoutputtype_js_1.MetadataOutputType$outboundSchema),
    customFieldData: z.optional(z.record(z.string(), z.nullable((0, smartUnion_js_1.smartUnion)([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    platformFeeAmount: z.int(),
    platformFeeCurrency: z.nullable(z.string()),
    customer: ordercustomer_js_1.OrderCustomer$outboundSchema,
    product: z.nullable(orderproduct_js_1.OrderProduct$outboundSchema),
    discount: z.nullable((0, smartUnion_js_1.smartUnion)([
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
    ])),
    subscription: z.nullable(ordersubscription_js_1.OrderSubscription$outboundSchema),
    items: z.array(orderitemschema_js_1.OrderItemSchema$outboundSchema),
    description: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subtotalAmount: "subtotal_amount",
        discountAmount: "discount_amount",
        netAmount: "net_amount",
        taxAmount: "tax_amount",
        totalAmount: "total_amount",
        appliedBalanceAmount: "applied_balance_amount",
        dueAmount: "due_amount",
        refundedAmount: "refunded_amount",
        refundedTaxAmount: "refunded_tax_amount",
        billingReason: "billing_reason",
        billingName: "billing_name",
        billingAddress: "billing_address",
        invoiceNumber: "invoice_number",
        isInvoiceGenerated: "is_invoice_generated",
        customerId: "customer_id",
        productId: "product_id",
        discountId: "discount_id",
        subscriptionId: "subscription_id",
        checkoutId: "checkout_id",
        customFieldData: "custom_field_data",
        platformFeeAmount: "platform_fee_amount",
        platformFeeCurrency: "platform_fee_currency",
    });
}));
function orderToJSON(order) {
    return JSON.stringify(exports.Order$outboundSchema.parse(order));
}
function orderFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Order$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Order' from JSON`);
}
