// @polar-sh/sdk/dist/commonjs/models/components/organizationdetails.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationDetails$outboundSchema = exports.SwitchingFrom$outboundSchema = exports.SwitchingFrom = void 0;
exports.organizationDetailsToJSON = organizationDetailsToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.SwitchingFrom = {
    Paddle: "paddle",
    LemonSqueezy: "lemon_squeezy",
    Gumroad: "gumroad",
    Stripe: "stripe",
    Other: "other",
};
/** @internal */
exports.SwitchingFrom$outboundSchema = z.enum(exports.SwitchingFrom);
/** @internal */
exports.OrganizationDetails$outboundSchema = z.pipe(z.object({
    about: z.optional(z.nullable(z.string())),
    productDescription: z.optional(z.nullable(z.string())),
    sellingCategories: z.optional(z.array(z.string())),
    pricingModels: z.optional(z.array(z.string())),
    intendedUse: z.optional(z.nullable(z.string())),
    customerAcquisition: z.optional(z.array(z.string())),
    futureAnnualRevenue: z.optional(z.nullable(z.int())),
    switching: z._default(z.boolean(), false),
    switchingFrom: z.optional(z.nullable(exports.SwitchingFrom$outboundSchema)),
    previousAnnualRevenue: z.optional(z.nullable(z.int())),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        productDescription: "product_description",
        sellingCategories: "selling_categories",
        pricingModels: "pricing_models",
        intendedUse: "intended_use",
        customerAcquisition: "customer_acquisition",
        futureAnnualRevenue: "future_annual_revenue",
        switchingFrom: "switching_from",
        previousAnnualRevenue: "previous_annual_revenue",
    });
}));
function organizationDetailsToJSON(organizationDetails) {
    return JSON.stringify(exports.OrganizationDetails$outboundSchema.parse(organizationDetails));
}
