// @polar-sh/sdk/dist/commonjs/models/components/organizationfeaturesettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationFeatureSettings$outboundSchema = exports.OrganizationFeatureSettings$inboundSchema = void 0;
exports.organizationFeatureSettingsToJSON = organizationFeatureSettingsToJSON;
exports.organizationFeatureSettingsFromJSON = organizationFeatureSettingsFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OrganizationFeatureSettings$inboundSchema = z.pipe(z.object({
    issue_funding_enabled: z._default(z.boolean(), false),
    seat_based_pricing_enabled: z._default(z.boolean(), false),
    revops_enabled: z._default(z.boolean(), false),
    wallets_enabled: z._default(z.boolean(), false),
    member_model_enabled: z._default(z.boolean(), false),
    tinybird_read: z._default(z.boolean(), false),
    tinybird_compare: z._default(z.boolean(), false),
    checkout_localization_enabled: z._default(z.boolean(), false),
    overview_metrics: z.optional(z.nullable(z.array(z.string()))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "issue_funding_enabled": "issueFundingEnabled",
        "seat_based_pricing_enabled": "seatBasedPricingEnabled",
        "revops_enabled": "revopsEnabled",
        "wallets_enabled": "walletsEnabled",
        "member_model_enabled": "memberModelEnabled",
        "tinybird_read": "tinybirdRead",
        "tinybird_compare": "tinybirdCompare",
        "checkout_localization_enabled": "checkoutLocalizationEnabled",
        "overview_metrics": "overviewMetrics",
    });
}));
/** @internal */
exports.OrganizationFeatureSettings$outboundSchema = z.pipe(z.object({
    issueFundingEnabled: z._default(z.boolean(), false),
    seatBasedPricingEnabled: z._default(z.boolean(), false),
    revopsEnabled: z._default(z.boolean(), false),
    walletsEnabled: z._default(z.boolean(), false),
    memberModelEnabled: z._default(z.boolean(), false),
    tinybirdRead: z._default(z.boolean(), false),
    tinybirdCompare: z._default(z.boolean(), false),
    checkoutLocalizationEnabled: z._default(z.boolean(), false),
    overviewMetrics: z.optional(z.nullable(z.array(z.string()))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        issueFundingEnabled: "issue_funding_enabled",
        seatBasedPricingEnabled: "seat_based_pricing_enabled",
        revopsEnabled: "revops_enabled",
        walletsEnabled: "wallets_enabled",
        memberModelEnabled: "member_model_enabled",
        tinybirdRead: "tinybird_read",
        tinybirdCompare: "tinybird_compare",
        checkoutLocalizationEnabled: "checkout_localization_enabled",
        overviewMetrics: "overview_metrics",
    });
}));
function organizationFeatureSettingsToJSON(organizationFeatureSettings) {
    return JSON.stringify(exports.OrganizationFeatureSettings$outboundSchema.parse(organizationFeatureSettings));
}
function organizationFeatureSettingsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationFeatureSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationFeatureSettings' from JSON`);
}
