// @polar-sh/sdk/dist/commonjs/models/components/organizationcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationCreate$outboundSchema = exports.CountryAlpha2Input$outboundSchema = exports.LegalEntity$outboundSchema = exports.CountryAlpha2Input = void 0;
exports.legalEntityToJSON = legalEntityToJSON;
exports.organizationCreateToJSON = organizationCreateToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const organizationcompanylegalentityschema_js_1 = require("./organizationcompanylegalentityschema.js");
const organizationcustomeremailsettings_js_1 = require("./organizationcustomeremailsettings.js");
const organizationcustomerportalsettings_js_1 = require("./organizationcustomerportalsettings.js");
const organizationdetails_js_1 = require("./organizationdetails.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
const organizationindividuallegalentityschema_js_1 = require("./organizationindividuallegalentityschema.js");
const organizationnotificationsettings_js_1 = require("./organizationnotificationsettings.js");
const organizationsociallink_js_1 = require("./organizationsociallink.js");
const organizationsubscriptionsettings_js_1 = require("./organizationsubscriptionsettings.js");
const presentmentcurrency_js_1 = require("./presentmentcurrency.js");
const taxbehavioroption_js_1 = require("./taxbehavioroption.js");
exports.CountryAlpha2Input = {
    Ad: "AD",
    Ae: "AE",
    Af: "AF",
    Ag: "AG",
    Ai: "AI",
    Al: "AL",
    Am: "AM",
    Ao: "AO",
    Aq: "AQ",
    Ar: "AR",
    As: "AS",
    At: "AT",
    Au: "AU",
    Aw: "AW",
    Ax: "AX",
    Az: "AZ",
    Ba: "BA",
    Bb: "BB",
    Bd: "BD",
    Be: "BE",
    Bf: "BF",
    Bg: "BG",
    Bh: "BH",
    Bi: "BI",
    Bj: "BJ",
    Bl: "BL",
    Bm: "BM",
    Bn: "BN",
    Bo: "BO",
    Bq: "BQ",
    Br: "BR",
    Bs: "BS",
    Bt: "BT",
    Bv: "BV",
    Bw: "BW",
    By: "BY",
    Bz: "BZ",
    Ca: "CA",
    Cc: "CC",
    Cd: "CD",
    Cf: "CF",
    Cg: "CG",
    Ch: "CH",
    Ci: "CI",
    Ck: "CK",
    Cl: "CL",
    Cm: "CM",
    Cn: "CN",
    Co: "CO",
    Cr: "CR",
    Cv: "CV",
    Cw: "CW",
    Cx: "CX",
    Cy: "CY",
    Cz: "CZ",
    De: "DE",
    Dj: "DJ",
    Dk: "DK",
    Dm: "DM",
    Do: "DO",
    Dz: "DZ",
    Ec: "EC",
    Ee: "EE",
    Eg: "EG",
    Eh: "EH",
    Er: "ER",
    Es: "ES",
    Et: "ET",
    Fi: "FI",
    Fj: "FJ",
    Fk: "FK",
    Fm: "FM",
    Fo: "FO",
    Fr: "FR",
    Ga: "GA",
    Gb: "GB",
    Gd: "GD",
    Ge: "GE",
    Gf: "GF",
    Gg: "GG",
    Gh: "GH",
    Gi: "GI",
    Gl: "GL",
    Gm: "GM",
    Gn: "GN",
    Gp: "GP",
    Gq: "GQ",
    Gr: "GR",
    Gs: "GS",
    Gt: "GT",
    Gu: "GU",
    Gw: "GW",
    Gy: "GY",
    Hk: "HK",
    Hm: "HM",
    Hn: "HN",
    Hr: "HR",
    Ht: "HT",
    Hu: "HU",
    Id: "ID",
    Ie: "IE",
    Il: "IL",
    Im: "IM",
    In: "IN",
    Io: "IO",
    Iq: "IQ",
    Is: "IS",
    It: "IT",
    Je: "JE",
    Jm: "JM",
    Jo: "JO",
    Jp: "JP",
    Ke: "KE",
    Kg: "KG",
    Kh: "KH",
    Ki: "KI",
    Km: "KM",
    Kn: "KN",
    Kr: "KR",
    Kw: "KW",
    Ky: "KY",
    Kz: "KZ",
    La: "LA",
    Lb: "LB",
    Lc: "LC",
    Li: "LI",
    Lk: "LK",
    Lr: "LR",
    Ls: "LS",
    Lt: "LT",
    Lu: "LU",
    Lv: "LV",
    Ly: "LY",
    Ma: "MA",
    Mc: "MC",
    Md: "MD",
    Me: "ME",
    Mf: "MF",
    Mg: "MG",
    Mh: "MH",
    Mk: "MK",
    Ml: "ML",
    Mm: "MM",
    Mn: "MN",
    Mo: "MO",
    Mp: "MP",
    Mq: "MQ",
    Mr: "MR",
    Ms: "MS",
    Mt: "MT",
    Mu: "MU",
    Mv: "MV",
    Mw: "MW",
    Mx: "MX",
    My: "MY",
    Mz: "MZ",
    Na: "NA",
    Nc: "NC",
    Ne: "NE",
    Nf: "NF",
    Ng: "NG",
    Ni: "NI",
    Nl: "NL",
    No: "NO",
    Np: "NP",
    Nr: "NR",
    Nu: "NU",
    Nz: "NZ",
    Om: "OM",
    Pa: "PA",
    Pe: "PE",
    Pf: "PF",
    Pg: "PG",
    Ph: "PH",
    Pk: "PK",
    Pl: "PL",
    Pm: "PM",
    Pn: "PN",
    Pr: "PR",
    Ps: "PS",
    Pt: "PT",
    Pw: "PW",
    Py: "PY",
    Qa: "QA",
    Re: "RE",
    Ro: "RO",
    Rs: "RS",
    Rw: "RW",
    Sa: "SA",
    Sb: "SB",
    Sc: "SC",
    Sd: "SD",
    Se: "SE",
    Sg: "SG",
    Sh: "SH",
    Si: "SI",
    Sj: "SJ",
    Sk: "SK",
    Sl: "SL",
    Sm: "SM",
    Sn: "SN",
    So: "SO",
    Sr: "SR",
    Ss: "SS",
    St: "ST",
    Sv: "SV",
    Sx: "SX",
    Sz: "SZ",
    Tc: "TC",
    Td: "TD",
    Tf: "TF",
    Tg: "TG",
    Th: "TH",
    Tj: "TJ",
    Tk: "TK",
    Tl: "TL",
    Tm: "TM",
    Tn: "TN",
    To: "TO",
    Tr: "TR",
    Tt: "TT",
    Tv: "TV",
    Tw: "TW",
    Tz: "TZ",
    Ua: "UA",
    Ug: "UG",
    Um: "UM",
    Us: "US",
    Uy: "UY",
    Uz: "UZ",
    Va: "VA",
    Vc: "VC",
    Ve: "VE",
    Vg: "VG",
    Vi: "VI",
    Vn: "VN",
    Vu: "VU",
    Wf: "WF",
    Ws: "WS",
    Ye: "YE",
    Yt: "YT",
    Za: "ZA",
    Zm: "ZM",
    Zw: "ZW",
};
/** @internal */
exports.LegalEntity$outboundSchema = z.union([
    organizationcompanylegalentityschema_js_1.OrganizationCompanyLegalEntitySchema$outboundSchema,
    organizationindividuallegalentityschema_js_1.OrganizationIndividualLegalEntitySchema$outboundSchema,
]);
function legalEntityToJSON(legalEntity) {
    return JSON.stringify(exports.LegalEntity$outboundSchema.parse(legalEntity));
}
/** @internal */
exports.CountryAlpha2Input$outboundSchema = z.enum(exports.CountryAlpha2Input);
/** @internal */
exports.OrganizationCreate$outboundSchema = z.pipe(z.object({
    name: z.string(),
    slug: z.string(),
    avatarUrl: z.optional(z.nullable(z.string())),
    legalEntity: z.optional(z.nullable(z.union([
        organizationcompanylegalentityschema_js_1.OrganizationCompanyLegalEntitySchema$outboundSchema,
        organizationindividuallegalentityschema_js_1.OrganizationIndividualLegalEntitySchema$outboundSchema,
    ]))),
    email: z.optional(z.nullable(z.string())),
    website: z.optional(z.nullable(z.string())),
    socials: z.optional(z.nullable(z.array(organizationsociallink_js_1.OrganizationSocialLink$outboundSchema))),
    details: z.optional(z.nullable(organizationdetails_js_1.OrganizationDetails$outboundSchema)),
    country: z.optional(z.nullable(exports.CountryAlpha2Input$outboundSchema)),
    featureSettings: z.optional(z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema)),
    subscriptionSettings: z.optional(z.nullable(organizationsubscriptionsettings_js_1.OrganizationSubscriptionSettings$outboundSchema)),
    notificationSettings: z.optional(z.nullable(organizationnotificationsettings_js_1.OrganizationNotificationSettings$outboundSchema)),
    customerEmailSettings: z.optional(z.nullable(organizationcustomeremailsettings_js_1.OrganizationCustomerEmailSettings$outboundSchema)),
    customerPortalSettings: z.optional(z.nullable(organizationcustomerportalsettings_js_1.OrganizationCustomerPortalSettings$outboundSchema)),
    defaultPresentmentCurrency: z.optional(presentmentcurrency_js_1.PresentmentCurrency$outboundSchema),
    defaultTaxBehavior: z.optional(taxbehavioroption_js_1.TaxBehaviorOption$outboundSchema),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        avatarUrl: "avatar_url",
        legalEntity: "legal_entity",
        featureSettings: "feature_settings",
        subscriptionSettings: "subscription_settings",
        notificationSettings: "notification_settings",
        customerEmailSettings: "customer_email_settings",
        customerPortalSettings: "customer_portal_settings",
        defaultPresentmentCurrency: "default_presentment_currency",
        defaultTaxBehavior: "default_tax_behavior",
    });
}));
function organizationCreateToJSON(organizationCreate) {
    return JSON.stringify(exports.OrganizationCreate$outboundSchema.parse(organizationCreate));
}
