// @polar-sh/sdk/dist/commonjs/models/components/pendingsubscriptionupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PendingSubscriptionUpdate$outboundSchema = exports.PendingSubscriptionUpdate$inboundSchema = void 0;
exports.pendingSubscriptionUpdateToJSON = pendingSubscriptionUpdateToJSON;
exports.pendingSubscriptionUpdateFromJSON = pendingSubscriptionUpdateFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.PendingSubscriptionUpdate$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    applies_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    product_id: z.nullable(z.string()),
    seats: z.nullable(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "applies_at": "appliesAt",
        "product_id": "productId",
    });
}));
/** @internal */
exports.PendingSubscriptionUpdate$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    appliesAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    productId: z.nullable(z.string()),
    seats: z.nullable(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        appliesAt: "applies_at",
        productId: "product_id",
    });
}));
function pendingSubscriptionUpdateToJSON(pendingSubscriptionUpdate) {
    return JSON.stringify(exports.PendingSubscriptionUpdate$outboundSchema.parse(pendingSubscriptionUpdate));
}
function pendingSubscriptionUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PendingSubscriptionUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PendingSubscriptionUpdate' from JSON`);
}
