// @polar-sh/sdk/dist/commonjs/models/components/productpriceseattiersoutput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceSeatTiersOutput$outboundSchema = exports.ProductPriceSeatTiersOutput$inboundSchema = void 0;
exports.productPriceSeatTiersOutputToJSON = productPriceSeatTiersOutputToJSON;
exports.productPriceSeatTiersOutputFromJSON = productPriceSeatTiersOutputFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const productpriceseattier_js_1 = require("./productpriceseattier.js");
const seattiertype_js_1 = require("./seattiertype.js");
/** @internal */
exports.ProductPriceSeatTiersOutput$inboundSchema = z.pipe(z.object({
    seat_tier_type: z.optional(seattiertype_js_1.SeatTierType$inboundSchema),
    tiers: z.array(productpriceseattier_js_1.ProductPriceSeatTier$inboundSchema),
    minimum_seats: z.int(),
    maximum_seats: z.nullable(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "seat_tier_type": "seatTierType",
        "minimum_seats": "minimumSeats",
        "maximum_seats": "maximumSeats",
    });
}));
/** @internal */
exports.ProductPriceSeatTiersOutput$outboundSchema = z.pipe(z.object({
    seatTierType: z.optional(seattiertype_js_1.SeatTierType$outboundSchema),
    tiers: z.array(productpriceseattier_js_1.ProductPriceSeatTier$outboundSchema),
    minimumSeats: z.int(),
    maximumSeats: z.nullable(z.int()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        seatTierType: "seat_tier_type",
        minimumSeats: "minimum_seats",
        maximumSeats: "maximum_seats",
    });
}));
function productPriceSeatTiersOutputToJSON(productPriceSeatTiersOutput) {
    return JSON.stringify(exports.ProductPriceSeatTiersOutput$outboundSchema.parse(productPriceSeatTiersOutput));
}
function productPriceSeatTiersOutputFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceSeatTiersOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatTiersOutput' from JSON`);
}
