// @polar-sh/sdk/dist/commonjs/models/components/productpriceseatbased.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPriceSeatBased$outboundSchema = exports.ProductPriceSeatBased$inboundSchema = void 0;
exports.productPriceSeatBasedToJSON = productPriceSeatBasedToJSON;
exports.productPriceSeatBasedFromJSON = productPriceSeatBasedFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const productpriceseattiersoutput_js_1 = require("./productpriceseattiersoutput.js");
const productpricesource_js_1 = require("./productpricesource.js");
const taxbehavioroption_js_1 = require("./taxbehavioroption.js");
/** @internal */
exports.ProductPriceSeatBased$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    source: productpricesource_js_1.ProductPriceSource$inboundSchema,
    amount_type: z.literal("seat_based"),
    price_currency: z.string(),
    tax_behavior: z.nullable(taxbehavioroption_js_1.TaxBehaviorOption$inboundSchema),
    is_archived: z.boolean(),
    product_id: z.string(),
    seat_tiers: productpriceseattiersoutput_js_1.ProductPriceSeatTiersOutput$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "tax_behavior": "taxBehavior",
        "is_archived": "isArchived",
        "product_id": "productId",
        "seat_tiers": "seatTiers",
    });
}));
/** @internal */
exports.ProductPriceSeatBased$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    source: productpricesource_js_1.ProductPriceSource$outboundSchema,
    amountType: z.literal("seat_based"),
    priceCurrency: z.string(),
    taxBehavior: z.nullable(taxbehavioroption_js_1.TaxBehaviorOption$outboundSchema),
    isArchived: z.boolean(),
    productId: z.string(),
    seatTiers: productpriceseattiersoutput_js_1.ProductPriceSeatTiersOutput$outboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        isArchived: "is_archived",
        productId: "product_id",
        seatTiers: "seat_tiers",
    });
}));
function productPriceSeatBasedToJSON(productPriceSeatBased) {
    return JSON.stringify(exports.ProductPriceSeatBased$outboundSchema.parse(productPriceSeatBased));
}
function productPriceSeatBasedFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductPriceSeatBased$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatBased' from JSON`);
}
