// @polar-sh/sdk/dist/commonjs/models/components/productcreateonetime.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCreateOneTime$outboundSchema = exports.ProductCreateOneTimePrices$outboundSchema = exports.ProductCreateOneTimeMetadata$outboundSchema = void 0;
exports.productCreateOneTimeMetadataToJSON = productCreateOneTimeMetadataToJSON;
exports.productCreateOneTimePricesToJSON = productCreateOneTimePricesToJSON;
exports.productCreateOneTimeToJSON = productCreateOneTimeToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const productpricemeteredunitcreate_js_1 = require("./productpricemeteredunitcreate.js");
const productpriceseatbasedcreate_js_1 = require("./productpriceseatbasedcreate.js");
const productvisibility_js_1 = require("./productvisibility.js");
/** @internal */
exports.ProductCreateOneTimeMetadata$outboundSchema = (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]);
function productCreateOneTimeMetadataToJSON(productCreateOneTimeMetadata) {
    return JSON.stringify(exports.ProductCreateOneTimeMetadata$outboundSchema.parse(productCreateOneTimeMetadata));
}
/** @internal */
exports.ProductCreateOneTimePrices$outboundSchema = z.union([
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
    productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
    productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema,
]);
function productCreateOneTimePricesToJSON(productCreateOneTimePrices) {
    return JSON.stringify(exports.ProductCreateOneTimePrices$outboundSchema.parse(productCreateOneTimePrices));
}
/** @internal */
exports.ProductCreateOneTime$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), (0, smartUnion_js_1.smartUnion)([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.string(),
    description: z.optional(z.nullable(z.string())),
    visibility: z.optional(productvisibility_js_1.ProductVisibility$outboundSchema),
    prices: z.array(z.union([
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
        productpricemeteredunitcreate_js_1.ProductPriceMeteredUnitCreate$outboundSchema,
        productpriceseatbasedcreate_js_1.ProductPriceSeatBasedCreate$outboundSchema,
    ])),
    medias: z.optional(z.nullable(z.array(z.string()))),
    attachedCustomFields: z.optional(z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)),
    organizationId: z.optional(z.nullable(z.string())),
    recurringInterval: z.optional(z.nullable(z.any())),
    recurringIntervalCount: z.optional(z.nullable(z.any())),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        recurringInterval: "recurring_interval",
        recurringIntervalCount: "recurring_interval_count",
    });
}));
function productCreateOneTimeToJSON(productCreateOneTime) {
    return JSON.stringify(exports.ProductCreateOneTime$outboundSchema.parse(productCreateOneTime));
}
