// @polar-sh/sdk/dist/commonjs/models/components/refunddispute.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundDispute$outboundSchema = exports.RefundDispute$inboundSchema = void 0;
exports.refundDisputeToJSON = refundDisputeToJSON;
exports.refundDisputeFromJSON = refundDisputeFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const disputestatus_js_1 = require("./disputestatus.js");
/** @internal */
exports.RefundDispute$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    status: disputestatus_js_1.DisputeStatus$inboundSchema,
    resolved: z.boolean(),
    closed: z.boolean(),
    amount: z.int(),
    tax_amount: z.int(),
    currency: z.string(),
    order_id: z.string(),
    payment_id: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "order_id": "orderId",
        "payment_id": "paymentId",
    });
}));
/** @internal */
exports.RefundDispute$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    status: disputestatus_js_1.DisputeStatus$outboundSchema,
    resolved: z.boolean(),
    closed: z.boolean(),
    amount: z.int(),
    taxAmount: z.int(),
    currency: z.string(),
    orderId: z.string(),
    paymentId: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        taxAmount: "tax_amount",
        orderId: "order_id",
        paymentId: "payment_id",
    });
}));
function refundDisputeToJSON(refundDispute) {
    return JSON.stringify(exports.RefundDispute$outboundSchema.parse(refundDispute));
}
function refundDisputeFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RefundDispute$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefundDispute' from JSON`);
}
