// @polar-sh/sdk/dist/commonjs/models/components/revoketokenrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RevokeTokenRequest$outboundSchema = exports.TokenTypeHint$outboundSchema = exports.TokenTypeHint = void 0;
exports.revokeTokenRequestToJSON = revokeTokenRequestToJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
exports.TokenTypeHint = {
    AccessToken: "access_token",
    RefreshToken: "refresh_token",
};
/** @internal */
exports.TokenTypeHint$outboundSchema = z.enum(exports.TokenTypeHint);
/** @internal */
exports.RevokeTokenRequest$outboundSchema = z.pipe(z.object({
    token: z.string(),
    tokenTypeHint: z.optional(z.nullable(exports.TokenTypeHint$outboundSchema)),
    clientId: z.string(),
    clientSecret: z.string(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        tokenTypeHint: "token_type_hint",
        clientId: "client_id",
        clientSecret: "client_secret",
    });
}));
function revokeTokenRequestToJSON(revokeTokenRequest) {
    return JSON.stringify(exports.RevokeTokenRequest$outboundSchema.parse(revokeTokenRequest));
}
