// @polar-sh/sdk/dist/commonjs/models/components/subscriptioncycledevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCycledEvent$inboundSchema = void 0;
exports.subscriptionCycledEventFromJSON = subscriptionCycledEventFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customer_js_1 = require("./customer.js");
const subscriptioncycledmetadata_js_1 = require("./subscriptioncycledmetadata.js");
/** @internal */
exports.SubscriptionCycledEvent$inboundSchema = z.pipe(z.object({
    id: z.string(),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(customer_js_1.Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    member_id: z.optional(z.nullable(z.string())),
    external_member_id: z.optional(z.nullable(z.string())),
    child_count: z._default(z.int(), 0),
    parent_id: z.optional(z.nullable(z.string())),
    label: z.string(),
    source: z.literal("system"),
    name: z.literal("subscription.cycled"),
    metadata: subscriptioncycledmetadata_js_1.SubscriptionCycledMetadata$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
        "member_id": "memberId",
        "external_member_id": "externalMemberId",
        "child_count": "childCount",
        "parent_id": "parentId",
    });
}));
function subscriptionCycledEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCycledEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCycledEvent' from JSON`);
}
