// @polar-sh/sdk/dist/commonjs/models/components/subscriptioncanceledmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionCanceledMetadata$inboundSchema = void 0;
exports.subscriptionCanceledMetadataFromJSON = subscriptionCanceledMetadataFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.SubscriptionCanceledMetadata$inboundSchema = z.pipe(z.object({
    subscription_id: z.string(),
    product_id: z.optional(z.string()),
    amount: z.int(),
    currency: z.string(),
    recurring_interval: z.string(),
    recurring_interval_count: z.int(),
    customer_cancellation_reason: z.optional(z.string()),
    customer_cancellation_comment: z.optional(z.string()),
    canceled_at: z.string(),
    ends_at: z.optional(z.string()),
    cancel_at_period_end: z.optional(z.boolean()),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "subscription_id": "subscriptionId",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "canceled_at": "canceledAt",
        "ends_at": "endsAt",
        "cancel_at_period_end": "cancelAtPeriodEnd",
    });
}));
function subscriptionCanceledMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCanceledMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCanceledMetadata' from JSON`);
}
