// @polar-sh/sdk/dist/commonjs/models/components/validationerror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationError$outboundSchema = exports.ValidationError$inboundSchema = exports.Context$outboundSchema = exports.Context$inboundSchema = exports.Loc$outboundSchema = exports.Loc$inboundSchema = void 0;
exports.locToJSON = locToJSON;
exports.locFromJSON = locFromJSON;
exports.contextToJSON = contextToJSON;
exports.contextFromJSON = contextFromJSON;
exports.validationErrorToJSON = validationErrorToJSON;
exports.validationErrorFromJSON = validationErrorFromJSON;
const z = __importStar(require("zod/v4-mini"));
const schemas_js_1 = require("../../lib/schemas.js");
const smartUnion_js_1 = require("../../types/smartUnion.js");
/** @internal */
exports.Loc$inboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
]);
/** @internal */
exports.Loc$outboundSchema = (0, smartUnion_js_1.smartUnion)([
    z.string(),
    z.int(),
]);
function locToJSON(loc) {
    return JSON.stringify(exports.Loc$outboundSchema.parse(loc));
}
function locFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Loc$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Loc' from JSON`);
}
/** @internal */
exports.Context$inboundSchema = z.object({});
/** @internal */
exports.Context$outboundSchema = z.object({});
function contextToJSON(context) {
    return JSON.stringify(exports.Context$outboundSchema.parse(context));
}
function contextFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Context$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Context' from JSON`);
}
/** @internal */
exports.ValidationError$inboundSchema = z.object({
    loc: z.array((0, smartUnion_js_1.smartUnion)([z.string(), z.int()])),
    msg: z.string(),
    type: z.string(),
    input: z.optional(z.any()),
    ctx: z.optional(z.lazy(() => exports.Context$inboundSchema)),
});
/** @internal */
exports.ValidationError$outboundSchema = z.object({
    loc: z.array((0, smartUnion_js_1.smartUnion)([z.string(), z.int()])),
    msg: z.string(),
    type: z.string(),
    input: z.optional(z.any()),
    ctx: z.optional(z.lazy(() => exports.Context$outboundSchema)),
});
function validationErrorToJSON(validationError) {
    return JSON.stringify(exports.ValidationError$outboundSchema.parse(validationError));
}
function validationErrorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ValidationError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidationError' from JSON`);
}
