// @polar-sh/sdk/dist/esm/sdk/customerseats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { customerSeatsAssignSeat } from "../funcs/customerSeatsAssignSeat.js";
import { customerSeatsClaimSeat } from "../funcs/customerSeatsClaimSeat.js";
import { customerSeatsGetClaimInfo } from "../funcs/customerSeatsGetClaimInfo.js";
import { customerSeatsListSeats } from "../funcs/customerSeatsListSeats.js";
import { customerSeatsResendInvitation } from "../funcs/customerSeatsResendInvitation.js";
import { customerSeatsRevokeSeat } from "../funcs/customerSeatsRevokeSeat.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class CustomerSeats extends ClientSDK {
    /**
     * List Seats
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async listSeats(request, options) {
        return unwrapAsync(customerSeatsListSeats(this, request, options));
    }
    /**
     * Assign Seat
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async assignSeat(request, options) {
        return unwrapAsync(customerSeatsAssignSeat(this, request, options));
    }
    /**
     * Revoke Seat
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async revokeSeat(request, options) {
        return unwrapAsync(customerSeatsRevokeSeat(this, request, options));
    }
    /**
     * Resend Invitation
     *
     * @remarks
     * **Scopes**: `customer_seats:write`
     */
    async resendInvitation(request, options) {
        return unwrapAsync(customerSeatsResendInvitation(this, request, options));
    }
    /**
     * Get Claim Info
     */
    async getClaimInfo(request, options) {
        return unwrapAsync(customerSeatsGetClaimInfo(this, request, options));
    }
    /**
     * Claim Seat
     */
    async claimSeat(request, options) {
        return unwrapAsync(customerSeatsClaimSeat(this, request, options));
    }
}
