// @polar-sh/sdk/dist/esm/sdk/metrics.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { metricsCreateDashboard } from "../funcs/metricsCreateDashboard.js";
import { metricsDeleteDashboard } from "../funcs/metricsDeleteDashboard.js";
import { metricsGet } from "../funcs/metricsGet.js";
import { metricsGetDashboard } from "../funcs/metricsGetDashboard.js";
import { metricsLimits } from "../funcs/metricsLimits.js";
import { metricsListDashboards } from "../funcs/metricsListDashboards.js";
import { metricsUpdateDashboard } from "../funcs/metricsUpdateDashboard.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class Metrics extends ClientSDK {
    /**
     * Get Metrics
     *
     * @remarks
     * Get metrics about your orders and subscriptions.
     *
     * Currency values are output in cents.
     *
     * **Scopes**: `metrics:read`
     */
    async get(request, options) {
        return unwrapAsync(metricsGet(this, request, options));
    }
    /**
     * Get Metrics Limits
     *
     * @remarks
     * Get the interval limits for the metrics endpoint.
     *
     * **Scopes**: `metrics:read`
     */
    async limits(options) {
        return unwrapAsync(metricsLimits(this, options));
    }
    /**
     * List Metric Dashboards
     *
     * @remarks
     * List user-defined metric dashboards.
     *
     * **Scopes**: `metrics:read`
     */
    async listDashboards(request, options) {
        return unwrapAsync(metricsListDashboards(this, request, options));
    }
    /**
     * Create Metric Dashboard
     *
     * @remarks
     * Create a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async createDashboard(request, options) {
        return unwrapAsync(metricsCreateDashboard(this, request, options));
    }
    /**
     * Get Metric Dashboard
     *
     * @remarks
     * Get a user-defined metric dashboard by ID.
     *
     * **Scopes**: `metrics:read`
     */
    async getDashboard(request, options) {
        return unwrapAsync(metricsGetDashboard(this, request, options));
    }
    /**
     * Delete Metric Dashboard
     *
     * @remarks
     * Delete a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async deleteDashboard(request, options) {
        return unwrapAsync(metricsDeleteDashboard(this, request, options));
    }
    /**
     * Update Metric Dashboard
     *
     * @remarks
     * Update a user-defined metric dashboard.
     *
     * **Scopes**: `metrics:write`
     */
    async updateDashboard(request, options) {
        return unwrapAsync(metricsUpdateDashboard(this, request, options));
    }
}
