// @polar-sh/sdk/dist/esm/sdk/seats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { customerPortalSeatsAssignSeat } from "../funcs/customerPortalSeatsAssignSeat.js";
import { customerPortalSeatsListClaimedSubscriptions } from "../funcs/customerPortalSeatsListClaimedSubscriptions.js";
import { customerPortalSeatsListSeats } from "../funcs/customerPortalSeatsListSeats.js";
import { customerPortalSeatsResendInvitation } from "../funcs/customerPortalSeatsResendInvitation.js";
import { customerPortalSeatsRevokeSeat } from "../funcs/customerPortalSeatsRevokeSeat.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Seats extends ClientSDK {
    /**
     * List Seats
     *
     * @remarks
     * **Scopes**: `customer_portal:read` `customer_portal:write`
     */
    async listSeats(security, request, options) {
        return unwrapAsync(customerPortalSeatsListSeats(this, security, request, options));
    }
    /**
     * Assign Seat
     */
    async assignSeat(security, request, options) {
        return unwrapAsync(customerPortalSeatsAssignSeat(this, security, request, options));
    }
    /**
     * Revoke Seat
     */
    async revokeSeat(security, request, options) {
        return unwrapAsync(customerPortalSeatsRevokeSeat(this, security, request, options));
    }
    /**
     * Resend Invitation
     */
    async resendInvitation(security, request, options) {
        return unwrapAsync(customerPortalSeatsResendInvitation(this, security, request, options));
    }
    /**
     * List Claimed Subscriptions
     *
     * @remarks
     * List all subscriptions where the authenticated customer has claimed a seat.
     *
     * **Scopes**: `customer_portal:read` `customer_portal:write`
     */
    async listClaimedSubscriptions(security, request, options) {
        return unwrapResultIterator(customerPortalSeatsListClaimedSubscriptions(this, security, request, options));
    }
}
