// @polar-sh/sdk/dist/esm/funcs/customerPortalSeatsAssignSeat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { encodeJSON } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { resolveSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import { CustomerSeat$inboundSchema, } from "../models/components/customerseat.js";
import { SeatAssign$outboundSchema, } from "../models/components/seatassign.js";
import { HTTPValidationError$inboundSchema, } from "../models/errors/httpvalidationerror.js";
import { APIPromise } from "../types/async.js";
/**
 * Assign Seat
 */
export function customerPortalSeatsAssignSeat(client, security, request, options) {
    return new APIPromise($do(client, security, request, options));
}
async function $do(client, security, request, options) {
    const parsed = safeParse(request, (value) => z.parse(SeatAssign$outboundSchema, value), "Input validation failed");
    if (!parsed.ok) {
        return [parsed, { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = encodeJSON("body", payload, { explode: true });
    const path = pathToFunc("/v1/customer-portal/seats")();
    const headers = new Headers(compactMap({
        "Content-Type": "application/json",
        Accept: "application/json",
    }));
    const requestSecurity = resolveSecurity([
        {
            fieldName: "Authorization",
            type: "http:bearer",
            value: security?.customerSession,
        },
    ], [
        {
            fieldName: "Authorization",
            type: "http:bearer",
            value: security?.memberSession,
        },
    ]);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "customer_portal:seats:assign_seat",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: security,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "POST",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["400", "401", "403", "404", "422", "4XX", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.json(200, CustomerSeat$inboundSchema), M.jsonErr(422, HTTPValidationError$inboundSchema), M.fail([400, 401, 403, 404, "4XX"]), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
