// @polar-sh/sdk/dist/esm/funcs/membersCreateMember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { encodeJSON } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import { Member$inboundSchema } from "../models/components/member.js";
import { MemberCreate$outboundSchema, } from "../models/components/membercreate.js";
import { HTTPValidationError$inboundSchema, } from "../models/errors/httpvalidationerror.js";
import { ResourceNotFound$inboundSchema, } from "../models/errors/resourcenotfound.js";
import { APIPromise } from "../types/async.js";
/**
 * Create Member
 *
 * @remarks
 * Create a new member for a customer.
 *
 * Only B2B customers with the member management feature enabled can add members.
 * The authenticated user or organization must have access to the customer's organization.
 *
 * **Scopes**: `members:write`
 */
export function membersCreateMember(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => z.parse(MemberCreate$outboundSchema, value), "Input validation failed");
    if (!parsed.ok) {
        return [parsed, { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = encodeJSON("body", payload, { explode: true });
    const path = pathToFunc("/v1/members/")();
    const headers = new Headers(compactMap({
        "Content-Type": "application/json",
        Accept: "application/json",
    }));
    const secConfig = await extractSecurity(client._options.accessToken);
    const securityInput = secConfig == null ? {} : { accessToken: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "members:create_member",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.accessToken,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "POST",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["403", "404", "422", "4XX", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.json(201, Member$inboundSchema), M.jsonErr(404, ResourceNotFound$inboundSchema), M.jsonErr(422, HTTPValidationError$inboundSchema), M.fail([403, "4XX"]), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
