// @polar-sh/sdk/dist/esm/models/components/authorizeresponseorganization.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { AuthorizeOrganization$inboundSchema, } from "./authorizeorganization.js";
import { OAuth2ClientPublic$inboundSchema, } from "./oauth2clientpublic.js";
import { Scope$inboundSchema } from "./scope.js";
/** @internal */
export const AuthorizeResponseOrganization$inboundSchema = z.pipe(z.object({
    client: OAuth2ClientPublic$inboundSchema,
    sub_type: z.literal("organization"),
    sub: z.nullable(AuthorizeOrganization$inboundSchema),
    scopes: z.array(Scope$inboundSchema),
    scope_display_names: z.optional(z.record(z.string(), z.string())),
    organizations: z.array(AuthorizeOrganization$inboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        "sub_type": "subType",
        "scope_display_names": "scopeDisplayNames",
    });
}));
export function authorizeResponseOrganizationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AuthorizeResponseOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AuthorizeResponseOrganization' from JSON`);
}
