// @polar-sh/sdk/dist/esm/models/components/authorizeresponseuser.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { AuthorizeUser$inboundSchema } from "./authorizeuser.js";
import { OAuth2ClientPublic$inboundSchema, } from "./oauth2clientpublic.js";
import { Scope$inboundSchema } from "./scope.js";
/** @internal */
export const AuthorizeResponseUser$inboundSchema = z.pipe(z.object({
    client: OAuth2ClientPublic$inboundSchema,
    sub_type: z.literal("user"),
    sub: z.nullable(AuthorizeUser$inboundSchema),
    scopes: z.array(Scope$inboundSchema),
    scope_display_names: z.optional(z.record(z.string(), z.string())),
}), z.transform((v) => {
    return remap$(v, {
        "sub_type": "subType",
        "scope_display_names": "scopeDisplayNames",
    });
}));
export function authorizeResponseUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AuthorizeResponseUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AuthorizeResponseUser' from JSON`);
}
