// @polar-sh/sdk/dist/esm/models/components/balancedisputemetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BalanceDisputeMetadata$inboundSchema = z.pipe(z.object({
    transaction_id: z.string(),
    dispute_id: z.string(),
    order_id: z.optional(z.string()),
    order_created_at: z.optional(z.string()),
    product_id: z.optional(z.string()),
    subscription_id: z.optional(z.string()),
    amount: z.int(),
    currency: z.string(),
    presentment_amount: z.int(),
    presentment_currency: z.string(),
    tax_amount: z.int(),
    tax_state: z.optional(z.nullable(z.string())),
    tax_country: z.optional(z.nullable(z.string())),
    fee: z.int(),
    exchange_rate: z.optional(z.number()),
}), z.transform((v) => {
    return remap$(v, {
        "transaction_id": "transactionId",
        "dispute_id": "disputeId",
        "order_id": "orderId",
        "order_created_at": "orderCreatedAt",
        "product_id": "productId",
        "subscription_id": "subscriptionId",
        "presentment_amount": "presentmentAmount",
        "presentment_currency": "presentmentCurrency",
        "tax_amount": "taxAmount",
        "tax_state": "taxState",
        "tax_country": "taxCountry",
        "exchange_rate": "exchangeRate",
    });
}));
export function balanceDisputeMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BalanceDisputeMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BalanceDisputeMetadata' from JSON`);
}
