// @polar-sh/sdk/dist/esm/models/components/benefitgrant.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Benefit$inboundSchema } from "./benefit.js";
import { BenefitGrantCustomProperties$inboundSchema, } from "./benefitgrantcustomproperties.js";
import { BenefitGrantDiscordProperties$inboundSchema, } from "./benefitgrantdiscordproperties.js";
import { BenefitGrantDownloadablesProperties$inboundSchema, } from "./benefitgrantdownloadablesproperties.js";
import { BenefitGrantError$inboundSchema, } from "./benefitgranterror.js";
import { BenefitGrantFeatureFlagProperties$inboundSchema, } from "./benefitgrantfeatureflagproperties.js";
import { BenefitGrantGitHubRepositoryProperties$inboundSchema, } from "./benefitgrantgithubrepositoryproperties.js";
import { BenefitGrantLicenseKeysProperties$inboundSchema, } from "./benefitgrantlicensekeysproperties.js";
import { Customer$inboundSchema } from "./customer.js";
import { Member$inboundSchema } from "./member.js";
/** @internal */
export const Properties$inboundSchema = smartUnion([
    BenefitGrantDiscordProperties$inboundSchema,
    BenefitGrantGitHubRepositoryProperties$inboundSchema,
    BenefitGrantDownloadablesProperties$inboundSchema,
    BenefitGrantLicenseKeysProperties$inboundSchema,
    BenefitGrantCustomProperties$inboundSchema,
    BenefitGrantFeatureFlagProperties$inboundSchema,
]);
export function propertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Properties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Properties' from JSON`);
}
/** @internal */
export const BenefitGrant$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    granted_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_granted: z.boolean(),
    revoked_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    customer_id: z.string(),
    member_id: z.optional(z.nullable(z.string())),
    benefit_id: z.string(),
    error: z.optional(z.nullable(BenefitGrantError$inboundSchema)),
    customer: Customer$inboundSchema,
    member: z.optional(z.nullable(Member$inboundSchema)),
    benefit: Benefit$inboundSchema,
    properties: smartUnion([
        BenefitGrantDiscordProperties$inboundSchema,
        BenefitGrantGitHubRepositoryProperties$inboundSchema,
        BenefitGrantDownloadablesProperties$inboundSchema,
        BenefitGrantLicenseKeysProperties$inboundSchema,
        BenefitGrantCustomProperties$inboundSchema,
        BenefitGrantFeatureFlagProperties$inboundSchema,
    ]),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "member_id": "memberId",
        "benefit_id": "benefitId",
    });
}));
export function benefitGrantFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrant$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrant' from JSON`);
}
