// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyactivationproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitLicenseKeyActivationProperties$inboundSchema = z.pipe(z.object({
    limit: z.int(),
    enable_customer_admin: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "enable_customer_admin": "enableCustomerAdmin",
    });
}));
/** @internal */
export const BenefitLicenseKeyActivationProperties$outboundSchema = z.pipe(z.object({
    limit: z.int(),
    enableCustomerAdmin: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        enableCustomerAdmin: "enable_customer_admin",
    });
}));
export function benefitLicenseKeyActivationPropertiesToJSON(benefitLicenseKeyActivationProperties) {
    return JSON.stringify(BenefitLicenseKeyActivationProperties$outboundSchema.parse(benefitLicenseKeyActivationProperties));
}
export function benefitLicenseKeyActivationPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeyActivationProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeyActivationProperties' from JSON`);
}
