// @polar-sh/sdk/dist/esm/models/components/benefitgrantgithubrepositorywebhook.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGitHubRepository$inboundSchema, BenefitGitHubRepository$outboundSchema, } from "./benefitgithubrepository.js";
import { BenefitGrantError$inboundSchema, BenefitGrantError$outboundSchema, } from "./benefitgranterror.js";
import { BenefitGrantGitHubRepositoryProperties$inboundSchema, BenefitGrantGitHubRepositoryProperties$outboundSchema, } from "./benefitgrantgithubrepositoryproperties.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { Member$inboundSchema, Member$outboundSchema, } from "./member.js";
/** @internal */
export const BenefitGrantGitHubRepositoryWebhook$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    granted_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_granted: z.boolean(),
    revoked_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    customer_id: z.string(),
    member_id: z.optional(z.nullable(z.string())),
    benefit_id: z.string(),
    error: z.optional(z.nullable(BenefitGrantError$inboundSchema)),
    customer: Customer$inboundSchema,
    member: z.optional(z.nullable(Member$inboundSchema)),
    benefit: BenefitGitHubRepository$inboundSchema,
    properties: BenefitGrantGitHubRepositoryProperties$inboundSchema,
    previous_properties: z.optional(z.nullable(BenefitGrantGitHubRepositoryProperties$inboundSchema)),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "customer_id": "customerId",
        "member_id": "memberId",
        "benefit_id": "benefitId",
        "previous_properties": "previousProperties",
    });
}));
/** @internal */
export const BenefitGrantGitHubRepositoryWebhook$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    grantedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    isGranted: z.boolean(),
    revokedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    isRevoked: z.boolean(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    customerId: z.string(),
    memberId: z.optional(z.nullable(z.string())),
    benefitId: z.string(),
    error: z.optional(z.nullable(BenefitGrantError$outboundSchema)),
    customer: Customer$outboundSchema,
    member: z.optional(z.nullable(Member$outboundSchema)),
    benefit: BenefitGitHubRepository$outboundSchema,
    properties: BenefitGrantGitHubRepositoryProperties$outboundSchema,
    previousProperties: z.optional(z.nullable(BenefitGrantGitHubRepositoryProperties$outboundSchema)),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        isGranted: "is_granted",
        revokedAt: "revoked_at",
        isRevoked: "is_revoked",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        customerId: "customer_id",
        memberId: "member_id",
        benefitId: "benefit_id",
        previousProperties: "previous_properties",
    });
}));
export function benefitGrantGitHubRepositoryWebhookToJSON(benefitGrantGitHubRepositoryWebhook) {
    return JSON.stringify(BenefitGrantGitHubRepositoryWebhook$outboundSchema.parse(benefitGrantGitHubRepositoryWebhook));
}
export function benefitGrantGitHubRepositoryWebhookFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantGitHubRepositoryWebhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantGitHubRepositoryWebhook' from JSON`);
}
