// @polar-sh/sdk/dist/esm/models/components/benefitgithubrepositoryproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * The permission level to grant. Read more about roles and their permissions on [GitHub documentation](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role).
 */
export const Permission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
export const Permission$inboundSchema = openEnums.inboundSchema(Permission);
/** @internal */
export const Permission$outboundSchema = openEnums.outboundSchema(Permission);
/** @internal */
export const BenefitGitHubRepositoryProperties$inboundSchema = z.pipe(z.object({
    repository_owner: z.string(),
    repository_name: z.string(),
    permission: Permission$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
    });
}));
/** @internal */
export const BenefitGitHubRepositoryProperties$outboundSchema = z.pipe(z.object({
    repositoryOwner: z.string(),
    repositoryName: z.string(),
    permission: Permission$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
    });
}));
export function benefitGitHubRepositoryPropertiesToJSON(benefitGitHubRepositoryProperties) {
    return JSON.stringify(BenefitGitHubRepositoryProperties$outboundSchema.parse(benefitGitHubRepositoryProperties));
}
export function benefitGitHubRepositoryPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGitHubRepositoryProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryProperties' from JSON`);
}
