// @polar-sh/sdk/dist/esm/models/components/benefitgrantmetercreditproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitGrantMeterCreditProperties$inboundSchema = z.pipe(z.object({
    last_credited_meter_id: z.optional(z.string()),
    last_credited_units: z.optional(z.int()),
    last_credited_at: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        "last_credited_meter_id": "lastCreditedMeterId",
        "last_credited_units": "lastCreditedUnits",
        "last_credited_at": "lastCreditedAt",
    });
}));
/** @internal */
export const BenefitGrantMeterCreditProperties$outboundSchema = z.pipe(z.object({
    lastCreditedMeterId: z.optional(z.string()),
    lastCreditedUnits: z.optional(z.int()),
    lastCreditedAt: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        lastCreditedMeterId: "last_credited_meter_id",
        lastCreditedUnits: "last_credited_units",
        lastCreditedAt: "last_credited_at",
    });
}));
export function benefitGrantMeterCreditPropertiesToJSON(benefitGrantMeterCreditProperties) {
    return JSON.stringify(BenefitGrantMeterCreditProperties$outboundSchema.parse(benefitGrantMeterCreditProperties));
}
export function benefitGrantMeterCreditPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantMeterCreditProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantMeterCreditProperties' from JSON`);
}
