// @polar-sh/sdk/dist/esm/models/components/benefitcustomcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitCustomCreateProperties$outboundSchema, } from "./benefitcustomcreateproperties.js";
/** @internal */
export const BenefitCustomCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitCustomCreateMetadataToJSON(benefitCustomCreateMetadata) {
    return JSON.stringify(BenefitCustomCreateMetadata$outboundSchema.parse(benefitCustomCreateMetadata));
}
/** @internal */
export const BenefitCustomCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("custom"),
    description: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: BenefitCustomCreateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function benefitCustomCreateToJSON(benefitCustomCreate) {
    return JSON.stringify(BenefitCustomCreate$outboundSchema.parse(benefitCustomCreate));
}
