// @polar-sh/sdk/dist/esm/models/components/benefitfeatureflagupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitFeatureFlagProperties$outboundSchema, } from "./benefitfeatureflagproperties.js";
/** @internal */
export const BenefitFeatureFlagUpdateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitFeatureFlagUpdateMetadataToJSON(benefitFeatureFlagUpdateMetadata) {
    return JSON.stringify(BenefitFeatureFlagUpdateMetadata$outboundSchema.parse(benefitFeatureFlagUpdateMetadata));
}
/** @internal */
export const BenefitFeatureFlagUpdate$outboundSchema = z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    description: z.optional(z.nullable(z.string())),
    type: z.literal("feature_flag"),
    properties: z.optional(z.nullable(BenefitFeatureFlagProperties$outboundSchema)),
});
export function benefitFeatureFlagUpdateToJSON(benefitFeatureFlagUpdate) {
    return JSON.stringify(BenefitFeatureFlagUpdate$outboundSchema.parse(benefitFeatureFlagUpdate));
}
