// @polar-sh/sdk/dist/esm/models/components/benefitpublic.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitType$inboundSchema, BenefitType$outboundSchema, } from "./benefittype.js";
/** @internal */
export const BenefitPublic$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: BenefitType$inboundSchema,
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const BenefitPublic$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    type: BenefitType$outboundSchema,
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function benefitPublicToJSON(benefitPublic) {
    return JSON.stringify(BenefitPublic$outboundSchema.parse(benefitPublic));
}
export function benefitPublicFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitPublic$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitPublic' from JSON`);
}
