// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablescreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitDownloadablesCreateProperties$outboundSchema, } from "./benefitdownloadablescreateproperties.js";
/** @internal */
export const BenefitDownloadablesCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitDownloadablesCreateMetadataToJSON(benefitDownloadablesCreateMetadata) {
    return JSON.stringify(BenefitDownloadablesCreateMetadata$outboundSchema.parse(benefitDownloadablesCreateMetadata));
}
/** @internal */
export const BenefitDownloadablesCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("downloadables"),
    description: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: BenefitDownloadablesCreateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function benefitDownloadablesCreateToJSON(benefitDownloadablesCreate) {
    return JSON.stringify(BenefitDownloadablesCreate$outboundSchema.parse(benefitDownloadablesCreate));
}
