// @polar-sh/sdk/dist/esm/models/components/benefitmetercredit.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitMeterCreditProperties$inboundSchema, BenefitMeterCreditProperties$outboundSchema, } from "./benefitmetercreditproperties.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const BenefitMeterCredit$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: z.literal("meter_credit"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    properties: BenefitMeterCreditProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const BenefitMeterCredit$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    type: z.literal("meter_credit"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    properties: BenefitMeterCreditProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function benefitMeterCreditToJSON(benefitMeterCredit) {
    return JSON.stringify(BenefitMeterCredit$outboundSchema.parse(benefitMeterCredit));
}
export function benefitMeterCreditFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCredit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCredit' from JSON`);
}
