// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BenefitMeterCreditProperties$inboundSchema = z.pipe(z.object({
    units: z.int(),
    rollover: z.boolean(),
    meter_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "meter_id": "meterId",
    });
}));
/** @internal */
export const BenefitMeterCreditProperties$outboundSchema = z.pipe(z.object({
    units: z.int(),
    rollover: z.boolean(),
    meterId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        meterId: "meter_id",
    });
}));
export function benefitMeterCreditPropertiesToJSON(benefitMeterCreditProperties) {
    return JSON.stringify(BenefitMeterCreditProperties$outboundSchema.parse(benefitMeterCreditProperties));
}
export function benefitMeterCreditPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCreditProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditProperties' from JSON`);
}
