// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitMeterCreditCreateProperties$outboundSchema, } from "./benefitmetercreditcreateproperties.js";
/** @internal */
export const BenefitMeterCreditUpdateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitMeterCreditUpdateMetadataToJSON(benefitMeterCreditUpdateMetadata) {
    return JSON.stringify(BenefitMeterCreditUpdateMetadata$outboundSchema.parse(benefitMeterCreditUpdateMetadata));
}
/** @internal */
export const BenefitMeterCreditUpdate$outboundSchema = z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    description: z.optional(z.nullable(z.string())),
    type: z.literal("meter_credit"),
    properties: z.optional(z.nullable(BenefitMeterCreditCreateProperties$outboundSchema)),
});
export function benefitMeterCreditUpdateToJSON(benefitMeterCreditUpdate) {
    return JSON.stringify(BenefitMeterCreditUpdate$outboundSchema.parse(benefitMeterCreditUpdate));
}
