// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyexpirationproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
export const Timeframe = {
    Year: "year",
    Month: "month",
    Day: "day",
};
/** @internal */
export const Timeframe$inboundSchema = openEnums.inboundSchema(Timeframe);
/** @internal */
export const Timeframe$outboundSchema = openEnums.outboundSchema(Timeframe);
/** @internal */
export const BenefitLicenseKeyExpirationProperties$inboundSchema = z.object({
    ttl: z.int(),
    timeframe: Timeframe$inboundSchema,
});
/** @internal */
export const BenefitLicenseKeyExpirationProperties$outboundSchema = z.object({
    ttl: z.int(),
    timeframe: Timeframe$outboundSchema,
});
export function benefitLicenseKeyExpirationPropertiesToJSON(benefitLicenseKeyExpirationProperties) {
    return JSON.stringify(BenefitLicenseKeyExpirationProperties$outboundSchema.parse(benefitLicenseKeyExpirationProperties));
}
export function benefitLicenseKeyExpirationPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeyExpirationProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeyExpirationProperties' from JSON`);
}
