// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeyssubscriberproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitLicenseKeyActivationProperties$inboundSchema, } from "./benefitlicensekeyactivationproperties.js";
import { BenefitLicenseKeyExpirationProperties$inboundSchema, } from "./benefitlicensekeyexpirationproperties.js";
/** @internal */
export const BenefitLicenseKeysSubscriberProperties$inboundSchema = z.pipe(z.object({
    prefix: z.nullable(z.string()),
    expires: z.nullable(BenefitLicenseKeyExpirationProperties$inboundSchema),
    activations: z.nullable(BenefitLicenseKeyActivationProperties$inboundSchema),
    limit_usage: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "limit_usage": "limitUsage",
    });
}));
export function benefitLicenseKeysSubscriberPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitLicenseKeysSubscriberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitLicenseKeysSubscriberProperties' from JSON`);
}
