// @polar-sh/sdk/dist/esm/models/components/benefitfeatureflagcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitFeatureFlagCreateProperties$outboundSchema, } from "./benefitfeatureflagcreateproperties.js";
/** @internal */
export const BenefitFeatureFlagCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitFeatureFlagCreateMetadataToJSON(benefitFeatureFlagCreateMetadata) {
    return JSON.stringify(BenefitFeatureFlagCreateMetadata$outboundSchema.parse(benefitFeatureFlagCreateMetadata));
}
/** @internal */
export const BenefitFeatureFlagCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("feature_flag"),
    description: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: BenefitFeatureFlagCreateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function benefitFeatureFlagCreateToJSON(benefitFeatureFlagCreate) {
    return JSON.stringify(BenefitFeatureFlagCreate$outboundSchema.parse(benefitFeatureFlagCreate));
}
