// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditsubscriber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitMeterCreditSubscriberProperties$inboundSchema, } from "./benefitmetercreditsubscriberproperties.js";
import { BenefitSubscriberOrganization$inboundSchema, } from "./benefitsubscriberorganization.js";
import { MetadataOutputType$inboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const BenefitMeterCreditSubscriber$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: z.literal("meter_credit"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    organization: BenefitSubscriberOrganization$inboundSchema,
    properties: BenefitMeterCreditSubscriberProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
export function benefitMeterCreditSubscriberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitMeterCreditSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitMeterCreditSubscriber' from JSON`);
}
