// @polar-sh/sdk/dist/esm/models/components/benefitgrantgithubrepositoryproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
export const BenefitGrantGitHubRepositoryPropertiesPermission = {
    Pull: "pull",
    Triage: "triage",
    Push: "push",
    Maintain: "maintain",
    Admin: "admin",
};
/** @internal */
export const BenefitGrantGitHubRepositoryPropertiesPermission$inboundSchema = openEnums.inboundSchema(BenefitGrantGitHubRepositoryPropertiesPermission);
/** @internal */
export const BenefitGrantGitHubRepositoryPropertiesPermission$outboundSchema = openEnums.outboundSchema(BenefitGrantGitHubRepositoryPropertiesPermission);
/** @internal */
export const BenefitGrantGitHubRepositoryProperties$inboundSchema = z.pipe(z.object({
    account_id: z.optional(z.nullable(z.string())),
    repository_owner: z.optional(z.string()),
    repository_name: z.optional(z.string()),
    permission: z.optional(BenefitGrantGitHubRepositoryPropertiesPermission$inboundSchema),
    granted_account_id: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        "account_id": "accountId",
        "repository_owner": "repositoryOwner",
        "repository_name": "repositoryName",
        "granted_account_id": "grantedAccountId",
    });
}));
/** @internal */
export const BenefitGrantGitHubRepositoryProperties$outboundSchema = z.pipe(z.object({
    accountId: z.optional(z.nullable(z.string())),
    repositoryOwner: z.optional(z.string()),
    repositoryName: z.optional(z.string()),
    permission: z.optional(BenefitGrantGitHubRepositoryPropertiesPermission$outboundSchema),
    grantedAccountId: z.optional(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        accountId: "account_id",
        repositoryOwner: "repository_owner",
        repositoryName: "repository_name",
        grantedAccountId: "granted_account_id",
    });
}));
export function benefitGrantGitHubRepositoryPropertiesToJSON(benefitGrantGitHubRepositoryProperties) {
    return JSON.stringify(BenefitGrantGitHubRepositoryProperties$outboundSchema.parse(benefitGrantGitHubRepositoryProperties));
}
export function benefitGrantGitHubRepositoryPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGrantGitHubRepositoryProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantGitHubRepositoryProperties' from JSON`);
}
