// @polar-sh/sdk/dist/esm/models/components/cardpayment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CardPaymentMetadata$inboundSchema, } from "./cardpaymentmetadata.js";
import { PaymentProcessor$inboundSchema, } from "./paymentprocessor.js";
import { PaymentStatus$inboundSchema } from "./paymentstatus.js";
/** @internal */
export const CardPayment$inboundSchema = z
    .pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    processor: PaymentProcessor$inboundSchema,
    status: PaymentStatus$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    method: z.literal("card"),
    decline_reason: z.nullable(z.string()),
    decline_message: z.nullable(z.string()),
    organization_id: z.string(),
    checkout_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    processor_metadata: z.optional(z.record(z.string(), z.any())),
    method_metadata: CardPaymentMetadata$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "decline_reason": "declineReason",
        "decline_message": "declineMessage",
        "organization_id": "organizationId",
        "checkout_id": "checkoutId",
        "order_id": "orderId",
        "processor_metadata": "processorMetadata",
        "method_metadata": "methodMetadata",
    });
}));
export function cardPaymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CardPayment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CardPayment' from JSON`);
}
