// @polar-sh/sdk/dist/esm/models/components/checkout.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
import { AttachedCustomField$inboundSchema, AttachedCustomField$outboundSchema, } from "./attachedcustomfield.js";
import { CheckoutBillingAddressFields$inboundSchema, CheckoutBillingAddressFields$outboundSchema, } from "./checkoutbillingaddressfields.js";
import { CheckoutDiscountFixedOnceForeverDuration$inboundSchema, CheckoutDiscountFixedOnceForeverDuration$outboundSchema, } from "./checkoutdiscountfixedonceforeverduration.js";
import { CheckoutDiscountFixedRepeatDuration$inboundSchema, CheckoutDiscountFixedRepeatDuration$outboundSchema, } from "./checkoutdiscountfixedrepeatduration.js";
import { CheckoutDiscountPercentageOnceForeverDuration$inboundSchema, CheckoutDiscountPercentageOnceForeverDuration$outboundSchema, } from "./checkoutdiscountpercentageonceforeverduration.js";
import { CheckoutDiscountPercentageRepeatDuration$inboundSchema, CheckoutDiscountPercentageRepeatDuration$outboundSchema, } from "./checkoutdiscountpercentagerepeatduration.js";
import { CheckoutProduct$inboundSchema, CheckoutProduct$outboundSchema, } from "./checkoutproduct.js";
import { CheckoutStatus$inboundSchema, CheckoutStatus$outboundSchema, } from "./checkoutstatus.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
import { PaymentProcessor$inboundSchema, PaymentProcessor$outboundSchema, } from "./paymentprocessor.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { TrialInterval$inboundSchema, TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CheckoutCustomFieldData$inboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
export const CheckoutCustomFieldData$outboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
export function checkoutCustomFieldDataToJSON(checkoutCustomFieldData) {
    return JSON.stringify(CheckoutCustomFieldData$outboundSchema.parse(checkoutCustomFieldData));
}
export function checkoutCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutCustomFieldData' from JSON`);
}
/** @internal */
export const CheckoutProductPrice$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CheckoutProductPrice$outboundSchema = smartUnion([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
export function checkoutProductPriceToJSON(checkoutProductPrice) {
    return JSON.stringify(CheckoutProductPrice$outboundSchema.parse(checkoutProductPrice));
}
export function checkoutProductPriceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductPrice' from JSON`);
}
/** @internal */
export const CheckoutPrices$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const CheckoutPrices$outboundSchema = smartUnion([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
export function checkoutPricesToJSON(checkoutPrices) {
    return JSON.stringify(CheckoutPrices$outboundSchema.parse(checkoutPrices));
}
export function checkoutPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPrices' from JSON`);
}
/** @internal */
export const CheckoutDiscount$inboundSchema = smartUnion([
    CheckoutDiscountFixedRepeatDuration$inboundSchema,
    CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
    CheckoutDiscountPercentageRepeatDuration$inboundSchema,
    CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
]);
/** @internal */
export const CheckoutDiscount$outboundSchema = smartUnion([
    CheckoutDiscountFixedRepeatDuration$outboundSchema,
    CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
    CheckoutDiscountPercentageRepeatDuration$outboundSchema,
    CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
]);
export function checkoutDiscountToJSON(checkoutDiscount) {
    return JSON.stringify(CheckoutDiscount$outboundSchema.parse(checkoutDiscount));
}
export function checkoutDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscount' from JSON`);
}
/** @internal */
export const CustomerMetadata$inboundSchema = smartUnion([z.string(), z.int(), z.boolean()]);
/** @internal */
export const CustomerMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.boolean()]);
export function customerMetadataToJSON(customerMetadata) {
    return JSON.stringify(CustomerMetadata$outboundSchema.parse(customerMetadata));
}
export function customerMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerMetadata' from JSON`);
}
/** @internal */
export const Checkout$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    payment_processor: PaymentProcessor$inboundSchema,
    status: CheckoutStatus$inboundSchema,
    client_secret: z.string(),
    url: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    success_url: z.string(),
    return_url: z.nullable(z.string()),
    embed_origin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    min_seats: z.optional(z.nullable(z.int())),
    max_seats: z.optional(z.nullable(z.int())),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.nullable(z.int()),
    total_amount: z.int(),
    currency: z.string(),
    allow_trial: z.nullable(z.boolean()),
    active_trial_interval: z.nullable(TrialInterval$inboundSchema),
    active_trial_interval_count: z.nullable(z.int()),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    organization_id: z.string(),
    product_id: z.nullable(z.string()),
    product_price_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    require_billing_address: z.boolean(),
    is_discount_applicable: z.boolean(),
    is_free_product_price: z.boolean(),
    is_payment_required: z.boolean(),
    is_payment_setup_required: z.boolean(),
    is_payment_form_required: z.boolean(),
    customer_id: z.nullable(z.string()),
    is_business_customer: z.boolean(),
    customer_name: z.nullable(z.string()),
    customer_email: z.nullable(z.string()),
    customer_ip_address: z.nullable(z.string()),
    customer_billing_name: z.nullable(z.string()),
    customer_billing_address: z.nullable(Address$inboundSchema),
    customer_tax_id: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    payment_processor_metadata: z.record(z.string(), z.string()),
    billing_address_fields: CheckoutBillingAddressFields$inboundSchema,
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.int()),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    products: z.array(CheckoutProduct$inboundSchema),
    product: z.nullable(CheckoutProduct$inboundSchema),
    product_price: z.nullable(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])))),
    discount: z.nullable(smartUnion([
        CheckoutDiscountFixedRepeatDuration$inboundSchema,
        CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
        CheckoutDiscountPercentageRepeatDuration$inboundSchema,
        CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
    ])),
    subscription_id: z.nullable(z.string()),
    attached_custom_fields: z.nullable(z.array(AttachedCustomField$inboundSchema)),
    customer_metadata: z.record(z.string(), smartUnion([z.string(), z.int(), z.boolean()])),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "expires_at": "expiresAt",
        "success_url": "successUrl",
        "return_url": "returnUrl",
        "embed_origin": "embedOrigin",
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "allow_trial": "allowTrial",
        "active_trial_interval": "activeTrialInterval",
        "active_trial_interval_count": "activeTrialIntervalCount",
        "trial_end": "trialEnd",
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "is_discount_applicable": "isDiscountApplicable",
        "is_free_product_price": "isFreeProductPrice",
        "is_payment_required": "isPaymentRequired",
        "is_payment_setup_required": "isPaymentSetupRequired",
        "is_payment_form_required": "isPaymentFormRequired",
        "customer_id": "customerId",
        "is_business_customer": "isBusinessCustomer",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_name": "customerBillingName",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "payment_processor_metadata": "paymentProcessorMetadata",
        "billing_address_fields": "billingAddressFields",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "external_customer_id": "externalCustomerId",
        "product_price": "productPrice",
        "subscription_id": "subscriptionId",
        "attached_custom_fields": "attachedCustomFields",
        "customer_metadata": "customerMetadata",
    });
}));
/** @internal */
export const Checkout$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    customFieldData: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    paymentProcessor: PaymentProcessor$outboundSchema,
    status: CheckoutStatus$outboundSchema,
    clientSecret: z.string(),
    url: z.string(),
    expiresAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    successUrl: z.string(),
    returnUrl: z.nullable(z.string()),
    embedOrigin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    minSeats: z.optional(z.nullable(z.int())),
    maxSeats: z.optional(z.nullable(z.int())),
    discountAmount: z.int(),
    netAmount: z.int(),
    taxAmount: z.nullable(z.int()),
    totalAmount: z.int(),
    currency: z.string(),
    allowTrial: z.nullable(z.boolean()),
    activeTrialInterval: z.nullable(TrialInterval$outboundSchema),
    activeTrialIntervalCount: z.nullable(z.int()),
    trialEnd: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    organizationId: z.string(),
    productId: z.nullable(z.string()),
    productPriceId: z.nullable(z.string()),
    discountId: z.nullable(z.string()),
    allowDiscountCodes: z.boolean(),
    requireBillingAddress: z.boolean(),
    isDiscountApplicable: z.boolean(),
    isFreeProductPrice: z.boolean(),
    isPaymentRequired: z.boolean(),
    isPaymentSetupRequired: z.boolean(),
    isPaymentFormRequired: z.boolean(),
    customerId: z.nullable(z.string()),
    isBusinessCustomer: z.boolean(),
    customerName: z.nullable(z.string()),
    customerEmail: z.nullable(z.string()),
    customerIpAddress: z.nullable(z.string()),
    customerBillingName: z.nullable(z.string()),
    customerBillingAddress: z.nullable(Address$outboundSchema),
    customerTaxId: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    paymentProcessorMetadata: z.record(z.string(), z.string()),
    billingAddressFields: CheckoutBillingAddressFields$outboundSchema,
    trialInterval: z.nullable(TrialInterval$outboundSchema),
    trialIntervalCount: z.nullable(z.int()),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    products: z.array(CheckoutProduct$outboundSchema),
    product: z.nullable(CheckoutProduct$outboundSchema),
    productPrice: z.nullable(smartUnion([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array(smartUnion([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ])))),
    discount: z.nullable(smartUnion([
        CheckoutDiscountFixedRepeatDuration$outboundSchema,
        CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
        CheckoutDiscountPercentageRepeatDuration$outboundSchema,
        CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
    ])),
    subscriptionId: z.nullable(z.string()),
    attachedCustomFields: z.nullable(z.array(AttachedCustomField$outboundSchema)),
    customerMetadata: z.record(z.string(), smartUnion([z.string(), z.int(), z.boolean()])),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customFieldData: "custom_field_data",
        paymentProcessor: "payment_processor",
        clientSecret: "client_secret",
        expiresAt: "expires_at",
        successUrl: "success_url",
        returnUrl: "return_url",
        embedOrigin: "embed_origin",
        minSeats: "min_seats",
        maxSeats: "max_seats",
        discountAmount: "discount_amount",
        netAmount: "net_amount",
        taxAmount: "tax_amount",
        totalAmount: "total_amount",
        allowTrial: "allow_trial",
        activeTrialInterval: "active_trial_interval",
        activeTrialIntervalCount: "active_trial_interval_count",
        trialEnd: "trial_end",
        organizationId: "organization_id",
        productId: "product_id",
        productPriceId: "product_price_id",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        isDiscountApplicable: "is_discount_applicable",
        isFreeProductPrice: "is_free_product_price",
        isPaymentRequired: "is_payment_required",
        isPaymentSetupRequired: "is_payment_setup_required",
        isPaymentFormRequired: "is_payment_form_required",
        customerId: "customer_id",
        isBusinessCustomer: "is_business_customer",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingName: "customer_billing_name",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        paymentProcessorMetadata: "payment_processor_metadata",
        billingAddressFields: "billing_address_fields",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        externalCustomerId: "external_customer_id",
        productPrice: "product_price",
        subscriptionId: "subscription_id",
        attachedCustomFields: "attached_custom_fields",
        customerMetadata: "customer_metadata",
    });
}));
export function checkoutToJSON(checkout) {
    return JSON.stringify(Checkout$outboundSchema.parse(checkout));
}
export function checkoutFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Checkout$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Checkout' from JSON`);
}
