// @polar-sh/sdk/dist/esm/models/components/checkoutlinkcreateproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CheckoutLinkCreateProductMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function checkoutLinkCreateProductMetadataToJSON(checkoutLinkCreateProductMetadata) {
    return JSON.stringify(CheckoutLinkCreateProductMetadata$outboundSchema.parse(checkoutLinkCreateProductMetadata));
}
/** @internal */
export const CheckoutLinkCreateProduct$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    trialInterval: z.optional(z.nullable(TrialInterval$outboundSchema)),
    trialIntervalCount: z.optional(z.nullable(z.int())),
    paymentProcessor: z.literal("stripe"),
    label: z.optional(z.nullable(z.string())),
    allowDiscountCodes: z._default(z.boolean(), true),
    requireBillingAddress: z._default(z.boolean(), false),
    discountId: z.optional(z.nullable(z.string())),
    successUrl: z.optional(z.nullable(z.string())),
    returnUrl: z.optional(z.nullable(z.string())),
    productId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
        returnUrl: "return_url",
        productId: "product_id",
    });
}));
export function checkoutLinkCreateProductToJSON(checkoutLinkCreateProduct) {
    return JSON.stringify(CheckoutLinkCreateProduct$outboundSchema.parse(checkoutLinkCreateProduct));
}
