// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountpercentagerepeatduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountPercentageRepeatDuration$inboundSchema = z.pipe(z.object({
    duration: DiscountDuration$inboundSchema,
    duration_in_months: z.int(),
    type: DiscountType$inboundSchema,
    basis_points: z.int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
    });
}));
/** @internal */
export const CheckoutDiscountPercentageRepeatDuration$outboundSchema = z.pipe(z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: DiscountType$outboundSchema,
    basisPoints: z.int(),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
    });
}));
export function checkoutDiscountPercentageRepeatDurationToJSON(checkoutDiscountPercentageRepeatDuration) {
    return JSON.stringify(CheckoutDiscountPercentageRepeatDuration$outboundSchema.parse(checkoutDiscountPercentageRepeatDuration));
}
export function checkoutDiscountPercentageRepeatDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountPercentageRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountPercentageRepeatDuration' from JSON`);
}
