// @polar-sh/sdk/dist/esm/models/components/checkoutlinkupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CheckoutLinkUpdateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function checkoutLinkUpdateMetadataToJSON(checkoutLinkUpdateMetadata) {
    return JSON.stringify(CheckoutLinkUpdateMetadata$outboundSchema.parse(checkoutLinkUpdateMetadata));
}
/** @internal */
export const CheckoutLinkUpdate$outboundSchema = z.pipe(z.object({
    trialInterval: z.optional(z.nullable(TrialInterval$outboundSchema)),
    trialIntervalCount: z.optional(z.nullable(z.int())),
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    products: z.optional(z.nullable(z.array(z.string()))),
    label: z.optional(z.nullable(z.string())),
    allowDiscountCodes: z.optional(z.nullable(z.boolean())),
    requireBillingAddress: z.optional(z.nullable(z.boolean())),
    discountId: z.optional(z.nullable(z.string())),
    successUrl: z.optional(z.nullable(z.string())),
    returnUrl: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        discountId: "discount_id",
        successUrl: "success_url",
        returnUrl: "return_url",
    });
}));
export function checkoutLinkUpdateToJSON(checkoutLinkUpdate) {
    return JSON.stringify(CheckoutLinkUpdate$outboundSchema.parse(checkoutLinkUpdate));
}
