// @polar-sh/sdk/dist/esm/models/components/checkoutpublic.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema } from "./address.js";
import { AttachedCustomField$inboundSchema, } from "./attachedcustomfield.js";
import { CheckoutBillingAddressFields$inboundSchema, } from "./checkoutbillingaddressfields.js";
import { CheckoutDiscountFixedOnceForeverDuration$inboundSchema, } from "./checkoutdiscountfixedonceforeverduration.js";
import { CheckoutDiscountFixedRepeatDuration$inboundSchema, } from "./checkoutdiscountfixedrepeatduration.js";
import { CheckoutDiscountPercentageOnceForeverDuration$inboundSchema, } from "./checkoutdiscountpercentageonceforeverduration.js";
import { CheckoutDiscountPercentageRepeatDuration$inboundSchema, } from "./checkoutdiscountpercentagerepeatduration.js";
import { CheckoutOrganization$inboundSchema, } from "./checkoutorganization.js";
import { CheckoutProduct$inboundSchema, } from "./checkoutproduct.js";
import { CheckoutStatus$inboundSchema, } from "./checkoutstatus.js";
import { LegacyRecurringProductPrice$inboundSchema, } from "./legacyrecurringproductprice.js";
import { PaymentProcessor$inboundSchema, } from "./paymentprocessor.js";
import { ProductPrice$inboundSchema } from "./productprice.js";
import { TrialInterval$inboundSchema } from "./trialinterval.js";
/** @internal */
export const CheckoutPublicCustomFieldData$inboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
export function checkoutPublicCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPublicCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicCustomFieldData' from JSON`);
}
/** @internal */
export const CheckoutPublicProductPrice$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
export function checkoutPublicProductPriceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPublicProductPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicProductPrice' from JSON`);
}
/** @internal */
export const CheckoutPublicPrices$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
export function checkoutPublicPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPublicPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicPrices' from JSON`);
}
/** @internal */
export const CheckoutPublicDiscount$inboundSchema = smartUnion([
    CheckoutDiscountFixedRepeatDuration$inboundSchema,
    CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
    CheckoutDiscountPercentageRepeatDuration$inboundSchema,
    CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
]);
export function checkoutPublicDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPublicDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublicDiscount' from JSON`);
}
/** @internal */
export const CheckoutPublic$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    payment_processor: PaymentProcessor$inboundSchema,
    status: CheckoutStatus$inboundSchema,
    client_secret: z.string(),
    url: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    success_url: z.string(),
    return_url: z.nullable(z.string()),
    embed_origin: z.nullable(z.string()),
    amount: z.int(),
    seats: z.optional(z.nullable(z.int())),
    min_seats: z.optional(z.nullable(z.int())),
    max_seats: z.optional(z.nullable(z.int())),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.nullable(z.int()),
    total_amount: z.int(),
    currency: z.string(),
    allow_trial: z.nullable(z.boolean()),
    active_trial_interval: z.nullable(TrialInterval$inboundSchema),
    active_trial_interval_count: z.nullable(z.int()),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    organization_id: z.string(),
    product_id: z.nullable(z.string()),
    product_price_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    require_billing_address: z.boolean(),
    is_discount_applicable: z.boolean(),
    is_free_product_price: z.boolean(),
    is_payment_required: z.boolean(),
    is_payment_setup_required: z.boolean(),
    is_payment_form_required: z.boolean(),
    customer_id: z.nullable(z.string()),
    is_business_customer: z.boolean(),
    customer_name: z.nullable(z.string()),
    customer_email: z.nullable(z.string()),
    customer_ip_address: z.nullable(z.string()),
    customer_billing_name: z.nullable(z.string()),
    customer_billing_address: z.nullable(Address$inboundSchema),
    customer_tax_id: z.nullable(z.string()),
    locale: z.optional(z.nullable(z.string())),
    payment_processor_metadata: z.record(z.string(), z.string()),
    billing_address_fields: CheckoutBillingAddressFields$inboundSchema,
    products: z.array(CheckoutProduct$inboundSchema),
    product: z.nullable(CheckoutProduct$inboundSchema),
    product_price: z.nullable(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    prices: z.nullable(z.record(z.string(), z.array(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])))),
    discount: z.nullable(smartUnion([
        CheckoutDiscountFixedRepeatDuration$inboundSchema,
        CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
        CheckoutDiscountPercentageRepeatDuration$inboundSchema,
        CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
    ])),
    organization: CheckoutOrganization$inboundSchema,
    attached_custom_fields: z.nullable(z.array(AttachedCustomField$inboundSchema)),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "expires_at": "expiresAt",
        "success_url": "successUrl",
        "return_url": "returnUrl",
        "embed_origin": "embedOrigin",
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "allow_trial": "allowTrial",
        "active_trial_interval": "activeTrialInterval",
        "active_trial_interval_count": "activeTrialIntervalCount",
        "trial_end": "trialEnd",
        "organization_id": "organizationId",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "is_discount_applicable": "isDiscountApplicable",
        "is_free_product_price": "isFreeProductPrice",
        "is_payment_required": "isPaymentRequired",
        "is_payment_setup_required": "isPaymentSetupRequired",
        "is_payment_form_required": "isPaymentFormRequired",
        "customer_id": "customerId",
        "is_business_customer": "isBusinessCustomer",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_name": "customerBillingName",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "payment_processor_metadata": "paymentProcessorMetadata",
        "billing_address_fields": "billingAddressFields",
        "product_price": "productPrice",
        "attached_custom_fields": "attachedCustomFields",
    });
}));
export function checkoutPublicFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutPublic$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPublic' from JSON`);
}
