// @polar-sh/sdk/dist/esm/models/components/checkoutlink.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CheckoutLinkProduct$inboundSchema, } from "./checkoutlinkproduct.js";
import { DiscountFixedOnceForeverDurationBase$inboundSchema, } from "./discountfixedonceforeverdurationbase.js";
import { DiscountFixedRepeatDurationBase$inboundSchema, } from "./discountfixedrepeatdurationbase.js";
import { DiscountPercentageOnceForeverDurationBase$inboundSchema, } from "./discountpercentageonceforeverdurationbase.js";
import { DiscountPercentageRepeatDurationBase$inboundSchema, } from "./discountpercentagerepeatdurationbase.js";
import { MetadataOutputType$inboundSchema, } from "./metadataoutputtype.js";
import { PaymentProcessor$inboundSchema, } from "./paymentprocessor.js";
import { TrialInterval$inboundSchema } from "./trialinterval.js";
/** @internal */
export const CheckoutLinkDiscount$inboundSchema = smartUnion([
    DiscountFixedRepeatDurationBase$inboundSchema,
    DiscountFixedOnceForeverDurationBase$inboundSchema,
    DiscountPercentageRepeatDurationBase$inboundSchema,
    DiscountPercentageOnceForeverDurationBase$inboundSchema,
]);
export function checkoutLinkDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLinkDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkDiscount' from JSON`);
}
/** @internal */
export const CheckoutLink$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.int()),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    payment_processor: PaymentProcessor$inboundSchema,
    client_secret: z.string(),
    success_url: z.nullable(z.string()),
    return_url: z.nullable(z.string()),
    label: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    require_billing_address: z.boolean(),
    discount_id: z.nullable(z.string()),
    organization_id: z.string(),
    products: z.array(CheckoutLinkProduct$inboundSchema),
    discount: z.nullable(smartUnion([
        DiscountFixedRepeatDurationBase$inboundSchema,
        DiscountFixedOnceForeverDurationBase$inboundSchema,
        DiscountPercentageRepeatDurationBase$inboundSchema,
        DiscountPercentageOnceForeverDurationBase$inboundSchema,
    ])),
    url: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "success_url": "successUrl",
        "return_url": "returnUrl",
        "allow_discount_codes": "allowDiscountCodes",
        "require_billing_address": "requireBillingAddress",
        "discount_id": "discountId",
        "organization_id": "organizationId",
    });
}));
export function checkoutLinkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutLink$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLink' from JSON`);
}
