// @polar-sh/sdk/dist/esm/models/components/checkoutcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { AddressInput$outboundSchema, } from "./addressinput.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
import { ProductPriceCustomCreate$outboundSchema, } from "./productpricecustomcreate.js";
import { ProductPriceFixedCreate$outboundSchema, } from "./productpricefixedcreate.js";
import { ProductPriceFreeCreate$outboundSchema, } from "./productpricefreecreate.js";
import { ProductPriceMeteredUnitCreate$outboundSchema, } from "./productpricemeteredunitcreate.js";
import { ProductPriceSeatBasedCreate$outboundSchema, } from "./productpriceseatbasedcreate.js";
import { TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const CheckoutCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function checkoutCreateMetadataToJSON(checkoutCreateMetadata) {
    return JSON.stringify(CheckoutCreateMetadata$outboundSchema.parse(checkoutCreateMetadata));
}
/** @internal */
export const CheckoutCreateCustomFieldData$outboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
export function checkoutCreateCustomFieldDataToJSON(checkoutCreateCustomFieldData) {
    return JSON.stringify(CheckoutCreateCustomFieldData$outboundSchema.parse(checkoutCreateCustomFieldData));
}
/** @internal */
export const CheckoutCreateCustomerMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function checkoutCreateCustomerMetadataToJSON(checkoutCreateCustomerMetadata) {
    return JSON.stringify(CheckoutCreateCustomerMetadata$outboundSchema.parse(checkoutCreateCustomerMetadata));
}
/** @internal */
export const CheckoutCreatePrices$outboundSchema = z.union([
    ProductPriceCustomCreate$outboundSchema,
    ProductPriceFixedCreate$outboundSchema,
    ProductPriceFreeCreate$outboundSchema,
    ProductPriceMeteredUnitCreate$outboundSchema,
    ProductPriceSeatBasedCreate$outboundSchema,
]);
export function checkoutCreatePricesToJSON(checkoutCreatePrices) {
    return JSON.stringify(CheckoutCreatePrices$outboundSchema.parse(checkoutCreatePrices));
}
/** @internal */
export const CheckoutCreate$outboundSchema = z.pipe(z.object({
    trialInterval: z.optional(z.nullable(TrialInterval$outboundSchema)),
    trialIntervalCount: z.optional(z.nullable(z.int())),
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    customFieldData: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    discountId: z.optional(z.nullable(z.string())),
    allowDiscountCodes: z._default(z.boolean(), true),
    requireBillingAddress: z._default(z.boolean(), false),
    amount: z.optional(z.nullable(z.int())),
    seats: z.optional(z.nullable(z.int())),
    minSeats: z.optional(z.nullable(z.int())),
    maxSeats: z.optional(z.nullable(z.int())),
    allowTrial: z._default(z.boolean(), true),
    customerId: z.optional(z.nullable(z.string())),
    isBusinessCustomer: z._default(z.boolean(), false),
    externalCustomerId: z.optional(z.nullable(z.string())),
    customerName: z.optional(z.nullable(z.string())),
    customerEmail: z.optional(z.nullable(z.string())),
    customerIpAddress: z.optional(z.nullable(z.string())),
    customerBillingName: z.optional(z.nullable(z.string())),
    customerBillingAddress: z.optional(z.nullable(AddressInput$outboundSchema)),
    customerTaxId: z.optional(z.nullable(z.string())),
    customerMetadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    subscriptionId: z.optional(z.nullable(z.string())),
    successUrl: z.optional(z.nullable(z.string())),
    returnUrl: z.optional(z.nullable(z.string())),
    embedOrigin: z.optional(z.nullable(z.string())),
    locale: z.optional(z.nullable(z.string())),
    currency: z.optional(z.nullable(PresentmentCurrency$outboundSchema)),
    products: z.array(z.string()),
    prices: z.optional(z.nullable(z.record(z.string(), z.array(z.union([
        ProductPriceCustomCreate$outboundSchema,
        ProductPriceFixedCreate$outboundSchema,
        ProductPriceFreeCreate$outboundSchema,
        ProductPriceMeteredUnitCreate$outboundSchema,
        ProductPriceSeatBasedCreate$outboundSchema,
    ]))))),
}), z.transform((v) => {
    return remap$(v, {
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        customFieldData: "custom_field_data",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        requireBillingAddress: "require_billing_address",
        minSeats: "min_seats",
        maxSeats: "max_seats",
        allowTrial: "allow_trial",
        customerId: "customer_id",
        isBusinessCustomer: "is_business_customer",
        externalCustomerId: "external_customer_id",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingName: "customer_billing_name",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        customerMetadata: "customer_metadata",
        subscriptionId: "subscription_id",
        successUrl: "success_url",
        returnUrl: "return_url",
        embedOrigin: "embed_origin",
    });
}));
export function checkoutCreateToJSON(checkoutCreate) {
    return JSON.stringify(CheckoutCreate$outboundSchema.parse(checkoutCreate));
}
