// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountfixedrepeatduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountFixedRepeatDuration$inboundSchema = z.pipe(z.object({
    duration: DiscountDuration$inboundSchema,
    duration_in_months: z.int(),
    type: DiscountType$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        "duration_in_months": "durationInMonths",
    });
}));
/** @internal */
export const CheckoutDiscountFixedRepeatDuration$outboundSchema = z.pipe(z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: DiscountType$outboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
    });
}));
export function checkoutDiscountFixedRepeatDurationToJSON(checkoutDiscountFixedRepeatDuration) {
    return JSON.stringify(CheckoutDiscountFixedRepeatDuration$outboundSchema.parse(checkoutDiscountFixedRepeatDuration));
}
export function checkoutDiscountFixedRepeatDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountFixedRepeatDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedRepeatDuration' from JSON`);
}
