// @polar-sh/sdk/dist/esm/models/components/checkoutdiscountfixedonceforeverduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
/** @internal */
export const CheckoutDiscountFixedOnceForeverDuration$inboundSchema = z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
/** @internal */
export const CheckoutDiscountFixedOnceForeverDuration$outboundSchema = z.object({
    duration: DiscountDuration$outboundSchema,
    type: DiscountType$outboundSchema,
    amount: z.int(),
    currency: z.string(),
    amounts: z.record(z.string(), z.int()),
    id: z.string(),
    name: z.string(),
    code: z.nullable(z.string()),
});
export function checkoutDiscountFixedOnceForeverDurationToJSON(checkoutDiscountFixedOnceForeverDuration) {
    return JSON.stringify(CheckoutDiscountFixedOnceForeverDuration$outboundSchema.parse(checkoutDiscountFixedOnceForeverDuration));
}
export function checkoutDiscountFixedOnceForeverDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutDiscountFixedOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscountFixedOnceForeverDuration' from JSON`);
}
