// @polar-sh/sdk/dist/esm/models/components/customfieldcreateselect.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomFieldSelectProperties$outboundSchema, } from "./customfieldselectproperties.js";
/** @internal */
export const CustomFieldCreateSelectMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldCreateSelectMetadataToJSON(customFieldCreateSelectMetadata) {
    return JSON.stringify(CustomFieldCreateSelectMetadata$outboundSchema.parse(customFieldCreateSelectMetadata));
}
/** @internal */
export const CustomFieldCreateSelect$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("select"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: CustomFieldSelectProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customFieldCreateSelectToJSON(customFieldCreateSelect) {
    return JSON.stringify(CustomFieldCreateSelect$outboundSchema.parse(customFieldCreateSelect));
}
