// @polar-sh/sdk/dist/esm/models/components/customerstatebenefitgrant.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitGrantCustomProperties$inboundSchema, BenefitGrantCustomProperties$outboundSchema, } from "./benefitgrantcustomproperties.js";
import { BenefitGrantDiscordProperties$inboundSchema, BenefitGrantDiscordProperties$outboundSchema, } from "./benefitgrantdiscordproperties.js";
import { BenefitGrantDownloadablesProperties$inboundSchema, BenefitGrantDownloadablesProperties$outboundSchema, } from "./benefitgrantdownloadablesproperties.js";
import { BenefitGrantFeatureFlagProperties$inboundSchema, BenefitGrantFeatureFlagProperties$outboundSchema, } from "./benefitgrantfeatureflagproperties.js";
import { BenefitGrantGitHubRepositoryProperties$inboundSchema, BenefitGrantGitHubRepositoryProperties$outboundSchema, } from "./benefitgrantgithubrepositoryproperties.js";
import { BenefitGrantLicenseKeysProperties$inboundSchema, BenefitGrantLicenseKeysProperties$outboundSchema, } from "./benefitgrantlicensekeysproperties.js";
import { BenefitType$inboundSchema, BenefitType$outboundSchema, } from "./benefittype.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const CustomerStateBenefitGrantProperties$inboundSchema = smartUnion([
    BenefitGrantDiscordProperties$inboundSchema,
    BenefitGrantGitHubRepositoryProperties$inboundSchema,
    BenefitGrantDownloadablesProperties$inboundSchema,
    BenefitGrantLicenseKeysProperties$inboundSchema,
    BenefitGrantCustomProperties$inboundSchema,
    BenefitGrantFeatureFlagProperties$inboundSchema,
]);
/** @internal */
export const CustomerStateBenefitGrantProperties$outboundSchema = smartUnion([
    BenefitGrantDiscordProperties$outboundSchema,
    BenefitGrantGitHubRepositoryProperties$outboundSchema,
    BenefitGrantDownloadablesProperties$outboundSchema,
    BenefitGrantLicenseKeysProperties$outboundSchema,
    BenefitGrantCustomProperties$outboundSchema,
    BenefitGrantFeatureFlagProperties$outboundSchema,
]);
export function customerStateBenefitGrantPropertiesToJSON(customerStateBenefitGrantProperties) {
    return JSON.stringify(CustomerStateBenefitGrantProperties$outboundSchema.parse(customerStateBenefitGrantProperties));
}
export function customerStateBenefitGrantPropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateBenefitGrantProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateBenefitGrantProperties' from JSON`);
}
/** @internal */
export const CustomerStateBenefitGrant$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    granted_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    benefit_id: z.string(),
    benefit_type: BenefitType$inboundSchema,
    benefit_metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    properties: smartUnion([
        BenefitGrantDiscordProperties$inboundSchema,
        BenefitGrantGitHubRepositoryProperties$inboundSchema,
        BenefitGrantDownloadablesProperties$inboundSchema,
        BenefitGrantLicenseKeysProperties$inboundSchema,
        BenefitGrantCustomProperties$inboundSchema,
        BenefitGrantFeatureFlagProperties$inboundSchema,
    ]),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "benefit_id": "benefitId",
        "benefit_type": "benefitType",
        "benefit_metadata": "benefitMetadata",
    });
}));
/** @internal */
export const CustomerStateBenefitGrant$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    grantedAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    benefitId: z.string(),
    benefitType: BenefitType$outboundSchema,
    benefitMetadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    properties: smartUnion([
        BenefitGrantDiscordProperties$outboundSchema,
        BenefitGrantGitHubRepositoryProperties$outboundSchema,
        BenefitGrantDownloadablesProperties$outboundSchema,
        BenefitGrantLicenseKeysProperties$outboundSchema,
        BenefitGrantCustomProperties$outboundSchema,
        BenefitGrantFeatureFlagProperties$outboundSchema,
    ]),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        benefitId: "benefit_id",
        benefitType: "benefit_type",
        benefitMetadata: "benefit_metadata",
    });
}));
export function customerStateBenefitGrantToJSON(customerStateBenefitGrant) {
    return JSON.stringify(CustomerStateBenefitGrant$outboundSchema.parse(customerStateBenefitGrant));
}
export function customerStateBenefitGrantFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateBenefitGrant$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateBenefitGrant' from JSON`);
}
