// @polar-sh/sdk/dist/esm/models/components/customersubscription.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerCancellationReason$inboundSchema, } from "./customercancellationreason.js";
import { CustomerSubscriptionMeter$inboundSchema, } from "./customersubscriptionmeter.js";
import { CustomerSubscriptionProduct$inboundSchema, } from "./customersubscriptionproduct.js";
import { LegacyRecurringProductPrice$inboundSchema, } from "./legacyrecurringproductprice.js";
import { PendingSubscriptionUpdate$inboundSchema, } from "./pendingsubscriptionupdate.js";
import { ProductPrice$inboundSchema } from "./productprice.js";
import { SubscriptionRecurringInterval$inboundSchema, } from "./subscriptionrecurringinterval.js";
import { SubscriptionStatus$inboundSchema, } from "./subscriptionstatus.js";
/** @internal */
export const CustomerSubscriptionPrices$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
export function customerSubscriptionPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionPrices' from JSON`);
}
/** @internal */
export const CustomerSubscription$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    amount: z.int(),
    currency: z.string(),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
    recurring_interval_count: z.int(),
    status: SubscriptionStatus$inboundSchema,
    current_period_start: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    current_period_end: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    trial_start: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    cancel_at_period_end: z.boolean(),
    canceled_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    started_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ended_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    customer_id: z.string(),
    product_id: z.string(),
    discount_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    seats: z.optional(z.nullable(z.int())),
    customer_cancellation_reason: z.nullable(CustomerCancellationReason$inboundSchema),
    customer_cancellation_comment: z.nullable(z.string()),
    product: CustomerSubscriptionProduct$inboundSchema,
    prices: z.array(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    meters: z.array(CustomerSubscriptionMeter$inboundSchema),
    pending_update: z.nullable(PendingSubscriptionUpdate$inboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "current_period_start": "currentPeriodStart",
        "current_period_end": "currentPeriodEnd",
        "trial_start": "trialStart",
        "trial_end": "trialEnd",
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "canceled_at": "canceledAt",
        "started_at": "startedAt",
        "ends_at": "endsAt",
        "ended_at": "endedAt",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "checkout_id": "checkoutId",
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "pending_update": "pendingUpdate",
    });
}));
export function customerSubscriptionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscription' from JSON`);
}
