// @polar-sh/sdk/dist/esm/models/components/customerorder.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Address$inboundSchema } from "./address.js";
import { CustomerOrderProduct$inboundSchema, } from "./customerorderproduct.js";
import { CustomerOrderSubscription$inboundSchema, } from "./customerordersubscription.js";
import { OrderBillingReason$inboundSchema, } from "./orderbillingreason.js";
import { OrderItemSchema$inboundSchema, } from "./orderitemschema.js";
import { OrderStatus$inboundSchema } from "./orderstatus.js";
/** @internal */
export const CustomerOrder$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    status: OrderStatus$inboundSchema,
    paid: z.boolean(),
    subtotal_amount: z.int(),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.int(),
    total_amount: z.int(),
    applied_balance_amount: z.int(),
    due_amount: z.int(),
    refunded_amount: z.int(),
    refunded_tax_amount: z.int(),
    currency: z.string(),
    billing_reason: OrderBillingReason$inboundSchema,
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    invoice_number: z.string(),
    is_invoice_generated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customer_id: z.string(),
    product_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    subscription_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    product: z.nullable(CustomerOrderProduct$inboundSchema),
    subscription: z.nullable(CustomerOrderSubscription$inboundSchema),
    items: z.array(OrderItemSchema$inboundSchema),
    description: z.string(),
    next_payment_attempt_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subtotal_amount": "subtotalAmount",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "applied_balance_amount": "appliedBalanceAmount",
        "due_amount": "dueAmount",
        "refunded_amount": "refundedAmount",
        "refunded_tax_amount": "refundedTaxAmount",
        "billing_reason": "billingReason",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "invoice_number": "invoiceNumber",
        "is_invoice_generated": "isInvoiceGenerated",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "next_payment_attempt_at": "nextPaymentAttemptAt",
    });
}));
export function customerOrderFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrder$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrder' from JSON`);
}
