// @polar-sh/sdk/dist/esm/models/components/customerorganization.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerOrganizationFeatureSettings$inboundSchema, } from "./customerorganizationfeaturesettings.js";
import { OrganizationCustomerPortalSettings$inboundSchema, } from "./organizationcustomerportalsettings.js";
import { SubscriptionProrationBehavior$inboundSchema, } from "./subscriptionprorationbehavior.js";
/** @internal */
export const CustomerOrganization$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()),
    proration_behavior: SubscriptionProrationBehavior$inboundSchema,
    allow_customer_updates: z.boolean(),
    customer_portal_settings: OrganizationCustomerPortalSettings$inboundSchema,
    organization_features: z.optional(CustomerOrganizationFeatureSettings$inboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "avatar_url": "avatarUrl",
        "proration_behavior": "prorationBehavior",
        "allow_customer_updates": "allowCustomerUpdates",
        "customer_portal_settings": "customerPortalSettings",
        "organization_features": "organizationFeatures",
    });
}));
export function customerOrganizationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerOrganization$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerOrganization' from JSON`);
}
