// @polar-sh/sdk/dist/esm/models/components/customerstateteam.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
import { CustomerStateBenefitGrant$inboundSchema, CustomerStateBenefitGrant$outboundSchema, } from "./customerstatebenefitgrant.js";
import { CustomerStateMeter$inboundSchema, CustomerStateMeter$outboundSchema, } from "./customerstatemeter.js";
import { CustomerStateSubscription$inboundSchema, CustomerStateSubscription$outboundSchema, } from "./customerstatesubscription.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
import { TaxIDFormat$inboundSchema, TaxIDFormat$outboundSchema, } from "./taxidformat.js";
/** @internal */
export const CustomerStateTeamTaxId$inboundSchema = smartUnion([z.string(), TaxIDFormat$inboundSchema]);
/** @internal */
export const CustomerStateTeamTaxId$outboundSchema = smartUnion([z.string(), TaxIDFormat$outboundSchema]);
export function customerStateTeamTaxIdToJSON(customerStateTeamTaxId) {
    return JSON.stringify(CustomerStateTeamTaxId$outboundSchema.parse(customerStateTeamTaxId));
}
export function customerStateTeamTaxIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateTeamTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateTeamTaxId' from JSON`);
}
/** @internal */
export const CustomerStateTeam$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    external_id: z.optional(z.nullable(z.string())),
    email: z.optional(z.nullable(z.string())),
    email_verified: z.boolean(),
    type: z.literal("team"),
    name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(smartUnion([z.string(), TaxIDFormat$inboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
    deleted_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    active_subscriptions: z.array(CustomerStateSubscription$inboundSchema),
    granted_benefits: z.array(CustomerStateBenefitGrant$inboundSchema),
    active_meters: z.array(CustomerStateMeter$inboundSchema),
    avatar_url: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "deleted_at": "deletedAt",
        "active_subscriptions": "activeSubscriptions",
        "granted_benefits": "grantedBenefits",
        "active_meters": "activeMeters",
        "avatar_url": "avatarUrl",
    });
}));
/** @internal */
export const CustomerStateTeam$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    externalId: z.optional(z.nullable(z.string())),
    email: z.optional(z.nullable(z.string())),
    emailVerified: z.boolean(),
    type: z.literal("team"),
    name: z.nullable(z.string()),
    billingAddress: z.nullable(Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable(smartUnion([z.string(), TaxIDFormat$outboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organizationId: z.string(),
    deletedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    activeSubscriptions: z.array(CustomerStateSubscription$outboundSchema),
    grantedBenefits: z.array(CustomerStateBenefitGrant$outboundSchema),
    activeMeters: z.array(CustomerStateMeter$outboundSchema),
    avatarUrl: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        externalId: "external_id",
        emailVerified: "email_verified",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
        deletedAt: "deleted_at",
        activeSubscriptions: "active_subscriptions",
        grantedBenefits: "granted_benefits",
        activeMeters: "active_meters",
        avatarUrl: "avatar_url",
    });
}));
export function customerStateTeamToJSON(customerStateTeam) {
    return JSON.stringify(CustomerStateTeam$outboundSchema.parse(customerStateTeam));
}
export function customerStateTeamFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateTeam$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateTeam' from JSON`);
}
