// @polar-sh/sdk/dist/esm/models/components/customfielddateproperties.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomFieldDateProperties$inboundSchema = z.pipe(z.object({
    form_label: z.optional(z.string()),
    form_help_text: z.optional(z.string()),
    form_placeholder: z.optional(z.string()),
    ge: z.optional(z.int()),
    le: z.optional(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
}));
/** @internal */
export const CustomFieldDateProperties$outboundSchema = z.pipe(z.object({
    formLabel: z.optional(z.string()),
    formHelpText: z.optional(z.string()),
    formPlaceholder: z.optional(z.string()),
    ge: z.optional(z.int()),
    le: z.optional(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
}));
export function customFieldDatePropertiesToJSON(customFieldDateProperties) {
    return JSON.stringify(CustomFieldDateProperties$outboundSchema.parse(customFieldDateProperties));
}
export function customFieldDatePropertiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldDateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldDateProperties' from JSON`);
}
