// @polar-sh/sdk/dist/esm/models/components/customersubscriptionmeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSubscriptionMeterMeter$inboundSchema, } from "./customersubscriptionmetermeter.js";
/** @internal */
export const CustomerSubscriptionMeter$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    consumed_units: z.number(),
    credited_units: z.int(),
    amount: z.int(),
    meter_id: z.string(),
    meter: CustomerSubscriptionMeterMeter$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
        "meter_id": "meterId",
    });
}));
export function customerSubscriptionMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionMeter' from JSON`);
}
