// @polar-sh/sdk/dist/esm/models/components/customerstatesubscription.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomerStateSubscriptionMeter$inboundSchema, CustomerStateSubscriptionMeter$outboundSchema, } from "./customerstatesubscriptionmeter.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
export const Status = {
    Active: "active",
    Trialing: "trialing",
};
/** @internal */
export const CustomerStateSubscriptionCustomFieldData$inboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
export const CustomerStateSubscriptionCustomFieldData$outboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
export function customerStateSubscriptionCustomFieldDataToJSON(customerStateSubscriptionCustomFieldData) {
    return JSON.stringify(CustomerStateSubscriptionCustomFieldData$outboundSchema.parse(customerStateSubscriptionCustomFieldData));
}
export function customerStateSubscriptionCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateSubscriptionCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateSubscriptionCustomFieldData' from JSON`);
}
/** @internal */
export const Status$inboundSchema = openEnums
    .inboundSchema(Status);
/** @internal */
export const Status$outboundSchema = openEnums
    .outboundSchema(Status);
/** @internal */
export const CustomerStateSubscription$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    custom_field_data: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    status: Status$inboundSchema,
    amount: z.int(),
    currency: z.string(),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
    current_period_start: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    current_period_end: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    trial_start: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_end: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    cancel_at_period_end: z.boolean(),
    canceled_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    started_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    product_id: z.string(),
    discount_id: z.nullable(z.string()),
    meters: z.array(CustomerStateSubscriptionMeter$inboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "recurring_interval": "recurringInterval",
        "current_period_start": "currentPeriodStart",
        "current_period_end": "currentPeriodEnd",
        "trial_start": "trialStart",
        "trial_end": "trialEnd",
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "canceled_at": "canceledAt",
        "started_at": "startedAt",
        "ends_at": "endsAt",
        "product_id": "productId",
        "discount_id": "discountId",
    });
}));
/** @internal */
export const CustomerStateSubscription$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    customFieldData: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    status: Status$outboundSchema,
    amount: z.int(),
    currency: z.string(),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
    currentPeriodStart: z.pipe(z.date(), z.transform(v => v.toISOString())),
    currentPeriodEnd: z.pipe(z.date(), z.transform(v => v.toISOString())),
    trialStart: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    trialEnd: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    cancelAtPeriodEnd: z.boolean(),
    canceledAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    startedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    endsAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    productId: z.string(),
    discountId: z.nullable(z.string()),
    meters: z.array(CustomerStateSubscriptionMeter$outboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customFieldData: "custom_field_data",
        recurringInterval: "recurring_interval",
        currentPeriodStart: "current_period_start",
        currentPeriodEnd: "current_period_end",
        trialStart: "trial_start",
        trialEnd: "trial_end",
        cancelAtPeriodEnd: "cancel_at_period_end",
        canceledAt: "canceled_at",
        startedAt: "started_at",
        endsAt: "ends_at",
        productId: "product_id",
        discountId: "discount_id",
    });
}));
export function customerStateSubscriptionToJSON(customerStateSubscription) {
    return JSON.stringify(CustomerStateSubscription$outboundSchema.parse(customerStateSubscription));
}
export function customerStateSubscriptionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerStateSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerStateSubscription' from JSON`);
}
