// @polar-sh/sdk/dist/esm/models/components/customfieldcheckbox.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldCheckboxProperties$inboundSchema, CustomFieldCheckboxProperties$outboundSchema, } from "./customfieldcheckboxproperties.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const CustomFieldCheckbox$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.string(),
    properties: CustomFieldCheckboxProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const CustomFieldCheckbox$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.string(),
    properties: CustomFieldCheckboxProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function customFieldCheckboxToJSON(customFieldCheckbox) {
    return JSON.stringify(CustomFieldCheckbox$outboundSchema.parse(customFieldCheckbox));
}
export function customFieldCheckboxFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldCheckbox' from JSON`);
}
