// @polar-sh/sdk/dist/esm/models/components/customerdeletedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema } from "./customer.js";
import { CustomerDeletedMetadata$inboundSchema, } from "./customerdeletedmetadata.js";
/** @internal */
export const CustomerDeletedEvent$inboundSchema = z.pipe(z.object({
    id: z.string(),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    member_id: z.optional(z.nullable(z.string())),
    external_member_id: z.optional(z.nullable(z.string())),
    child_count: z._default(z.int(), 0),
    parent_id: z.optional(z.nullable(z.string())),
    label: z.string(),
    source: z.literal("system"),
    name: z.literal("customer.deleted"),
    metadata: CustomerDeletedMetadata$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
        "member_id": "memberId",
        "external_member_id": "externalMemberId",
        "child_count": "childCount",
        "parent_id": "parentId",
    });
}));
export function customerDeletedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerDeletedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerDeletedEvent' from JSON`);
}
