// @polar-sh/sdk/dist/esm/models/components/customerportalmember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { MemberRole$inboundSchema } from "./memberrole.js";
/** @internal */
export const CustomerPortalMember$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    email: z.string(),
    name: z.nullable(z.string()),
    role: MemberRole$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
    });
}));
export function customerPortalMemberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalMember$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalMember' from JSON`);
}
