// @polar-sh/sdk/dist/esm/models/components/customerindividualcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { AddressInput$outboundSchema, } from "./addressinput.js";
import { MemberOwnerCreate$outboundSchema, } from "./memberownercreate.js";
/** @internal */
export const CustomerIndividualCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function customerIndividualCreateMetadataToJSON(customerIndividualCreateMetadata) {
    return JSON.stringify(CustomerIndividualCreateMetadata$outboundSchema.parse(customerIndividualCreateMetadata));
}
/** @internal */
export const CustomerIndividualCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    externalId: z.optional(z.nullable(z.string())),
    name: z.optional(z.nullable(z.string())),
    billingAddress: z.optional(z.nullable(AddressInput$outboundSchema)),
    taxId: z.optional(z.nullable(z.string())),
    locale: z.optional(z.nullable(z.string())),
    organizationId: z.optional(z.nullable(z.string())),
    owner: z.optional(z.nullable(MemberOwnerCreate$outboundSchema)),
    type: z._default(z.literal("individual"), "individual"),
    email: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        externalId: "external_id",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
    });
}));
export function customerIndividualCreateToJSON(customerIndividualCreate) {
    return JSON.stringify(CustomerIndividualCreate$outboundSchema.parse(customerIndividualCreate));
}
