// @polar-sh/sdk/dist/esm/models/components/customfieldcreatecheckbox.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CustomFieldCheckboxProperties$outboundSchema, } from "./customfieldcheckboxproperties.js";
/** @internal */
export const CustomFieldCreateCheckboxMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldCreateCheckboxMetadataToJSON(customFieldCreateCheckboxMetadata) {
    return JSON.stringify(CustomFieldCreateCheckboxMetadata$outboundSchema.parse(customFieldCreateCheckboxMetadata));
}
/** @internal */
export const CustomFieldCreateCheckbox$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("checkbox"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: CustomFieldCheckboxProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customFieldCreateCheckboxToJSON(customFieldCreateCheckbox) {
    return JSON.stringify(CustomFieldCreateCheckbox$outboundSchema.parse(customFieldCreateCheckbox));
}
