// @polar-sh/sdk/dist/esm/models/components/customerportalsubscriptionsettings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalSubscriptionSettings$inboundSchema = z.pipe(z.object({
    update_seats: z.boolean(),
    update_plan: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "update_seats": "updateSeats",
        "update_plan": "updatePlan",
    });
}));
/** @internal */
export const CustomerPortalSubscriptionSettings$outboundSchema = z.pipe(z.object({
    updateSeats: z.boolean(),
    updatePlan: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        updateSeats: "update_seats",
        updatePlan: "update_plan",
    });
}));
export function customerPortalSubscriptionSettingsToJSON(customerPortalSubscriptionSettings) {
    return JSON.stringify(CustomerPortalSubscriptionSettings$outboundSchema.parse(customerPortalSubscriptionSettings));
}
export function customerPortalSubscriptionSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalSubscriptionSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalSubscriptionSettings' from JSON`);
}
