// @polar-sh/sdk/dist/esm/models/components/customersubscriptionproduct.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { BenefitPublic$inboundSchema } from "./benefitpublic.js";
import { CustomerOrganization$inboundSchema, } from "./customerorganization.js";
import { LegacyRecurringProductPrice$inboundSchema, } from "./legacyrecurringproductprice.js";
import { ProductMediaFileRead$inboundSchema, } from "./productmediafileread.js";
import { ProductPrice$inboundSchema } from "./productprice.js";
import { ProductVisibility$inboundSchema, } from "./productvisibility.js";
import { SubscriptionRecurringInterval$inboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$inboundSchema } from "./trialinterval.js";
/** @internal */
export const CustomerSubscriptionProductPrices$inboundSchema = smartUnion([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
export function customerSubscriptionProductPricesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionProductPrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionProductPrices' from JSON`);
}
/** @internal */
export const CustomerSubscriptionProduct$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    trial_interval: z.nullable(TrialInterval$inboundSchema),
    trial_interval_count: z.nullable(z.int()),
    name: z.string(),
    description: z.nullable(z.string()),
    visibility: ProductVisibility$inboundSchema,
    recurring_interval: z.nullable(SubscriptionRecurringInterval$inboundSchema),
    recurring_interval_count: z.nullable(z.int()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(smartUnion([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ])),
    benefits: z.array(BenefitPublic$inboundSchema),
    medias: z.array(ProductMediaFileRead$inboundSchema),
    organization: CustomerOrganization$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "trial_interval": "trialInterval",
        "trial_interval_count": "trialIntervalCount",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
}));
export function customerSubscriptionProductFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSubscriptionProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSubscriptionProduct' from JSON`);
}
