// @polar-sh/sdk/dist/esm/models/components/discountpercentagerepeatdurationbase.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$inboundSchema, DiscountType$outboundSchema, } from "./discounttype.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const DiscountPercentageRepeatDurationBase$inboundSchema = z.pipe(z.object({
    duration: DiscountDuration$inboundSchema,
    duration_in_months: z.int(),
    type: DiscountType$inboundSchema,
    basis_points: z.int(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    max_redemptions: z.nullable(z.int()),
    redemptions_count: z.int(),
    organization_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "duration_in_months": "durationInMonths",
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const DiscountPercentageRepeatDurationBase$outboundSchema = z.pipe(z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: DiscountType$outboundSchema,
    basisPoints: z.int(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    name: z.string(),
    code: z.nullable(z.string()),
    startsAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    endsAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    maxRedemptions: z.nullable(z.int()),
    redemptionsCount: z.int(),
    organizationId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        redemptionsCount: "redemptions_count",
        organizationId: "organization_id",
    });
}));
export function discountPercentageRepeatDurationBaseToJSON(discountPercentageRepeatDurationBase) {
    return JSON.stringify(DiscountPercentageRepeatDurationBase$outboundSchema.parse(discountPercentageRepeatDurationBase));
}
export function discountPercentageRepeatDurationBaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountPercentageRepeatDurationBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageRepeatDurationBase' from JSON`);
}
