// @polar-sh/sdk/dist/esm/models/components/discountpercentagerepeatdurationcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$outboundSchema } from "./discounttype.js";
/** @internal */
export const DiscountPercentageRepeatDurationCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function discountPercentageRepeatDurationCreateMetadataToJSON(discountPercentageRepeatDurationCreateMetadata) {
    return JSON.stringify(DiscountPercentageRepeatDurationCreateMetadata$outboundSchema.parse(discountPercentageRepeatDurationCreateMetadata));
}
/** @internal */
export const DiscountPercentageRepeatDurationCreate$outboundSchema = z.pipe(z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: DiscountType$outboundSchema,
    basisPoints: z.int(),
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.string(),
    code: z.optional(z.nullable(z.string())),
    startsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    maxRedemptions: z.optional(z.nullable(z.int())),
    products: z.optional(z.nullable(z.array(z.string()))),
    organizationId: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        organizationId: "organization_id",
    });
}));
export function discountPercentageRepeatDurationCreateToJSON(discountPercentageRepeatDurationCreate) {
    return JSON.stringify(DiscountPercentageRepeatDurationCreate$outboundSchema.parse(discountPercentageRepeatDurationCreate));
}
