// @polar-sh/sdk/dist/esm/models/components/discountupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$outboundSchema } from "./discounttype.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
/** @internal */
export const DiscountUpdateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function discountUpdateMetadataToJSON(discountUpdateMetadata) {
    return JSON.stringify(DiscountUpdateMetadata$outboundSchema.parse(discountUpdateMetadata));
}
/** @internal */
export const DiscountUpdate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.optional(z.nullable(z.string())),
    code: z.optional(z.nullable(z.string())),
    startsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    maxRedemptions: z.optional(z.nullable(z.int())),
    duration: z.optional(z.nullable(DiscountDuration$outboundSchema)),
    durationInMonths: z.optional(z.nullable(z.int())),
    type: z.optional(z.nullable(DiscountType$outboundSchema)),
    amount: z.optional(z.nullable(z.int())),
    currency: z.optional(z.nullable(PresentmentCurrency$outboundSchema)),
    amounts: z.optional(z.nullable(z.record(z.string(), z.int()))),
    basisPoints: z.optional(z.nullable(z.int())),
    products: z.optional(z.nullable(z.array(z.string()))),
}), z.transform((v) => {
    return remap$(v, {
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        durationInMonths: "duration_in_months",
        basisPoints: "basis_points",
    });
}));
export function discountUpdateToJSON(discountUpdate) {
    return JSON.stringify(DiscountUpdate$outboundSchema.parse(discountUpdate));
}
