// @polar-sh/sdk/dist/esm/models/components/discountpercentageonceforeverduration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DiscountDuration$inboundSchema, } from "./discountduration.js";
import { DiscountProduct$inboundSchema, } from "./discountproduct.js";
import { DiscountType$inboundSchema } from "./discounttype.js";
import { MetadataOutputType$inboundSchema, } from "./metadataoutputtype.js";
/** @internal */
export const DiscountPercentageOnceForeverDuration$inboundSchema = z.pipe(z.object({
    duration: DiscountDuration$inboundSchema,
    type: DiscountType$inboundSchema,
    basis_points: z.int(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    name: z.string(),
    code: z.nullable(z.string()),
    starts_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    ends_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    max_redemptions: z.nullable(z.int()),
    redemptions_count: z.int(),
    organization_id: z.string(),
    products: z.array(DiscountProduct$inboundSchema),
}), z.transform((v) => {
    return remap$(v, {
        "basis_points": "basisPoints",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "starts_at": "startsAt",
        "ends_at": "endsAt",
        "max_redemptions": "maxRedemptions",
        "redemptions_count": "redemptionsCount",
        "organization_id": "organizationId",
    });
}));
export function discountPercentageOnceForeverDurationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DiscountPercentageOnceForeverDuration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountPercentageOnceForeverDuration' from JSON`);
}
