// @polar-sh/sdk/dist/esm/models/components/discountfixedrepeatdurationcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { DiscountDuration$outboundSchema, } from "./discountduration.js";
import { DiscountType$outboundSchema } from "./discounttype.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
/** @internal */
export const DiscountFixedRepeatDurationCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function discountFixedRepeatDurationCreateMetadataToJSON(discountFixedRepeatDurationCreateMetadata) {
    return JSON.stringify(DiscountFixedRepeatDurationCreateMetadata$outboundSchema.parse(discountFixedRepeatDurationCreateMetadata));
}
/** @internal */
export const DiscountFixedRepeatDurationCreate$outboundSchema = z.pipe(z.object({
    duration: DiscountDuration$outboundSchema,
    durationInMonths: z.int(),
    type: DiscountType$outboundSchema,
    amount: z.optional(z.nullable(z.int())),
    currency: z.optional(z.nullable(PresentmentCurrency$outboundSchema)),
    amounts: z.optional(z.nullable(z.record(z.string(), z.int()))),
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.string(),
    code: z.optional(z.nullable(z.string())),
    startsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endsAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    maxRedemptions: z.optional(z.nullable(z.int())),
    products: z.optional(z.nullable(z.array(z.string()))),
    organizationId: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        durationInMonths: "duration_in_months",
        startsAt: "starts_at",
        endsAt: "ends_at",
        maxRedemptions: "max_redemptions",
        organizationId: "organization_id",
    });
}));
export function discountFixedRepeatDurationCreateToJSON(discountFixedRepeatDurationCreate) {
    return JSON.stringify(DiscountFixedRepeatDurationCreate$outboundSchema.parse(discountFixedRepeatDurationCreate));
}
