// @polar-sh/sdk/dist/esm/models/components/dispute.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DisputeStatus$inboundSchema } from "./disputestatus.js";
/** @internal */
export const Dispute$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    status: DisputeStatus$inboundSchema,
    resolved: z.boolean(),
    closed: z.boolean(),
    amount: z.int(),
    tax_amount: z.int(),
    currency: z.string(),
    order_id: z.string(),
    payment_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "order_id": "orderId",
        "payment_id": "paymentId",
    });
}));
export function disputeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Dispute$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Dispute' from JSON`);
}
