// @polar-sh/sdk/dist/esm/models/components/eventtypewithstats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { EventSource$inboundSchema } from "./eventsource.js";
/** @internal */
export const EventTypeWithStats$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    name: z.string(),
    label: z.string(),
    label_property_selector: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
    source: EventSource$inboundSchema,
    occurrences: z.int(),
    first_seen: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    last_seen: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "label_property_selector": "labelPropertySelector",
        "organization_id": "organizationId",
        "first_seen": "firstSeen",
        "last_seen": "lastSeen",
    });
}));
export function eventTypeWithStatsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventTypeWithStats$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventTypeWithStats' from JSON`);
}
