// @polar-sh/sdk/dist/esm/models/components/eventtype.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const EventType$inboundSchema = z
    .pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    name: z.string(),
    label: z.string(),
    label_property_selector: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "label_property_selector": "labelPropertySelector",
        "organization_id": "organizationId",
    });
}));
export function eventTypeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventType$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventType' from JSON`);
}
