// @polar-sh/sdk/dist/esm/models/components/filterclause.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { FilterOperator$inboundSchema, FilterOperator$outboundSchema, } from "./filteroperator.js";
/** @internal */
export const Value$inboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
]);
/** @internal */
export const Value$outboundSchema = smartUnion([z.string(), z.int(), z.boolean()]);
export function valueToJSON(value) {
    return JSON.stringify(Value$outboundSchema.parse(value));
}
export function valueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Value' from JSON`);
}
/** @internal */
export const FilterClause$inboundSchema = z.object({
    property: z.string(),
    operator: FilterOperator$inboundSchema,
    value: smartUnion([z.string(), z.int(), z.boolean()]),
});
/** @internal */
export const FilterClause$outboundSchema = z.object({
    property: z.string(),
    operator: FilterOperator$outboundSchema,
    value: smartUnion([z.string(), z.int(), z.boolean()]),
});
export function filterClauseToJSON(filterClause) {
    return JSON.stringify(FilterClause$outboundSchema.parse(filterClause));
}
export function filterClauseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilterClause$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilterClause' from JSON`);
}
