// @polar-sh/sdk/dist/esm/models/components/filter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { FilterClause$inboundSchema, FilterClause$outboundSchema, } from "./filterclause.js";
import { FilterConjunction$inboundSchema, FilterConjunction$outboundSchema, } from "./filterconjunction.js";
/** @internal */
export const Filter$inboundSchema = z.object({
    conjunction: FilterConjunction$inboundSchema,
    clauses: z.array(smartUnion([
        FilterClause$inboundSchema,
        z.lazy(() => Filter$inboundSchema),
    ])),
});
/** @internal */
export const Filter$outboundSchema = z
    .object({
    conjunction: FilterConjunction$outboundSchema,
    clauses: z.array(smartUnion([
        FilterClause$outboundSchema,
        z.lazy(() => Filter$outboundSchema),
    ])),
});
export function filterToJSON(filter) {
    return JSON.stringify(Filter$outboundSchema.parse(filter));
}
export function filterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Filter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Filter' from JSON`);
}
/** @internal */
export const Clauses$inboundSchema = smartUnion([FilterClause$inboundSchema, z.lazy(() => Filter$inboundSchema)]);
/** @internal */
export const Clauses$outboundSchema = smartUnion([
    FilterClause$outboundSchema,
    z.lazy(() => Filter$outboundSchema),
]);
export function clausesToJSON(clauses) {
    return JSON.stringify(Clauses$outboundSchema.parse(clauses));
}
export function clausesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Clauses$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Clauses' from JSON`);
}
