// @polar-sh/sdk/dist/esm/models/components/introspecttokenrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
export const IntrospectTokenRequestTokenTypeHint = {
    AccessToken: "access_token",
    RefreshToken: "refresh_token",
};
/** @internal */
export const IntrospectTokenRequestTokenTypeHint$outboundSchema = z.enum(IntrospectTokenRequestTokenTypeHint);
/** @internal */
export const IntrospectTokenRequest$outboundSchema = z.pipe(z.object({
    token: z.string(),
    tokenTypeHint: z.optional(z.nullable(IntrospectTokenRequestTokenTypeHint$outboundSchema)),
    clientId: z.string(),
    clientSecret: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        tokenTypeHint: "token_type_hint",
        clientId: "client_id",
        clientSecret: "client_secret",
    });
}));
export function introspectTokenRequestToJSON(introspectTokenRequest) {
    return JSON.stringify(IntrospectTokenRequest$outboundSchema.parse(introspectTokenRequest));
}
