// @polar-sh/sdk/dist/esm/models/components/member.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { MemberRole$inboundSchema, MemberRole$outboundSchema, } from "./memberrole.js";
/** @internal */
export const Member$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    customer_id: z.string(),
    email: z.string(),
    name: z.nullable(z.string()),
    external_id: z.nullable(z.string()),
    role: MemberRole$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "customer_id": "customerId",
        "external_id": "externalId",
    });
}));
/** @internal */
export const Member$outboundSchema = z
    .pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    customerId: z.string(),
    email: z.string(),
    name: z.nullable(z.string()),
    externalId: z.nullable(z.string()),
    role: MemberRole$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customerId: "customer_id",
        externalId: "external_id",
    });
}));
export function memberToJSON(member) {
    return JSON.stringify(Member$outboundSchema.parse(member));
}
export function memberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Member$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Member' from JSON`);
}
