// @polar-sh/sdk/dist/esm/models/components/metercreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { CountAggregation$outboundSchema, } from "./countaggregation.js";
import { Filter$outboundSchema } from "./filter.js";
import { PropertyAggregation$outboundSchema, } from "./propertyaggregation.js";
import { UniqueAggregation$outboundSchema, } from "./uniqueaggregation.js";
/** @internal */
export const MeterCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function meterCreateMetadataToJSON(meterCreateMetadata) {
    return JSON.stringify(MeterCreateMetadata$outboundSchema.parse(meterCreateMetadata));
}
/** @internal */
export const MeterCreateAggregation$outboundSchema = z.union([
    z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
    CountAggregation$outboundSchema,
    z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
    z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
    z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
    UniqueAggregation$outboundSchema,
]);
export function meterCreateAggregationToJSON(meterCreateAggregation) {
    return JSON.stringify(MeterCreateAggregation$outboundSchema.parse(meterCreateAggregation));
}
/** @internal */
export const MeterCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.string(),
    filter: Filter$outboundSchema,
    aggregation: z.union([
        z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
        CountAggregation$outboundSchema,
        z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
        z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
        z.intersection(PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
        UniqueAggregation$outboundSchema,
    ]),
    organizationId: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function meterCreateToJSON(meterCreate) {
    return JSON.stringify(MeterCreate$outboundSchema.parse(meterCreate));
}
