// @polar-sh/sdk/dist/esm/models/components/metricstotals.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const MetricsTotalsActiveSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsActiveSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsActiveSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsActiveSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsCommittedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCommittedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCommittedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCommittedSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsMonthlyRecurringRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsMonthlyRecurringRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsCommittedMonthlyRecurringRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCommittedMonthlyRecurringRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCommittedMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCommittedMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsAverageRevenuePerUser$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsAverageRevenuePerUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsAverageRevenuePerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsAverageRevenuePerUser' from JSON`);
}
/** @internal */
export const MetricsTotalsCheckouts$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCheckoutsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCheckouts' from JSON`);
}
/** @internal */
export const MetricsTotalsSucceededCheckouts$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsSucceededCheckoutsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsSucceededCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsSucceededCheckouts' from JSON`);
}
/** @internal */
export const MetricsTotalsChurnedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsChurnedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsChurnedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsChurnedSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsChurnRate$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsChurnRateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsChurnRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsChurnRate' from JSON`);
}
/** @internal */
export const MetricsTotalsOrders$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsOrdersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsOrders$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOrders' from JSON`);
}
/** @internal */
export const MetricsTotalsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsNetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsCumulativeRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCumulativeRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCumulativeRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsNetCumulativeRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsNetCumulativeRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNetCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetCumulativeRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsCosts$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCostsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCosts' from JSON`);
}
/** @internal */
export const MetricsTotalsCumulativeCosts$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCumulativeCostsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCumulativeCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCumulativeCosts' from JSON`);
}
/** @internal */
export const MetricsTotalsAverageOrderValue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsAverageOrderValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsAverageOrderValue' from JSON`);
}
/** @internal */
export const MetricsTotalsNetAverageOrderValue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsNetAverageOrderValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNetAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNetAverageOrderValue' from JSON`);
}
/** @internal */
export const MetricsTotalsCostPerUser$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCostPerUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCostPerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCostPerUser' from JSON`);
}
/** @internal */
export const MetricsTotalsActiveUserByEvent$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsActiveUserByEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsActiveUserByEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsActiveUserByEvent' from JSON`);
}
/** @internal */
export const MetricsTotalsOneTimeProducts$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsOneTimeProductsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsOneTimeProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProducts' from JSON`);
}
/** @internal */
export const MetricsTotalsOneTimeProductsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsOneTimeProductsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsOneTimeProductsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProductsRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsOneTimeProductsNetRevenue$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function metricsTotalsOneTimeProductsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsOneTimeProductsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsOneTimeProductsNetRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsNewSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsNewSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNewSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsNewSubscriptionsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsNewSubscriptionsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNewSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptionsRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsNewSubscriptionsNetRevenue$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function metricsTotalsNewSubscriptionsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsNewSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsNewSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsRenewedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsRenewedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsRenewedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsRenewedSubscriptionsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsRenewedSubscriptionsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsRenewedSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptionsRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsRenewedSubscriptionsNetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsRenewedSubscriptionsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsRenewedSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsRenewedSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptions' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsCustomerService$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsCustomerServiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsCustomerService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsCustomerService' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsLowQuality$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsLowQualityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsLowQuality$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsLowQuality' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsMissingFeatures$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsMissingFeaturesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsMissingFeatures$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsMissingFeatures' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsSwitchedService$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsSwitchedServiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsSwitchedService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsSwitchedService' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsTooComplex$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsTooComplexFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsTooComplex$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsTooComplex' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsTooExpensive$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsTooExpensiveFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsTooExpensive$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsTooExpensive' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsUnused$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCanceledSubscriptionsUnusedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsUnused$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsUnused' from JSON`);
}
/** @internal */
export const MetricsTotalsCanceledSubscriptionsOther$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function metricsTotalsCanceledSubscriptionsOtherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCanceledSubscriptionsOther$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCanceledSubscriptionsOther' from JSON`);
}
/** @internal */
export const MetricsTotalsCheckoutsConversion$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCheckoutsConversionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCheckoutsConversion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCheckoutsConversion' from JSON`);
}
/** @internal */
export const MetricsTotalsLtv$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsLtvFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsLtv$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsLtv' from JSON`);
}
/** @internal */
export const MetricsTotalsGrossMargin$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsGrossMarginFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsGrossMargin$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsGrossMargin' from JSON`);
}
/** @internal */
export const MetricsTotalsGrossMarginPercentage$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsGrossMarginPercentageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsGrossMarginPercentage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsGrossMarginPercentage' from JSON`);
}
/** @internal */
export const MetricsTotalsCashflow$inboundSchema = smartUnion([z.int(), z.number()]);
export function metricsTotalsCashflowFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotalsCashflow$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotalsCashflow' from JSON`);
}
/** @internal */
export const MetricsTotals$inboundSchema = z.pipe(z.object({
    active_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    committed_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    monthly_recurring_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    committed_monthly_recurring_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    average_revenue_per_user: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    checkouts: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    succeeded_checkouts: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    churned_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    churn_rate: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    orders: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cumulative_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_cumulative_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    costs: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cumulative_costs: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    average_order_value: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_average_order_value: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cost_per_user: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    active_user_by_event: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_customer_service: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_low_quality: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_missing_features: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_switched_service: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_too_complex: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_too_expensive: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_unused: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_other: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    checkouts_conversion: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    ltv: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    gross_margin: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    gross_margin_percentage: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cashflow: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
}), z.transform((v) => {
    return remap$(v, {
        "active_subscriptions": "activeSubscriptions",
        "committed_subscriptions": "committedSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "succeeded_checkouts": "succeededCheckouts",
        "churned_subscriptions": "churnedSubscriptions",
        "churn_rate": "churnRate",
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "checkouts_conversion": "checkoutsConversion",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
export function metricsTotalsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricsTotals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricsTotals' from JSON`);
}
