// @polar-sh/sdk/dist/esm/models/components/metricperiod.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const ActiveSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function activeSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActiveSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActiveSubscriptions' from JSON`);
}
/** @internal */
export const CommittedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function committedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CommittedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CommittedSubscriptions' from JSON`);
}
/** @internal */
export const MonthlyRecurringRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function monthlyRecurringRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MonthlyRecurringRevenue' from JSON`);
}
/** @internal */
export const CommittedMonthlyRecurringRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function committedMonthlyRecurringRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CommittedMonthlyRecurringRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CommittedMonthlyRecurringRevenue' from JSON`);
}
/** @internal */
export const AverageRevenuePerUser$inboundSchema = smartUnion([z.int(), z.number()]);
export function averageRevenuePerUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AverageRevenuePerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AverageRevenuePerUser' from JSON`);
}
/** @internal */
export const Checkouts$inboundSchema = smartUnion([z.int(), z.number()]);
export function checkoutsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Checkouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Checkouts' from JSON`);
}
/** @internal */
export const SucceededCheckouts$inboundSchema = smartUnion([z.int(), z.number()]);
export function succeededCheckoutsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SucceededCheckouts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SucceededCheckouts' from JSON`);
}
/** @internal */
export const ChurnedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function churnedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChurnedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChurnedSubscriptions' from JSON`);
}
/** @internal */
export const ChurnRate$inboundSchema = smartUnion([z.int(), z.number()]);
export function churnRateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChurnRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChurnRate' from JSON`);
}
/** @internal */
export const Orders$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function ordersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Orders$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Orders' from JSON`);
}
/** @internal */
export const Revenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function revenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Revenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Revenue' from JSON`);
}
/** @internal */
export const NetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function netRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetRevenue' from JSON`);
}
/** @internal */
export const CumulativeRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function cumulativeRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CumulativeRevenue' from JSON`);
}
/** @internal */
export const NetCumulativeRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function netCumulativeRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NetCumulativeRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetCumulativeRevenue' from JSON`);
}
/** @internal */
export const Costs$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function costsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Costs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Costs' from JSON`);
}
/** @internal */
export const CumulativeCosts$inboundSchema = smartUnion([z.int(), z.number()]);
export function cumulativeCostsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CumulativeCosts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CumulativeCosts' from JSON`);
}
/** @internal */
export const AverageOrderValue$inboundSchema = smartUnion([z.int(), z.number()]);
export function averageOrderValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AverageOrderValue' from JSON`);
}
/** @internal */
export const NetAverageOrderValue$inboundSchema = smartUnion([z.int(), z.number()]);
export function netAverageOrderValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NetAverageOrderValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NetAverageOrderValue' from JSON`);
}
/** @internal */
export const CostPerUser$inboundSchema = smartUnion([z.int(), z.number()]);
export function costPerUserFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CostPerUser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CostPerUser' from JSON`);
}
/** @internal */
export const ActiveUserByEvent$inboundSchema = smartUnion([z.int(), z.number()]);
export function activeUserByEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActiveUserByEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActiveUserByEvent' from JSON`);
}
/** @internal */
export const OneTimeProducts$inboundSchema = smartUnion([z.int(), z.number()]);
export function oneTimeProductsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OneTimeProducts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProducts' from JSON`);
}
/** @internal */
export const OneTimeProductsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function oneTimeProductsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OneTimeProductsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProductsRevenue' from JSON`);
}
/** @internal */
export const OneTimeProductsNetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function oneTimeProductsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OneTimeProductsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneTimeProductsNetRevenue' from JSON`);
}
/** @internal */
export const NewSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function newSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NewSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptions' from JSON`);
}
/** @internal */
export const NewSubscriptionsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function newSubscriptionsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NewSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptionsRevenue' from JSON`);
}
/** @internal */
export const NewSubscriptionsNetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function newSubscriptionsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NewSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NewSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
export const RenewedSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function renewedSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewedSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptions' from JSON`);
}
/** @internal */
export const RenewedSubscriptionsRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function renewedSubscriptionsRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewedSubscriptionsRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptionsRevenue' from JSON`);
}
/** @internal */
export const RenewedSubscriptionsNetRevenue$inboundSchema = smartUnion([z.int(), z.number()]);
export function renewedSubscriptionsNetRevenueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RenewedSubscriptionsNetRevenue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RenewedSubscriptionsNetRevenue' from JSON`);
}
/** @internal */
export const CanceledSubscriptions$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptions' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsCustomerService$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsCustomerServiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsCustomerService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsCustomerService' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsLowQuality$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsLowQualityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsLowQuality$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsLowQuality' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsMissingFeatures$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsMissingFeaturesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsMissingFeatures$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsMissingFeatures' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsSwitchedService$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsSwitchedServiceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsSwitchedService$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsSwitchedService' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsTooComplex$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsTooComplexFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsTooComplex$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsTooComplex' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsTooExpensive$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsTooExpensiveFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsTooExpensive$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsTooExpensive' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsUnused$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsUnusedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsUnused$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsUnused' from JSON`);
}
/** @internal */
export const CanceledSubscriptionsOther$inboundSchema = smartUnion([z.int(), z.number()]);
export function canceledSubscriptionsOtherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanceledSubscriptionsOther$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanceledSubscriptionsOther' from JSON`);
}
/** @internal */
export const CheckoutsConversion$inboundSchema = smartUnion([z.int(), z.number()]);
export function checkoutsConversionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CheckoutsConversion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutsConversion' from JSON`);
}
/** @internal */
export const Ltv$inboundSchema = smartUnion([
    z.int(),
    z.number(),
]);
export function ltvFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Ltv$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ltv' from JSON`);
}
/** @internal */
export const GrossMargin$inboundSchema = smartUnion([z.int(), z.number()]);
export function grossMarginFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GrossMargin$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrossMargin' from JSON`);
}
/** @internal */
export const GrossMarginPercentage$inboundSchema = smartUnion([z.int(), z.number()]);
export function grossMarginPercentageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GrossMarginPercentage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GrossMarginPercentage' from JSON`);
}
/** @internal */
export const Cashflow$inboundSchema = smartUnion([z.int(), z.number()]);
export function cashflowFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Cashflow$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Cashflow' from JSON`);
}
/** @internal */
export const MetricPeriod$inboundSchema = z.pipe(z.object({
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    active_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    committed_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    monthly_recurring_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    committed_monthly_recurring_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    average_revenue_per_user: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    checkouts: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    succeeded_checkouts: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    churned_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    churn_rate: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    orders: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cumulative_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_cumulative_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    costs: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cumulative_costs: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    average_order_value: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    net_average_order_value: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cost_per_user: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    active_user_by_event: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    one_time_products_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    new_subscriptions_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    renewed_subscriptions_net_revenue: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_customer_service: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_low_quality: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_missing_features: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_switched_service: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_too_complex: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_too_expensive: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_unused: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    canceled_subscriptions_other: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    checkouts_conversion: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    ltv: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    gross_margin: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    gross_margin_percentage: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
    cashflow: z.optional(z.nullable(smartUnion([z.int(), z.number()]))),
}), z.transform((v) => {
    return remap$(v, {
        "active_subscriptions": "activeSubscriptions",
        "committed_subscriptions": "committedSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "succeeded_checkouts": "succeededCheckouts",
        "churned_subscriptions": "churnedSubscriptions",
        "churn_rate": "churnRate",
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "checkouts_conversion": "checkoutsConversion",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
export function metricPeriodFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MetricPeriod$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MetricPeriod' from JSON`);
}
