// @polar-sh/sdk/dist/esm/models/components/orderitemschema.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrderItemSchema$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    label: z.string(),
    amount: z.int(),
    tax_amount: z.int(),
    proration: z.boolean(),
    product_price_id: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "product_price_id": "productPriceId",
    });
}));
/** @internal */
export const OrderItemSchema$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    label: z.string(),
    amount: z.int(),
    taxAmount: z.int(),
    proration: z.boolean(),
    productPriceId: z.nullable(z.string()),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        taxAmount: "tax_amount",
        productPriceId: "product_price_id",
    });
}));
export function orderItemSchemaToJSON(orderItemSchema) {
    return JSON.stringify(OrderItemSchema$outboundSchema.parse(orderItemSchema));
}
export function orderItemSchemaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderItemSchema$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderItemSchema' from JSON`);
}
