// @polar-sh/sdk/dist/esm/models/components/order.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
import { DiscountFixedOnceForeverDurationBase$inboundSchema, DiscountFixedOnceForeverDurationBase$outboundSchema, } from "./discountfixedonceforeverdurationbase.js";
import { DiscountFixedRepeatDurationBase$inboundSchema, DiscountFixedRepeatDurationBase$outboundSchema, } from "./discountfixedrepeatdurationbase.js";
import { DiscountPercentageOnceForeverDurationBase$inboundSchema, DiscountPercentageOnceForeverDurationBase$outboundSchema, } from "./discountpercentageonceforeverdurationbase.js";
import { DiscountPercentageRepeatDurationBase$inboundSchema, DiscountPercentageRepeatDurationBase$outboundSchema, } from "./discountpercentagerepeatdurationbase.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
import { OrderBillingReason$inboundSchema, OrderBillingReason$outboundSchema, } from "./orderbillingreason.js";
import { OrderCustomer$inboundSchema, OrderCustomer$outboundSchema, } from "./ordercustomer.js";
import { OrderItemSchema$inboundSchema, OrderItemSchema$outboundSchema, } from "./orderitemschema.js";
import { OrderProduct$inboundSchema, OrderProduct$outboundSchema, } from "./orderproduct.js";
import { OrderStatus$inboundSchema, OrderStatus$outboundSchema, } from "./orderstatus.js";
import { OrderSubscription$inboundSchema, OrderSubscription$outboundSchema, } from "./ordersubscription.js";
/** @internal */
export const OrderCustomFieldData$inboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
]);
/** @internal */
export const OrderCustomFieldData$outboundSchema = smartUnion([
    z.string(),
    z.int(),
    z.boolean(),
    z.pipe(z.date(), z.transform(v => v.toISOString())),
]);
export function orderCustomFieldDataToJSON(orderCustomFieldData) {
    return JSON.stringify(OrderCustomFieldData$outboundSchema.parse(orderCustomFieldData));
}
export function orderCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderCustomFieldData' from JSON`);
}
/** @internal */
export const OrderDiscount$inboundSchema = smartUnion([
    DiscountFixedRepeatDurationBase$inboundSchema,
    DiscountFixedOnceForeverDurationBase$inboundSchema,
    DiscountPercentageRepeatDurationBase$inboundSchema,
    DiscountPercentageOnceForeverDurationBase$inboundSchema,
]);
/** @internal */
export const OrderDiscount$outboundSchema = smartUnion([
    DiscountFixedRepeatDurationBase$outboundSchema,
    DiscountFixedOnceForeverDurationBase$outboundSchema,
    DiscountPercentageRepeatDurationBase$outboundSchema,
    DiscountPercentageOnceForeverDurationBase$outboundSchema,
]);
export function orderDiscountToJSON(orderDiscount) {
    return JSON.stringify(OrderDiscount$outboundSchema.parse(orderDiscount));
}
export function orderDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrderDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderDiscount' from JSON`);
}
/** @internal */
export const Order$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    status: OrderStatus$inboundSchema,
    paid: z.boolean(),
    subtotal_amount: z.int(),
    discount_amount: z.int(),
    net_amount: z.int(),
    tax_amount: z.int(),
    total_amount: z.int(),
    applied_balance_amount: z.int(),
    due_amount: z.int(),
    refunded_amount: z.int(),
    refunded_tax_amount: z.int(),
    currency: z.string(),
    billing_reason: OrderBillingReason$inboundSchema,
    billing_name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    invoice_number: z.string(),
    is_invoice_generated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customer_id: z.string(),
    product_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    subscription_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    custom_field_data: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    ])))),
    platform_fee_amount: z.int(),
    platform_fee_currency: z.nullable(z.string()),
    customer: OrderCustomer$inboundSchema,
    product: z.nullable(OrderProduct$inboundSchema),
    discount: z.nullable(smartUnion([
        DiscountFixedRepeatDurationBase$inboundSchema,
        DiscountFixedOnceForeverDurationBase$inboundSchema,
        DiscountPercentageRepeatDurationBase$inboundSchema,
        DiscountPercentageOnceForeverDurationBase$inboundSchema,
    ])),
    subscription: z.nullable(OrderSubscription$inboundSchema),
    items: z.array(OrderItemSchema$inboundSchema),
    description: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "subtotal_amount": "subtotalAmount",
        "discount_amount": "discountAmount",
        "net_amount": "netAmount",
        "tax_amount": "taxAmount",
        "total_amount": "totalAmount",
        "applied_balance_amount": "appliedBalanceAmount",
        "due_amount": "dueAmount",
        "refunded_amount": "refundedAmount",
        "refunded_tax_amount": "refundedTaxAmount",
        "billing_reason": "billingReason",
        "billing_name": "billingName",
        "billing_address": "billingAddress",
        "invoice_number": "invoiceNumber",
        "is_invoice_generated": "isInvoiceGenerated",
        "customer_id": "customerId",
        "product_id": "productId",
        "discount_id": "discountId",
        "subscription_id": "subscriptionId",
        "checkout_id": "checkoutId",
        "custom_field_data": "customFieldData",
        "platform_fee_amount": "platformFeeAmount",
        "platform_fee_currency": "platformFeeCurrency",
    });
}));
/** @internal */
export const Order$outboundSchema = z
    .pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    status: OrderStatus$outboundSchema,
    paid: z.boolean(),
    subtotalAmount: z.int(),
    discountAmount: z.int(),
    netAmount: z.int(),
    taxAmount: z.int(),
    totalAmount: z.int(),
    appliedBalanceAmount: z.int(),
    dueAmount: z.int(),
    refundedAmount: z.int(),
    refundedTaxAmount: z.int(),
    currency: z.string(),
    billingReason: OrderBillingReason$outboundSchema,
    billingName: z.nullable(z.string()),
    billingAddress: z.nullable(Address$outboundSchema),
    invoiceNumber: z.string(),
    isInvoiceGenerated: z.boolean(),
    seats: z.optional(z.nullable(z.int())),
    customerId: z.string(),
    productId: z.nullable(z.string()),
    discountId: z.nullable(z.string()),
    subscriptionId: z.nullable(z.string()),
    checkoutId: z.nullable(z.string()),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    customFieldData: z.optional(z.record(z.string(), z.nullable(smartUnion([
        z.string(),
        z.int(),
        z.boolean(),
        z.pipe(z.date(), z.transform(v => v.toISOString())),
    ])))),
    platformFeeAmount: z.int(),
    platformFeeCurrency: z.nullable(z.string()),
    customer: OrderCustomer$outboundSchema,
    product: z.nullable(OrderProduct$outboundSchema),
    discount: z.nullable(smartUnion([
        DiscountFixedRepeatDurationBase$outboundSchema,
        DiscountFixedOnceForeverDurationBase$outboundSchema,
        DiscountPercentageRepeatDurationBase$outboundSchema,
        DiscountPercentageOnceForeverDurationBase$outboundSchema,
    ])),
    subscription: z.nullable(OrderSubscription$outboundSchema),
    items: z.array(OrderItemSchema$outboundSchema),
    description: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        subtotalAmount: "subtotal_amount",
        discountAmount: "discount_amount",
        netAmount: "net_amount",
        taxAmount: "tax_amount",
        totalAmount: "total_amount",
        appliedBalanceAmount: "applied_balance_amount",
        dueAmount: "due_amount",
        refundedAmount: "refunded_amount",
        refundedTaxAmount: "refunded_tax_amount",
        billingReason: "billing_reason",
        billingName: "billing_name",
        billingAddress: "billing_address",
        invoiceNumber: "invoice_number",
        isInvoiceGenerated: "is_invoice_generated",
        customerId: "customer_id",
        productId: "product_id",
        discountId: "discount_id",
        subscriptionId: "subscription_id",
        checkoutId: "checkout_id",
        customFieldData: "custom_field_data",
        platformFeeAmount: "platform_fee_amount",
        platformFeeCurrency: "platform_fee_currency",
    });
}));
export function orderToJSON(order) {
    return JSON.stringify(Order$outboundSchema.parse(order));
}
export function orderFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Order$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Order' from JSON`);
}
