// @polar-sh/sdk/dist/esm/models/components/organizationfeaturesettings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrganizationFeatureSettings$inboundSchema = z.pipe(z.object({
    issue_funding_enabled: z._default(z.boolean(), false),
    seat_based_pricing_enabled: z._default(z.boolean(), false),
    revops_enabled: z._default(z.boolean(), false),
    wallets_enabled: z._default(z.boolean(), false),
    member_model_enabled: z._default(z.boolean(), false),
    tinybird_read: z._default(z.boolean(), false),
    tinybird_compare: z._default(z.boolean(), false),
    checkout_localization_enabled: z._default(z.boolean(), false),
    overview_metrics: z.optional(z.nullable(z.array(z.string()))),
}), z.transform((v) => {
    return remap$(v, {
        "issue_funding_enabled": "issueFundingEnabled",
        "seat_based_pricing_enabled": "seatBasedPricingEnabled",
        "revops_enabled": "revopsEnabled",
        "wallets_enabled": "walletsEnabled",
        "member_model_enabled": "memberModelEnabled",
        "tinybird_read": "tinybirdRead",
        "tinybird_compare": "tinybirdCompare",
        "checkout_localization_enabled": "checkoutLocalizationEnabled",
        "overview_metrics": "overviewMetrics",
    });
}));
/** @internal */
export const OrganizationFeatureSettings$outboundSchema = z.pipe(z.object({
    issueFundingEnabled: z._default(z.boolean(), false),
    seatBasedPricingEnabled: z._default(z.boolean(), false),
    revopsEnabled: z._default(z.boolean(), false),
    walletsEnabled: z._default(z.boolean(), false),
    memberModelEnabled: z._default(z.boolean(), false),
    tinybirdRead: z._default(z.boolean(), false),
    tinybirdCompare: z._default(z.boolean(), false),
    checkoutLocalizationEnabled: z._default(z.boolean(), false),
    overviewMetrics: z.optional(z.nullable(z.array(z.string()))),
}), z.transform((v) => {
    return remap$(v, {
        issueFundingEnabled: "issue_funding_enabled",
        seatBasedPricingEnabled: "seat_based_pricing_enabled",
        revopsEnabled: "revops_enabled",
        walletsEnabled: "wallets_enabled",
        memberModelEnabled: "member_model_enabled",
        tinybirdRead: "tinybird_read",
        tinybirdCompare: "tinybird_compare",
        checkoutLocalizationEnabled: "checkout_localization_enabled",
        overviewMetrics: "overview_metrics",
    });
}));
export function organizationFeatureSettingsToJSON(organizationFeatureSettings) {
    return JSON.stringify(OrganizationFeatureSettings$outboundSchema.parse(organizationFeatureSettings));
}
export function organizationFeatureSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationFeatureSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationFeatureSettings' from JSON`);
}
