// @polar-sh/sdk/dist/esm/models/components/organizationsociallink.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { OrganizationSocialPlatforms$inboundSchema, OrganizationSocialPlatforms$outboundSchema, } from "./organizationsocialplatforms.js";
/** @internal */
export const OrganizationSocialLink$inboundSchema = z.object({
    platform: OrganizationSocialPlatforms$inboundSchema,
    url: z.string(),
});
/** @internal */
export const OrganizationSocialLink$outboundSchema = z.object({
    platform: OrganizationSocialPlatforms$outboundSchema,
    url: z.string(),
});
export function organizationSocialLinkToJSON(organizationSocialLink) {
    return JSON.stringify(OrganizationSocialLink$outboundSchema.parse(organizationSocialLink));
}
export function organizationSocialLinkFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationSocialLink$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationSocialLink' from JSON`);
}
