// @polar-sh/sdk/dist/esm/models/components/organizationcustomeremailsettings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrganizationCustomerEmailSettings$inboundSchema = z.pipe(z.object({
    order_confirmation: z.boolean(),
    subscription_cancellation: z.boolean(),
    subscription_confirmation: z.boolean(),
    subscription_cycled: z.boolean(),
    subscription_cycled_after_trial: z.boolean(),
    subscription_past_due: z.boolean(),
    subscription_renewal_reminder: z.boolean(),
    subscription_revoked: z.boolean(),
    subscription_trial_conversion_reminder: z.boolean(),
    subscription_uncanceled: z.boolean(),
    subscription_updated: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "order_confirmation": "orderConfirmation",
        "subscription_cancellation": "subscriptionCancellation",
        "subscription_confirmation": "subscriptionConfirmation",
        "subscription_cycled": "subscriptionCycled",
        "subscription_cycled_after_trial": "subscriptionCycledAfterTrial",
        "subscription_past_due": "subscriptionPastDue",
        "subscription_renewal_reminder": "subscriptionRenewalReminder",
        "subscription_revoked": "subscriptionRevoked",
        "subscription_trial_conversion_reminder": "subscriptionTrialConversionReminder",
        "subscription_uncanceled": "subscriptionUncanceled",
        "subscription_updated": "subscriptionUpdated",
    });
}));
/** @internal */
export const OrganizationCustomerEmailSettings$outboundSchema = z.pipe(z.object({
    orderConfirmation: z.boolean(),
    subscriptionCancellation: z.boolean(),
    subscriptionConfirmation: z.boolean(),
    subscriptionCycled: z.boolean(),
    subscriptionCycledAfterTrial: z.boolean(),
    subscriptionPastDue: z.boolean(),
    subscriptionRenewalReminder: z.boolean(),
    subscriptionRevoked: z.boolean(),
    subscriptionTrialConversionReminder: z.boolean(),
    subscriptionUncanceled: z.boolean(),
    subscriptionUpdated: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        orderConfirmation: "order_confirmation",
        subscriptionCancellation: "subscription_cancellation",
        subscriptionConfirmation: "subscription_confirmation",
        subscriptionCycled: "subscription_cycled",
        subscriptionCycledAfterTrial: "subscription_cycled_after_trial",
        subscriptionPastDue: "subscription_past_due",
        subscriptionRenewalReminder: "subscription_renewal_reminder",
        subscriptionRevoked: "subscription_revoked",
        subscriptionTrialConversionReminder: "subscription_trial_conversion_reminder",
        subscriptionUncanceled: "subscription_uncanceled",
        subscriptionUpdated: "subscription_updated",
    });
}));
export function organizationCustomerEmailSettingsToJSON(organizationCustomerEmailSettings) {
    return JSON.stringify(OrganizationCustomerEmailSettings$outboundSchema.parse(organizationCustomerEmailSettings));
}
export function organizationCustomerEmailSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationCustomerEmailSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationCustomerEmailSettings' from JSON`);
}
