// @polar-sh/sdk/dist/esm/models/components/organizationcustomerportalsettings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { CustomerPortalSubscriptionSettings$inboundSchema, CustomerPortalSubscriptionSettings$outboundSchema, } from "./customerportalsubscriptionsettings.js";
import { CustomerPortalUsageSettings$inboundSchema, CustomerPortalUsageSettings$outboundSchema, } from "./customerportalusagesettings.js";
/** @internal */
export const OrganizationCustomerPortalSettings$inboundSchema = z.object({
    usage: CustomerPortalUsageSettings$inboundSchema,
    subscription: CustomerPortalSubscriptionSettings$inboundSchema,
});
/** @internal */
export const OrganizationCustomerPortalSettings$outboundSchema = z.object({
    usage: CustomerPortalUsageSettings$outboundSchema,
    subscription: CustomerPortalSubscriptionSettings$outboundSchema,
});
export function organizationCustomerPortalSettingsToJSON(organizationCustomerPortalSettings) {
    return JSON.stringify(OrganizationCustomerPortalSettings$outboundSchema.parse(organizationCustomerPortalSettings));
}
export function organizationCustomerPortalSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationCustomerPortalSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationCustomerPortalSettings' from JSON`);
}
