// @polar-sh/sdk/dist/esm/models/components/organizationnotificationsettings.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrganizationNotificationSettings$inboundSchema = z.pipe(z.object({
    new_order: z.boolean(),
    new_subscription: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "new_order": "newOrder",
        "new_subscription": "newSubscription",
    });
}));
/** @internal */
export const OrganizationNotificationSettings$outboundSchema = z.pipe(z.object({
    newOrder: z.boolean(),
    newSubscription: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        newOrder: "new_order",
        newSubscription: "new_subscription",
    });
}));
export function organizationNotificationSettingsToJSON(organizationNotificationSettings) {
    return JSON.stringify(OrganizationNotificationSettings$outboundSchema.parse(organizationNotificationSettings));
}
export function organizationNotificationSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationNotificationSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationNotificationSettings' from JSON`);
}
