// @polar-sh/sdk/dist/esm/models/components/organizationdetails.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
export const SwitchingFrom = {
    Paddle: "paddle",
    LemonSqueezy: "lemon_squeezy",
    Gumroad: "gumroad",
    Stripe: "stripe",
    Other: "other",
};
/** @internal */
export const SwitchingFrom$outboundSchema = z.enum(SwitchingFrom);
/** @internal */
export const OrganizationDetails$outboundSchema = z.pipe(z.object({
    about: z.optional(z.nullable(z.string())),
    productDescription: z.optional(z.nullable(z.string())),
    sellingCategories: z.optional(z.array(z.string())),
    pricingModels: z.optional(z.array(z.string())),
    intendedUse: z.optional(z.nullable(z.string())),
    customerAcquisition: z.optional(z.array(z.string())),
    futureAnnualRevenue: z.optional(z.nullable(z.int())),
    switching: z._default(z.boolean(), false),
    switchingFrom: z.optional(z.nullable(SwitchingFrom$outboundSchema)),
    previousAnnualRevenue: z.optional(z.nullable(z.int())),
}), z.transform((v) => {
    return remap$(v, {
        productDescription: "product_description",
        sellingCategories: "selling_categories",
        pricingModels: "pricing_models",
        intendedUse: "intended_use",
        customerAcquisition: "customer_acquisition",
        futureAnnualRevenue: "future_annual_revenue",
        switchingFrom: "switching_from",
        previousAnnualRevenue: "previous_annual_revenue",
    });
}));
export function organizationDetailsToJSON(organizationDetails) {
    return JSON.stringify(OrganizationDetails$outboundSchema.parse(organizationDetails));
}
