// @polar-sh/sdk/dist/esm/models/components/organizationavatarfileread.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrganizationAvatarFileRead$inboundSchema = z.pipe(z.object({
    id: z.string(),
    organization_id: z.string(),
    name: z.string(),
    path: z.string(),
    mime_type: z.string(),
    size: z.int(),
    storage_version: z.nullable(z.string()),
    checksum_etag: z.nullable(z.string()),
    checksum_sha256_base64: z.nullable(z.string()),
    checksum_sha256_hex: z.nullable(z.string()),
    last_modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    version: z.nullable(z.string()),
    service: z.literal("organization_avatar"),
    is_uploaded: z.boolean(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    size_readable: z.string(),
    public_url: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "mime_type": "mimeType",
        "storage_version": "storageVersion",
        "checksum_etag": "checksumEtag",
        "checksum_sha256_base64": "checksumSha256Base64",
        "checksum_sha256_hex": "checksumSha256Hex",
        "last_modified_at": "lastModifiedAt",
        "is_uploaded": "isUploaded",
        "created_at": "createdAt",
        "size_readable": "sizeReadable",
        "public_url": "publicUrl",
    });
}));
export function organizationAvatarFileReadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrganizationAvatarFileRead$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationAvatarFileRead' from JSON`);
}
