// @polar-sh/sdk/dist/esm/models/components/organizationavatarfilecreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { S3FileCreateMultipart$outboundSchema, } from "./s3filecreatemultipart.js";
/** @internal */
export const OrganizationAvatarFileCreate$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(z.string())),
    name: z.string(),
    mimeType: z.string(),
    size: z.int(),
    checksumSha256Base64: z.optional(z.nullable(z.string())),
    upload: S3FileCreateMultipart$outboundSchema,
    service: z.literal("organization_avatar"),
    version: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        mimeType: "mime_type",
        checksumSha256Base64: "checksum_sha256_base64",
    });
}));
export function organizationAvatarFileCreateToJSON(organizationAvatarFileCreate) {
    return JSON.stringify(OrganizationAvatarFileCreate$outboundSchema.parse(organizationAvatarFileCreate));
}
