// @polar-sh/sdk/dist/esm/models/components/pendingsubscriptionupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const PendingSubscriptionUpdate$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    applies_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    product_id: z.nullable(z.string()),
    seats: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "applies_at": "appliesAt",
        "product_id": "productId",
    });
}));
/** @internal */
export const PendingSubscriptionUpdate$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    appliesAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    productId: z.nullable(z.string()),
    seats: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        appliesAt: "applies_at",
        productId: "product_id",
    });
}));
export function pendingSubscriptionUpdateToJSON(pendingSubscriptionUpdate) {
    return JSON.stringify(PendingSubscriptionUpdate$outboundSchema.parse(pendingSubscriptionUpdate));
}
export function pendingSubscriptionUpdateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PendingSubscriptionUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PendingSubscriptionUpdate' from JSON`);
}
