// @polar-sh/sdk/dist/esm/models/components/productpriceseatbased.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSeatTiersOutput$inboundSchema, ProductPriceSeatTiersOutput$outboundSchema, } from "./productpriceseattiersoutput.js";
import { ProductPriceSource$inboundSchema, ProductPriceSource$outboundSchema, } from "./productpricesource.js";
import { TaxBehaviorOption$inboundSchema, TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceSeatBased$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    source: ProductPriceSource$inboundSchema,
    amount_type: z.literal("seat_based"),
    price_currency: z.string(),
    tax_behavior: z.nullable(TaxBehaviorOption$inboundSchema),
    is_archived: z.boolean(),
    product_id: z.string(),
    seat_tiers: ProductPriceSeatTiersOutput$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "tax_behavior": "taxBehavior",
        "is_archived": "isArchived",
        "product_id": "productId",
        "seat_tiers": "seatTiers",
    });
}));
/** @internal */
export const ProductPriceSeatBased$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    source: ProductPriceSource$outboundSchema,
    amountType: z.literal("seat_based"),
    priceCurrency: z.string(),
    taxBehavior: z.nullable(TaxBehaviorOption$outboundSchema),
    isArchived: z.boolean(),
    productId: z.string(),
    seatTiers: ProductPriceSeatTiersOutput$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        isArchived: "is_archived",
        productId: "product_id",
        seatTiers: "seat_tiers",
    });
}));
export function productPriceSeatBasedToJSON(productPriceSeatBased) {
    return JSON.stringify(ProductPriceSeatBased$outboundSchema.parse(productPriceSeatBased));
}
export function productPriceSeatBasedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatBased$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatBased' from JSON`);
}
