// @polar-sh/sdk/dist/esm/models/components/productpriceseattier.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ProductPriceSeatTier$inboundSchema = z.pipe(z.object({
    min_seats: z.int(),
    max_seats: z.optional(z.nullable(z.int())),
    price_per_seat: z.int(),
}), z.transform((v) => {
    return remap$(v, {
        "min_seats": "minSeats",
        "max_seats": "maxSeats",
        "price_per_seat": "pricePerSeat",
    });
}));
/** @internal */
export const ProductPriceSeatTier$outboundSchema = z.pipe(z.object({
    minSeats: z.int(),
    maxSeats: z.optional(z.nullable(z.int())),
    pricePerSeat: z.int(),
}), z.transform((v) => {
    return remap$(v, {
        minSeats: "min_seats",
        maxSeats: "max_seats",
        pricePerSeat: "price_per_seat",
    });
}));
export function productPriceSeatTierToJSON(productPriceSeatTier) {
    return JSON.stringify(ProductPriceSeatTier$outboundSchema.parse(productPriceSeatTier));
}
export function productPriceSeatTierFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatTier$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatTier' from JSON`);
}
