// @polar-sh/sdk/dist/esm/models/components/productpriceseatbasedcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
import { ProductPriceSeatTiersInput$outboundSchema, } from "./productpriceseattiersinput.js";
import { TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceSeatBasedCreate$outboundSchema = z.pipe(z.object({
    amountType: z.literal("seat_based"),
    priceCurrency: z.optional(PresentmentCurrency$outboundSchema),
    taxBehavior: z.optional(z.nullable(TaxBehaviorOption$outboundSchema)),
    seatTiers: ProductPriceSeatTiersInput$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        seatTiers: "seat_tiers",
    });
}));
export function productPriceSeatBasedCreateToJSON(productPriceSeatBasedCreate) {
    return JSON.stringify(ProductPriceSeatBasedCreate$outboundSchema.parse(productPriceSeatBasedCreate));
}
