// @polar-sh/sdk/dist/esm/models/components/productpricecustomcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
import { TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceCustomCreate$outboundSchema = z.pipe(z.object({
    amountType: z.literal("custom"),
    priceCurrency: z.optional(PresentmentCurrency$outboundSchema),
    taxBehavior: z.optional(z.nullable(TaxBehaviorOption$outboundSchema)),
    minimumAmount: z._default(z.int(), 50),
    maximumAmount: z.optional(z.nullable(z.int())),
    presetAmount: z.optional(z.nullable(z.int())),
}), z.transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
}));
export function productPriceCustomCreateToJSON(productPriceCustomCreate) {
    return JSON.stringify(ProductPriceCustomCreate$outboundSchema.parse(productPriceCustomCreate));
}
