// @polar-sh/sdk/dist/esm/models/components/productpriceseattiersoutput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSeatTier$inboundSchema, ProductPriceSeatTier$outboundSchema, } from "./productpriceseattier.js";
import { SeatTierType$inboundSchema, SeatTierType$outboundSchema, } from "./seattiertype.js";
/** @internal */
export const ProductPriceSeatTiersOutput$inboundSchema = z.pipe(z.object({
    seat_tier_type: z.optional(SeatTierType$inboundSchema),
    tiers: z.array(ProductPriceSeatTier$inboundSchema),
    minimum_seats: z.int(),
    maximum_seats: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "seat_tier_type": "seatTierType",
        "minimum_seats": "minimumSeats",
        "maximum_seats": "maximumSeats",
    });
}));
/** @internal */
export const ProductPriceSeatTiersOutput$outboundSchema = z.pipe(z.object({
    seatTierType: z.optional(SeatTierType$outboundSchema),
    tiers: z.array(ProductPriceSeatTier$outboundSchema),
    minimumSeats: z.int(),
    maximumSeats: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        seatTierType: "seat_tier_type",
        minimumSeats: "minimum_seats",
        maximumSeats: "maximum_seats",
    });
}));
export function productPriceSeatTiersOutputToJSON(productPriceSeatTiersOutput) {
    return JSON.stringify(ProductPriceSeatTiersOutput$outboundSchema.parse(productPriceSeatTiersOutput));
}
export function productPriceSeatTiersOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceSeatTiersOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceSeatTiersOutput' from JSON`);
}
