// @polar-sh/sdk/dist/esm/models/components/productcreaterecurring.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { AttachedCustomFieldCreate$outboundSchema, } from "./attachedcustomfieldcreate.js";
import { ProductPriceCustomCreate$outboundSchema, } from "./productpricecustomcreate.js";
import { ProductPriceFixedCreate$outboundSchema, } from "./productpricefixedcreate.js";
import { ProductPriceFreeCreate$outboundSchema, } from "./productpricefreecreate.js";
import { ProductPriceMeteredUnitCreate$outboundSchema, } from "./productpricemeteredunitcreate.js";
import { ProductPriceSeatBasedCreate$outboundSchema, } from "./productpriceseatbasedcreate.js";
import { ProductVisibility$outboundSchema, } from "./productvisibility.js";
import { SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { TrialInterval$outboundSchema, } from "./trialinterval.js";
/** @internal */
export const ProductCreateRecurringMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function productCreateRecurringMetadataToJSON(productCreateRecurringMetadata) {
    return JSON.stringify(ProductCreateRecurringMetadata$outboundSchema.parse(productCreateRecurringMetadata));
}
/** @internal */
export const ProductCreateRecurringPrices$outboundSchema = z.union([
    ProductPriceCustomCreate$outboundSchema,
    ProductPriceFixedCreate$outboundSchema,
    ProductPriceFreeCreate$outboundSchema,
    ProductPriceMeteredUnitCreate$outboundSchema,
    ProductPriceSeatBasedCreate$outboundSchema,
]);
export function productCreateRecurringPricesToJSON(productCreateRecurringPrices) {
    return JSON.stringify(ProductCreateRecurringPrices$outboundSchema.parse(productCreateRecurringPrices));
}
/** @internal */
export const ProductCreateRecurring$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    name: z.string(),
    description: z.optional(z.nullable(z.string())),
    visibility: z.optional(ProductVisibility$outboundSchema),
    prices: z.array(z.union([
        ProductPriceCustomCreate$outboundSchema,
        ProductPriceFixedCreate$outboundSchema,
        ProductPriceFreeCreate$outboundSchema,
        ProductPriceMeteredUnitCreate$outboundSchema,
        ProductPriceSeatBasedCreate$outboundSchema,
    ])),
    medias: z.optional(z.nullable(z.array(z.string()))),
    attachedCustomFields: z.optional(z.array(AttachedCustomFieldCreate$outboundSchema)),
    organizationId: z.optional(z.nullable(z.string())),
    trialInterval: z.optional(z.nullable(TrialInterval$outboundSchema)),
    trialIntervalCount: z.optional(z.nullable(z.int())),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
    recurringIntervalCount: z._default(z.int(), 1),
}), z.transform((v) => {
    return remap$(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
        trialInterval: "trial_interval",
        trialIntervalCount: "trial_interval_count",
        recurringInterval: "recurring_interval",
        recurringIntervalCount: "recurring_interval_count",
    });
}));
export function productCreateRecurringToJSON(productCreateRecurring) {
    return JSON.stringify(ProductCreateRecurring$outboundSchema.parse(productCreateRecurring));
}
