// @polar-sh/sdk/dist/esm/models/components/productpricefree.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSource$inboundSchema, ProductPriceSource$outboundSchema, } from "./productpricesource.js";
import { TaxBehaviorOption$inboundSchema, TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceFree$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    source: ProductPriceSource$inboundSchema,
    amount_type: z.literal("free"),
    price_currency: z.string(),
    tax_behavior: z.nullable(TaxBehaviorOption$inboundSchema),
    is_archived: z.boolean(),
    product_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "tax_behavior": "taxBehavior",
        "is_archived": "isArchived",
        "product_id": "productId",
    });
}));
/** @internal */
export const ProductPriceFree$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    source: ProductPriceSource$outboundSchema,
    amountType: z.literal("free"),
    priceCurrency: z.string(),
    taxBehavior: z.nullable(TaxBehaviorOption$outboundSchema),
    isArchived: z.boolean(),
    productId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        isArchived: "is_archived",
        productId: "product_id",
    });
}));
export function productPriceFreeToJSON(productPriceFree) {
    return JSON.stringify(ProductPriceFree$outboundSchema.parse(productPriceFree));
}
export function productPriceFreeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceFree$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceFree' from JSON`);
}
