// @polar-sh/sdk/dist/esm/models/components/productpricemeteredunitcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
import { TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const UnitAmount$outboundSchema = smartUnion([z.number(), z.string()]);
export function unitAmountToJSON(unitAmount) {
    return JSON.stringify(UnitAmount$outboundSchema.parse(unitAmount));
}
/** @internal */
export const ProductPriceMeteredUnitCreate$outboundSchema = z.pipe(z.object({
    amountType: z.literal("metered_unit"),
    priceCurrency: z.optional(PresentmentCurrency$outboundSchema),
    taxBehavior: z.optional(z.nullable(TaxBehaviorOption$outboundSchema)),
    meterId: z.string(),
    unitAmount: smartUnion([z.number(), z.string()]),
    capAmount: z.optional(z.nullable(z.int())),
}), z.transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        meterId: "meter_id",
        unitAmount: "unit_amount",
        capAmount: "cap_amount",
    });
}));
export function productPriceMeteredUnitCreateToJSON(productPriceMeteredUnitCreate) {
    return JSON.stringify(ProductPriceMeteredUnitCreate$outboundSchema.parse(productPriceMeteredUnitCreate));
}
