// @polar-sh/sdk/dist/esm/models/components/productpricecustom.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ProductPriceSource$inboundSchema, ProductPriceSource$outboundSchema, } from "./productpricesource.js";
import { TaxBehaviorOption$inboundSchema, TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceCustom$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    source: ProductPriceSource$inboundSchema,
    amount_type: z.literal("custom"),
    price_currency: z.string(),
    tax_behavior: z.nullable(TaxBehaviorOption$inboundSchema),
    is_archived: z.boolean(),
    product_id: z.string(),
    minimum_amount: z.int(),
    maximum_amount: z.nullable(z.int()),
    preset_amount: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "amount_type": "amountType",
        "price_currency": "priceCurrency",
        "tax_behavior": "taxBehavior",
        "is_archived": "isArchived",
        "product_id": "productId",
        "minimum_amount": "minimumAmount",
        "maximum_amount": "maximumAmount",
        "preset_amount": "presetAmount",
    });
}));
/** @internal */
export const ProductPriceCustom$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    source: ProductPriceSource$outboundSchema,
    amountType: z.literal("custom"),
    priceCurrency: z.string(),
    taxBehavior: z.nullable(TaxBehaviorOption$outboundSchema),
    isArchived: z.boolean(),
    productId: z.string(),
    minimumAmount: z.int(),
    maximumAmount: z.nullable(z.int()),
    presetAmount: z.nullable(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        isArchived: "is_archived",
        productId: "product_id",
        minimumAmount: "minimum_amount",
        maximumAmount: "maximum_amount",
        presetAmount: "preset_amount",
    });
}));
export function productPriceCustomToJSON(productPriceCustom) {
    return JSON.stringify(ProductPriceCustom$outboundSchema.parse(productPriceCustom));
}
export function productPriceCustomFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProductPriceCustom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductPriceCustom' from JSON`);
}
