// @polar-sh/sdk/dist/esm/models/components/productpricefixedcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { PresentmentCurrency$outboundSchema, } from "./presentmentcurrency.js";
import { TaxBehaviorOption$outboundSchema, } from "./taxbehavioroption.js";
/** @internal */
export const ProductPriceFixedCreate$outboundSchema = z.pipe(z.object({
    amountType: z.literal("fixed"),
    priceCurrency: z.optional(PresentmentCurrency$outboundSchema),
    taxBehavior: z.optional(z.nullable(TaxBehaviorOption$outboundSchema)),
    priceAmount: z.int(),
}), z.transform((v) => {
    return remap$(v, {
        amountType: "amount_type",
        priceCurrency: "price_currency",
        taxBehavior: "tax_behavior",
        priceAmount: "price_amount",
    });
}));
export function productPriceFixedCreateToJSON(productPriceFixedCreate) {
    return JSON.stringify(ProductPriceFixedCreate$outboundSchema.parse(productPriceFixedCreate));
}
