// @polar-sh/sdk/dist/esm/models/components/propertyaggregation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
export const Func = {
    Sum: "sum",
    Max: "max",
    Min: "min",
    Avg: "avg",
};
/** @internal */
export const Func$inboundSchema = openEnums
    .inboundSchema(Func);
/** @internal */
export const Func$outboundSchema = openEnums
    .outboundSchema(Func);
/** @internal */
export const PropertyAggregation$inboundSchema = z.object({
    func: Func$inboundSchema,
    property: z.string(),
});
/** @internal */
export const PropertyAggregation$outboundSchema = z.object({
    func: Func$outboundSchema,
    property: z.string(),
});
export function propertyAggregationToJSON(propertyAggregation) {
    return JSON.stringify(PropertyAggregation$outboundSchema.parse(propertyAggregation));
}
export function propertyAggregationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PropertyAggregation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyAggregation' from JSON`);
}
