// @polar-sh/sdk/dist/esm/models/components/refundcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
import { RefundReason$outboundSchema } from "./refundreason.js";
/** @internal */
export const RefundCreateMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function refundCreateMetadataToJSON(refundCreateMetadata) {
    return JSON.stringify(RefundCreateMetadata$outboundSchema.parse(refundCreateMetadata));
}
/** @internal */
export const RefundCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    orderId: z.string(),
    reason: RefundReason$outboundSchema,
    amount: z.int(),
    comment: z.optional(z.nullable(z.string())),
    revokeBenefits: z._default(z.boolean(), false),
}), z.transform((v) => {
    return remap$(v, {
        orderId: "order_id",
        revokeBenefits: "revoke_benefits",
    });
}));
export function refundCreateToJSON(refundCreate) {
    return JSON.stringify(RefundCreate$outboundSchema.parse(refundCreate));
}
