// @polar-sh/sdk/dist/esm/models/components/refunddispute.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { DisputeStatus$inboundSchema, DisputeStatus$outboundSchema, } from "./disputestatus.js";
/** @internal */
export const RefundDispute$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    status: DisputeStatus$inboundSchema,
    resolved: z.boolean(),
    closed: z.boolean(),
    amount: z.int(),
    tax_amount: z.int(),
    currency: z.string(),
    order_id: z.string(),
    payment_id: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "order_id": "orderId",
        "payment_id": "paymentId",
    });
}));
/** @internal */
export const RefundDispute$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    status: DisputeStatus$outboundSchema,
    resolved: z.boolean(),
    closed: z.boolean(),
    amount: z.int(),
    taxAmount: z.int(),
    currency: z.string(),
    orderId: z.string(),
    paymentId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        taxAmount: "tax_amount",
        orderId: "order_id",
        paymentId: "payment_id",
    });
}));
export function refundDisputeToJSON(refundDispute) {
    return JSON.stringify(RefundDispute$outboundSchema.parse(refundDispute));
}
export function refundDisputeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RefundDispute$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefundDispute' from JSON`);
}
