// @polar-sh/sdk/dist/esm/models/components/s3fileuploadpart.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const S3FileUploadPart$inboundSchema = z.pipe(z.object({
    number: z.int(),
    chunk_start: z.int(),
    chunk_end: z.int(),
    checksum_sha256_base64: z.optional(z.nullable(z.string())),
    url: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    headers: z.optional(z.record(z.string(), z.string())),
}), z.transform((v) => {
    return remap$(v, {
        "chunk_start": "chunkStart",
        "chunk_end": "chunkEnd",
        "checksum_sha256_base64": "checksumSha256Base64",
        "expires_at": "expiresAt",
    });
}));
export function s3FileUploadPartFromJSON(jsonString) {
    return safeParse(jsonString, (x) => S3FileUploadPart$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'S3FileUploadPart' from JSON`);
}
