// @polar-sh/sdk/dist/esm/models/components/seatassign.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const SeatAssign$outboundSchema = z.pipe(z.object({
    subscriptionId: z.optional(z.nullable(z.string())),
    checkoutId: z.optional(z.nullable(z.string())),
    orderId: z.optional(z.nullable(z.string())),
    email: z.optional(z.nullable(z.string())),
    externalCustomerId: z.optional(z.nullable(z.string())),
    customerId: z.optional(z.nullable(z.string())),
    externalMemberId: z.optional(z.nullable(z.string())),
    memberId: z.optional(z.nullable(z.string())),
    metadata: z.optional(z.nullable(z.record(z.string(), z.any()))),
    immediateClaim: z._default(z.boolean(), false),
}), z.transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
        checkoutId: "checkout_id",
        orderId: "order_id",
        externalCustomerId: "external_customer_id",
        customerId: "customer_id",
        externalMemberId: "external_member_id",
        memberId: "member_id",
        immediateClaim: "immediate_claim",
    });
}));
export function seatAssignToJSON(seatAssign) {
    return JSON.stringify(SeatAssign$outboundSchema.parse(seatAssign));
}
