// @polar-sh/sdk/dist/esm/models/components/subscriptioncreatedmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionCreatedMetadata$inboundSchema = z.pipe(z.object({
    subscription_id: z.string(),
    product_id: z.string(),
    amount: z.int(),
    currency: z.string(),
    recurring_interval: z.string(),
    recurring_interval_count: z.int(),
    started_at: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "started_at": "startedAt",
    });
}));
export function subscriptionCreatedMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCreatedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCreatedMetadata' from JSON`);
}
