// @polar-sh/sdk/dist/esm/models/components/subscriptionmeter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Meter$inboundSchema, Meter$outboundSchema, } from "./meter.js";
/** @internal */
export const SubscriptionMeter$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    consumed_units: z.number(),
    credited_units: z.int(),
    amount: z.int(),
    meter_id: z.string(),
    meter: Meter$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "consumed_units": "consumedUnits",
        "credited_units": "creditedUnits",
        "meter_id": "meterId",
    });
}));
/** @internal */
export const SubscriptionMeter$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    consumedUnits: z.number(),
    creditedUnits: z.int(),
    amount: z.int(),
    meterId: z.string(),
    meter: Meter$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        consumedUnits: "consumed_units",
        creditedUnits: "credited_units",
        meterId: "meter_id",
    });
}));
export function subscriptionMeterToJSON(subscriptionMeter) {
    return JSON.stringify(SubscriptionMeter$outboundSchema.parse(subscriptionMeter));
}
export function subscriptionMeterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionMeter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionMeter' from JSON`);
}
