// @polar-sh/sdk/dist/esm/models/components/subscriptioncreateexternalcustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const SubscriptionCreateExternalCustomerMetadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function subscriptionCreateExternalCustomerMetadataToJSON(subscriptionCreateExternalCustomerMetadata) {
    return JSON.stringify(SubscriptionCreateExternalCustomerMetadata$outboundSchema.parse(subscriptionCreateExternalCustomerMetadata));
}
/** @internal */
export const SubscriptionCreateExternalCustomer$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    productId: z.string(),
    externalCustomerId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        productId: "product_id",
        externalCustomerId: "external_customer_id",
    });
}));
export function subscriptionCreateExternalCustomerToJSON(subscriptionCreateExternalCustomer) {
    return JSON.stringify(SubscriptionCreateExternalCustomer$outboundSchema.parse(subscriptionCreateExternalCustomer));
}
