// @polar-sh/sdk/dist/esm/models/components/subscriptionrevokedmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionRevokedMetadata$inboundSchema = z.pipe(z.object({
    subscription_id: z.string(),
    product_id: z.optional(z.string()),
    amount: z.optional(z.int()),
    currency: z.optional(z.string()),
    recurring_interval: z.optional(z.string()),
    recurring_interval_count: z.optional(z.int()),
}), z.transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
    });
}));
export function subscriptionRevokedMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionRevokedMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionRevokedMetadata' from JSON`);
}
