// @polar-sh/sdk/dist/esm/models/components/subscriptioncanceledmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SubscriptionCanceledMetadata$inboundSchema = z.pipe(z.object({
    subscription_id: z.string(),
    product_id: z.optional(z.string()),
    amount: z.int(),
    currency: z.string(),
    recurring_interval: z.string(),
    recurring_interval_count: z.int(),
    customer_cancellation_reason: z.optional(z.string()),
    customer_cancellation_comment: z.optional(z.string()),
    canceled_at: z.string(),
    ends_at: z.optional(z.string()),
    cancel_at_period_end: z.optional(z.boolean()),
}), z.transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "product_id": "productId",
        "recurring_interval": "recurringInterval",
        "recurring_interval_count": "recurringIntervalCount",
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "canceled_at": "canceledAt",
        "ends_at": "endsAt",
        "cancel_at_period_end": "cancelAtPeriodEnd",
    });
}));
export function subscriptionCanceledMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCanceledMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCanceledMetadata' from JSON`);
}
