// @polar-sh/sdk/dist/esm/models/components/subscriptioncreatecustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const Metadata$outboundSchema = smartUnion([z.string(), z.int(), z.number(), z.boolean()]);
export function metadataToJSON(metadata) {
    return JSON.stringify(Metadata$outboundSchema.parse(metadata));
}
/** @internal */
export const SubscriptionCreateCustomer$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), smartUnion([z.string(), z.int(), z.number(), z.boolean()]))),
    productId: z.string(),
    customerId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        productId: "product_id",
        customerId: "customer_id",
    });
}));
export function subscriptionCreateCustomerToJSON(subscriptionCreateCustomer) {
    return JSON.stringify(SubscriptionCreateCustomer$outboundSchema.parse(subscriptionCreateCustomer));
}
