// @polar-sh/sdk/dist/esm/models/components/subscriptionupdatedevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Customer$inboundSchema } from "./customer.js";
import { SubscriptionUpdatedBillingPeriodMetadata$inboundSchema, } from "./subscriptionupdatedbillingperiodmetadata.js";
import { SubscriptionUpdatedDiscountMetadata$inboundSchema, } from "./subscriptionupdateddiscountmetadata.js";
import { SubscriptionUpdatedProductMetadata$inboundSchema, } from "./subscriptionupdatedproductmetadata.js";
import { SubscriptionUpdatedSeatsMetadata$inboundSchema, } from "./subscriptionupdatedseatsmetadata.js";
import { SubscriptionUpdatedTrialMetadata$inboundSchema, } from "./subscriptionupdatedtrialmetadata.js";
/** @internal */
export const SubscriptionUpdatedEventMetadata$inboundSchema = smartUnion([
    SubscriptionUpdatedProductMetadata$inboundSchema,
    SubscriptionUpdatedSeatsMetadata$inboundSchema,
    SubscriptionUpdatedDiscountMetadata$inboundSchema,
    SubscriptionUpdatedTrialMetadata$inboundSchema,
    SubscriptionUpdatedBillingPeriodMetadata$inboundSchema,
]);
export function subscriptionUpdatedEventMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionUpdatedEventMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionUpdatedEventMetadata' from JSON`);
}
/** @internal */
export const SubscriptionUpdatedEvent$inboundSchema = z.pipe(z.object({
    id: z.string(),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    member_id: z.optional(z.nullable(z.string())),
    external_member_id: z.optional(z.nullable(z.string())),
    child_count: z._default(z.int(), 0),
    parent_id: z.optional(z.nullable(z.string())),
    label: z.string(),
    source: z.literal("system"),
    name: z.literal("subscription.updated"),
    metadata: smartUnion([
        SubscriptionUpdatedProductMetadata$inboundSchema,
        SubscriptionUpdatedSeatsMetadata$inboundSchema,
        SubscriptionUpdatedDiscountMetadata$inboundSchema,
        SubscriptionUpdatedTrialMetadata$inboundSchema,
        SubscriptionUpdatedBillingPeriodMetadata$inboundSchema,
    ]),
}), z.transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
        "member_id": "memberId",
        "external_member_id": "externalMemberId",
        "child_count": "childCount",
        "parent_id": "parentId",
    });
}));
export function subscriptionUpdatedEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionUpdatedEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionUpdatedEvent' from JSON`);
}
