// @polar-sh/sdk/dist/esm/models/components/validatedlicensekey.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { LicenseKeyActivationBase$inboundSchema, } from "./licensekeyactivationbase.js";
import { LicenseKeyCustomer$inboundSchema, } from "./licensekeycustomer.js";
import { LicenseKeyStatus$inboundSchema, } from "./licensekeystatus.js";
/** @internal */
export const ValidatedLicenseKey$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    organization_id: z.string(),
    customer_id: z.string(),
    customer: LicenseKeyCustomer$inboundSchema,
    benefit_id: z.string(),
    key: z.string(),
    display_key: z.string(),
    status: LicenseKeyStatus$inboundSchema,
    limit_activations: z.nullable(z.int()),
    usage: z.int(),
    limit_usage: z.nullable(z.int()),
    validations: z.int(),
    last_validated_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    expires_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    activation: z.optional(z.nullable(LicenseKeyActivationBase$inboundSchema)),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "benefit_id": "benefitId",
        "display_key": "displayKey",
        "limit_activations": "limitActivations",
        "limit_usage": "limitUsage",
        "last_validated_at": "lastValidatedAt",
        "expires_at": "expiresAt",
    });
}));
export function validatedLicenseKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ValidatedLicenseKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidatedLicenseKey' from JSON`);
}
