// @polar-sh/sdk/dist/esm/models/components/webhookorderpaidpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderPaidPayload$inboundSchema = z.object({
    type: z.literal("order.paid"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderPaidPayload$outboundSchema = z.object({
    type: z.literal("order.paid"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Order$outboundSchema,
});
export function webhookOrderPaidPayloadToJSON(webhookOrderPaidPayload) {
    return JSON.stringify(WebhookOrderPaidPayload$outboundSchema.parse(webhookOrderPaidPayload));
}
export function webhookOrderPaidPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderPaidPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderPaidPayload' from JSON`);
}
