// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantUpdatedPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantUpdatedPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: BenefitGrantWebhook$outboundSchema,
});
export function webhookBenefitGrantUpdatedPayloadToJSON(webhookBenefitGrantUpdatedPayload) {
    return JSON.stringify(WebhookBenefitGrantUpdatedPayload$outboundSchema.parse(webhookBenefitGrantUpdatedPayload));
}
export function webhookBenefitGrantUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantUpdatedPayload' from JSON`);
}
