// @polar-sh/sdk/dist/esm/models/components/webhooksubscriptionpastduepayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Subscription$inboundSchema, Subscription$outboundSchema, } from "./subscription.js";
/** @internal */
export const WebhookSubscriptionPastDuePayload$inboundSchema = z.object({
    type: z.literal("subscription.past_due"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Subscription$inboundSchema,
});
/** @internal */
export const WebhookSubscriptionPastDuePayload$outboundSchema = z.object({
    type: z.literal("subscription.past_due"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Subscription$outboundSchema,
});
export function webhookSubscriptionPastDuePayloadToJSON(webhookSubscriptionPastDuePayload) {
    return JSON.stringify(WebhookSubscriptionPastDuePayload$outboundSchema.parse(webhookSubscriptionPastDuePayload));
}
export function webhookSubscriptionPastDuePayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookSubscriptionPastDuePayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionPastDuePayload' from JSON`);
}
