// @polar-sh/sdk/dist/esm/models/components/webhookendpoint.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { WebhookEventType$inboundSchema, } from "./webhookeventtype.js";
import { WebhookFormat$inboundSchema } from "./webhookformat.js";
/** @internal */
export const WebhookEndpoint$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    url: z.string(),
    name: z.optional(z.nullable(z.string())),
    format: WebhookFormat$inboundSchema,
    secret: z.string(),
    organization_id: z.string(),
    events: z.array(WebhookEventType$inboundSchema),
    enabled: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
export function webhookEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEndpoint' from JSON`);
}
