// @polar-sh/sdk/dist/esm/models/components/webhookmemberupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Member$inboundSchema, Member$outboundSchema, } from "./member.js";
/** @internal */
export const WebhookMemberUpdatedPayload$inboundSchema = z.object({
    type: z.literal("member.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Member$inboundSchema,
});
/** @internal */
export const WebhookMemberUpdatedPayload$outboundSchema = z.object({
    type: z.literal("member.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Member$outboundSchema,
});
export function webhookMemberUpdatedPayloadToJSON(webhookMemberUpdatedPayload) {
    return JSON.stringify(WebhookMemberUpdatedPayload$outboundSchema.parse(webhookMemberUpdatedPayload));
}
export function webhookMemberUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookMemberUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookMemberUpdatedPayload' from JSON`);
}
