// @polar-sh/sdk/dist/esm/models/components/webhookorderrefundedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderRefundedPayload$inboundSchema = z.object({
    type: z.literal("order.refunded"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderRefundedPayload$outboundSchema = z.object({
    type: z.literal("order.refunded"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Order$outboundSchema,
});
export function webhookOrderRefundedPayloadToJSON(webhookOrderRefundedPayload) {
    return JSON.stringify(WebhookOrderRefundedPayload$outboundSchema.parse(webhookOrderRefundedPayload));
}
export function webhookOrderRefundedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderRefundedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderRefundedPayload' from JSON`);
}
