// @polar-sh/sdk/dist/esm/models/components/webhookorderupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Order$inboundSchema, Order$outboundSchema, } from "./order.js";
/** @internal */
export const WebhookOrderUpdatedPayload$inboundSchema = z.object({
    type: z.literal("order.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Order$inboundSchema,
});
/** @internal */
export const WebhookOrderUpdatedPayload$outboundSchema = z.object({
    type: z.literal("order.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Order$outboundSchema,
});
export function webhookOrderUpdatedPayloadToJSON(webhookOrderUpdatedPayload) {
    return JSON.stringify(WebhookOrderUpdatedPayload$outboundSchema.parse(webhookOrderUpdatedPayload));
}
export function webhookOrderUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookOrderUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookOrderUpdatedPayload' from JSON`);
}
