// @polar-sh/sdk/dist/esm/models/components/webhookrefundupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Refund$inboundSchema, Refund$outboundSchema, } from "./refund.js";
/** @internal */
export const WebhookRefundUpdatedPayload$inboundSchema = z.object({
    type: z.literal("refund.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Refund$inboundSchema,
});
/** @internal */
export const WebhookRefundUpdatedPayload$outboundSchema = z.object({
    type: z.literal("refund.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Refund$outboundSchema,
});
export function webhookRefundUpdatedPayloadToJSON(webhookRefundUpdatedPayload) {
    return JSON.stringify(WebhookRefundUpdatedPayload$outboundSchema.parse(webhookRefundUpdatedPayload));
}
export function webhookRefundUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookRefundUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookRefundUpdatedPayload' from JSON`);
}
