// @polar-sh/sdk/dist/esm/models/components/webhookmembercreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Member$inboundSchema, Member$outboundSchema, } from "./member.js";
/** @internal */
export const WebhookMemberCreatedPayload$inboundSchema = z.object({
    type: z.literal("member.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Member$inboundSchema,
});
/** @internal */
export const WebhookMemberCreatedPayload$outboundSchema = z.object({
    type: z.literal("member.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Member$outboundSchema,
});
export function webhookMemberCreatedPayloadToJSON(webhookMemberCreatedPayload) {
    return JSON.stringify(WebhookMemberCreatedPayload$outboundSchema.parse(webhookMemberCreatedPayload));
}
export function webhookMemberCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookMemberCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookMemberCreatedPayload' from JSON`);
}
