// @polar-sh/sdk/dist/esm/models/components/webhookcustomerseatclaimedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSeat$inboundSchema, CustomerSeat$outboundSchema, } from "./customerseat.js";
/** @internal */
export const WebhookCustomerSeatClaimedPayload$inboundSchema = z.object({
    type: z.literal("customer_seat.claimed"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: CustomerSeat$inboundSchema,
});
/** @internal */
export const WebhookCustomerSeatClaimedPayload$outboundSchema = z.object({
    type: z.literal("customer_seat.claimed"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: CustomerSeat$outboundSchema,
});
export function webhookCustomerSeatClaimedPayloadToJSON(webhookCustomerSeatClaimedPayload) {
    return JSON.stringify(WebhookCustomerSeatClaimedPayload$outboundSchema.parse(webhookCustomerSeatClaimedPayload));
}
export function webhookCustomerSeatClaimedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerSeatClaimedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerSeatClaimedPayload' from JSON`);
}
