// @polar-sh/sdk/dist/esm/models/components/webhookmemberdeletedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Member$inboundSchema, Member$outboundSchema, } from "./member.js";
/** @internal */
export const WebhookMemberDeletedPayload$inboundSchema = z.object({
    type: z.literal("member.deleted"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Member$inboundSchema,
});
/** @internal */
export const WebhookMemberDeletedPayload$outboundSchema = z.object({
    type: z.literal("member.deleted"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Member$outboundSchema,
});
export function webhookMemberDeletedPayloadToJSON(webhookMemberDeletedPayload) {
    return JSON.stringify(WebhookMemberDeletedPayload$outboundSchema.parse(webhookMemberDeletedPayload));
}
export function webhookMemberDeletedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookMemberDeletedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookMemberDeletedPayload' from JSON`);
}
