// @polar-sh/sdk/dist/esm/models/components/webhookbenefitcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Benefit$inboundSchema, Benefit$outboundSchema, } from "./benefit.js";
/** @internal */
export const WebhookBenefitCreatedPayload$inboundSchema = z.object({
    type: z.literal("benefit.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Benefit$inboundSchema,
});
/** @internal */
export const WebhookBenefitCreatedPayload$outboundSchema = z.object({
    type: z.literal("benefit.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Benefit$outboundSchema,
});
export function webhookBenefitCreatedPayloadToJSON(webhookBenefitCreatedPayload) {
    return JSON.stringify(WebhookBenefitCreatedPayload$outboundSchema.parse(webhookBenefitCreatedPayload));
}
export function webhookBenefitCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitCreatedPayload' from JSON`);
}
