// @polar-sh/sdk/dist/esm/models/components/webhookbenefitgrantcycledpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGrantWebhook$inboundSchema, BenefitGrantWebhook$outboundSchema, } from "./benefitgrantwebhook.js";
/** @internal */
export const WebhookBenefitGrantCycledPayload$inboundSchema = z.object({
    type: z.literal("benefit_grant.cycled"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: BenefitGrantWebhook$inboundSchema,
});
/** @internal */
export const WebhookBenefitGrantCycledPayload$outboundSchema = z.object({
    type: z.literal("benefit_grant.cycled"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: BenefitGrantWebhook$outboundSchema,
});
export function webhookBenefitGrantCycledPayloadToJSON(webhookBenefitGrantCycledPayload) {
    return JSON.stringify(WebhookBenefitGrantCycledPayload$outboundSchema.parse(webhookBenefitGrantCycledPayload));
}
export function webhookBenefitGrantCycledPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookBenefitGrantCycledPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookBenefitGrantCycledPayload' from JSON`);
}
