// @polar-sh/sdk/dist/esm/models/components/webhookrefundcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Refund$inboundSchema, Refund$outboundSchema, } from "./refund.js";
/** @internal */
export const WebhookRefundCreatedPayload$inboundSchema = z.object({
    type: z.literal("refund.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Refund$inboundSchema,
});
/** @internal */
export const WebhookRefundCreatedPayload$outboundSchema = z.object({
    type: z.literal("refund.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Refund$outboundSchema,
});
export function webhookRefundCreatedPayloadToJSON(webhookRefundCreatedPayload) {
    return JSON.stringify(WebhookRefundCreatedPayload$outboundSchema.parse(webhookRefundCreatedPayload));
}
export function webhookRefundCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookRefundCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookRefundCreatedPayload' from JSON`);
}
