// @polar-sh/sdk/dist/esm/models/components/webhookproductupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
/** @internal */
export const WebhookProductUpdatedPayload$inboundSchema = z.object({
    type: z.literal("product.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Product$inboundSchema,
});
/** @internal */
export const WebhookProductUpdatedPayload$outboundSchema = z.object({
    type: z.literal("product.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Product$outboundSchema,
});
export function webhookProductUpdatedPayloadToJSON(webhookProductUpdatedPayload) {
    return JSON.stringify(WebhookProductUpdatedPayload$outboundSchema.parse(webhookProductUpdatedPayload));
}
export function webhookProductUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookProductUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookProductUpdatedPayload' from JSON`);
}
