// @polar-sh/sdk/dist/esm/models/components/webhooksubscriptioncreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Subscription$inboundSchema, Subscription$outboundSchema, } from "./subscription.js";
/** @internal */
export const WebhookSubscriptionCreatedPayload$inboundSchema = z.object({
    type: z.literal("subscription.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Subscription$inboundSchema,
});
/** @internal */
export const WebhookSubscriptionCreatedPayload$outboundSchema = z.object({
    type: z.literal("subscription.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Subscription$outboundSchema,
});
export function webhookSubscriptionCreatedPayloadToJSON(webhookSubscriptionCreatedPayload) {
    return JSON.stringify(WebhookSubscriptionCreatedPayload$outboundSchema.parse(webhookSubscriptionCreatedPayload));
}
export function webhookSubscriptionCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookSubscriptionCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionCreatedPayload' from JSON`);
}
