// @polar-sh/sdk/dist/esm/models/components/webhookdelivery.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { WebhookEvent$inboundSchema } from "./webhookevent.js";
/** @internal */
export const WebhookDelivery$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    succeeded: z.boolean(),
    http_code: z.nullable(z.int()),
    response: z.nullable(z.string()),
    webhook_event: WebhookEvent$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "http_code": "httpCode",
        "webhook_event": "webhookEvent",
    });
}));
export function webhookDeliveryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookDelivery' from JSON`);
}
