// @polar-sh/sdk/dist/esm/models/components/webhookcustomerdeletedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const WebhookCustomerDeletedPayload$inboundSchema = z.object({
    type: z.literal("customer.deleted"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Customer$inboundSchema,
});
/** @internal */
export const WebhookCustomerDeletedPayload$outboundSchema = z.object({
    type: z.literal("customer.deleted"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Customer$outboundSchema,
});
export function webhookCustomerDeletedPayloadToJSON(webhookCustomerDeletedPayload) {
    return JSON.stringify(WebhookCustomerDeletedPayload$outboundSchema.parse(webhookCustomerDeletedPayload));
}
export function webhookCustomerDeletedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerDeletedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerDeletedPayload' from JSON`);
}
