// @polar-sh/sdk/dist/esm/models/components/webhookcustomerseatrevokedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSeat$inboundSchema, CustomerSeat$outboundSchema, } from "./customerseat.js";
/** @internal */
export const WebhookCustomerSeatRevokedPayload$inboundSchema = z.object({
    type: z.literal("customer_seat.revoked"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: CustomerSeat$inboundSchema,
});
/** @internal */
export const WebhookCustomerSeatRevokedPayload$outboundSchema = z.object({
    type: z.literal("customer_seat.revoked"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: CustomerSeat$outboundSchema,
});
export function webhookCustomerSeatRevokedPayloadToJSON(webhookCustomerSeatRevokedPayload) {
    return JSON.stringify(WebhookCustomerSeatRevokedPayload$outboundSchema.parse(webhookCustomerSeatRevokedPayload));
}
export function webhookCustomerSeatRevokedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerSeatRevokedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerSeatRevokedPayload' from JSON`);
}
