// @polar-sh/sdk/dist/esm/models/components/webhookcheckoutexpiredpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Checkout$inboundSchema, Checkout$outboundSchema, } from "./checkout.js";
/** @internal */
export const WebhookCheckoutExpiredPayload$inboundSchema = z.object({
    type: z.literal("checkout.expired"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Checkout$inboundSchema,
});
/** @internal */
export const WebhookCheckoutExpiredPayload$outboundSchema = z.object({
    type: z.literal("checkout.expired"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Checkout$outboundSchema,
});
export function webhookCheckoutExpiredPayloadToJSON(webhookCheckoutExpiredPayload) {
    return JSON.stringify(WebhookCheckoutExpiredPayload$outboundSchema.parse(webhookCheckoutExpiredPayload));
}
export function webhookCheckoutExpiredPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCheckoutExpiredPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCheckoutExpiredPayload' from JSON`);
}
