// @polar-sh/sdk/dist/esm/models/components/webhooksubscriptionactivepayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Subscription$inboundSchema, Subscription$outboundSchema, } from "./subscription.js";
/** @internal */
export const WebhookSubscriptionActivePayload$inboundSchema = z.object({
    type: z.literal("subscription.active"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Subscription$inboundSchema,
});
/** @internal */
export const WebhookSubscriptionActivePayload$outboundSchema = z.object({
    type: z.literal("subscription.active"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Subscription$outboundSchema,
});
export function webhookSubscriptionActivePayloadToJSON(webhookSubscriptionActivePayload) {
    return JSON.stringify(WebhookSubscriptionActivePayload$outboundSchema.parse(webhookSubscriptionActivePayload));
}
export function webhookSubscriptionActivePayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookSubscriptionActivePayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionActivePayload' from JSON`);
}
