// @polar-sh/sdk/dist/esm/models/components/webhooksubscriptionuncanceledpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Subscription$inboundSchema, Subscription$outboundSchema, } from "./subscription.js";
/** @internal */
export const WebhookSubscriptionUncanceledPayload$inboundSchema = z.object({
    type: z.literal("subscription.uncanceled"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Subscription$inboundSchema,
});
/** @internal */
export const WebhookSubscriptionUncanceledPayload$outboundSchema = z.object({
    type: z.literal("subscription.uncanceled"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Subscription$outboundSchema,
});
export function webhookSubscriptionUncanceledPayloadToJSON(webhookSubscriptionUncanceledPayload) {
    return JSON.stringify(WebhookSubscriptionUncanceledPayload$outboundSchema.parse(webhookSubscriptionUncanceledPayload));
}
export function webhookSubscriptionUncanceledPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookSubscriptionUncanceledPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionUncanceledPayload' from JSON`);
}
