// @polar-sh/sdk/dist/esm/models/components/webhookcustomerstatechangedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { CustomerState$inboundSchema, CustomerState$outboundSchema, } from "./customerstate.js";
/** @internal */
export const WebhookCustomerStateChangedPayload$inboundSchema = z.object({
    type: z.literal("customer.state_changed"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: CustomerState$inboundSchema,
});
/** @internal */
export const WebhookCustomerStateChangedPayload$outboundSchema = z.object({
    type: z.literal("customer.state_changed"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: CustomerState$outboundSchema,
});
export function webhookCustomerStateChangedPayloadToJSON(webhookCustomerStateChangedPayload) {
    return JSON.stringify(WebhookCustomerStateChangedPayload$outboundSchema.parse(webhookCustomerStateChangedPayload));
}
export function webhookCustomerStateChangedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerStateChangedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerStateChangedPayload' from JSON`);
}
