// @polar-sh/sdk/dist/esm/models/components/webhookproductcreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
/** @internal */
export const WebhookProductCreatedPayload$inboundSchema = z.object({
    type: z.literal("product.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Product$inboundSchema,
});
/** @internal */
export const WebhookProductCreatedPayload$outboundSchema = z.object({
    type: z.literal("product.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Product$outboundSchema,
});
export function webhookProductCreatedPayloadToJSON(webhookProductCreatedPayload) {
    return JSON.stringify(WebhookProductCreatedPayload$outboundSchema.parse(webhookProductCreatedPayload));
}
export function webhookProductCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookProductCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookProductCreatedPayload' from JSON`);
}
