// @polar-sh/sdk/dist/esm/models/components/webhookcustomercreatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const WebhookCustomerCreatedPayload$inboundSchema = z.object({
    type: z.literal("customer.created"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Customer$inboundSchema,
});
/** @internal */
export const WebhookCustomerCreatedPayload$outboundSchema = z.object({
    type: z.literal("customer.created"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Customer$outboundSchema,
});
export function webhookCustomerCreatedPayloadToJSON(webhookCustomerCreatedPayload) {
    return JSON.stringify(WebhookCustomerCreatedPayload$outboundSchema.parse(webhookCustomerCreatedPayload));
}
export function webhookCustomerCreatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerCreatedPayload' from JSON`);
}
