// @polar-sh/sdk/dist/esm/models/components/webhookcustomerupdatedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
/** @internal */
export const WebhookCustomerUpdatedPayload$inboundSchema = z.object({
    type: z.literal("customer.updated"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: Customer$inboundSchema,
});
/** @internal */
export const WebhookCustomerUpdatedPayload$outboundSchema = z.object({
    type: z.literal("customer.updated"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: Customer$outboundSchema,
});
export function webhookCustomerUpdatedPayloadToJSON(webhookCustomerUpdatedPayload) {
    return JSON.stringify(WebhookCustomerUpdatedPayload$outboundSchema.parse(webhookCustomerUpdatedPayload));
}
export function webhookCustomerUpdatedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerUpdatedPayload' from JSON`);
}
