// @polar-sh/sdk/dist/esm/models/components/webhookcustomerseatassignedpayload.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSeat$inboundSchema, CustomerSeat$outboundSchema, } from "./customerseat.js";
/** @internal */
export const WebhookCustomerSeatAssignedPayload$inboundSchema = z.object({
    type: z.literal("customer_seat.assigned"),
    timestamp: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    data: CustomerSeat$inboundSchema,
});
/** @internal */
export const WebhookCustomerSeatAssignedPayload$outboundSchema = z.object({
    type: z.literal("customer_seat.assigned"),
    timestamp: z.pipe(z.date(), z.transform(v => v.toISOString())),
    data: CustomerSeat$outboundSchema,
});
export function webhookCustomerSeatAssignedPayloadToJSON(webhookCustomerSeatAssignedPayload) {
    return JSON.stringify(WebhookCustomerSeatAssignedPayload$outboundSchema.parse(webhookCustomerSeatAssignedPayload));
}
export function webhookCustomerSeatAssignedPayloadFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookCustomerSeatAssignedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCustomerSeatAssignedPayload' from JSON`);
}
