// @polar-sh/sdk/dist/esm/models/components/webhookevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { WebhookEventType$inboundSchema, } from "./webhookeventtype.js";
/** @internal */
export const WebhookEvent$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    last_http_code: z.optional(z.nullable(z.int())),
    succeeded: z.optional(z.nullable(z.boolean())),
    skipped: z.boolean(),
    payload: z.nullable(z.string()),
    type: WebhookEventType$inboundSchema,
    is_archived: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "last_http_code": "lastHttpCode",
        "is_archived": "isArchived",
    });
}));
export function webhookEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhookEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookEvent' from JSON`);
}
