// @polar-sh/sdk/dist/esm/models/components/webtokenrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
export const WebTokenRequestSubType = {
    User: "user",
    Organization: "organization",
};
/** @internal */
export const WebTokenRequestSubType$outboundSchema = z.enum(WebTokenRequestSubType);
/** @internal */
export const WebTokenRequest$outboundSchema = z.pipe(z.object({
    grantType: z.literal("web"),
    clientId: z.string(),
    clientSecret: z.string(),
    sessionToken: z.string(),
    subType: z._default(WebTokenRequestSubType$outboundSchema, "user"),
    sub: z.optional(z.nullable(z.string())),
    scope: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        grantType: "grant_type",
        clientId: "client_id",
        clientSecret: "client_secret",
        sessionToken: "session_token",
        subType: "sub_type",
    });
}));
export function webTokenRequestToJSON(webTokenRequest) {
    return JSON.stringify(WebTokenRequest$outboundSchema.parse(webTokenRequest));
}
